<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<meta name="generator" content="http://www.nongnu.org/elyxer/"/>
<meta name="create-date" content="2020-07-21"/>
<link rel="stylesheet" href="http://elyxer.nongnu.org/lyx.css" type="text/css" media="all"/>
<title>CPPTRAJ</title>
</head>
<body>
<div id="globalWrapper">
<h1 class="title">
CPPTRAJ
</h1>
<h2 class="author">
Daniel R. Roe
</h2>
<div class="Unindented">
<div class="center">
<a class="FlexURL" href="https://github.com/Amber-MD/cpptraj">https://github.com/Amber-MD/cpptraj</a>
</div>

</div>
<div class="fulltoc">
<div class="tocheader">
Table of Contents
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Section-1">Section 1: Introduction</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-1.1">Subsection 1.1: Manual Syntax Format</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-2">Section 2: Running Cpptraj</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-2.1">Subsection 2.1: Command Line Syntax</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-2.2">Subsection 2.2: Commands</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-2.3">Subsection 2.3: Getting Help</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-2.4">Subsection 2.4: Batch mode</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-2.5">Subsection 2.5: Interactive mode</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-2.6">Subsection 2.6: Trajectory Processing Run</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection-2.6.1">Subsubsection 2.6.1: Actions and multiple topologies</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-2.7">Subsection 2.7: Parallelization</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection-2.7.1">Subsubsection 2.7.1: MPI Trajectory Parallelization</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-2.7.2">Subsubsection 2.7.2: OpenMP Parallelization</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-2.7.3">Subsubsection 2.7.3: CUDA Parallelization</a>
</div>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-3">Section 3: General Concepts</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-3.1">Subsection 3.1: Units</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-3.2">Subsection 3.2: Atom Mask Selection Syntax</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--1">Subsubsection: Distance-based Masks</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-3.3">Subsection 3.3: Ranges</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-3.4">Subsection 3.4: Parameter/Reference Tagging</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-4">Section 4: Variables and Control Structures</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-4.1">Subsection 4.1: for</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-4.2">Subsection 4.2: set</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-4.3">Subsection 4.3: show</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-5">Section 5: Data Sets and Data Files</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-5.1">Subsection 5.1: Data Set Selection Syntax</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-5.2">Subsection 5.2: Data Set Math</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-6">Section 6: Data File Options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-6.1">Subsection 6.1: Standard Data File Options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--2">Subsubsection: Write</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--3">Subsubsection: Read</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.2">Subsection 6.2: Grace Data File Options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--4">Subsubsection: Write</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--5">Subsubsection: Read</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.3">Subsection 6.3: Gnuplot Data File Options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--6">Subsubsection: Write</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.4">Subsection 6.4: Amber REM Log Options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--7">Subsubsection: Read</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.5">Subsection 6.5: Amber MDOUT Options</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.6">Subsection 6.6: Evecs File Options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--8">Subsubsection: Read</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.7">Subsection 6.7: Vector psuedo-traj Options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--9">Subsubsection: Write</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.8">Subsection 6.8: OpenDX file options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--10">Subsubsection: Write</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.9">Subsection 6.9: CCP4 file options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--11">Subsubsection: Write</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.10">Subsection 6.10: Charmm REPD log options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--12">Subsubsection: Read</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-6.11">Subsection 6.11: Amber Constant pH Out options</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--13">Subsubsection: Read</a>
</div>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-7">Section 7: Coordinates (COORDS) Data Set Commands</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-7.1">Subsection 7.1: catcrd</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.2">Subsection 7.2: combinecrd</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.3">Subsection 7.3: crdaction</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.4">Subsection 7.4: createcrd</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.5">Subsection 7.5: crdout</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.6">Subsection 7.6: loadcrd</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.7">Subsection 7.7: loadtraj</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.8">Subsection 7.8: permutedihedrals</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.9">Subsection 7.9: reference</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.10">Subsection 7.10: rotatedihedral</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-7.11">Subsection 7.11: splitcoords</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-8">Section 8: General Commands</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-8.1">Subsection 8.1: activeref</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.2">Subsection 8.2: calc</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.3">Subsection 8.3: clear</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.4">Subsection 8.4: create</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.5">Subsection 8.5: createset</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.6">Subsection 8.6: datafile</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.7">Subsection 8.7: datafilter</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.8">Subsection 8.8: dataset</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.9">Subsection 8.9: debug | prnlev</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.10">Subsection 8.10: ensextension</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.11">Subsection 8.11: exit | quit</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.12">Subsection 8.12: go | run</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.13">Subsection 8.13: help</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.14">Subsection 8.14: list</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.15">Subsection 8.15: noexitonerror</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.16">Subsection 8.16: noprogress</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.17">Subsection 8.17: parallelanalysis</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.18">Subsection 8.18: precision</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.19">Subsection 8.19: readdata</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.20">Subsection 8.20: readensembledata</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.21">Subsection 8.21: readinput</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.22">Subsection 8.22: removedata</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.23">Subsection 8.23: rst</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.24">Subsection 8.24: runanalysis</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.25">Subsection 8.25: select</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.26">Subsection 8.26: selectds</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.27">Subsection 8.27: sortensembledata</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.28">Subsection 8.28: usediskcache</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.29">Subsection 8.29: write | writedata</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-8.30">Subsection 8.30: System Commands</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-9">Section 9: Topology File Commands</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-9.1">Subsection 9.1: angleinfo | angles | printangles</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.2">Subsection 9.2: atominfo | atoms | printatoms</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.3">Subsection 9.3: bondinfo | bonds | printbonds</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.4">Subsection 9.4: change</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.5">Subsection 9.5: charge</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.6">Subsection 9.6: comparetop</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.7">Subsection 9.7: dihedralinfo | dihedrals | printdihedrals</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.8">Subsection 9.8: improperinfo | impropers | printimpropers</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.9">Subsection 9.9: mass</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.10">Subsection 9.10: molinfo</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.11">Subsection 9.11: parm</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection-9.11.1">Subsubsection 9.11.1: <i>PDB format:</i></a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-9.11.2">Subsubsection 9.11.2: Charmm PSF:</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.12">Subsection 9.12: parmbox</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.13">Subsection 9.13: parminfo </a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.14">Subsection 9.14: parmstrip</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.15">Subsection 9.15: parmwrite</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection-9.15.1">Subsubsection 9.15.1: Amber Topology</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-9.15.2">Subsubsection 9.15.2: Charmm PSF</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.16">Subsection 9.16: printub | ubinfo</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.17">Subsection 9.17: resinfo</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.18">Subsection 9.18: scaledihedralk</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.19">Subsection 9.19: solvent</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-9.20">Subsection 9.20: updateparameters</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-10">Section 10: Trajectory File Commands</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-10.1">Subsection 10.1: ensemble</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-10.2">Subsection 10.2: ensemblesize</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-10.3">Subsection 10.3: reference</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-10.4">Subsection 10.4: trajin </a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--14">Subsubsection: Replica Trajectory Processing</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.4.1">Subsubsection 10.4.1: <i>Options for Amber NetCDF, Amber NC Restart, Amber Restart:</i></a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.4.2">Subsubsection 10.4.2: <i>Options for CHARMM DCD:</i></a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-10.5">Subsection 10.5: trajout</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.1">Subsubsection 10.5.1</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.2">Subsubsection 10.5.2</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.3">Subsubsection 10.5.3</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.4">Subsubsection 10.5.4</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.5">Subsubsection 10.5.5: <i>Options for CHARMM COORdinates:</i></a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.6">Subsubsection 10.5.6: <i>Options for CHARMM DCD:</i></a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.7">Subsubsection 10.5.7: <i>Options for GROMACS TRX/XTC format:</i></a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.8">Subsubsection 10.5.8</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.9">Subsubsection 10.5.9: Options for SQM input format:</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-10.5.10">Subsubsection 10.5.10: Options for XYZ format:</a>
</div>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-11">Section 11: Action Commands</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-11.1">Subsection 11.1: align</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.2">Subsection 11.2: angle</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.3">Subsection 11.3: areapermol</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.4">Subsection 11.4: atomiccorr</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.5">Subsection 11.5: atomicfluct | rmsf</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.6">Subsection 11.6: atommap</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.7">Subsection 11.7: autoimage</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.8">Subsection 11.8: average</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.9">Subsection 11.9: avgcoord</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.10">Subsection 11.10: bounds</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.11">Subsection 11.11: box</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.12">Subsection 11.12: center</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.13">Subsection 11.13: check | checkoverlap | checkstructure</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.14">Subsection 11.14: checkchirality</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.15">Subsection 11.15: closest | closestwaters</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.16">Subsection 11.16: cluster</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.17">Subsection 11.17: clusterdihedral</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.18">Subsection 11.18: contacts</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.19">Subsection 11.19: createcrd</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.20">Subsection 11.20: createreservoir</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.21">Subsection 11.21: density</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.22">Subsection 11.22: diffusion</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.23">Subsection 11.23: dihedral </a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.24">Subsection 11.24: dihedralrms | dihrms</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.25">Subsection 11.25: dihedralscan</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.26">Subsection 11.26: dipole</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.27">Subsection 11.27: distance </a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.28">Subsection 11.28: drms | drmsd (distance RMSD)</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.29">Subsection 11.29: dssp</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.30">Subsection 11.30: energy</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.31">Subsection 11.31: esander</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.32">Subsection 11.32: filter</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.33">Subsection 11.33: fixatomorder</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.34">Subsection 11.34: fiximagedbonds</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.35">Subsection 11.35: gist (Grid Inhomogeneous Solvation Theory)</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--15">Subsubsection:  </a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.36">Subsection 11.36: grid</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--16">Subsubsection: Examples</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.37">Subsection 11.37: hbond</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--17">Subsubsection: hbond Examples</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.38">Subsection 11.38: image</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.39">Subsection 11.39: jcoupling</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.40">Subsection 11.40: lessplit</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.41">Subsection 11.41: lie</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.42">Subsection 11.42: lipidorder</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.43">Subsection 11.43: lipidscd</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.44">Subsection 11.44: makestructure</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--18">Subsubsection: &lt;sstype keyword&gt;:&lt;res range&gt;</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--19">Subsubsection: &lt;custom ss name&gt;:&lt;res range&gt;[:&lt;phi&gt;:&lt;psi&gt;]</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--20">Subsubsection: &lt;custom turn name&gt;:&lt;res range&gt;[:&lt;phi1&gt;:&lt;psi1&gt;:&lt;phi2&gt;:&lt;psi2&gt;]</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--21">Subsubsection: &lt;custom dih name&gt;:&lt;res range&gt;[:&lt;dih type&gt;:&lt;angle&gt;]</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--22">Subsubsection: &lt;custom dih name&gt;:&lt;res range&gt;[:&lt;at0&gt;:&lt;at1&gt;:&lt;at2&gt;:&lt;at3&gt;:&lt;angle&gt;[:&lt;offset&gt;]]</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--23">Subsubsection: ref:&lt;range&gt;:&lt;refname&gt;[:&lt;ref range&gt;[:&lt;dih types&gt;]] [refvalsout &lt;file&gt;] [founddihout &lt;file&gt;]</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--24">Subsubsection: Examples</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.45">Subsection 11.45: mask</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.46">Subsection 11.46: matrix</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.47">Subsection 11.47: mindist</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.48">Subsection 11.48: minimage</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.49">Subsection 11.49: molsurf</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.50">Subsection 11.50: multidihedral</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.51">Subsection 11.51: multivector</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.52">Subsection 11.52: nastruct</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--25">Subsubsection: Custom Nucleic Acid Base References</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.53">Subsection 11.53: nativecontacts</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.54">Subsection 11.54: outtraj</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.55">Subsection 11.55: pairdist</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.56">Subsection 11.56: pairwise</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.57">Subsection 11.57: principal</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.58">Subsection 11.58: projection</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.59">Subsection 11.59: pucker </a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.60">Subsection 11.60: radgyr | rog</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.61">Subsection 11.61: radial | rdf</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.62">Subsection 11.62: randomizeions</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.63">Subsection 11.63: replicatecell</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.64">Subsection 11.64: rms | rmsd</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--26">Subsubsection: Per-residue RMSD calculation</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.65">Subsection 11.65: rms2d | 2drms</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.66">Subsection 11.66: rmsavgcorr</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.67">Subsection 11.67: rmsf | atomicfluct</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.68">Subsection 11.68: rotate</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.69">Subsection 11.69: rotdif</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.70">Subsection 11.70: runavg | runningaverage</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.71">Subsection 11.71: scale</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.72">Subsection 11.72: secstruct</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.73">Subsection 11.73: spam</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.74">Subsection 11.74: setvelocity</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.75">Subsection 11.75: stfcdiffusion</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.76">Subsection 11.76: strip</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.77">Subsection 11.77: surf</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.78">Subsection 11.78: symmrmsd</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.79">Subsection 11.79: temperature</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.80">Subsection 11.80: time</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.81">Subsection 11.81: trans | translate</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.82">Subsection 11.82: unstrip</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.83">Subsection 11.83: unwrap</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.84">Subsection 11.84: vector</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.85">Subsection 11.85: velocityautocorr</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.86">Subsection 11.86: volmap</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.87">Subsection 11.87: volume</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.88">Subsection 11.88: watershell</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-11.89">Subsection 11.89: xtalsymm</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-12">Section 12: Analysis Commands</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-12.1">Subsection 12.1: autocorr</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.2">Subsection 12.2: avg</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.3">Subsection 12.3: calcstate</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.4">Subsection 12.4: cluster</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--27">Subsubsection: Clustering Metrics</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--28">Subsubsection: Hints for setting DBSCAN parameters with ’kdist’</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--29">Subsubsection: Using ’dpeaks’ clustering</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection--30">Subsubsection: The CpptrajPairDist file format</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.5">Subsection 12.5: cphstats</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.6">Subsection 12.6: corr | correlationcoe</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.7">Subsection 12.7: crank | crankshaft</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.8">Subsection 12.8: crdfluct</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.9">Subsection 12.9: crosscorr</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.10">Subsection 12.10: curvefit</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.11">Subsection 12.11: diagmatrix</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--31">Subsubsection: Output Format</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.12">Subsection 12.12: divergence</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.13">Subsection 12.13: fft</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.14">Subsection 12.14: hausdorff</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.15">Subsection 12.15: hist | histogram</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.16">Subsection 12.16: integrate</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.17">Subsection 12.17: ired</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--32">Subsubsection: Example of IRED in Cpptraj</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.18">Subsection 12.18: kde</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.19">Subsection 12.19: lifetime</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--33">Subsubsection: Example: hbond lifetime analysis</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.20">Subsection 12.20: lowestcurve</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.21">Subsection 12.21: meltcurve</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.22">Subsection 12.22: modes</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.23">Subsection 12.23: multicurve</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.24">Subsection 12.24: multihist</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.25">Subsection 12.25: phipsi</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.26">Subsection 12.26: regress</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.27">Subsection 12.27: remlog</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.28">Subsection 12.28: rms2d | 2drms</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.29">Subsection 12.29: rmsavgcorr</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.30">Subsection 12.30: rotdif</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--34">Subsubsection: Example</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.31">Subsection 12.31: runningavg</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.32">Subsection 12.32: slope</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.33">Subsection 12.33: spline</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.34">Subsection 12.34: statistics | stat</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection-12.34.1">Subsubsection 12.34.1: Torsion Analysis</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-12.34.2">Subsubsection 12.34.2: Distance Analysis</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsubsection-12.34.3">Subsubsection 12.34.3: Pucker Analysis</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.35">Subsection 12.35: ti</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.36">Subsection 12.36: timecorr</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--35">Subsubsection: Examples</a>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.37">Subsection 12.37: vectormath</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-12.38">Subsection 12.38: wavelet</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsubsection--36">Subsubsection: Wavelet Analysis Feature Extraction</a>
</div>
</div>
</div>
<div class="toc">
<a class="Link" href="#toc-Section-13">Section 13: Analysis Examples</a>
</div>
<div class="tocindent">
<div class="toc">
<a class="Link" href="#toc-Subsection-13.1">Subsection 13.1</a>
</div>
<div class="toc">
<a class="Link" href="#toc-Subsection-13.2">Subsection 13.2: Dihedral covariance matrix calculation and projection for backbone phi/psi (PCA)</a>
</div>
</div>
</div>

</div>
<div class="Unindented">

</div>
<h1 class="Section">
<a class="toc" name="toc-Section-1">1</a> Introduction
</h1>
<div class="Unindented">
<i>Cpptraj</i><span class="bibcites">[<a class="bibliocite" name="cite-31" href="#biblio-31"><span class="bib-index">31</span></a>]</span> (the successor to <i>ptraj</i>) is the main program in Amber for processing coordinate trajectories and data files. <i>Cpptraj</i> has a wide range of functionality, and makes use of OpenMP/MPI to speed up many calculations, including processing ensembles of trajectories and/or conducting multiple analyses in parallel with MPI.<span class="bibcites">[<a class="bibliocite" name="cite-30" href="#biblio-30"><span class="bib-index">30</span></a>]</span>
</div>
<div class="Indented">
Here are several notable features of <i>cpptraj</i>:
</div>
<ol>
<li>
Trajectories with different topologies can be processed in the same run.
</li>
<li>
Several actions/analyses in <i>cpptraj</i> are OpenMP parallelized; see section <a class="Reference" href="#subsec:CpptrajOpenMP">2.7.2↓</a> for more details.
</li>
<li>
Trajectory and ensemble reads can be MPI parallelized.
</li>
<li>
Almost any file read or written by <i>cpptraj</i> can be compressed (with the exception of the NetCDF trajectory format). So for example gzipped/bzipped topology files can be read, and data files can be written out as gzip/bzip2 files. Compression is detected automatically when reading, and is determined by the filename extension (.gz and .bz2 respectively) on writing.
</li>
<li>
The format of output data files can be specified by extension. For example, data files can be written in xmgrace format if the filename given has a ’.agr’ extension. A trajectory can be written in DCD format if the ’.dcd’ extension is used.
</li>
<li>
Multiple output trajectories can be specified, and can be written during action processing (as opposed to only after) via the <b><i>outtraj</i></b> command. In addition, output files can be directed to write only specific frames from the input trajectories.
</li>
<li>
Multiple reference structures can be specified. Specific frames from trajectories may be used as a reference structure.
</li>
<li>
The <b><i>rmsd</i></b> action allows specification of a separate mask for the reference structure. In addition, per-residue RMSD can be calculated easily. 
</li>
<li>
Actions that modify coordinates and topology such as the <b><i>strip</i></b>/<b><i>closest</i></b> actions can often write an accompanying fully-functional stripped topology file. 
</li>
<li>
Users usually are able to fine-tune the output format of data files declared in actions using the &ldquo;<b>out</b>&rdquo; keyword (for example, the precision of the numbers can be changed). In addition, users can control which data sets are written to which files (e.g. if two actions specify the same data file with the ’out’ keyword, data from both actions will be written to that data file).
</li>
<li>
Users can manipulate data sets using mathematical expressions (with some limitations), see <a class="Reference" href="#subsec:cpptraj-Data-Set-Math">5.2 on page 1↓</a> for details.
</li>
<li>
There is some support for creating internal loops over e.g. mask expressions and setting internal variables (see <b><i>for</i></b>, <b><i>set</i></b>, and <b><i>show</i></b> commands).
</li>

</ol>
<div class="Unindented">
See the README.md file in the <i>cpptraj</i> home directory for information on how to build, authors, and so on.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-1.1">1.1</a> Manual Syntax Format
</h2>
<div class="Unindented">
The syntax presented in this manual uses the following conventions:
</div>
<div class="Description">
<span class="Description-entry">&lt;&gt;</span> Denotes a variable.
</div>
<div class="Description">
<span class="Description-entry">[]</span> Denotes something is optional.
</div>
<div class="Description">
<span class="Description-entry">{|}</span> Denotes several choices separated by the ’|’ character; one of the choices must be specified.
</div>
<div class="Description">
<span class="Description-entry">...</span> Denotes the preceding option can be repeated.
</div>
<div class="Unindented">
Everything else is as printed. 
</div>
<h1 class="Section">
<a class="toc" name="toc-Section-2">2</a> Running Cpptraj
</h1>
<div class="Unindented">
<i>Cpptraj</i> can be run in either &ldquo;interactive mode&rdquo; or in &ldquo;batch mode&rdquo;. 
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-2.1">2.1</a> Command Line Syntax
</h2>
<pre class="LyX-Code">
cpptraj [-p &lt;Top0&gt;] [-i &lt;Input0&gt;] [-y &lt;trajin&gt;] [-x &lt;trajout&gt;]
        [-ya &lt;args&gt;] [-xa &lt;args&gt;] [&lt;file&gt;]
        [-c &lt;reference&gt;] [-d &lt;datain&gt;] [-w &lt;dataout&gt;] [-o &lt;output&gt;]
        [-h | --help] [-V | --version] [--defines] [-debug &lt;#&gt;]
        [--interactive] [--log &lt;logfile&gt;] [-tl]
        [-ms &lt;mask&gt;] [-mr &lt;mask&gt;] [--mask &lt;mask&gt;] [--resmask &lt;mask&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Unindented">
<b>*</b> denotes a flag may be specified multiple times.
</div>
<div class="Description">
<span class="Description-entry">-p &lt;Top0&gt;*</span> Load &lt;Top0&gt; as a topology file.
</div>
<div class="Description">
<span class="Description-entry">-i &lt;Input0&gt;*</span> Read input from &lt;Input0&gt;.
</div>
<div class="Description">
<span class="Description-entry">-y &lt;trajin&gt;*</span> Read from trajectory file &lt;trajin&gt;; same as input ’trajin &lt;trajin&gt;’.
</div>
<div class="Description">
<span class="Description-entry">-x &lt;trajout&gt;*</span> Write trajectory file &lt;trajout&gt;; same as input ’trajout &lt;trajout&gt;’.
</div>
<div class="Description">
<span class="Description-entry">-ya &lt;args&gt;*</span> Input trajectory file arguments.
</div>
<div class="Description">
<span class="Description-entry">-xa &lt;args&gt;*</span> Output trajectory file arguments.
</div>
<div class="Description">
<span class="Description-entry">&lt;file&gt;*</span> A topology, input trajectory, or file containing cpptraj input.
</div>
<div class="Description">
<span class="Description-entry">-c &lt;reference&gt;*</span> Read &lt;reference&gt; as reference coordinates; same as input ’reference &lt;reference&gt;’.
</div>
<div class="Description">
<span class="Description-entry">-d &lt;datain&gt;*</span> Read data in from file &lt;datain&gt; (’readdata &lt;datain&gt;’).
</div>
<div class="Description">
<span class="Description-entry">-w &lt;dataout&gt;</span> Write data from &lt;datain&gt; as file &lt;dataout&gt; (’writedata &lt;dataout&gt;).
</div>
<div class="Description">
<span class="Description-entry">-o &lt;output&gt;</span> Write CPPTRAJ STDOUT output to file &lt;output&gt;.
</div>
<div class="Description">
<span class="Description-entry">-h | –help</span> Print command line help and exit.
</div>
<div class="Description">
<span class="Description-entry">-V | –version</span> Print version and exit.
</div>
<div class="Description">
<span class="Description-entry">–defines</span> Print compiler defines and exit.
</div>
<div class="Description">
<span class="Description-entry">-debug &lt;#&gt;</span> Set global debug level to &lt;#&gt;; same as input ’debug &lt;#&gt;’.
</div>
<div class="Description">
<span class="Description-entry">–interactive</span> Force interactive mode.
</div>
<div class="Description">
<span class="Description-entry">–log &lt;logfile&gt;</span> Record commands to &lt;logfile&gt; (interactive mode only). Default is ’cpptraj.log’.
</div>
<div class="Description">
<span class="Description-entry">-tl</span> Print length of trajectories specified with ’-y’ to STDOUT. The total number of frames is written out as ’Frames: &lt;X&gt;’
</div>
<div class="Description">
<span class="Description-entry">-ms &lt;mask&gt;</span> Print selected atom numbers to STDOUT. Selected atoms are written out as ’Selected= 1 2 3 ...’
</div>
<div class="Description">
<span class="Description-entry">-mr</span> &lt;mask&gt; : Print selected residue numbers to STDOUT. Selected residues are written out as ’Selected= 1 2 3 ...’
</div>
<div class="Description">
<span class="Description-entry">–mask &lt;mask&gt;</span> Print detailed atom selection to STDOUT.
</div>
<div class="Description">
<span class="Description-entry">–resmask</span> &lt;mask&gt; : Print detailed residue selection to STDOUT.
</div>

</li>

</ul>
<div class="Unindented">
Note that unlike <i>ptraj</i>, in <i>cpptraj</i> it is not required that a topology file be specified on the command line as long as one is specified in the input file with the ’parm’ keyword. Multiple topology/input files can be specified by use of multiple ’-p’ and ’-i’ flags. All topology and coordinate flags will be processed before any input flags.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-2.2">2.2</a> Commands
</h2>
<div class="Unindented">
Input to <i>cpptraj</i> is in the form of commands, which can be categorized in to 2 types: immediate and queued. Immediate commands are executed as soon as they are encountered. Queued commands are initialized when they are encountered, but are not executed until a Run is executed via a <b><i>run</i></b> or <b><i>go</i></b> command. Actions, Analyses, and Trajectory commands (except <b><i>reference</i></b>) are queued commands; however, they can also be run immediately via commands such as <b><i>crdaction</i></b>, <b><i>runanalysis</i></b>, <b><i>loadcrd</i></b>, etc. See <a class="Reference" href="#sec:Coordinates-as-a">7 on page 1↓</a> for more details.
</div>
<div class="Indented">
Commands fall into seven categories:
</div>
<div class="Description">
<span class="Description-entry">General</span> (Immediate) These commands are executed immediately when entered.
</div>
<div class="Description">
<span class="Description-entry">System</span> (Immediate) These are unix system commands (e.g. ’ls’, ’pwd’, etc).
</div>
<div class="Description">
<span class="Description-entry">Coords</span> (Immediate) These commands are used to manipulate COORDS data sets; see <a class="Reference" href="#sec:Coordinates-as-a">7 on page 1↓</a> for more details.
</div>
<div class="Description">
<span class="Description-entry">Trajectory</span> (Queued) These commands prepare cpptraj for reading or writing trajectories during a Run.
</div>
<div class="Description">
<span class="Description-entry">Topology</span> (Immediate) These commands are used to read, write, and modify topology information.
</div>
<div class="Description">
<span class="Description-entry">Action</span> (Queued) These commands specify actions that will be performed on coordinate frames read in from trajectories during a Run.
</div>
<div class="Description">
<span class="Description-entry">Analysis</span> (Queued) These commands specify analyses that will be performed on data that has been either generated from a Run or read in from an external source.
</div>
<div class="Description">
<span class="Description-entry">Control</span> (Immediate) These commands set up control blocks that can be used to e.g. loop over a set of commands.
</div>
<div class="Unindented">
In addition to normal commands, <i>cpptraj</i> now has the ability to perform certain basic math operations, even on data sets. See <a class="Reference" href="#subsec:cpptraj-Data-Set-Math">5.2 on page 1↓</a> for more details.
</div>
<div class="Indented">
Commands in <i>cpptraj</i> can be read in from an input file or from the interactive command prompt. A ’#’ anywhere on a line denotes a comment; anything after ’#’ will be ignored no matter where it occurs. A ’\’ allows the continuation of one line to another. For example, the input:
</div>
<pre class="LyX-Code">
# Sample input
trajin mdcrd # This is a trajectory
rms first out rmsd.dat \
    :1-10
</pre>
<div class="Unindented">
Translates to:
</div>
<pre class="LyX-Code">
trajin mdcrd
rms first out rmsd.dat :1-10
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-2.3">2.3</a> Getting Help
</h2>
<div class="Unindented">
If in interactive mode, ’help &lt;command&gt;’ can be used to get the associated keywords as well as an abbreviated description of the command. Most commands have a corresponding test which also serves as an example of how to use the command. See $AMBERHOME/AmberTools/test/cpptraj/README for more details.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-2.4">2.4</a> Batch mode
</h2>
<div class="Unindented">
In &ldquo;batch&rdquo; mode, cpptraj is executed from the command line with one or more input files containing commands to be processed or STDIN. The syntax of &lt;input file&gt; is similar to that of <i>ptraj</i>. Keywords specifying different commands are given one per line. Lines beginning with ’#’ are ignored as comments. Lines can also be continued through use of the ’\’ character. This is the only allowed mode for cpptraj.MPI.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-2.5">2.5</a> Interactive mode
</h2>
<div class="Unindented">
In &ldquo;interactive mode&rdquo; users can enter commands in a UNIX-like shell. Interactive mode is useful for running short and simple analyses or for trying out new kinds of analyses. If <i>cpptraj</i> is run with ’–interactive’, no arguments, or no specified input file:
</div>
<pre class="LyX-Code">
cpptraj
cpptraj --interactive
cpptraj &lt;parm file&gt;
cpptraj -p &lt;parm file&gt;
</pre>
<div class="Unindented">
this brings up the interactive interface. This interface supports command history (via the up and down arrows) and tab completion for commands and file names. If no log file name has been given (with ’–log &lt;logfile&gt;’), all commands used in interactive mode will be logged to a file named ’cpptraj.log’, which can subsequently be used as input if desired. When starting cpptraj, command histories will be read from any existing logs.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-2.6">2.6</a> Trajectory Processing &ldquo;Run&rdquo;
</h2>
<div class="Unindented">
Like <i>ptraj</i>, a trajectory processing &ldquo;Run&rdquo; is one of the main ways to run <i>cpptraj</i>. First the Run is set up via commands read in from an input file or the interactive prompt. Trajectories are then read in one frame at a time (or in the case of ensemble processing all frames from a given step are read). Actions are performed on the coordinates stored in the frame, after which any output coordinates are written. At the end of the run, any data sets generated are written, and any queued Analyses are performed.
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-2.6.1">2.6.1</a> Actions and multiple topologies
</h3>
<div class="Unindented">
Since <i>cpptraj</i> supports multiple topology files, during a Run actions are set up every time the topology changes in order to recalculate things like what atoms are in a mask etc. Actions that are not valid for the current topology are skipped for that topology. So for example given two topology files with 100 residues, if the first topology file processed includes a ligand named MOL and the second one does not, the action:
</div>
<pre class="LyX-Code">
distance :80 :MOL out D_80-to-MOL.dat
</pre>
<div class="Unindented">
will be valid for the first topology but not for the second, so it will be skipped as long as the second topology is active.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-2.7">2.7</a> Parallelization
</h2>
<div class="Unindented">
<i>Cpptraj</i> has many levels of parallelization that can be enabled via the ’-mpi’, ’-openmp’, and/or ’-cuda’ configure flags for MPI, OpenMP, and CUDA parallelization respectively. At the highest level, trajectory and ensemble reads are parallelized with MPI. In addition, certain time consuming actions have been parallelized with OpenMP and/or CUDA.
</div>
<div class="Indented">
Note that any combination of the ’-openmp’, ’-cuda’, and ’-mpi’ flags may be used to generate a hybrid MPI/OpenMP/CUDA binary; however this may require additional runtime setup (e.g. setting OMP_NUM_THREADS for OpenMP) to work properly and not oversubscribe cores.
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-2.7.1">2.7.1</a> MPI Trajectory Parallelization<a class="Label" name="subsec:CpptrajMPI"> </a>
</h3>
<div class="Unindented">
<i>Cpptraj</i> has two levels of MPI parallelization for reading input trajectories. The first is for ’<b><i>trajin</i></b>’ trajectory input, where the trajectory read is divided as evenly as possible among all input frames (across-trajectory parallelism). For example, if given two trajectories of 1000 frames each and 4 MPI processes, process 0 reads frames 1-500 of trajectory 1, process 1 reads frames 501-1000 of trajectory 1, process 2 reads frames 1-500 of trajectory 2, and process 3 reads frames 501-1000 of trajectory 2. Most Actions will work with across-trajectory parallelization with the exception of the following:
</div>
<div class="Indented">
’clusterdihedral’, ’contacts’, ’createreservoir’, ’gist’, ’lipidorder’, ’pairwise’, ’stfcdiffusion’, ’unwrap’, and ’xtalsymm’. Note that ’diffusion’ will only work with across-trajectory parallelism if no imaging is to be performed.
</div>
<div class="Indented">
<i><i>The second is for <b></b></i>’ensemble’</i> trajectory input, where the reading/processing/writing of each member of the ensemble is divided up among MPI processes. The number of MPI processes must be a multiple of the ensemble size. If the number of processes is greater than the ensemble size then the processing of each ensemble member will be divided among MPI processes (i.e. across-trajectory parallelism will be used). For example, given an ensemble of 4 trajectories and 8 processes, processes 0 and 1 are assigned to the first ensemble trajectory, processes 2 and 3 are assigned to the second ensemble trajectory, and so on. When using ensemble mode in parallel it is recommended that the <b><i>ensemblesize</i></b> command be used prior to any ensemble command as this will make set up far more efficient.
</div>
<div class="Indented">
In order to use the MPI version, Amber/<i>cpptraj</i> should be configured with the ’-mpi’ flag. You can tell if <i>cpptraj</i> has been compiled with MPI as it will print ’MPI’ in the title, and/or by calling ’cpptraj —defines’ and looking for ’-DMPI’. 
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-2.7.2">2.7.2</a> OpenMP Parallelization<a class="Label" name="subsec:CpptrajOpenMP"> </a>
</h3>
<div class="Unindented">
Some of the more time-consuming actions/analyses in <i>cpptraj</i> have been parallelized with OpenMP to take advantage of machines with multiple cores. In order to use OpenMP parallelization Amber/<i>cpptraj</i> should be configured with the ’-openmp’ flag. You can easily tell if <i>cpptraj</i> has been compiled with OpenMP as it will print ’OpenMP’ in the title, and/or by calling ’cpptraj —defines’ and looking for ’-D_OPENMP’. The following actions/analyses have been OpenMP parallelized:
</div>
<pre class="LyX-Code">
2drms/rms2d
atomiccorr
checkstructure
closest
cluster (pair-wise distance calculation and sieved frame restore only)
dssp/secstruct
energy
gist (non-bonded calculation)
hbond
kde
lipidscd
mask (distance-based masks only)
matrix (coordinate covariance matrices only)
minimage
radial
replicatecell
rmsavgcorr
spam
surf
velocityautocorr
volmap
watershell
wavelet
</pre>
<div class="Unindented">
By default OpenMP <i>cpptraj</i> will use all available cores. The number of OpenMP threads can be controlled by setting the OMP_NUM_THREADS environment variable.
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-2.7.3">2.7.3</a> CUDA Parallelization
</h3>
<div class="Unindented">
Some time-consuming actions in cpptraj have been parallelized with CUDA to take advantage of machines with NVIDIA GPUs. In order to use CUDA parallelization Amber/cpptraj should be configured with the ’-cuda’ flag. You can easily tell if cpptraj has been compiled with CUDA as it will print ’CUDA’ and details on the current graphics device in the title, and/or by calling ’cpptraj —defines’ and looking for ’-DCUDA’. The following actions have been CUDA parallelized:
</div>
<pre class="LyX-Code">
closest
watershell
gist
</pre>
<h1 class="Section">
<a class="toc" name="toc-Section-3">3</a> General Concepts
</h1>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-3.1">3.1</a> Units
</h2>
<div class="Unindented">
Cpptraj uses the AKMA system of units. The execption is time, which is typically expressed in ps (except where noted).\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
<b>Variable</b>
</td>
<td align="center" valign="top">
<b>Unit</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
Length
</td>
<td align="center" valign="top">
Angstrom
</td>

</tr>
<tr>
<td align="center" valign="top">
Energy
</td>
<td align="center" valign="top">
kcal/mol
</td>

</tr>
<tr>
<td align="center" valign="top">
Mass
</td>
<td align="center" valign="top">
AMU
</td>

</tr>
<tr>
<td align="center" valign="top">
Charge
</td>
<td align="center" valign="top">
electron
</td>

</tr>
<tr>
<td align="center" valign="top">
Time
</td>
<td align="center" valign="top">
ps (typically)
</td>

</tr>
<tr>
<td align="center" valign="top">
Force
</td>
<td align="center" valign="top">
kcal/mol*Angstrom
</td>

</tr>

</table>

</div>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-3.2">3.2</a> Atom Mask Selection Syntax
</h2>
<div class="Unindented">
The mask syntax is similar to <i>ptraj</i>. Note that the characters ’<tt>:</tt>’, ’<tt>@</tt>’, and ’<tt>*</tt>’ are reserved for masks and should not be used in output file or data set names. All masks are case-sensitive. Either names or numbers can be used. Masks can contain ranges (denoted with ’<tt>-</tt>’) and comma separated lists. The logical operands ’<tt>&amp;</tt>’ (and), ’<tt>|</tt>’ (or), and ’<tt>!</tt>’ (not) are also supported. 
</div>
<div class="Indented">
The syntax for elementary selections is the following:
</div>
<div class="Description">
<span class="Description-entry">@{atom numlist}</span> e.g. ’@12,17’, ’@54-85’, ’@12,54-85,90’
</div>
<div class="Description">
<span class="Description-entry">@{atom namelist}</span> e.g. ’@CA’, ’@CA,C,O,N,H’
</div>
<div class="Description">
<span class="Description-entry">@%{atom type name}</span> e.g. ’@%CT’
</div>
<div class="Description">
<span class="Description-entry">@/{atom_element_name}</span> e.g. ’@/N’
</div>
<div class="Description">
<span class="Description-entry">:{residue numlist}</span> e.g. ’:1-10’, ’:1,3,5’, ’:1-3,5,7-9’
</div>
<div class="Description">
<span class="Description-entry">:{residue namelist}</span> e.g. ’:LYS’, ’:ARG,ALA,GLY’
</div>
<div class="Description">
<span class="Description-entry">::{chain id}</span> e.g. ’::B’, ’::A,D’. Requires chain ID information be present in the topology.
</div>
<div class="Description">
<span class="Description-entry">:;{pdb residue number}</span> e.g. ’:;2-4,8’. Requires a PDB loaded as topology, or Amber topology with embedded PDB information (see <a class="Reference" href="#subsec:parmed_addPDB"> on page 1↓</a>).
</div>
<div class="Description">
<span class="Description-entry">^{molecule numlist}</span> e.g. ’^1-10’, ’:23,84,111’
</div>
<div class="Description">
<span class="Description-entry">&lt;mask&gt;&lt;distance op&gt;&lt;distance&gt;</span> Selection by distance, see below.
</div>
<div class="Unindented">
Several wildcard characters are supported:
</div>
<div class="Description">
<span class="Description-entry">’*’</span> Zero or more characters.
</div>
<div class="Description">
<span class="Description-entry">’=’</span> Same as ’*’
</div>
<div class="Description">
<span class="Description-entry">’?’</span> One character.
</div>
<div class="Unindented">
The wildcards can also be used with numbers or other mask characters, e.g. ’:?0’ means &ldquo;:10,20,30,40,50,60,70,80,90&rdquo;, ’:*’ means all residues and ’@*’ means all atoms. If the atom name (or type name) contains a wildcard character like an asterisk, it can be explictly selected by escaping (i.e. preceding) the wildcard character with a backslash ’\’. So for example:
</div>
<pre class="LyX-Code">
atoms @C?*
</pre>
<div class="Unindented">
would select atoms named C5, C4*, C422, etc., but:
</div>
<pre class="LyX-Code">
atoms @C?\*
</pre>
<div class="Unindented">
would only select C4* out of the above 3 atoms.
</div>
<div class="Indented">
Compound expressions of the following type are allowed:
</div>
<pre class="LyX-Code">
:{residue numlist | namelist}@{atom namelist | numlist}
</pre>
<div class="Unindented">
and are processed as: 
</div>
<pre class="LyX-Code">
:{residue numlist | namelist} &amp; @{atom namelist | numlist}
</pre>
<div class="Unindented">
e.g. ’:1-10@CA’ is equivalent to &ldquo;:1-10 &amp; @CA&rdquo;.
</div>
<div class="Indented">
More examples: 
</div>
<div class="Description">
<span class="Description-entry">:ALA,TRP</span> All alanine and tryptophan residues. 
</div>
<div class="Description">
<span class="Description-entry">:5,10@CA</span> CA carbon in residues 5 and 10.
</div>
<div class="Description">
<span class="Description-entry">:*&amp;!@H=</span> All non-hydrogen atoms (equivalent to "!@H="). 
</div>
<div class="Description">
<span class="Description-entry">@CA,C,O,N,H</span> All backbone atoms. 
</div>
<div class="Description">
<span class="Description-entry">!@CA,C,O,N,H</span> All non-backbone atoms (=sidechains for proteins only). 
</div>
<div class="Description">
<span class="Description-entry">:1-500@O&amp;!(:WAT|:LYS,ARG)</span> All backbone oxygens in residues 1-500 but not in water, lysine or arginine residues.
</div>
<div class="Description">
<span class="Description-entry">^1-2:ASP</span> All residues named ’ASP’ in the first two molecules.
</div>
<div class="Description">
<span class="Description-entry">::A,D@CA</span> All atoms named ’CA’ in chains A and D.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--1"></a>Distance-based Masks
</h3>
<div class="Unindented">
There are two very important things to keep in mind when using distance based masks:
</div>
<ol>
<li>
Distance-based masks that update each frame are currently only supported by the <b><i>mask</i></b> action.
</li>
<li>
Selection by distance for everything but the <b><i>mask</i></b> action requires defining a reference frame with <b><i>reference</i></b>; distances are then calculated using the <u>specified reference frame only</u>. This reference frame can be changed using the <b><i>activeref</i></b> command.
</li>

</ol>
<div class="Unindented">
The syntax for selection by distance is a <b>&lt;mask&gt;</b> expression followed by a <b>&lt;distance operator&gt;</b> followed by a <b>&lt;distance&gt;</b> (which is in Angstroms). The <b>&lt;distance operator&gt;</b> consists of 2 characters: ’<b>&lt;</b>’ (within) or ’<b>&gt;</b>’ (without) followed by either ’<b>^</b>’ (molecules), ’<b>:</b>’ (residues), or ’<b>@</b>’ (atoms). For example, ’<b>&lt;:</b>3.0’ means &ldquo;residues within 3.0 Angstroms&rdquo; etc. For residue- and molecule-based distance selection, if any atom in that residue/molecule matches the given distance criterion, the entire residue/molecule is selected.
</div>
<div class="Indented">
In plain language, the entire distance mask can be read as &ldquo;Select <b>&lt;distance operator&gt;</b> <b>&lt;distance&gt;</b> of <b>&lt;mask&gt;</b>&rdquo;. So for example, the mask expression:
</div>
<pre class="LyX-Code">
:11-17&lt;@2.4
</pre>
<div class="Unindented">
Means &ldquo;Select atoms within 2.4 Å distance of atoms selected by ’:11-17’ (residues numbered 11 through 17)&rdquo;. 
</div>
<div class="Indented">
To strip everything outside 3.0 Å (i.e. without 3.0 Å) from residue 4 using specified reference coordinates:
</div>
<pre class="LyX-Code">
reference mol.rst7
trajin mol.rst7
strip !(:4&lt;:3.0)
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-3.3">3.3</a> Ranges
</h2>
<div class="Unindented">
For several commands some arguments are ranges (e.g. ’trajout onlyframes &lt;range&gt;’, ’nastruct resrange &lt;range&gt;’, ’rmsd perres range &lt;range&gt;’); <b>THESE ARE NOT ATOM MASKS</b>. They are simple number ranges using ’-’ to specify a range and ’,’ to separate different ranges. For example 1-2,4-6,9 specifies 1 to 2, 4 to 6, and 9, i.e. ’1 2 4 5 6 9’.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-3.4">3.4</a> Parameter/Reference Tagging<a class="Label" name="subsec:Parameter/Reference-Tagging"> </a>
</h2>
<div class="Unindented">
Parameter and reference files may be ’tagged’ (i.e. given a nickname); these tags can then be used in place of the file name itself. A tag in <i>cpptraj</i> is recognized by being bounded by brackets (’[’ and ’]’). This can be particularly useful when reading in many parameter or reference files. For example, when reading in multiple reference structures:
</div>
<pre class="LyX-Code">
trajin Test1.crd
reference 1LE1.NoWater.Xray.rst7 [xray]
reference Test1.crd lastframe [last]
reference Test2.crd 225 [open]
rms Xray ref [xray] :2-12@CA out rmsd.dat
rms Last ref [last] :2-12@CA out rmsd.dat
rms Open ref [open] :2-12@CA out rmsd.dat
</pre>
<div class="Unindented">
This defines three reference structures and gives them tags [xray], [last], and [open]. These reference structures can then be referred to by their tags instead of their filenames by any action that uses reference structures (in this case the RMSD action).
</div>
<div class="Indented">
Similarly, this can be useful when reading in multiple parameter files:
</div>
<pre class="LyX-Code">
parm tz2.ff99sb.tip3p.truncoct.parm7 [tz2-water]
parm tz2.ff99sb.mbondi2.parm7 [tz2-nowater]
trajin tz2.run1.explicit.nc parm [tz2-water]
reference tz2.dry.rst7 parm [tz2-nowater] [tz2]
rms ref [tz2] !(:WAT) out rmsd.dat
</pre>
<div class="Unindented">
This defines two parm files and gives them tags [tz2-water] and [tz2-nowater], then reads in a trajectory associated with one, and a reference structure associated with the other. Note that in the ’reference’ command there are two tags; the first goes along with the ’parm’ keyword and specifies what parameter file the reference should use, the second is the tag given to the reference itself (as in the previous example) and is referred to in the subsequent RMSD action.
</div>
<h1 class="Section">
<a class="toc" name="toc-Section-4">4</a> Variables and Control Structures
</h1>
<div class="Unindented">
As of version 18, CPPTRAJ has limited support for &ldquo;script&rdquo; variables and ’for’ loops. Script variables are referred to by a dollar sign (’$’) prefix and are replaced when they are processed. These are stored in the master data set list like other data and are assigned the type &ldquo;string variable&rdquo;. <b>Note that to use script variables in CPPTRAJ input that is inside another script (e.g. a BASH script), they must be escaped with the ’\’ character,</b> e.g.
</div>
<pre class="LyX-Code">
#!/bin/bash
TOP=MyTop.parm7
cpptraj &lt;&lt;EOF
set topname=$TOP # TOP is a BASH script variable
parm \$topname   # topname is a CPPTRAJ script variable
EOF
</pre>
<div class="Unindented">
Note that regular CPPTRAJ 1D Data Sets that contain a single value can be used as script variables (if the Data Set contains more than 1 value only the first value will be used).
</div>
<div class="Indented">
<table>
<tr>
<td align="center" valign="top">
<b>Command</b>
</td>
<td align="center" valign="top">
<b>Description</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
for
</td>
<td align="center" valign="top">
Create a ’for’ loop.
</td>

</tr>
<tr>
<td align="center" valign="top">
set
</td>
<td align="center" valign="top">
Set or update a script variable.
</td>

</tr>
<tr>
<td align="center" valign="top">
show
</td>
<td align="center" valign="top">
Show all current script variables and their values.
</td>

</tr>

</table>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-4.1">4.1</a> for
</h2>
<pre class="LyX-Code">
for { {atoms|residues|molecules|molfirstres|mollastres}
	    &lt;var&gt; inmask &lt;mask&gt; [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] |
      &lt;var&gt; in &lt;list&gt; |
      &lt;var&gt; oversets &lt;list&gt; |
      &lt;var&gt; datasetblocks &lt;set&gt; blocksize &lt;#&gt; [blockoffset &lt;#&gt;]
        [cumulative [firstblock &lt;#&gt;]] |
	  &lt;var&gt;=&lt;start&gt;;[&lt;var&gt;&lt;end OP&gt;&lt;end&gt;;]&lt;var&gt;&lt;increment OP&gt;[&lt;value&gt;] ... }
	END KEYWORD: ’done’
  Available ’end OP’       : ’&lt;’ ’&gt;’ ’&lt;=’ ’&gt;=’
  Available ’increment OP’ : ’++’, ’--’, ’+=’, ’-=’
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">atoms|residues|molecules|molfirstres|mollastres &lt;var&gt; inmask &lt;mask&gt;</span> Loop over atoms/residues/molecules/first residue in molecules/last residue in molecules selected by the given mask expression, set as script variable <b>&lt;var&gt;</b>.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;name&gt; | parmindex &lt;#&gt; &lt;#&gt;</span> Select topology that <b>&lt;mask&gt;</b> should be based on (default first topology).
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">&lt;var&gt; in &lt;list&gt;</span> Loop over a comma-separated list of strings. File name wildcards can be used.
</div>
<div class="Description">
<span class="Description-entry">&lt;var&gt; oversets &lt;list&gt;</span> Loop over sets selected by comma-separated list of names. Data set wildcards can be used.
</div>
<div class="Description">
<span class="Description-entry">&lt;var&gt; datasetblocks &lt;set&gt;</span> Loop over blocks in specified DataSet.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">blocksize &lt;#&gt;</span> Size of blocks to use.
</div>
<div class="Description">
<span class="Description-entry">[blockoffset &lt;#&gt;]</span> Offset between blocks.
</div>
<div class="Description">
<span class="Description-entry">[cumulative]</span> Instead of blocks of fixed size, use blocks of increasing size incremented by blocksize.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[firstblock &lt;#&gt;]</span> When cumulative, the size of the first block (default is first data set element).
</div>

</li>

</ul>

</li>

</ul>
<div class="Description">
<span class="Description-entry">&lt;var&gt;=&lt;start&gt;;[&lt;var&gt;&lt;end OP&gt;&lt;end&gt;;]&lt;var&gt;&lt;increment OP&gt;[&lt;value&gt;]</span> Loop over integer script variable <b>&lt;var&gt;</b> starting from <b>&lt;start&gt;</b>, optionally ending at <b>&lt;end&gt;</b>, increment by <b>&lt;value&gt;</b>.
</div>
<div class="Unindented">
Data Sets Created (datasetblocks loops):
</div>
<div class="Description">
<span class="Description-entry">&lt;var&gt;[block]:&lt;start idx&gt;</span> (Data set blocks only) Data set block of blocksize starting at &lt;start idx&gt;.
</div>
<div class="Description">
<span class="Description-entry">&lt;var&gt;[cumul]:&lt;end idx&gt;</span> (Cumulative data set blocks only) Data set block starting at firstblock and ending at &lt;end idx&gt;.
</div>

</li>

</ul>
<div class="Unindented">
Create a for loop using one or more mask expressions, integers, etc. Loops can be nested inside each other. Integer loops may be used without an end condition, but in that case at least one descriptor in the loop should have an end condition or refer to a mask. Loops are ended by the <b>done</b> keyword. 
</div>
<div class="Indented">
Note that non-integer variables (e.g. ’inmask’ loops) are NOT incremented after the final loop iteration, i.e. these loop variables always retain their final value.
</div>
<div class="Indented">
For example:
</div>
<pre class="LyX-Code">
for atoms A0 inmask :1-3@CA i=1;i++
  distance d$i :TCS $A0 out $i.dat
done
</pre>
<div class="Unindented">
This loops over all atoms in the mask expression ’:1-3@CA’ (all atoms named CA in residues 1 to 3) and creates a variable named ’i’ that starts from 1 and is incremented by 1 each iteration. Inside the loop, the mask selection is referred to by <b>$A0</b> and the integer by <b>$i</b>. This is equivalent to doing 3 distance commands like so:
</div>
<pre class="LyX-Code">
distance d1 :TCS :1@CA out 1.dat
distance d2 :TCS :2@CA out 2.dat
distance d3 :TCS :3@CA out 3.dat
</pre>
<div class="Unindented">
To loop over files named trajA*.nc and trajB*.nc:
</div>
<pre class="LyX-Code">
for TRAJ in trajA*.nc,trajB*.nc
  trajin $TRAJ 1 last 10
done 
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-4.2">4.2</a> set
</h2>
<pre class="LyX-Code">
set { &lt;variable&gt; &lt;OP&gt; &lt;value&gt; |
	  &lt;variable&gt; &lt;OP&gt; {atoms|residues|molecules} inmask &lt;mask&gt;
	    [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]
	  &lt;variable&gt; &lt;OP&gt; trajinframes }
  Available &lt;OP&gt; : ’=’, ’+=’
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;variable&gt; &lt;OP&gt; &lt;value&gt;</span> Set or append a script variable.
</div>
<div class="Description">
<span class="Description-entry">&lt;variable&gt; &lt;OP&gt; {atoms|residues|molecules} inmask &lt;mask&gt;</span> Set/append a script variable to/by the total number of atoms/residues/molecules selected by given mask expression.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;</span> Topology to which mask should correspond (default first).
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">&lt;variable&gt; &lt;OP&gt; trajinframes</span> Set/append a script variable to/by the total number of frames in trajectories currently loaded by <b><i>trajin</i></b> commands.
</div>

</li>

</ul>
<div class="Unindented">
Set (&lt;OP&gt; = ’=’) or append (&lt;OP&gt; = ’+=’) a script variable. Script variables are character strings, and are referred to in CPPTRAJ input by using a dollar sign ’$’ prefix.
</div>
<div class="Indented">
For example, the following input will load files my.parm7 and my.rst7:
</div>
<pre class="LyX-Code">
set PREFIX = my
trajin $PREFIX.parm7
trajin $PREFIX.rst7
</pre>
<div class="Unindented">
For example, the following input will print info for the last 10 atoms in a topology to ’last10.dat’:
</div>
<pre class="LyX-Code">
set Natom = atoms inmask * 
last10 = $Natom - 10
show
atoms "@$last10 - $Natom" out last10.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-4.3">4.3</a> show
</h2>
<pre class="LyX-Code">
show [&lt;var1&gt; ...]
</pre>
<div class="Unindented">
If no variable names specified, show all current script variables and their values. Otherwise, show the values of the specified script variables.
</div>
<h1 class="Section">
<a class="toc" name="toc-Section-5">5</a> Data Sets and Data Files
</h1>
<div class="Unindented">
In <i>cpptraj</i>, Actions and Analyses can generate one or more data sets which are available for further processing. For example, the <b><i>distance</i></b> command creates a data set containing distances vs time. The data set can be named by the user simply by specifying a non-keyword string as an additional argument. If no name is given, a default one will be generated based on the action name and data set number. For example:
</div>
<pre class="LyX-Code">
distance d1-2 :1 :2 out d1-2.dat
</pre>
<div class="Unindented">
will create a data set named &ldquo;d1-2&rdquo;. If a name is not specified, e.g.:
</div>
<pre class="LyX-Code">
distance :1 :2 out d1-2.dat
</pre>
<div class="Unindented">
the data set will be named &ldquo;Dis_00000&rdquo;.
</div>
<div class="Indented">
Data files are created automatically by most commands, usually via the &ldquo;<b>out</b>&rdquo; keyword. Data files can also be explicitly created with the <b><i>write</i>/<i>writedata</i></b> and <b><i>create</i></b> commands. Data can also be read in from files via the <b><i>readdata</i></b> command. Cpptraj currently recognizes the formats listed in <a class="Reference" href="#tab:cpptraj-DataFile-formats">1↓</a>, although it cannot write in all formats. In addition, a data set must be valid for the data file format. For example, 3D data (such as a grid) can be written to an OpenDX format file but not a Grace format file. 
</div>
<div class="Indented">
The default file format is called ’Standard’, which simply has data in columns, like <i>ptraj</i>, although multiple data sets can be directed to the same output file. The format of a file can be changed either by specifying a recognized keyword (either on the command line itself or later via a ’datafile’ command) or by giving the file an extension corresponding to te format, so ’filename.agr’ will output in Grace format, and ’filename.gnu’ will output in Gnuplot contour, and so on. The xmgrace/gnuplot output is particularly nice for the secstruct sumout and rmsd perresout files. Additional options for data files can be found in <a class="Reference" href="#sec:Datafile-Commands">6 on page 1↓</a>.
</div>
<div class="Indented">
Any action using the &ldquo;out&rdquo; keyword will allow data sets from separate commands to be written into the same file. For example, the commands:
</div>
<pre class="LyX-Code">
dihedral phi :1@C :2@N :2@CA :2@C out phipsi.dat
dihedral psi :2@N :2@CA :2@C :3@N out phipsi.dat
</pre>
<div class="Unindented">
will assign the &ldquo;phi&rdquo; and &ldquo;psi&rdquo; data sets generated from each action to the standard data output file &ldquo;phipsi.dat&rdquo;:
</div>
<pre class="LyX-Code">
#Frame   phi   psi
</pre>
<div class="Unindented">
<div class="float">
<a class="Label" name="tab:cpptraj-DataFile-formats"> </a><div class="table">
<div class="center">
<table>
<tr>
<td align="center" valign="top" style="width: 20%;">
<b>Format</b>
</td>
<td align="center" valign="top" style="width: 20%;">
<b>Filename Extensions</b>
</td>
<td align="center" valign="top" style="width: 20%;">
<b>Keyword</b>
</td>
<td align="center" valign="top" style="width: 20%;">
<b>Valid Dimensions</b>
</td>
<td align="center" valign="top" style="width: 20%;">
<b>Notes</b>
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Standard
</td>
<td align="center" valign="top" style="width: 20%;">
.dat
</td>
<td align="center" valign="top" style="width: 20%;">
dat
</td>
<td align="center" valign="top" style="width: 20%;">
1D, 2D, 3D
</td>
<td align="center" valign="top" style="width: 20%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Grace
</td>
<td align="center" valign="top" style="width: 20%;">
.agr, .xmgr
</td>
<td align="center" valign="top" style="width: 20%;">
grace
</td>
<td align="center" valign="top" style="width: 20%;">
1D
</td>
<td align="center" valign="top" style="width: 20%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Gnuplot
</td>
<td align="center" valign="top" style="width: 20%;">
.gnu
</td>
<td align="center" valign="top" style="width: 20%;">
gnu
</td>
<td align="center" valign="top" style="width: 20%;">
1D, 2D
</td>
<td align="center" valign="top" style="width: 20%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Xplor
</td>
<td align="center" valign="top" style="width: 20%;">
.xplor, .grid
</td>
<td align="center" valign="top" style="width: 20%;">
xplor
</td>
<td align="center" valign="top" style="width: 20%;">
3D
</td>
<td align="center" valign="top" style="width: 20%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
OpenDX
</td>
<td align="center" valign="top" style="width: 20%;">
.dx
</td>
<td align="center" valign="top" style="width: 20%;">
opendx
</td>
<td align="center" valign="top" style="width: 20%;">
3D
</td>
<td align="center" valign="top" style="width: 20%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Amber REM log
</td>
<td align="center" valign="top" style="width: 20%;">
.log
</td>
<td align="center" valign="top" style="width: 20%;">
remlog
</td>
<td align="center" valign="top" style="width: 20%;">
-
</td>
<td align="center" valign="top" style="width: 20%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Amber MDOUT
</td>
<td align="center" valign="top" style="width: 20%;">
.mdout
</td>
<td align="center" valign="top" style="width: 20%;">
mdout
</td>
<td align="center" valign="top" style="width: 20%;">
-
</td>
<td align="center" valign="top" style="width: 20%;">
Energy information, Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Amber Evecs
</td>
<td align="center" valign="top" style="width: 20%;">
.evecs
</td>
<td align="center" valign="top" style="width: 20%;">
evecs
</td>
<td align="center" valign="top" style="width: 20%;">
Modes data set only
</td>
<td align="center" valign="top" style="width: 20%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Amber Constant pH output
</td>
<td align="center" valign="top" style="width: 20%;">
.cpout
</td>
<td align="center" valign="top" style="width: 20%;">
cpout
</td>
<td align="center" valign="top" style="width: 20%;">
pH data only
</td>
<td align="center" valign="top" style="width: 20%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Vector pseudo-traj
</td>
<td align="center" valign="top" style="width: 20%;">
.vectraj
</td>
<td align="center" valign="top" style="width: 20%;">
vectraj
</td>
<td align="center" valign="top" style="width: 20%;">
Vector data set only.
</td>
<td align="center" valign="top" style="width: 20%;">
Write Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Gromacs XVG
</td>
<td align="center" valign="top" style="width: 20%;">
.xvg
</td>
<td align="center" valign="top" style="width: 20%;">
xvg
</td>
<td align="center" valign="top" style="width: 20%;">
-
</td>
<td align="center" valign="top" style="width: 20%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
CCP4
</td>
<td align="center" valign="top" style="width: 20%;">
.ccp4
</td>
<td align="center" valign="top" style="width: 20%;">
ccp4
</td>
<td align="center" valign="top" style="width: 20%;">
3D
</td>
<td align="center" valign="top" style="width: 20%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Charmm REPD log
</td>
<td align="center" valign="top" style="width: 20%;">
.exch
</td>
<td align="center" valign="top" style="width: 20%;">
charmmrepd
</td>
<td align="center" valign="top" style="width: 20%;">
-
</td>
<td align="center" valign="top" style="width: 20%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 20%;">
Charmm Output
</td>
<td align="center" valign="top" style="width: 20%;">
.charmmout
</td>
<td align="center" valign="top" style="width: 20%;">
charmmout
</td>
<td align="center" valign="top" style="width: 20%;">
-
</td>
<td align="center" valign="top" style="width: 20%;">
Energy information, Read Only
</td>

</tr>

</table>

</div>
<div class="caption">
Table 1 DataFile formats recognized by <i>cpptraj</i>. ’Valid Dimensions’ shows what dimensions the format is valid for (e.g. you cannot write a 1D data set with OpenDX format).
</div>

</div>

</div>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-5.1">5.1</a> Data Set Selection Syntax
</h2>
<div class="Unindented">
Many analysis commands can be used to analyze multiple data sets. The general format for selecting data sets is:
</div>
<pre class="LyX-Code">
&lt;name&gt;[&lt;aspect&gt;]:&lt;index&gt;
</pre>
<div class="Unindented">
The ’*’ character can be used as a wild-card for <i>entire</i> names (no partial matches).
</div>
<ul>
<li>
<b>&lt;name&gt;:</b> The data set name, usually specified in the action (e.g. in ’distance d0 @1 @2’ the data set name is &ldquo;d0&rdquo;).
</li>
<li>
<b>&lt;aspect&gt;:</b> Optional; this is set for certain data sets internally in order to easily select subsets of data. <b>The brackets are required</b>. For example, when using ’hbond series’, both solute-solute and solute-solvent hydrogen bond time series may be generated. To select all solute-solute hydrogen bonds one would use the aspect &ldquo;[solutehb]&rdquo;; to select solute-solvent hydrogen bonds the aspect &ldquo;[solventhb]&rdquo; would be used. Aspects are hard-coded and are listed in the commands that use them.
</li>
<li>
<b>&lt;index&gt;:</b> Optional; for actions that generate many data sets (such as ’rmsd perres’) an index is used. Depending on the action, the index may correspond to atom #s, residue #s, etc. A number range (comma and/or dash separated) may be used.
</li>

</ul>
<div class="Unindented">
For example: to select all data sets with aspect &ldquo;[shear]&rdquo; named NA_00000:
</div>
<pre class="LyX-Code">
NA_00000[shear]
</pre>
<div class="Unindented">
To select all data sets with aspect &ldquo;[stagger]&rdquo; with any name, indices 1 and 3:
</div>
<pre class="LyX-Code">
*[stagger]:1,3
</pre>
<div class="Unindented">
In ensemble mode, data set selection has additional syntax:
</div>
<pre class="LyX-Code">
&lt;name&gt;[&lt;aspect&gt;]:&lt;index&gt;%&lt;member&gt;
</pre>
<div class="Unindented">
Where &lt;member&gt; is the ensemble member number starting from 0.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-5.2">5.2</a> Data Set Math<a class="Label" name="subsec:cpptraj-Data-Set-Math"> </a>
</h2>
<div class="Unindented">
As of version 15, <i>cpptraj</i> can perform basic math operations, even on data sets (with some limitations). Currently recognized operations are:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
<b>Operation</b>
</td>
<td align="center" valign="top">
<b>Symbol</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
Minus
</td>
<td align="center" valign="top">
-
</td>

</tr>
<tr>
<td align="center" valign="top">
Plus
</td>
<td align="center" valign="top">
+
</td>

</tr>
<tr>
<td align="center" valign="top">
Divide
</td>
<td align="center" valign="top">
/
</td>

</tr>
<tr>
<td align="center" valign="top">
Multiply
</td>
<td align="center" valign="top">
*
</td>

</tr>
<tr>
<td align="center" valign="top">
Power
</td>
<td align="center" valign="top">
^
</td>

</tr>
<tr>
<td align="center" valign="top">
Negate
</td>
<td align="center" valign="top">
-
</td>

</tr>
<tr>
<td align="center" valign="top">
Assign
</td>
<td align="center" valign="top">
=
</td>

</tr>

</table>

</div>

</div>
<div class="Indented">
Several functions are also supported:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
<b>Function</b>
</td>
<td align="center" valign="top">
<b>Form</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
Square Root
</td>
<td align="center" valign="top">
sqrt()
</td>

</tr>
<tr>
<td align="center" valign="top">
Exponential
</td>
<td align="center" valign="top">
exp()
</td>

</tr>
<tr>
<td align="center" valign="top">
Natural Logarithm
</td>
<td align="center" valign="top">
ln()
</td>

</tr>
<tr>
<td align="center" valign="top">
Absolute Value
</td>
<td align="center" valign="top">
abs()
</td>

</tr>
<tr>
<td align="center" valign="top">
Sine
</td>
<td align="center" valign="top">
sin()
</td>

</tr>
<tr>
<td align="center" valign="top">
Cosine
</td>
<td align="center" valign="top">
cos()
</td>

</tr>
<tr>
<td align="center" valign="top">
Tangent
</td>
<td align="center" valign="top">
tan()
</td>

</tr>
<tr>
<td align="center" valign="top">
Summation
</td>
<td align="center" valign="top">
sum()
</td>

</tr>
<tr>
<td align="center" valign="top">
Average
</td>
<td align="center" valign="top">
avg()
</td>

</tr>
<tr>
<td align="center" valign="top">
Standard Deviation
</td>
<td align="center" valign="top">
stdev()
</td>

</tr>
<tr>
<td align="center" valign="top">
Minimum
</td>
<td align="center" valign="top">
min()
</td>

</tr>
<tr>
<td align="center" valign="top">
Maximum
</td>
<td align="center" valign="top">
max()
</td>

</tr>

</table>

</div>

</div>
<div class="Indented">
Numbers can be expressed in scientific notation using &ldquo;E&rdquo; notation, e.g. 1E-5 = 0.00001. The parser also recognizes PI as the number pi. Expressions can also be enclosed in parentheses. So for example, the following expression is valid:
</div>
<pre class="LyX-Code">
&gt; 1 - ln(sin(PI/4) * 2)^2 
Result: 0.879887
</pre>
<div class="Unindented">
Results of numerical calculations like the above can be assigned to a variable (essentially a data set of size 1) for use in subsequent calculations, e.g.
</div>
<pre class="LyX-Code">
&gt; R = 1 - ln(sin(PI/4) * 2)^2 
Result stored in ’R’ 
&gt; R + 1 Result: 1.879887
</pre>
<div class="Unindented">
Data sets can be specified in expressions as well. Currently data sets in an expression must be of the same type and only 1D, 2D, and 3D data sets are supported. Functions are applied to each member of the data set. So for example, given two 1D data sets of the same size named D0 and D1, the following expression:
</div>
<pre class="LyX-Code">
&gt; D2 = sqrt( D0 ) + D1
</pre>
<div class="Unindented">
would take the square root of each member of D0, add it to the corresponding member of D1, and assign the result to D2. The following table lists which operations are valid for data set types. If a type is not listed it is not supported:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top" style="width: 25%;">
<b>Data Set Type</b>
</td>
<td align="center" valign="top" style="width: 25%;">
<b>Supported Ops</b>
</td>
<td align="center" valign="top" style="width: 25%;">
<b>Supported Funcs</b>
</td>
<td align="center" valign="top" style="width: 25%;">
<b>Notes</b>
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
1D (integer, double, float)
</td>
<td align="center" valign="top" style="width: 25%;">
All
</td>
<td align="center" valign="top" style="width: 25%;">
All
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
1D (vector)
</td>
<td align="center" valign="top" style="width: 25%;">
+, -, *, /, =
</td>
<td align="center" valign="top" style="width: 25%;">
None
</td>
<td align="center" valign="top" style="width: 25%;">
’*’ is dot product
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
2D (matrices)
</td>
<td align="center" valign="top" style="width: 25%;">
+, -, /, *, =
</td>
<td align="center" valign="top" style="width: 25%;">
sum, avg, stdev, min, max
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
3D (grids)
</td>
<td align="center" valign="top" style="width: 25%;">
+, -, /, *, =
</td>
<td align="center" valign="top" style="width: 25%;">
sum, avg, stdev, min, max
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>

</table>

</div>

</div>
<h1 class="Section">
<a class="toc" name="toc-Section-6">6</a> Data File Options<a class="Label" name="sec:Datafile-Commands"> </a>
</h1>
<div class="Unindented">
Data file output can be handled multiple ways in cpptraj. Output data files can be created by Actions/Analyses/Commands, or can be explicitly created with <b><i>writedata</i></b> (<a class="Reference" href="#subsec:cpptraj_writedata">8.29 on page 1↓</a>) or <b><i>create</i></b> (<a class="Reference" href="#subsec:cpptraj_create">8.4 on page 1↓</a>) commands. Reading data from files is only done via the <b><i>readdata</i></b> command (<a class="Reference" href="#subsec:cpptraj_readdata">8.19 on page 1↓</a>).
</div>
<div class="Indented">
In general, data files which have been declared with an <b>’out’</b> keyword will recognize data file write keywords on the same command line. For example, the <b>’time’</b> argument can be passed directly to the output from a <b><i>distance</i></b> command:
</div>
<pre class="LyX-Code">
distance d0 :1 :2 out d0.agr time 0.001
</pre>
<div class="Unindented">
The data file format can be changed from standard implicitly by using specific filename extensions or keywords. If the extension is not recognized or no keyword is give the default format is ’Standard’. Keywords and extensions for data file formats recognized by <i>cpptraj</i> are shown in <a class="Reference" href="#tab:cpptraj-DataFile-formats">1↑</a>. Note that the use of certain options may be restricted for certain data file formats. These options can also be passed to data files via the <b><i>datafile</i></b> command (<a class="Reference" href="#subsec:cpptraj_datafile_command">8.6 on page 1↓</a>).
</div>
<pre class="LyX-Code">
[&lt;format keyword&gt;]
[{xlabel|ylabel|zlabel} &lt;label&gt;] [{xmin|ymin|zmin} &lt;min&gt;] [sort]
[{xstep|ystep|zstep} &lt;step&gt;] [time &lt;dt&gt;] [prec &lt;width&gt;[.&lt;precision&gt;]]
[xprec &lt;width&gt;[.&lt;precision&gt;]] [xfmt {double|scientific|general}]
[noensextension]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">{xlabel | ylabel | zlabel} &lt;label&gt;</span> Set the x-axis label for the specified datafile to &lt;label&gt;. For regular data files this is the header for the first column of data. If the data is at least 2-dimensional ’datafile ylabel &lt;label&gt;’ will likewise set the y-axis label.
</div>
<div class="Description">
<span class="Description-entry">{xmin | ymin | zmin} &lt;min&gt;</span> Set the starting X coordinate value to &lt;min&gt;. If the data is at least 2-dimensional ’datafile ymin &lt;min&gt;’ will likewise set the starting Y coordinate value.
</div>
<div class="Description">
<span class="Description-entry">sort</span> Sort data sets prior to write. Ordering is by name, aspect, then index (all descending).
</div>
<div class="Description">
<span class="Description-entry">{xstep | ystep | zstep} &lt;step&gt;</span> Multiply each frame number by &lt;step&gt; (x coordinates). If the data is at least 2-dimensional ’datafile ystep &lt;step&gt;’ will likewise multiply y coordinates by &lt;step&gt;.
</div>
<div class="Description">
<span class="Description-entry">time &lt;dt&gt;</span> Equivalent to the <i>ptraj</i> argument ’time’ that could be specified with many actions. Multiplies frame numbers (x-axis) by &lt;dt&gt;.
</div>
<div class="Description">
<span class="Description-entry">prec &lt;width&gt;[.&lt;precision&gt;]</span> Change the output format width (and optionally precision) of all sets <i>subsequently</i> added to the data file (i.e. does not change the precision of any data sets currently in the file). For example,
</div>
<pre class="LyX-Code">
prec 12.4
prec 10
</pre>
<div class="Description">
<span class="Description-entry">xprec &lt;width&gt;[.&lt;precision&gt;]</span> Change output ordinate width and precision.
</div>
<div class="Description">
<span class="Description-entry">xfmt {double|scientific|general}</span> Change output ordinate format.
</div>
<div class="Description">
<span class="Description-entry">[noensextension]</span> Omit ensemble extension in ensemble processing mode. <b>NOTE: THIS OPTION HAS NOT BEEN FULLY TESTED IN PARALLEL.</b>
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.1">6.1</a> Standard Data File Options
</h2>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--2"></a>Write
</h3>
<pre class="LyX-Code">
[invert] [noxcol] [groupby &lt;type&gt;] [noheader] [square2d|nosquare2d]\begin_inset Separator latexpar\end_inset
[nosparse|sparse [cut &lt;cutoff&gt;]]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">invert</span> Normally, data is written out with X-values pertaining to frames (i.e. data over all trajectories is printed in columns). This command flips that behavior so that X-values pertain to data sets (i.e. data over all trajectories is printed in rows).
</div>
<div class="Description">
<span class="Description-entry">groupby &lt;type&gt;</span> (1D) group data sets by &lt;type&gt;:
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">name</span> Group by name.
</div>
<div class="Description">
<span class="Description-entry">aspect</span> Group by aspect.
</div>
<div class="Description">
<span class="Description-entry">idx</span> Group by index.
</div>
<div class="Description">
<span class="Description-entry">ens</span> Group by ensemble number.
</div>
<div class="Description">
<span class="Description-entry">dim</span> Group by dimension.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">xcol</span> Write indices for the specified datafile. This is usually the default behavior.
</div>
<div class="Description">
<span class="Description-entry">noxcol</span> Prevent printing of indices (i.e. the #Frame column in most datafiles) for the specified datafile. Useful e.g. if one would like a 2D plot such as phi vs psi. For example, given the input:\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<pre class="LyX-Code">
dihedral phi :1@C :2@N :2@CA :2@C out phipsi.dat
dihedral psi :2@N :2@CA :2@C :3@N out phipsi.dat
datafile phipsi.dat noxcol
</pre>
<div class="Unindented">
<i>Cpptraj</i> will write a 2 column datafile containing only phi and psi, no frame numbers will be written.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">header</span> Write header line at beginning of data file. This is usually the default behavior.
</div>
<div class="Description">
<span class="Description-entry">noheader</span> Prevent printing of header line (e.g. ’#Frame    D1’) at the beginning of data file.
</div>
<div class="Description">
<span class="Description-entry">square2d</span> Write 2D data as a square matrix, e.g.:\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<pre class="LyX-Code">
&lt;1,1&gt; &lt;2,1&gt; &lt;3,1&gt;
&lt;1,2&gt; &lt;2,2&gt; &lt;3,2&gt;
</pre>

</li>

</ul>
<div class="Description">
<span class="Description-entry">nosquare2d</span> Write 2D data in 3 columns as:\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<pre class="LyX-Code">
&lt;X&gt; &lt;Y&gt; &lt;Value&gt;
</pre>

</li>

</ul>
<div class="Description">
<span class="Description-entry">sparse</span> Only write 3D grid voxels with value &gt; cutoff (default 0).
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">cut &lt;cut&gt;</span> Cutoff for ’sparse’; default 0.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">nosparse</span> Write all 3D voxels (default).
</div>

</li>

</ul>
<pre class="LyX-Code">

</pre>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--3"></a>Read
</h3>
<pre class="LyX-Code">
[ read1d [index &lt;col&gt;] [onlycols &lt;range&gt;] ]
[read2d] [vector] [mat3x3]\begin_inset Separator latexpar\end_inset
[ read3d [dims &lt;nx&gt;,&lt;ny&gt;,&lt;nz&gt;] [origin &lt;ox&gt;,&lt;oy&gt;,&lt;oz&gt;]
  [delta &lt;dx&gt;,&lt;dy&gt;,dz&gt;] [prec {dbl|flt}] [bin {center|corner} ]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">index &lt;col&gt;</span> Use column &lt;col&gt; (starting from 1) as index column (1D data only).
</div>
<div class="Description">
<span class="Description-entry">read1d</span> Read data as 1D data sets (default).
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">index &lt;col&gt;</span> Use column &lt;col&gt; (starting from 1) as index column (1D data only).
</div>
<div class="Description">
<span class="Description-entry">onlycols &lt;range&gt;</span> Only read columns in range.
</div>
<div class="Description">
<span class="Description-entry">floatcols &lt;range&gt;</span> Force specified columns to be read as single-precision floats.
</div>
<div class="Description">
<span class="Description-entry">intcols &lt;range&gt;</span> Force specified columns to be read as integers.
</div>
<div class="Description">
<span class="Description-entry">stringcols &lt;range&gt;</span> Force specified columns to be read as strings.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">read2d</span> Read data as 2D square matrix.
</div>
<div class="Description">
<span class="Description-entry">vector</span> Read data as vector. If indices are present they will be skipped. Assume first 3 columns after the index volumn are vector X, Y, and Z, and (if present) the next 3 columns contain vector origin X, Y, and Z.
</div>
<div class="Description">
<span class="Description-entry">mat3x3</span> Read data as 3x3 matrix. If indices are present they will be skipped. Assume matrices are in row major order on each line, i.e. M(1,1) M(1,2) ... M(3,2) M(3,3).
</div>
<div class="Description">
<span class="Description-entry">read3d</span> Read data as 3D grid. If no dimension data in file must also speify ’dims’.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">dims &lt;dx&gt;,&lt;dy&gt;,&lt;dz&gt;</span> Grid dimensions.
</div>
<div class="Description">
<span class="Description-entry">origin &lt;ox&gt;,&lt;oy&gt;,&lt;oz&gt;</span> Grid origins (default 0,0,0).
</div>
<div class="Description">
<span class="Description-entry">delta &lt;dx&gt;,&lt;dy&gt;,dz&gt;</span> Grid spacings (default 1,1,1).
</div>
<div class="Description">
<span class="Description-entry">prec {dbl|flt}</span> Grid precision, double or float (default float).
</div>
<div class="Description">
<span class="Description-entry">bin {center|corner}</span> Coords specify bin centers or corners (default corners).
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
By default, standard data files are assumed to contain 1D data in columns. Data set legends will be read in if the file has a header line (denoted by ’#’). Columns labeled ’#Frame’ are automatically considered the ’index’ column and skipped. Data sets are stored as &lt;name&gt;:&lt;idx&gt; where &lt;name&gt; is the given data set name (the file name if not specified) and &lt;idx&gt; corresponds to the column the data was read from starting from 1. <i>Cpptraj</i> assumes the data increases monotonically and will automatically attempt to determine the dimensions of the data set(s); a warning will be printed if this is not successful.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.2">6.2</a> Grace Data File Options
</h2>
<div class="Unindented">
For more information on Grace see <a class="URL" href="http://plasma-gate.weizmann.ac.il/Grace/">http://plasma-gate.weizmann.ac.il/Grace/</a>.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--4"></a>Write
</h3>
<pre class="LyX-Code">
[invert] [xydy] [&lt;label set&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">invert</span> Normally, data is written out with X-values pertaining to frames (i.e. data over all trajectories is printed in columns). This command flips that behavior so that X-values pertain to data sets.
</div>
<div class="Description">
<span class="Description-entry">xydy</span> Combine consecutive pairs of sets into XYDY sets.
</div>
<div class="Description">
<span class="Description-entry">&lt;label set&gt;</span> If a string dataset is specified, assume it has data point labels.
</div>

</li>

</ul>
<div class="Unindented">
If a single string data set is specified when writing Grace format, it is assumed they are data point labels.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--5"></a>Read
</h3>
<div class="Unindented">
Cpptraj will read set legends from grace files, and data sets are stored as &lt;name&gt;:&lt;idx&gt; where &lt;name&gt; is the given data set name (the file name if not specified) and &lt;idx&gt; corresponds to the set number the data was read from starting from 0.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.3">6.3</a> Gnuplot Data File Options
</h2>
<div class="Unindented">
For more information on these options it helps to look at the PM3D options in the Gnuplot manual (see <a class="URL" href="http://www.gnuplot.info/">http://www.gnuplot.info/</a>).
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--6"></a>Write
</h3>
<pre class="LyX-Code">
[nolabels] [usemap] [pm3d] [nopm3d] [title &lt;title&gt;]
[jpeg] [noheader] [{xlabels|ylabels|zlabels} &lt;labellist&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">nolabels</span> Do not print axis labels.
</div>
<div class="Description">
<span class="Description-entry">usemap</span> pm3d output with 1 extra empty row/col (may improve look).
</div>
<div class="Description">
<span class="Description-entry">pm3d</span> Normal pm3d map output.
</div>
<div class="Description">
<span class="Description-entry">nopm3d</span> Turn off pm3d 
</div>
<div class="Description">
<span class="Description-entry">jpeg</span> Plot will write to a JPEG file when used with gnuplot.
</div>
<div class="Description">
<span class="Description-entry">title &lt;title&gt;</span> Set plot title (default is file name).
</div>
<div class="Description">
<span class="Description-entry">binary</span> Plot will be written in binary format.
</div>
<div class="Description">
<span class="Description-entry">header</span> Format the plot so it can be directly processed by gnuplot. This is usually the default behavior.
</div>
<div class="Description">
<span class="Description-entry">noheader</span> Do not format plot; data output only.
</div>
<div class="Description">
<span class="Description-entry">palette &lt;arg&gt;</span> Change gnuplot pm3d palette to &lt;arg&gt;:
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">’rgb’</span> Red, yellow, green, cyan, blue, magenta, red.
</div>
<div class="Description">
<span class="Description-entry">’kbvyw’</span> Black, blue, violet, yellow, white.
</div>
<div class="Description">
<span class="Description-entry">’bgyr’</span> Blue, green, yellow, red.
</div>
<div class="Description">
<span class="Description-entry">’gray’</span> Grayscale. 
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">xlabels|ylabels|zlabels &lt;labellist&gt;</span> Set x, y, or z axis labels with comma-separated list, e.g. ’xlabels X1,X2,X3’.
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.4">6.4</a> Amber REM Log Options
</h2>
<div class="Unindented">
Note that multiple REM logs can be specified in a single <b><i>readdata</i></b> command. See <a class="Reference" href="#subsec:cpptraj-remlog">12.27 on page 1↓</a> for more on replica log analysis.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--7"></a>Read
</h3>
<pre class="LyX-Code">
[nosearch] [dimfile &lt;file&gt;] [crdidx &lt;crd indices&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[nosearch]</span> If specified do not automatically search for MREMD dimension logs.
</div>
<div class="Description">
<span class="Description-entry">[dimfile &lt;file&gt;]</span> remd.dim file for processing MREMD logs.
</div>
<div class="Description">
<span class="Description-entry">[crdidx &lt;crd indices&gt;]</span> Use comma-separated list of indices as the initial coordinate indices (H-REMD only). For example (4 replicas):\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<pre class="LyX-Code">
crdidx 4,2,3,1
</pre>

</li>

</ul>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.5">6.5</a> Amber MDOUT Options
</h2>
<div class="Unindented">
Note that multiple MDOUT files can be specified in a single <b><i>readdata</i></b> command.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.6">6.6</a> Evecs File Options
</h2>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--8"></a>Read
</h3>
<pre class="LyX-Code">
[ibeg &lt;firstmode&gt;] [iend &lt;lastmode&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">ibeg &lt;firstmode&gt;</span> Number of the first mode (or principal component) to read from evecs file. Default 1.
</div>
<div class="Description">
<span class="Description-entry">iend &lt;lastmode&gt;</span> Number of the last mode (or principal component) to read from evecs file. Default is to read all for newer evecs files (generated by <i>cpptraj</i> version &gt; 12), 50 for older evecs files.
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.7">6.7</a> Vector psuedo-traj Options
</h2>
<div class="Unindented">
This can be used to write out a representation of a vector data set which can then be visualized. See <a class="Reference" href="#subsec:cpptraj-vector">11.84 on page 1↓</a> for more on generating vector data sets.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--9"></a>Write
</h3>
<pre class="LyX-Code">
[trajfmt &lt;format&gt;] [parmout &lt;file&gt;] [noorigin]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">trajfmt &lt;format&gt;</span> Output pseudo-trajectory format. See <a class="Reference" href="#tab:cpptraj-trajformats">10 on page 1↓</a> for trajectory format keywords.
</div>
<div class="Description">
<span class="Description-entry">parmout &lt;file&gt;</span> File to write pseudo-trajectory topology to.
</div>
<div class="Description">
<span class="Description-entry">[noorigin]</span> Do not write vector origin coordinates.
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.8">6.8</a> OpenDX file options
</h2>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--10"></a>Write
</h3>
<pre class="LyX-Code">
[bincenter] [gridwrap] [gridext]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">bincenter</span> Center grid points on bin centers instead of corners.
</div>
<div class="Description">
<span class="Description-entry">gridwrap</span> Like <b>’bincenter’</b>, but also wrap grid density. Useful when grid encompasses unit cell.
</div>
<div class="Description">
<span class="Description-entry">gridext</span> Like <b>’bincenter’</b>, but also print extra layer of empty bins.
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.9">6.9</a> CCP4 file options
</h2>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--11"></a>Write
</h3>
<pre class="LyX-Code">
[title &lt;title&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[title &lt;title&gt;]</span> Set CCP4 output title.
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.10">6.10</a> Charmm REPD log options
</h2>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--12"></a>Read
</h3>
<pre class="LyX-Code">
[nrep &lt;#&gt;] [crdidx &lt;crd indices&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">nrep &lt;#&gt;</span> Total number of replicas.
</div>
<div class="Description">
<span class="Description-entry">crdidx &lt;crd indices&gt;</span> Comma-separated list of indices to use as initial coordinate indices.
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-6.11">6.11</a> Amber Constant pH Out options<a class="Label" name="subsec:cpptraj_cphdatafile"> </a>
</h2>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--13"></a>Read
</h3>
<pre class="LyX-Code">
cpin &lt;file&gt;
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">cpin &lt;file&gt;</span> Constant pH input (CPIN) file name.
</div>

</li>

</ul>
<div class="Unindented">
Note that when reading in constant pH data the data set aspect will be set to the residue name and the index will be set to the residue number. When reading in constant pH REMD data the data is unsorted, and sortensembledata should be used to create sorted constant pH data sets (see <a class="Reference" href="#subsec:cpptraj_sortensembledata">8.27 on page 1↓</a>).
</div>
<h1 class="Section">
<a class="toc" name="toc-Section-7">7</a> Coordinates (COORDS) Data Set Commands<a class="Label" name="sec:Coordinates-as-a"> </a>
</h1>
<div class="Unindented">
Coordinate I/O tends to be the most time-consuming part of trajectory analysis. In addition, many types of analyses (for example two-dimensional RMSD and cluster analysis) require using coordinate frames multiple times. To simplify this, trajectory coordinates may be saved as a separate data set via the <b><i>loadcrd</i></b> command or <b><i>createcrd</i></b> action. Any action can then be performed on the COORDS data set with the <b><i>crdaction</i></b> command. The <b><i>crdout</i></b> command can be used to write coordinates to an output trajectory (similar to <b><i>trajout</i></b>).
</div>
<div class="Indented">
Although COORDS data sets store everything internally with single-precision, they can still use a large amount of memory. Because of this there is a specialized type of COORDS data set called a TRAJ data set (trajectory), which functions exactly like a COORDS data set except all data is stored on disk. TRAJ data sets can be created with the <b><i>loadtraj</i></b> command. <i>TRAJ data sets cannot be modified.</i>
</div>
<div class="Indented">
There are several analyses that can be performed using COORDS data sets, either as part of the normal analysis list or via the <b><i>runanalysis</i></b> command. Note that while these analyses can be run on specified COORDS data sets, if one is not specified a default COORDS data set will be created, made up of frames from <b><i>trajin</i></b> commands.
</div>
<div class="Indented">
As an example of where this might be useful is in the calculation of atomic positional fluctuations. Previously this required two steps: one to generate an average structure, then a second to rms-fit to that average structure prior to calculating the fluctuations. This can now be done in one pass with the following input:
</div>
<pre class="LyX-Code">
parm topology.parm7
loadcrd mdcrd.nc
# Generate average structure PDB, @CA only
crdaction mdcrd.nc average avg.pdb @CA
# Load average structure PDB as reference
parm avg.pdb
reference avg.pdb parm avg.pdb
# RMS-fit to average structure PDB
crdaction mdcrd.nc rms reference @CA
# Calculate atomic fluctuations for @CA only
crdaction mdcrd.nc atomicfluct out fluct.dat bfactor @CA
</pre>
<div class="Unindented">
The following COORDS data set commands are available:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top" style="width: 25%;">
<b>Command</b>
</td>
<td align="center" valign="top" style="width: 75%;">
<b>Description</b>
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
catcrd
</td>
<td align="center" valign="top" style="width: 75%;">
Concatenate two or more COORDS sets.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
combinecrd
</td>
<td align="center" valign="top" style="width: 75%;">
Combine two or more COORDS sets.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
crdaction
</td>
<td align="center" valign="top" style="width: 75%;">
Run a single Action on a COORDS set.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
crdout
</td>
<td align="center" valign="top" style="width: 75%;">
Write a COORDS set to a file.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
createcrd
</td>
<td align="center" valign="top" style="width: 75%;">
(Action) Create a COORDS set during a Run.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
loadcrd
</td>
<td align="center" valign="top" style="width: 75%;">
Create or append to a COORDS set from a file.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
loadtraj
</td>
<td align="center" valign="top" style="width: 75%;">
Create special COORDS set where frames remain on disk.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
permutedihedrals
</td>
<td align="center" valign="top" style="width: 75%;">
Rotate specified dihedral(s) in given COORDS set by specific interval or to random values.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
reference
</td>
<td align="center" valign="top" style="width: 75%;">
Load a single trajectory frame as a reference.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
rotatedihedral
</td>
<td align="center" valign="top" style="width: 75%;">
Rotate specified dihedral to specified value or by given increment.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
splitcoords
</td>
<td align="center" valign="top" style="width: 75%;">
Split molecules in a COORDS set into a trajectory.
</td>

</tr>

</table>

</div>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.1">7.1</a> catcrd
</h2>
<pre class="LyX-Code">
catcrd &lt;crd1&gt; &lt;crd2&gt; ... name &lt;name&gt;
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;crdX&gt;</span> COORDS data sets to concatentate, specify 2 or more.
</div>
<div class="Description">
<span class="Description-entry">name &lt;name&gt;</span> New COORDS set name
</div>

</li>

</ul>
<div class="Unindented">
Concatentate two or more COORDS data sets into a single COORDS data set. The topologies must have the same number of atoms for this to work. If the topologies differ in other ways, the topology of the first COORDS set takes priority.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.2">7.2</a> combinecrd
</h2>
<pre class="LyX-Code">
combinecrd &lt;crd1&gt; &lt;crd2&gt; ... [parmname &lt;topname&gt;] [crdname &lt;crdname&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;crdX&gt;</span> COORDS data sets to combine, specify 2 or more.
</div>
<div class="Description">
<span class="Description-entry">[parmname &lt;topname&gt;]</span> Name of combined Topology.
</div>
<div class="Description">
<span class="Description-entry">[crdname &lt;crdname&gt;]</span> Name of combined COORDS data set.
</div>

</li>

</ul>
<div class="Unindented">
Combined two or more COORDS data sets into a single COORDS data set. Note that the resulting topology will most likely <b>not</b> be usable for MD simulations. Box information will be retained - the largest box dimensions will be used.
</div>
<div class="Indented">
For example, to load two MOL2 files as COORDS data sets, combine them, and write them out as a single MOL2:
</div>
<pre class="LyX-Code">
loadcrd Tyr.mol2 CRD1
loadcrd Pry.mol2 CRD2
combinedcrd CRD1 CRD2 parmname Parm-1-2 crdname CRD-1-2
crdout CRD-1-2 Tyr.Pry.mol2
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.3">7.3</a> crdaction<a class="Label" name="subsec:cpptraj_crdaction"> </a>
</h2>
<pre class="LyX-Code">
crdaction &lt;crd set&gt; &lt;actioncmd&gt; [&lt;action args&gt;] [crdframes &lt;start&gt;,&lt;stop&gt;,&lt;offset&gt;]
</pre>
<div class="Unindented">
Perform action &lt;actioncmd&gt; on COORDS data set &lt;crd set&gt;. A subset of frames in the COORDS data set can be specified with ’crdframes’.
</div>
<div class="Indented">
For example, to calculate RMSD for a previously created COORDS data set named crd1 using frames 1 to the last, skipping every 10:
</div>
<pre class="LyX-Code">
crdaction crd1 rmsd first @CA out rmsd-ca.agr crdframes 1,last,10
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.4">7.4</a> createcrd
</h2>
<div class="Unindented">
This command is actually an Action that can be used to create COORDS data sets during trajectory processing, see <a class="Reference" href="#subsec:cpptraj_createcrd">11.19 on page 1↓</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.5">7.5</a> crdout
</h2>
<pre class="LyX-Code">
crdout &lt;crd set&gt; &lt;filename&gt; [&lt;trajout args&gt;] [crdframes &lt;start&gt;,&lt;stop&gt;,&lt;offset&gt;]
</pre>
<div class="Unindented">
Write COORDS data set &lt;crd set&gt; to trajectory named &lt;filename&gt;. A subset of frames in the COORDS data set can be specified with ’crdframes’.
</div>
<div class="Indented">
For example, to write frames 1 to 10 from a previously created COORDS data set named &ldquo;crd1&rdquo; to separate PDB files:
</div>
<pre class="LyX-Code">
crdout crd1 crd1.pdb multi crdframes 1,10
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.6">7.6</a> loadcrd
</h2>
<pre class="LyX-Code">
loadcrd &lt;filename&gt; [parm &lt;parm&gt; | parmindex&lt;#&gt;] [&lt;trajin args&gt;] [name &lt;name&gt;]
</pre>
<div class="Unindented">
Immediately load trajectory &lt;filename&gt; as a COORDS data set named &lt;name&gt; (default base name of &lt;filename&gt;). If &lt;name&gt; is already present the coordinates will be appended to the existing data set.
</div>
<div class="Indented">
For example, to load frames from trajectories named ’traj1.nc’ and ’traj2.nc’ into a COORDS data set named Crd1:
</div>
<pre class="LyX-Code">
loadcrd traj1.nc name Crd1
loadcrd traj2.nc name Crd2
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.7">7.7</a> loadtraj
</h2>
<pre class="LyX-Code">
loadtraj name &lt;setname&gt; [&lt;filename&gt;]
</pre>
<div class="Unindented">
This command functions in two ways. If &lt;filename&gt; is not provided, all currently loaded input trajectories (from <b><i>trajin</i></b> commands) are added to TRAJ data set named &lt;setname&gt;. <b>Note that if the input trajectory list is cleared (via ’clear trajin’) this will invalidate the TRAJ data set.</b> In addition, currently all trajectories must have the same number of atoms. Otherwise add trajectory &lt;filename&gt; to TRAJ data set &lt;setname&gt;.
</div>
<div class="Indented">
TRAJ data sets cannot be modified.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.8">7.8</a> permutedihedrals<a class="Label" name="subsec:cpptraj_permutedihedrals"> </a>
</h2>
<pre class="LyX-Code">
permutedihedrals crdset &lt;COORDS set&gt; resrange &lt;range&gt; [{interval | random}]
             	[outtraj &lt;filename&gt; [&lt;outfmt&gt;]] [crdout &lt;output COORDS&gt;] 
                 [&lt;dihedral types&gt;]
  Options for ’random’:
	[rseed &lt;rseed&gt;] [out &lt;# problems file&gt; [&lt;set name&gt;]]
	[ check [cutoff &lt;cutoff&gt;] [rescutoff &lt;rescutoff&gt;] [checkallresidues]
	  [backtrack &lt;backtrack&gt;] [increment &lt;increment&gt;] [maxfactor &lt;max_factor&gt;] ]
  Options for ’interval’:
	&lt;interval deg&gt;
  &lt;dihedral types&gt; =  alpha beta gamma delta epsilon zeta nu1 nu2 h1p c2p chin 
                      phi psi chip omega     \begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">crdset &lt;COORDS set&gt;</span> COORDS data set to operate on.
</div>
<div class="Description">
<span class="Description-entry">resrange &lt;range&gt;</span> Residue range to search for dihedrals.
</div>
<div class="Description">
<span class="Description-entry">interval</span> Rotate found dihedrals by &lt;interval&gt;. This is done in an ordered fashion so that every combination of dihedral rotations is sampled at least once.
</div>
<div class="Description">
<span class="Description-entry">random</span> Rotate each found dihedral randomly.
</div>
<div class="Description">
<span class="Description-entry">[outtraj &lt;filename&gt;]</span> Trajectory file to write coordinates to.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;outfmt&gt;]</span> Trajectory file format.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[crdout &lt;output COORDS&gt;]</span> COORDS data set to write coordinates to.
</div>
<div class="Description">
<span class="Description-entry">&lt;dihedral type&gt;</span> One or more dihedral types to search for.
</div>
<div class="Unindented">
Options for ’interval:
</div>
<div class="Description">
<span class="Description-entry">&lt;interval deg&gt;</span> Amount to rotate dihedral by each step.
</div>
<div class="Unindented">
Options for ’random’:
</div>
<div class="Description">
<span class="Description-entry">[rseed &lt;rseed&gt;]</span> Random number seed.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;# problems file&gt;]</span> File to write number of problems (clashes) each frame to.
</div>
<div class="Description">
<span class="Description-entry">[&lt;set name&gt;]</span> Number of problems data set name.
</div>
<div class="Description">
<span class="Description-entry">[check]</span> Check randomly rotated structure for clashes.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[cutoff &lt;cutoff&gt;]</span> Atom cutoff for checking for clashes (default 0.8 Å).
</div>
<div class="Description">
<span class="Description-entry">[rescutoff &lt;cutoff&gt;]</span> Residue cutoff for checking for clashes (defualt 10.0 Å).
</div>
<div class="Description">
<span class="Description-entry">[checkallresidues]</span> If specified all residues checked for clashes, otherwise only residues up to the currently rotated dihedral check.
</div>
<div class="Description">
<span class="Description-entry">[backtrack &lt;backtrack&gt;]</span> If a clash is encountered at dihedral N and cannot be resolved, go to dihedral N-&lt;backtrack&gt; to try and resolve the clash (default 4).
</div>
<div class="Description">
<span class="Description-entry">[increment &lt;increment&gt;]</span> If a clash is encountered, first attempt to rotate dihedral by increment to resolve it; if it cannot be resolved by a full rotation the calculation will backtrack (default 1). 
</div>
<div class="Description">
<span class="Description-entry">[maxfactor &lt;max_factor&gt;]</span> The maximum number of total attempted rotations will be &lt;max_factor&gt; * &lt;total # of dihedrals&gt; (default 2).
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Create a trajectory by rotating specified dihedrals in a structure by regular intervals (<b>interval</b>), or create 1 structure by randomly rotating specified dihedrals (<b>random</b>). When randomly rotating dihedrals steric clashes will be checked if <b>check</b> is specified; in such cases the algorithm will attempt to resolve the clash as best it can. If clashes are not being resolved you can increase the number of rotation attempts <tt><i>cpptraj</i></tt> will make by increasing <b>maxfactor</b>.
</div>
<div class="Indented">
For example, to rotate all backbone dihedrals in a protein with coordinates in a file named tz2.rst7 in -120 degree intervals and write the resulting trajectory in Amber format to rotations.mdcrd:
</div>
<pre class="LyX-Code">
reference tz2.rst7 [TZ2]
permutedihedrals crdset [TZ2] interval -120 outtraj rotations.mdcrd phi psi
</pre>
<div class="Unindented">
To randomly rotate backbone dihedrals for the same structure and write to file random.mol2 in MOL2 format:
</div>
<pre class="LyX-Code">
reference tz2.rst7 [TZ2]
permutedihedrals crdset [TZ2] random rseed 1 check maxfactor 10 phi psi \
                 outtraj random.mol2 multi
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.9">7.9</a> reference
</h2>
<div class="Unindented">
Reference coordinates can now be used and manipulated like COORDS data sets. See <a class="Reference" href="#subsec:cpptraj_reference">10.3 on page 1↓</a> for command syntax.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.10">7.10</a> rotatedihedral
</h2>
<pre class="LyX-Code">
rotatedihedral crdset &lt;COORDS set&gt; [frame &lt;#&gt;] [name &lt;output set name&gt;]
               {value &lt;value&gt; | increment &lt;increment&gt;}
               { &lt;mask1&gt; &lt;mask2&gt; &lt;mask3&gt; &lt;mask4&gt; |
                 res &lt;#&gt; type &lt;dih type&gt; }
&lt;dih type&gt; = alpha beta gamma delta epsilon zeta nu1 nu2 h1p c2p chin
             phi psi chip omega
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">crdset &lt;COORDS set&gt;</span> Coordinates data set to work on. If a TRAJ data set is specified, <b>name</b> must also be specified.
</div>
<div class="Description">
<span class="Description-entry">[frame &lt;#&gt;]</span> Frame of the COORDS set to work on.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;output set name&gt;]</span> Output COORDS set. If not specified the input COORDS set will be modified.
</div>
<div class="Description">
<span class="Description-entry">value &lt;value&gt;</span> Set specified dihedral to given value in degrees.
</div>
<div class="Description">
<span class="Description-entry">increment &lt;increment&gt;</span> Increment specified dihedral by increment in degrees.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask1&gt; &lt;mask2&gt; &lt;mask3&gt; &lt;mask4&gt;</span> Define dihedral by atom masks. Each mask should only select one atom.
</div>
<div class="Description">
<span class="Description-entry">res &lt;#&gt;</span> Rotate dihedral specified by type in residue number &lt;#&gt;.
</div>
<div class="Description">
<span class="Description-entry">type &lt;dih type&gt;</span> Dihedral type to rotate in specified residue.
</div>

</li>

</ul>
<div class="Unindented">
Rotate the specified dihedral in given COORDS set to a target value or by given increment. For example, to set the protein chi dihedral in residue 8 to 35 degrees and write out to a mol2 file:
</div>
<pre class="LyX-Code">
parm ../tz2.parm7
loadcrd ../tz2.nc 1 1 name TZ2
rotatedihedral crdset TZ2 value 35 res 8 type chip
crdout TZ2 tz2.rotate.1.mol2
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-7.11">7.11</a> splitcoords
</h2>
<pre class="LyX-Code">
splitcoords &lt;crd set&gt; name &lt;output set name&gt;
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;crd set&gt;</span> COORDS set to split.
</div>
<div class="Description">
<span class="Description-entry">name &lt;output set name&gt;</span> Name of new set to create.
</div>

</li>

</ul>
<div class="Unindented">
Split trajectory specified by &lt;crd set&gt; by molecule into a new COORDS set. All molecules in &lt;crd set&gt; must be the same size. For example, if there are 10 molecules and 10 frames in COORDS set &ldquo;Set0&rdquo;, the following would create a new COORDS set with 100 frames (original molecules 1-10 frame 1, original molecules 1-10 frame 2, etc):
</div>
<pre class="LyX-Code">
splitcoords Set0 name Set0Split
</pre>
<h1 class="Section">
<a class="toc" name="toc-Section-8">8</a> General Commands
</h1>
<div class="Unindented">
The following general commands are available:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
<b>Command</b>
</td>
<td align="center" valign="top">
<b>Description</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
activeref
</td>
<td align="center" valign="top">
Select the reference for distance-based masks.
</td>

</tr>
<tr>
<td align="center" valign="top">
calc
</td>
<td align="center" valign="top">
Evaluate the given mathematical expression.
</td>

</tr>
<tr>
<td align="center" valign="top">
clear
</td>
<td align="center" valign="top">
Clear various objects from the cpptraj state.
</td>

</tr>
<tr>
<td align="center" valign="top">
create
</td>
<td align="center" valign="top">
Create (but do not yet write) a data file.
</td>

</tr>
<tr>
<td align="center" valign="top">
createset
</td>
<td align="center" valign="top">
Create a dataset from a simple mathematical expression.
</td>

</tr>
<tr>
<td align="center" valign="top">
datafile
</td>
<td align="center" valign="top">
Used to manipulate data files.
</td>

</tr>
<tr>
<td align="center" valign="top">
datafilter
</td>
<td align="center" valign="top">
Filter data sets based on given criteria.
</td>

</tr>
<tr>
<td align="center" valign="top">
dataset
</td>
<td align="center" valign="top">
Use to manipulate data sets.
</td>

</tr>
<tr>
<td align="center" valign="top">
debug | prnlev
</td>
<td align="center" valign="top">
Set debug level. Higher levels give more info.
</td>

</tr>
<tr>
<td align="center" valign="top">
ensextension
</td>
<td align="center" valign="top">
Enable/disable ensemble number extension for files in ensemble mode.
</td>

</tr>
<tr>
<td align="center" valign="top">
exit | quit
</td>
<td align="center" valign="top">
Quit cpptraj.
</td>

</tr>
<tr>
<td align="center" valign="top">
go | run
</td>
<td align="center" valign="top">
Start a trajectory processing Run.
</td>

</tr>
<tr>
<td align="center" valign="top">
help
</td>
<td align="center" valign="top">
Provide help for commands.
</td>

</tr>
<tr>
<td align="center" valign="top">
list
</td>
<td align="center" valign="top">
List various objects in the cpptraj state.
</td>

</tr>
<tr>
<td align="center" valign="top">
noexitonerror
</td>
<td align="center" valign="top">
Attempt to continue even if errors are encountered.
</td>

</tr>
<tr>
<td align="center" valign="top">
noprogress
</td>
<td align="center" valign="top">
Do not print a progress bar during a Run.
</td>

</tr>
<tr>
<td align="center" valign="top">
parallelanalysis
</td>
<td align="center" valign="top">
(MPI only) Divide current Analyses among MPI processes.
</td>

</tr>
<tr>
<td align="center" valign="top">
precision
</td>
<td align="center" valign="top">
Change the output precision of data sets.
</td>

</tr>
<tr>
<td align="center" valign="top">
printdata
</td>
<td align="center" valign="top">
Print data set to screen.
</td>

</tr>
<tr>
<td align="center" valign="top">
readdata
</td>
<td align="center" valign="top">
Read data sets from files.
</td>

</tr>
<tr>
<td align="center" valign="top">
readensembledata
</td>
<td align="center" valign="top">
Read data files in ensemble mode.
</td>

</tr>
<tr>
<td align="center" valign="top">
readinput
</td>
<td align="center" valign="top">
Read cpptraj input from a file.
</td>

</tr>
<tr>
<td align="center" valign="top">
removedata
</td>
<td align="center" valign="top">
Remove specified data set(s).
</td>

</tr>
<tr>
<td align="center" valign="top">
rst
</td>
<td align="center" valign="top">
Generate Amber-style distance/angle/torsion restraints.
</td>

</tr>
<tr>
<td align="center" valign="top">
runanalysis
</td>
<td align="center" valign="top">
Run an analysis immediately or run all queued analyses.
</td>

</tr>
<tr>
<td align="center" valign="top">
select
</td>
<td align="center" valign="top">
Print the results of an atom mask expression.
</td>

</tr>
<tr>
<td align="center" valign="top">
selectds
</td>
<td align="center" valign="top">
Print the results of a data set selection expression.
</td>

</tr>
<tr>
<td align="center" valign="top">
silenceactions
</td>
<td align="center" valign="top">
Prevent Actions from writing information to STDOUT.
</td>

</tr>
<tr>
<td align="center" valign="top">
sortensembledata
</td>
<td align="center" valign="top">
Sort data sets using replica information (currently constant pH only).
</td>

</tr>
<tr>
<td align="center" valign="top">
usediskcache
</td>
<td align="center" valign="top">
Turn caching of data sets to disk on or off.
</td>

</tr>
<tr>
<td align="center" valign="top">
write | writedata
</td>
<td align="center" valign="top">
Immediately write data to a file or write to all current data files.
</td>

</tr>

</table>

</div>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.1">8.1</a> activeref
</h2>
<pre class="LyX-Code">
activeref<a class="IndexReference" name="entry-activeref-0" href="#index-activeref">↓</a> &lt;#&gt;
</pre>
<div class="Unindented">
Set which reference structure should be used when setting up distance-based masks for everything but the ’mask’ action. Numbering starts from 0, so ’activeref 0’ selects the first reference structure read in, ’activeref 1’ selects the second, and so on.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.2">8.2</a> calc
</h2>
<pre class="LyX-Code">
calc &lt;expression&gt;
	[prec &lt;width&gt;.&lt;precision&gt;] [format {double|general|scientific}]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;expression&gt;</span> Mathematical expression to evaluate. See <a class="Reference" href="#subsec:cpptraj-Data-Set-Math">5.2 on page 1↑</a> for details.
</div>
<div class="Description">
<span class="Description-entry">prec &lt;width&gt;.&lt;precision&gt;</span> Set the width and precision of the result.
</div>
<div class="Description">
<span class="Description-entry">format {double|general|scientific}</span> Set the format of the result.
</div>

</li>

</ul>
<div class="Unindented">
Evaluate the given mathematical expression. This version gives more control over the format of the output.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.3">8.3</a> clear
</h2>
<pre class="LyX-Code">
clear [{all | &lt;type&gt;}]
      (&lt;type&gt; = actions,trajin,trajout,ref,parm,analysis,datafile,dataset)
</pre>
<div class="Unindented">
Clear list of indicated type, or all lists if ’all’ specified. Note that when clearing actions or analyses, associated data sets and data files are not cleared and vice versa.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.4">8.4</a> create<a class="Label" name="subsec:cpptraj_create"> </a>
</h2>
<pre class="LyX-Code">
create<a class="IndexReference" name="entry-create-0" href="#index-create">↓</a> &lt;filename&gt; &lt;datasetname0&gt; [&lt;datasetname1&gt; ...] [&lt;DataFile Options&gt;]
</pre>
<div class="Unindented">
Add specified data sets to the data file named &lt;filename&gt;; if the file does not exist, it will be added to the DataFileList. Data files created in this way are only written at the end of coordinate processing, analyses, or via the <b><i>’writedata’</i></b> command. See <a class="Reference" href="#sec:Datafile-Commands">6 on page 1↑</a> for more data file format options.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.5">8.5</a> createset
</h2>
<pre class="LyX-Code">
createset &lt;expression&gt; [xmin &lt;min&gt;] xstep &lt;step&gt; nx &lt;nxvals&gt;
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">expression</span> Simple mathematical expression, must contain equals sign, can contain X (e.g. Y=2*X). If not enclosed in quotes must not contain whitespace.
</div>
<div class="Description">
<span class="Description-entry">xmin &lt;min&gt;</span> Minimum X value.
</div>
<div class="Description">
<span class="Description-entry">xstep &lt;step&gt;</span> X step.
</div>
<div class="Description">
<span class="Description-entry">nx &lt;nxvals&gt;</span> Number of X values.
</div>

</li>

</ul>
<div class="Unindented">
Generate a data set from a simple mathematical expression.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.6">8.6</a> datafile<a class="Label" name="subsec:cpptraj_datafile_command"> </a>
</h2>
<pre class="LyX-Code">
datafile<a class="IndexReference" name="entry-datafile-0" href="#index-datafile">↓</a> &lt;filename&gt; &lt;datafile arg&gt;
</pre>
<div class="Unindented">
Pass &lt;datafile arg&gt; to data file &lt;filename&gt;. See <a class="Reference" href="#sec:Datafile-Commands">6 on page 1↑</a> for more details.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.7">8.7</a> datafilter<a class="Label" name="subsec:cpptraj-datafilter"> </a>
</h2>
<pre class="LyX-Code">
datafilter {&lt;dataset arg&gt; min &lt;min&gt; max &lt;max&gt; ...} [out &lt;file&gt;] [name &lt;setname&gt;]\begin_inset Separator latexpar\end_inset
           [{multi | filterset &lt;set&gt; [newset &lt;newname&gt;]}]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dataset arg&gt; min &lt;min&gt; max &lt;max&gt;</span> Data set name and min/max cutoffs to use; can specify more than one.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> Write out to file named &lt;file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;setname&gt;]</span> Name of filter data set containing 1 when cutoffs satisfied, 0 otherwise.
</div>
<div class="Description">
<span class="Description-entry">[multi]</span> Filter each set separately instead of all together (creates filter set for each input set). Cannot be used with ’filterset’.
</div>
<div class="Description">
<span class="Description-entry">[filterset &lt;set&gt;]</span> If specified, &lt;set&gt; will be filtered to only contain data that satisfies cutoffs. Cannot be used with ’multi’.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[newset &lt;newname&gt;]</span> If specified a new set will be created from ’filterset’ instead of replacing ’filterset’.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Create a data set (optionally named &lt;setname&gt;) containing 1 for data within given <b>&lt;min&gt;</b> and <b>&lt;max&gt;</b> criteria for each specified data set. There must be at least one <b>&lt;min&gt;</b> and <b>&lt;max&gt;</b> argument, and can be as many as there are specified data sets. If <b>’multi’</b> is specified then only filter data sets will be created for each data set instead. If <b>’filterset’</b> is specified, the specified <b>&lt;set&gt;</b> will be modified to only contain ’1’ frames; cannot be used with <b>’multi’</b>. If <b>’newset’</b> is also specified, a new set will be created containing the ’1’ frames instead. The ’filterset’ functionality only works for 1D scalar sets.
</div>
<div class="Indented">
For example, to read in data from two separate files (d1.dat and a1.dat) and generate a filter data set named FILTER having 1 when d1 is between 0.0 and 3.0 and a1 is between 135.0 and 180.0:
</div>
<pre class="LyX-Code">
readdata a1.dat name a1
readdata d1.dat name d1
datafilter d1 min 0.0 max 3.0 a1 min 135.0 max 180.0 out filter.dat name FILTER
</pre>
<div class="Unindented">
Note that a similar command that can be used with data generated by Actions during trajectory processing is <b><i>filter</i></b> (see page <a class="Reference" href="#subsec:cpptraj-filter">1↓</a>).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.8">8.8</a> dataset
</h2>
<pre class="LyX-Code">
dataset { legend &lt;legend&gt; &lt;set&gt; |
          makexy &lt;Xset&gt; &lt;Yset&gt; [name &lt;name&gt;] |
          vectorcoord {X|Y|Z} &lt;set&gt; [name &lt;name&gt;] |
    	  cat &lt;set0&gt; &lt;set1&gt; ... [name &lt;name&gt;] [nooffset] |
          make2d &lt;1D set&gt; cols &lt;ncols&gt; rows &lt;nrows&gt; [name &lt;name&gt;] |
          {drop|keep}points {range &lt;range arg&gt; | [start &lt;#&gt;] [stop &lt;#&gt;] [offset &lt;#&gt;]}
                   [name &lt;output set&gt;] &lt;set arg1&gt; ... |
          remove &lt;criterion&gt; &lt;select&gt; &lt;value&gt; [and &lt;value2&gt;] [&lt;set selection&gt;] |
          dim {xdim|ydim|zdim|ndim &lt;#&gt;} [label &lt;label&gt;] [min &lt;min&gt;] [step &lt;step&gt;] |
          outformat {double|scientific|general} &lt;set arg1&gt; [&lt;set arg 2&gt; ...] |
          invert &lt;set arg0&gt; ... name &lt;new name&gt; [legendset &lt;set&gt;] |
          shift [above &lt;value&gt; by &lt;offset&gt;] [below &lt;value&gt; by &lt;offset&gt;] &lt;set arg0&gt; ...
    	  [mode &lt;mode&gt;] [type &lt;type&gt;] &lt;set arg1&gt; [&lt;set arg 2&gt; ...]
        }
  &lt;mode&gt;:  ’distance’ ’angle’ ’torsion’ ’pucker’ ’rms’ ’matrix’ ’vector’
  &lt;type&gt;:  ’alpha’ ’beta’ ’gamma’ ’delta’ ’epsilon’ ’zeta’ ’nu0’ ’nu1’ ’nu2’ ’nu3’
           ’nu4’ ’h1p’ ’c2p’ ’chin’ ’phi’ ’psi’ ’chip’ ’omega’ ’chi2’ ’chi3’ ’chi4’
           ’chi5’ ’pucker’ ’noe’ ’distance’ ’covariance’ ’mass-weighted covariance’
           ’correlation’ ’distance covariance’ ’IDEA’ ’IRED’ ’dihedral covariance’
	    Options for ’type noe’:
    	  [bound &lt;lower&gt; bound &lt;upper&gt;] [rexp &lt;expected&gt;] [noe_strong] [noe_medium]
          [noe_weak]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[name &lt;name&gt;]</span> New data set name for <b>makexy</b>/<b>vectorcoord/cat</b>/<b>make2d</b>/<b>droppoints</b>/<b>keeppoints</b>.
</div>
<div class="Description">
<span class="Description-entry">legend &lt;legend&gt; &lt;set&gt;</span> Set the legend for data set &lt;set&gt; to &lt;legend&gt;.
</div>
<div class="Description">
<span class="Description-entry">makexy &lt;Xset&gt; &lt;Yset&gt;</span> Create a new data set (optionally named &lt;name&gt;) with X values from &lt;Xset&gt; and Y values from &lt;Yset&gt;.
</div>
<div class="Description">
<span class="Description-entry">vectorcoord {X|Y|Z} &lt;set&gt;</span> Extract X/Y/Z coordinates from vector data set into a new 1D data set.
</div>
<div class="Description">
<span class="Description-entry">cat &lt;set0&gt; &lt;set1&gt; ...</span> Concatenate two or more data sets into a new data set (optionally named &lt;name&gt;).
</div>
<div class="Description">
<span class="Description-entry">make2d &lt;1D set&gt; cols &lt;ncols&gt; rows &lt;nrows&gt;</span> Convert 1D data set into row-major 2D data set with specified number of rows and columns.
</div>
<div class="Description">
<span class="Description-entry">{drop|keep}points &lt;set arg1&gt; ...</span> Drop or keep specified points from data set(s), optionally creating a new data set.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">range &lt;range arg&gt;</span> Range of points to drop/keep.
</div>
<div class="Description">
<span class="Description-entry">[start &lt;#&gt;] [stop &lt;#&gt;] [offset &lt;#&gt;]</span> Start/stop/offset values of points to drop/keep.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">remove &lt;criterion&gt; &lt;select&gt; &lt;value&gt; [and &lt;value2&gt;] [&lt;set selection&gt;]</span> Remove data sets from &lt;set selection&gt; according to specified criterion and selection.
</div>
<pre class="LyX-Code">
&lt;criterion&gt;:  ’ifaverage’ ’ifsize’ ’ifmode’ ’iftype’
&lt;select&gt;   :  ’equal’ ’==’ ’notequal’ ’!=’ ’lessthan’ ’&lt;’ 
              ’greaterthan’ ’&gt;’ ’between’ ’outside’
</pre>
<div class="Description">
<span class="Description-entry">dim {xdim|ydim|zdim|ndim &lt;#&gt;}</span> Change specified dimension in set(s).
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">label &lt;label&gt;</span> Change dimension label to &lt;label&gt;
</div>
<div class="Description">
<span class="Description-entry">min &lt;min&gt;</span> Change dimension minimum to &lt;min&gt;.
</div>
<div class="Description">
<span class="Description-entry">step &lt;step&gt;</span> Change dimension step to &lt;step&gt;.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">invert &lt;set arg0&gt; ... name &lt;new name&gt; [legendset &lt;set&gt;]</span> 
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;set arg0&gt; ...</span> Specify sets to invert.
</div>
<div class="Description">
<span class="Description-entry">name &lt;new name&gt;</span> Inverted output set name.
</div>
<div class="Description">
<span class="Description-entry">[legendset &lt;set&gt;]</span> String data set containing legends
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">shift </span> 
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[above &lt;value&gt; by &lt;offset&gt;]</span> Values in set(s) above &lt;value&gt; will be shifted by &lt;offset&gt;.
</div>
<div class="Description">
<span class="Description-entry">[below &lt;value&gt; by &lt;offset&gt;]</span> Values in set(s) below &lt;value&gt; will be shifted by &lt;offset&gt;.
</div>
<div class="Description">
<span class="Description-entry">&lt;set arg0&gt; ...</span> Set(s) to shift.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[mode &lt;mode&gt;]</span> Set data set(s) mode to &lt;mode&gt;.
</div>
<div class="Description">
<span class="Description-entry">[type &lt;type&gt;]</span> Set data set(s) type to ’type’, useful for e.g. analysis with <b><i>statistics</i></b>. Note this can also be done with ’type &lt;type&gt;’ for certain commands (<b><i>distance</i></b>, <b><i>dihedral</i></b>, <b><i>pucker</i></b> etc). Note that not every &lt;type&gt; is compatible with a given &lt;mode&gt;.
</div>
<div class="Unindented">
Options for type noe only:
</div>
<div class="Description">
<span class="Description-entry">[bound &lt;lower&gt; bound &lt;upper&gt;]</span> Lower and upper bounds for NOE (in Angstroms); must specify both.
</div>
<div class="Description">
<span class="Description-entry">[rexp &lt;expected&gt;]</span> Expected value for NOE (in Angstroms); if not given ’(&lt;lower&gt; + &lt;upper&gt;)’ / 2.0 is used.
</div>
<div class="Description">
<span class="Description-entry">[noe_strong]</span> Set lower and upper bounds to 1.8 and 2.9 Å respectively.
</div>
<div class="Description">
<span class="Description-entry">[noe_medium]</span> Set lower and upper bounds to 2.9 and 3.5 Å respectively.
</div>
<div class="Description">
<span class="Description-entry">[noe_weak]</span> Set lower and upper bounds to 3.5 and 5.0 Å respectively.
</div>

</li>

</ul>
<div class="Unindented">
Either set the legend for a single data set, create a new set with X values from one set and Y values from another, concatenate 2 or more sets, make a 2D set from 1D set, remove sets according to a certain criterion, or change the mode/type for one or more data sets.
</div>
<div class="Indented">
Setting the mode/type can be useful for cases where the data set is being read in from a file; for example when reading in a dihedral data set the type can be set to ’dihedral’ so that various Analysis routines like <b><i>statistics</i></b> know to treat it as periodic. A brief description of possible modes and types follows:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
<b>Mode</b>
</td>
<td align="center" valign="top">
<b>Type</b>
</td>
<td align="center" valign="top">
<b>Description</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
distance
</td>
<td align="center" valign="top">
noe
</td>
<td align="center" valign="top">
NOE distance.
</td>

</tr>
<tr>
<td align="center" valign="top">
angle
</td>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
Angle.
</td>

</tr>
<tr>
<td align="center" valign="top">
torsion
</td>
<td align="center" valign="top">
alpha
</td>
<td align="center" valign="top">
Nucleic acid alpha.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
beta
</td>
<td align="center" valign="top">
Nucleic acid beta.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
gamma
</td>
<td align="center" valign="top">
Nucleic acid gamma.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
delta
</td>
<td align="center" valign="top">
Nucleic acid delta.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
epsilon
</td>
<td align="center" valign="top">
Nucleic acid epsilon.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
zeta
</td>
<td align="center" valign="top">
Nucleic acid zeta.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
nu1
</td>
<td align="center" valign="top">
Nucleic pucker (O4’).
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
nu2
</td>
<td align="center" valign="top">
Nucleic pucker (C4’).
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
h1p
</td>
<td align="center" valign="top">
Nucleic acid H1’.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
c2p
</td>
<td align="center" valign="top">
Nucleic acid C2’.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
chin
</td>
<td align="center" valign="top">
Nucleic acid chi.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
phi
</td>
<td align="center" valign="top">
Protein Phi.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
psi
</td>
<td align="center" valign="top">
Protein psi.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
chip
</td>
<td align="center" valign="top">
Protein chi.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
omega
</td>
<td align="center" valign="top">
Protein omega.
</td>

</tr>
<tr>
<td align="center" valign="top">
pucker
</td>
<td align="center" valign="top">
pucker
</td>
<td align="center" valign="top">
Sugar pucker.
</td>

</tr>
<tr>
<td align="center" valign="top">
rms
</td>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
RMSD.
</td>

</tr>
<tr>
<td align="center" valign="top">
matrix
</td>
<td align="center" valign="top">
distance
</td>
<td align="center" valign="top">
Distance matrix.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
covariance
</td>
<td align="center" valign="top">
Cartesian covariance matrix.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
’mass-weighted covariance’
</td>
<td align="center" valign="top">
Mass weighted Cartesian covariance matrix.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
correlation
</td>
<td align="center" valign="top">
Dynamic cross correlation matrix.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
’distance covariance’
</td>
<td align="center" valign="top">
Distance covariance matrix.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
IDEA
</td>
<td align="center" valign="top">
IDEA matrix.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
IRED
</td>
<td align="center" valign="top">
IRED matrix.
</td>

</tr>
<tr>
<td align="center" valign="top">

</td>
<td align="center" valign="top">
’dihedral covariance’
</td>
<td align="center" valign="top">
Dihedral covariance matrix.
</td>

</tr>
<tr>
<td align="center" valign="top">
vector
</td>
<td align="center" valign="top">
IRED
</td>
<td align="center" valign="top">
IRED vector.
</td>

</tr>

</table>

</div>

</div>
<div class="Indented">
The invert mode takes a group of M 1D data sets of size N and create N new "inverted" data sets of size M. This is similar to the invert keyword already available for standard and Grace data writes, but operates directly on data sets. For example, given the following two data sets:
</div>
<pre class="LyX-Code">
D0 D1
 1  4
 2  5
 3  6
</pre>
<div class="Unindented">
The new data sets will be laid out like so:
</div>
<pre class="LyX-Code">
N0 N1 N2
 1  2  3
 4  5  6
</pre>
<div class="Unindented">
The dataset invert command can be useful if you want to easily view output from multiple analysis commands in a single graph. For example, to view state counts from two different simulations side by side:
</div>
<pre class="LyX-Code">
calcstate name Sim1 state bound1,dist1,0.0,2.0
calcstate name Sim2 state bound1,dist1,0.0,2.0
runanalysis dataset invert Sim*[Count] name Inverted legendset Sim1[Name]
dataset dim xdim label Simulation min 1 step 1 Inverted*
writedata statecount.agr Inverted* 
</pre>
<div class="Unindented">
The dataset shift command can be used for wrapping circular values, such as torsions. For example, to ensure a pucker has a range from 0 to 360 instead of -180 to 180:
</div>
<pre class="LyX-Code">
pucker Furanoid @C2 @C3 @C4 @C5 @O2 cremer out CremerF.dat amplitude
run
dataset shift Furanoid below 0 by 360
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.9">8.9</a> debug | prnlev
</h2>
<pre class="LyX-Code">
debug<a class="IndexReference" name="entry-debug-0" href="#index-debug">↓</a> [&lt;type&gt;] &lt;#&gt;
      (&lt;type&gt; = actions,trajin,trajout,ref,parm,analysis,datafile,dataset)
</pre>
<div class="Unindented">
Set the level of debug information to print. In general the higher the &lt;#&gt; the more information that is printed. If &lt;type&gt; is specified only set the debug level for a specific area of <i>cpptraj.</i>
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.10">8.10</a> ensextension
</h2>
<pre class="LyX-Code">
ensextension {on|off}
</pre>
<div class="Unindented">
Turn printing of ensemble member number filename extensions on or off. By default ensemble extensions are printed in parallel and not in serial.
</div>
<div class="Indented">
<b>NOTE: THE ’ensextension off’ OPTION HAS NOT BEEN FULLY TESTED IN PARALLEL AND IS NOT CURRENTLY RECOMMENDED.</b>
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.11">8.11</a> exit | quit
</h2>
<div class="Unindented">
Exit normally.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.12">8.12</a> go | run
</h2>
<div class="Unindented">
Begin trajectory processing, followed by analysis and datafile write.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.13">8.13</a> help
</h2>
<pre class="LyX-Code">
help<a class="IndexReference" name="entry-help-0" href="#index-help">↓</a> [ { All |
	     &lt;cmd&gt; |
 	    &lt;command category&gt; |
 	    Form[ats] [{read|write}] |
 	    Form[ats] [{trajin|trajout|readdata|writedata|parm|parmwrite} [&lt;fmt key&gt;]] |
 	    Mask 	   } ]
	Command Categories: Gen[eral] Sys[tem] Coor[ds] Traj[ectory] Top[ology]
	                    Act[ion] Ana[lysis] Con[trol]
  All                : Print all known commands.
  &lt;cmd&gt;              : Print help for command &lt;cmd&gt;.
  &lt;command category&gt; : Print all commands in specified category.
  Form[ats]          : Help for file formats.
  Mask               : Help for mask syntax.
</pre>
<div class="Unindented">
If ’All’ is specifed, list all commands known to <i>cpptraj</i>. If given with a command, print help for that command. Otherwise, list all commands of a certain category (General, System, Coords, Trajectory, Topology, Action, Analysis, or Control), help for various file formats, or help with atom mask syntax.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.14">8.14</a> list
</h2>
<pre class="LyX-Code">
list &lt;type&gt;
     (&lt;type&gt; = actions,trajin,trajout,ref,parm,analysis,datafile,dataset)
</pre>
<div class="Unindented">
List the currently loaded objects of &lt;type&gt;. If no type is given then list all loaded objects.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.15">8.15</a> noexitonerror
</h2>
<pre class="LyX-Code">
noexitonerror<a class="IndexReference" name="entry-noexitonerror-0" href="#index-noexitonerror">↓</a>
</pre>
<div class="Unindented">
Normally <i>cpptraj</i> will exit if actions fail to initialize properly. If <b><i>noexitonerror</i></b> is specified, <i>cpptraj</i> will attempt to continue past such errors. This is the default if in interactive mode.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.16">8.16</a> noprogress
</h2>
<pre class="LyX-Code">
noprogress<a class="IndexReference" name="entry-noprogress-0" href="#index-noprogress">↓</a>
</pre>
<div class="Unindented">
Do not display read progress during trajectory processing.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.17">8.17</a> parallelanalysis
</h2>
<pre class="LyX-Code">
parallelanalysis [sync]
</pre>
<div class="Unindented">
MPI only. Divide all currently set up analyses as evenly as possible among available MPI processes and execute. Each analysis will get a single MPI process. If <b>sync</b> is specified all data will be synced back to the master thread (for e.g. subsequent analysis). For an example of how to use the parallelanalysis command, see <a class="Reference" href="#subsec:cpptraj_hausdorff">12.14 on page 1↓</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.18">8.18</a> precision
</h2>
<pre class="LyX-Code">
precision<a class="IndexReference" name="entry-precision-0" href="#index-precision">↓</a> {&lt;filename&gt; | &lt;dataset arg&gt;} [&lt;width&gt;] [&lt;precision&gt;]
</pre>
<div class="Unindented">
Set the precision for all data sets in data file &lt;filename&gt; or data set(s) specified by &lt;dataset arg&gt; to <i>width.precision</i>, where width is the column width and precision is the number of digits after the decimal point. Note that the &lt;precision&gt; argument only applies to floating-point data sets.
</div>
<div class="Indented">
For example, if one wanted to set the precision of the output of an Rmsd calculation to 8.3, the input could be:
</div>
<pre class="LyX-Code">
trajin ../run0.nc 
rms first :10-260 out prec.dat 
precision prec.dat 8 3
</pre>
<div class="Unindented">
and the output would look like:
</div>
<pre class="LyX-Code">
#Frame RMSD_00000
1 0.000
2 0.630
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.19">8.19</a> readdata<a class="Label" name="subsec:cpptraj_readdata"> </a>
</h2>
<pre class="LyX-Code">
readdata<a class="IndexReference" name="entry-readdata-0" href="#index-readdata">↓</a> &lt;filename&gt;  [name &lt;dsname&gt;] [as &lt;fmt&gt;] [separate] [&lt;format options&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">name &lt;dsname&gt;</span> Name for read-in data set(s). Default is &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">as &lt;fmt&gt;</span> Force &lt;filename&gt; to be read as a specific format using given format keyword.
</div>
<div class="Description">
<span class="Description-entry">separate</span> Read each file specified into separate data sets indexed from 0.
</div>

</li>

</ul>
<div class="Unindented">
Read data from file &lt;filename&gt; and store as data sets. For more information on formats currently recognized by cpptraj see <a class="Reference" href="#tab:cpptraj-DataFile-formats">1 on page 1↑</a>. For format-specific options see <a class="Reference" href="#sec:Datafile-Commands">6↑</a>. For example, given the file calc.dat:
</div>
<pre class="LyX-Code">
#Frame  R0   D1
1      1.7 2.22
</pre>
<div class="Unindented">
The command ’<tt>readdata calc.dat</tt>’ would read data into two data sets, calc.dat:2 (legend set to &ldquo;R0&rdquo;) and calc.dat:3 (legend set to &ldquo;D1&rdquo;).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.20">8.20</a> readensembledata<a class="Label" name="subsec:cpptraj_readensembledata"> </a>
</h2>
<pre class="LyX-Code">
readensembledata &lt;filename&gt; [filenames &lt;additional files&gt;] [&lt;readdata args&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;filename&gt;</span> Lowest replica file name.
</div>
<div class="Description">
<span class="Description-entry">filenames &lt;additional files&gt;</span> Specified additional members of the ensemble. If not specified ensemble members will be search for using numerical extensions.
</div>
<div class="Description">
<span class="Description-entry">&lt;readdata args&gt;</span> Additional data file arguments.
</div>

</li>

</ul>
<div class="Unindented">
Read data sets as an ensemble, i.e. each file is a different member of an ensemble. This command is MPI-aware.
</div>
<div class="Indented">
If one filename is given, it is assumed it is the "lowest" member of an ensemble with a numerical extension, e.g. ’file.001’ and the remaining files are searched for automatically. Otherwise all other members of the ensemble can be specified with <b>’filenames’</b> and a comma-separated list e.g. ’file.001 filenames file.002,file.003,file.004. For additional ’readdata’ arguments that can be passed in see <a class="Reference" href="#sec:Datafile-Commands">6 on page 1↑</a>.
</div>
<div class="Indented">
For example, to read in data files named cpout.001 to cpout.006 automatically:
</div>
<pre class="LyX-Code">
readensembledata cpout.001 cpin cpin name PH
</pre>
<div class="Unindented">
Or specified:
</div>
<pre class="LyX-Code">
readensembledata cpout.001 \
                 filenames cpout.002,cpout.003,cpout.004,cpout.005,cpout.006 \
                 cpin cpin name PH 
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.21">8.21</a> readinput
</h2>
<pre class="LyX-Code">
readinput<a class="IndexReference" name="entry-readinput-0" href="#index-readinput">↓</a> &lt;filename&gt;
</pre>
<div class="Unindented">
Read <i>cpptraj</i> commands from file &lt;filename&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.22">8.22</a> removedata
</h2>
<pre class="LyX-Code">
removedata &lt;arg&gt;
</pre>
<div class="Unindented">
Remove data set corresponding to &lt;arg&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.23">8.23</a> rst
</h2>
<pre class="LyX-Code">
rst &lt;mask1&gt; &lt;mask2&gt; [&lt;mask3&gt;] [&lt;mask4&gt;]
    r1 &lt;r1&gt; r2 &lt;r2&gt; r3 &lt;r3&gt; r4 &lt;r4&gt; rk2 &lt;rk2&gt; rk3 &lt;rk3&gt;
    {[parm &lt;parmfile / tag&gt; | parmindex &lt;#&gt;]}
    [{ref &lt;refname&gt; | refindex &lt;#&gt; | reference} [offset &lt;off&gt;] [width &lt;width&gt;]]
    [out &lt;outfile&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask1&gt;</span> (Required) First atom mask.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask2&gt;</span> (Required) Second atom mask. If only two masks assume distance restraint.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask3&gt;]</span> (Optional) Third atom mask. If 3 atom masks assume angle restraint.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask4&gt;]</span> (Optional) Fourth atom mask. If 4 atom masks assume dihedral restraint.
</div>
<div class="Description">
<span class="Description-entry">rX &lt;rX&gt;</span> Value of RX (X=1-4, default 0.0)
</div>
<div class="Description">
<span class="Description-entry">rk2 &lt;rk2&gt;</span> Value of RK2 (force constant to be applied when R is R1 &lt;= R &lt; R2)
</div>
<div class="Description">
<span class="Description-entry">rk3 &lt;rk3&gt;</span> Value of RK3 (force constant to be applied when R is R3 &lt;= R &lt; R4)
</div>
<div class="Description">
<span class="Description-entry">[parm &lt;parmfile / tag&gt; | parmindex &lt;#&gt;]</span> Topology to be used for atom masks.
</div>
<div class="Description">
<span class="Description-entry">{ref &lt;refname&gt; | refindex &lt;#&gt; | reference}</span> Use distance/angle/dihedral in reference structure to determine values for r1, r2, r3, and r4. The value of r2 is set to &lt;r2&gt; + &lt;off&gt;, r3 = r2, r1 = r2 - &lt;width&gt;, r4 = r3 + &lt;width&gt;.
</div>
<div class="Description">
<span class="Description-entry">[offset &lt;off&gt;]</span> (Reference only) Value to offset distance/angle/torsion in reference by (default 0.0).
</div>
<div class="Description">
<span class="Description-entry">[width &lt;width&gt;]</span> (Reference only) Width between r1 and r2, r3 and r4 (default 0.5).
</div>
<div class="Description">
<span class="Description-entry">[out &lt;outfile&gt;]</span> Write restraints to outfile. If not specified, write to STDOUT.
</div>

</li>

</ul>
<div class="Unindented">
Generate Amber-style distance restraints for use with nmropt=1. This is particularly useful for generating distance restraints based off of reference coordinates. For example to generate a distance restraint between two C5’ atoms using the current distance between them in a reference structure, offsetting the distance by 1.0 Ang.:
</div>
<pre class="LyX-Code">
parm 30bp-longbox-tip3p-na.parm7
reference 30bp-longbox.rst7 
rst :1@C5’ :31@C5’ reference offset 1.0 rk2 10.0 rk3 10.0 out output
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.24">8.24</a> runanalysis<a class="Label" name="subsec:cpptraj_runanalysis"> </a>
</h2>
<pre class="LyX-Code">
runanalysis [&lt;analysiscmd&gt; [&lt;analysis args&gt;]]
</pre>
<div class="Unindented">
Run given analysis command immediately and write any data generated. If no command is given run any analysis currently set up. NOTE: When ’runanalysis’ is specified alone, data is not automatically written; to write data generated with ’runanalysis’ use the ’writedata’ command (this allows multiple analysis runs between output if desired).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.25">8.25</a> select
</h2>
<pre class="LyX-Code">
select<a class="IndexReference" name="entry-select-0" href="#index-select">↓</a> &lt;mask&gt;
</pre>
<div class="Unindented">
Prints the number of selected atoms corresponding to the given mask, as well as the atom numbers with format:
</div>
<pre class="LyX-Code">
Selected= &lt;#atom1&gt; &lt;#atom2&gt; ...
</pre>
<div class="Unindented">
This does not affect the state in any way, but is intended for use in scripts etc. for testing the results of a mask expression.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.26">8.26</a> selectds
</h2>
<pre class="LyX-Code">
selectds &lt;dataset arg&gt;
</pre>
<div class="Unindented">
Show the results of a data set selection. Data set selection has the format:
</div>
<pre class="LyX-Code">
&lt;name&gt;[&lt;aspect&gt;]:&lt;index&gt;
</pre>
<div class="Unindented">
Either the [&lt;aspect&gt;] or the &lt;index&gt; arguments may be omitted. A ’*’ can be used in place of &lt;name&gt; or [&lt;aspect&gt;] as a wildcard. The &lt;index&gt; argument can be a single number or a range separated by ’-’ and ’,’.
</div>
<div class="Indented">
This command does not affect the state in any way, but is particularly useful in interactive mode for determining the results of a dataset argument.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.27">8.27</a> sortensembledata<a class="Label" name="subsec:cpptraj_sortensembledata"> </a>
</h2>
<pre class="LyX-Code">
sortensembledata &lt;dset arg0&gt; [&lt;dset arg1&gt; ...]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dset arg0&gt; [&lt;dset arg1&gt; ...]</span> Data set(s) to sort.
</div>

</li>

</ul>
<div class="Unindented">
Sort unsorted data sets. Currently only works for constant pH REMD data.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.28">8.28</a> usediskcache
</h2>
<pre class="LyX-Code">
usediskcache {on|off}
</pre>
<div class="Unindented">
If on, CPPTRAJ will attempt to cache data sets to disk if possible. This currently only works for integer data sets (e.g. <b><i>hbond</i></b> <b>series</b> data sets, etc).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.29">8.29</a> write | writedata<a class="Label" name="subsec:cpptraj_writedata"> </a>
</h2>
<pre class="LyX-Code">
write [&lt;filename&gt; &lt;datasetname0&gt; [&lt;datasetname1&gt; ...]] [&lt;DataFile Options&gt;]
</pre>
<div class="Unindented">
With no arguments, write all files currently in the data file list. Otherwise, write specified data set(s) to &lt;filename&gt;. This is like the ’create’ command except a data file is not added to the data file list; it is written immediately. See <a class="Reference" href="#sec:Datafile-Commands">6 on page 1↑</a> for more data file format options.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-8.30">8.30</a> System Commands
</h2>
<div class="Unindented">
These commands call the equivalent external system commands. 
</div>
<div class="Description">
<span class="Description-entry">gnuplot &lt;args&gt;</span> Call <tt>gnuplot</tt> (if it is installed on your system) with the given arguments.
</div>
<div class="Description">
<span class="Description-entry">head &lt;args&gt;</span> Call <tt>head</tt>, which lists the first few lines of a file.
</div>
<div class="Description">
<span class="Description-entry">less &lt;args&gt;</span> Call <tt>less</tt>, which can be used to view the contents of a file.
</div>
<div class="Description">
<span class="Description-entry">ls &lt;args&gt;</span> List the contents of a directory.
</div>
<div class="Description">
<span class="Description-entry">pwd &lt;args&gt;</span> Print the current working directory.
</div>
<div class="Description">
<span class="Description-entry">xmgrace &lt;args&gt;</span> Call <tt>xmgrace</tt> (if it is installed on your system) with the given arguments.
</div>
<h1 class="Section">
<a class="toc" name="toc-Section-9">9</a> Topology File Commands
</h1>
<div class="Unindented">
These commands control the reading and writing of topology files. Cpptraj supports the following topology file formats:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
<b>Format</b>
</td>
<td align="center" valign="top">
<b>Keyword</b>
</td>
<td align="center" valign="top">
<b>Extension</b>
</td>
<td align="center" valign="top">
<b>Notes</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
Amber Topology
</td>
<td align="center" valign="top">
amber
</td>
<td align="center" valign="top">
.parm7
</td>
<td align="center" valign="top">
Only fully-supported format for write.
</td>

</tr>
<tr>
<td align="center" valign="top">
PDB 
</td>
<td align="center" valign="top">
pdb
</td>
<td align="center" valign="top">
.pdb
</td>
<td align="center" valign="top">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top">
Mol2
</td>
<td align="center" valign="top">
mol2
</td>
<td align="center" valign="top">
.mol2
</td>
<td align="center" valign="top">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top">
CIF
</td>
<td align="center" valign="top">
cif
</td>
<td align="center" valign="top">
.cif
</td>
<td align="center" valign="top">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top">
Charmm PSF
</td>
<td align="center" valign="top">
psf
</td>
<td align="center" valign="top">
.psf
</td>
<td align="center" valign="top">
Limited Write
</td>

</tr>
<tr>
<td align="center" valign="top">
Gromacs Topology
</td>
<td align="center" valign="top">
gromacs
</td>
<td align="center" valign="top">
.top
</td>
<td align="center" valign="top">
Read only
</td>

</tr>
<tr>
<td align="center" valign="top">
SDF
</td>
<td align="center" valign="top">
sdf
</td>
<td align="center" valign="top">
.sdf
</td>
<td align="center" valign="top">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top">
Tinker ARC
</td>
<td align="center" valign="top">
arc
</td>
<td align="center" valign="top">
.arc
</td>
<td align="center" valign="top">
Read Only
</td>

</tr>

</table>
<a class="Label" name="tab:cpptraj-Topology-formats"> </a>
</div>

</div>
<div class="Indented">
For most commands that require a topology one can be specified via two keywords:
</div>
<div class="Description">
<span class="Description-entry">parm [&lt;name&gt;]</span> Select topology corresponding to given file name, tag, or name.
</div>
<div class="Description">
<span class="Description-entry">parmindex [&lt;#&gt;]</span> Select topology by order in which it was loaded, starting from 0.
</div>
<div class="Unindented">
The following topology related commands are available:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
<b>Command</b>
</td>
<td align="center" valign="top">
<b>Description</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
angleinfo, angles, printangles
</td>
<td align="center" valign="top">
Print angle info for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
atominfo, atoms, printatoms
</td>
<td align="center" valign="top">
Print details for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
bondinfo, bonds, printbonds
</td>
<td align="center" valign="top">
Print bond info for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
change
</td>
<td align="center" valign="top">
Change specified parts of a topology.
</td>

</tr>
<tr>
<td align="center" valign="top">
charge
</td>
<td align="center" valign="top">
Print total charge for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
comparetop
</td>
<td align="center" valign="top">
Compare two topologies and report differences.
</td>

</tr>
<tr>
<td align="center" valign="top">
dihedralinfo, dihedrals, printdihedrals
</td>
<td align="center" valign="top">
Print dihedral info for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
improperinfo, impropers, printimpropers
</td>
<td align="center" valign="top">
Print improper info for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
mass
</td>
<td align="center" valign="top">
Print total mass for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
molinfo
</td>
<td align="center" valign="top">
Print molecule info for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
parm
</td>
<td align="center" valign="top">
Load a topology file.
</td>

</tr>
<tr>
<td align="center" valign="top">
parmbox
</td>
<td align="center" valign="top">
Modify box info for a loaded topology.
</td>

</tr>
<tr>
<td align="center" valign="top">
parminfo
</td>
<td align="center" valign="top">
Print details for selected topology.
</td>

</tr>
<tr>
<td align="center" valign="top">
parmstrip
</td>
<td align="center" valign="top">
Remove selected atoms from topology.
</td>

</tr>
<tr>
<td align="center" valign="top">
parmwrite
</td>
<td align="center" valign="top">
Write selected topology to file.
</td>

</tr>
<tr>
<td align="center" valign="top">
printub, ubinfo
</td>
<td align="center" valign="top">
Print Urey-Bradley info for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
resinfo
</td>
<td align="center" valign="top">
Print residue info for selected atoms.
</td>

</tr>
<tr>
<td align="center" valign="top">
scaledihedralk
</td>
<td align="center" valign="top">
Scale selected dihedral force constants.
</td>

</tr>
<tr>
<td align="center" valign="top">
solvent
</td>
<td align="center" valign="top">
Change which molecules are considered solvent.
</td>

</tr>
<tr>
<td align="center" valign="top">
updateparameters
</td>
<td align="center" valign="top">
Update/add parameters in/to a topology.
</td>

</tr>

</table>

</div>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.1">9.1</a> angleinfo | angles | printangles
</h2>
<pre class="LyX-Code">
angleinfo [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask1&gt;] [&lt;mask2&gt; &lt;mask3&gt;]
          [out &lt;file&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Name/tag or index of topology. Default is first loaded topology.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask1&gt;]</span> Mask to print angle info for.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask2&gt; &lt;mask3&gt;]</span> If specified, angles must match all masks.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to print to (default STDOUT).
</div>

</li>

</ul>
<div class="Unindented">
Print angle information of atoms in &lt;mask&gt; for selected topology (first loaded topology by default) with format:
</div>
<pre class="LyX-Code">
# Angle Kthet degrees atom names (numbers)
</pre>
<div class="Unindented">
Where <tt>Angle</tt> is the internal angle index, <tt>Kthet</tt> is the angle force constant, <tt>degrees</tt> is the angle equilibrium value, <tt>atom names</tt> shows the atoms involved in the angle with format <tt>:&lt;residue num&gt;@&lt;atom name&gt;</tt>, and <tt>(numbers)</tt> shows the atom indices involved in a comma-separated list. Atom types will be shown in the last column.
</div>
<div class="Indented">
If 3 masks are given instead of 1, print info for angles with first atom in &lt;mask1&gt;, second atom in &lt;mask2&gt;, and third atom in &lt;mask3&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.2">9.2</a> atominfo | atoms | printatoms
</h2>
<pre class="LyX-Code">
atominfo [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask&gt;] [out &lt;file&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Name/tag or index of topology. Default is first loaded topology.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Mask selecting atoms to print info for.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to print to (default STDOUT).
</div>

</li>

</ul>
<div class="Unindented">
Print information on atoms in &lt;mask&gt; for selected topology (first loaded topology by default) with format:
</div>
<pre class="LyX-Code">
#Atom Name #Res Name #Mol Type Charge Mass GBradius El [rVDW] [eVDW]
</pre>
<div class="Unindented">
where <tt>#Atom</tt> is the internal atom index, the first <tt>Name</tt> column is the atom name, <tt>#Res</tt> is the atom’s residue number, the second <tt>Name</tt> column is residue name, <tt>#Mol</tt> is the atom’s molecule number, <tt>Type</tt> is the atom’s type (certain topologies only), <tt>Charge</tt> is the atom charge (in units of electron charge), <tt>Mass</tt> is the atom’s mass (in amu), <tt>GBradius</tt> is the generalized Born radius of the atom (Amber topologies only), and <tt>El</tt> is the 2 character element string. The final two columns are only shown if the topology contains non-bonded parameters: <tt>rVDW</tt> is the atom’s Lennard-Jones radius and <tt>eVDW</tt> is the atom’s Lennard-Jones epsilon.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.3">9.3</a> bondinfo | bonds | printbonds
</h2>
<pre class="LyX-Code">
bondinfo [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] 
         [&lt;mask1&gt;] [&lt;mask2&gt;] [out &lt;file&gt;] [nointrares]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Name/tag or index of topology. Default is first loaded topology.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask1&gt;]</span> Mask to print bond info for.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask2&gt;]</span> If specified, bonds must match both masks.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to print to (default STDOUT).
</div>
<div class="Description">
<span class="Description-entry">[nointrares]</span> Do not print intra-residue bonds.
</div>

</li>

</ul>
<div class="Unindented">
Print bond information for atoms in &lt;mask&gt; for selected topology (first loaded topology by default) with format:
</div>
<pre class="LyX-Code">
# Bond Kb Req atom names (numbers)
</pre>
<div class="Unindented">
where <tt>Bond</tt> is the internal bond index, <tt>Kb</tt> is the bond force constant, <tt>Req</tt> is the bond equilibrium value (in Angstroms), <tt>atom names</tt> shows both atom names with format <tt>:&lt;residue num&gt;@&lt;atom name&gt;</tt>, and <tt>(numbers)</tt> shows both atom numbers in a comma-separated list. Atom types will be shown in the last column.
</div>
<div class="Indented">
If 2 masks are given instead of 1, print info for bonds with first atom in &lt;mask1&gt; and second atom in &lt;mask2&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.4">9.4</a> change
</h2>
<pre class="LyX-Code">
change [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt; |
        crdset &lt;COORDS set&gt; ]
	{ resname from &lt;mask&gt; to &lt;value&gt; |
      chainid of &lt;mask&gt; to &lt;value&gt; |
	  atomname from &lt;mask&gt; to &lt;value&gt; |
      addbond &lt;mask1&gt; &lt;mask2&gt; [req &lt;length&gt; &lt;rk&gt; &lt;force constant&gt;]
      removebonds &lt;mask1&gt; [&lt;mask2&gt;] [out &lt;file&gt;] }
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt; | crdset &lt;COORDS set&gt;</span> Topology to change.
</div>
<div class="Description">
<span class="Description-entry">resname from &lt;mask&gt; to &lt;value&gt;</span> Change residue names for residues in &lt;mask&gt; to the given &lt;value&gt;.
</div>
<div class="Description">
<span class="Description-entry">chainid of &lt;mask&gt; to &lt;value&gt;</span> Change the chain ID of residues in &lt;mask&gt; to given &lt;value&gt;.
</div>
<div class="Description">
<span class="Description-entry">atomname from &lt;mask&gt; to &lt;value&gt;</span> Change atom names for atoms in &lt;mask&gt; to the given &lt;value&gt;.
</div>
<div class="Description">
<span class="Description-entry">addbond &lt;mask1&gt; &lt;mask2&gt;</span> Add bond between atom specified by &lt;mask1&gt; and atom specified by &lt;mask2&gt;.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[req &lt;length&gt;]</span> The equilibrium bond length in Angstroms.
</div>
<div class="Description">
<span class="Description-entry">[rk &lt;force constant&gt;]</span> The bond force constant in kcal/mol*Angstrom.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">removebonds &lt;mask1&gt; [&lt;mask2&gt;]</span> Remove bonds from atoms in &lt;mask1&gt;. If &lt;mask2&gt; also given, remove bonds between atoms in &lt;mask1&gt; and atoms in &lt;mask2&gt;.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> If specified, write removed bonds to &lt;file&gt; with format ’&lt;residue name&gt; &lt;residue num&gt; &lt;atom name&gt; &lt;atom num&gt;’.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Change specified parts of the specified topology. For example, to change atoms named ’HN’ to ’H’ in topology 0:
</div>
<pre class="LyX-Code">
change parmindex 0 atomname from @HN to H
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.5">9.5</a> charge
</h2>
<pre class="LyX-Code">
charge [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask&gt;] [out &lt;file&gt;] [name &lt;set&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;name&gt; | parmindex &lt;#&gt;</span> Topology to calculate charge from.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atom(s) to calculate total charge for (default all).
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to write total charge to.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;set&gt;]</span> If specified, a data set named &lt;set&gt; will be cretaed containing total charge.
</div>

</li>

</ul>
<div class="Unindented">
Print the total charge of atoms in &lt;mask&gt; (in units of electron charge) for selected topology (first loaded topology by default).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.6">9.6</a> comparetop
</h2>
<pre class="LyX-Code">
comparetop {parm &lt;name&gt; | parmindex &lt;#&gt;} {parm &lt;name&gt; | parmindex &lt;#&gt;} [out &lt;file&gt;]
           [atype] [lj] [bnd] [ang] [dih] [atoms]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;name&gt; | parmindex &lt;#&gt;</span> Topologies to compare.
</div>
<div class="Description">
<span class="Description-entry">out &lt;file&gt;</span> Print results to file instead of screen.
</div>
<div class="Description">
<span class="Description-entry">[atype]</span> Only report atom type differences.
</div>
<div class="Description">
<span class="Description-entry">[lj]</span> Only report differences in Lennard-Jones parameters.
</div>
<div class="Description">
<span class="Description-entry">[bnd]</span> Only report differences in bond parameters.
</div>
<div class="Description">
<span class="Description-entry">[ang]</span> Only report differences in angle parameters.
</div>
<div class="Description">
<span class="Description-entry">[dih]</span> Only report differences in dihedral parmeters.
</div>
<div class="Description">
<span class="Description-entry">[atoms]</span> Only report differences in atom properties.
</div>

</li>

</ul>
<div class="Unindented">
Compare and report differences in atoms/parameters between two topologies. Differences are reported in standard ’diff’ format, with ’&lt;’ prefix indicating the parameter is from the first topology and ’&gt;’ prefix indicating the parameter is from the second topology.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.7">9.7</a> dihedralinfo | dihedrals | printdihedrals
</h2>
<pre class="LyX-Code">
dihedralinfo [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask1&gt;] [&lt;mask2&gt; &lt;mask3&gt; &lt;mask4&gt;]
             [out &lt;file&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Name/tag or index of topology. Default is first loaded topology.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask1&gt;]</span> Mask to print dihedral info for.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask2&gt; &lt;mask3&gt; &lt;mask4&gt;]</span> If specified, dihedrals must match all masks.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to print to (default STDOUT).
</div>

</li>

</ul>
<div class="Unindented">
Print dihedral information of atoms in &lt;mask&gt; for selected topology (first loaded topology by default) with format:
</div>
<pre class="LyX-Code">
#Dihedral pk phase pn atoms
</pre>
<div class="Unindented">
where <tt>#Dihedral</tt> is the internal dihedral index, <tt>pk</tt> is the dihedral force constant, <tt>phase</tt> is the dihedral phase, <tt>pn</tt> is the dihedral periodicity, and <tt>atoms</tt> shows the names of the atoms involved in the angle with format <tt>:&lt;residue num&gt;@&lt;atom name&gt;</tt>, followed by the atom indices involved in a comma-separated list. In addition if the dihedral is an end dihedral, improper dihedral, or both it will be prefaced with an <tt>E</tt>, <tt>I</tt>, or <tt>B</tt> respectively. Atom types will be shown in the last column.
</div>
<div class="Indented">
If 4 masks are given instead of 1, print info for dihedrals with first atom in &lt;mask1&gt;, second atom in &lt;mask2&gt;, third atom in &lt;mask3&gt;, and fourth atom in &lt;mask4&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.8">9.8</a> improperinfo | impropers | printimpropers
</h2>
<pre class="LyX-Code">
improperinfo [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask1&gt;] [&lt;mask2&gt; &lt;mask3&gt; &lt;mask4&gt;]
             [out &lt;file&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Name/tag or index of topology. Default is first loaded topology.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask1&gt;]</span> Mask to print improper info for.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask2&gt; &lt;mask3&gt; &lt;mask4&gt;]</span> If specified, impropers must match all masks.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to print to (default STDOUT).
</div>

</li>

</ul>
<div class="Unindented">
For specified topology (first by default) either print CHARMM improper info for all atoms in &lt;mask1&gt;, or print info for dihedrals with first atom in &lt;mask1&gt;, second atom in &lt;mask2&gt;, third atom in &lt;mask3&gt;, and fourth atom in &lt;mask4&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.9">9.9</a> mass
</h2>
<pre class="LyX-Code">
[&lt;parmindex&gt;] [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask&gt;] [out &lt;file&gt;] [name &lt;set&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;name&gt; | parmindex &lt;#&gt;</span> Topology to calculate mass from.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atom(s) to calculate total mass for (default all).
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to write total mass to.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;set&gt;]</span> If specified, a data set named &lt;set&gt; will be cretaed containing total mass.
</div>

</li>

</ul>
<div class="Unindented">
Print the total mass of atoms in &lt;mask&gt; (in amu) for selected topology (first loaded topology by default).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.10">9.10</a> molinfo
</h2>
<pre class="LyX-Code">
molinfo [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask&gt;] [out &lt;file&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Name/tag or index of topology. Default is first loaded topology.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Mask selecting molecules to print info for.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to print to (default STDOUT).
</div>

</li>

</ul>
<div class="Unindented">
Print molecule information for atoms in &lt;mask&gt; for selected topology (first loaded topology by default) with format:
</div>
<pre class="LyX-Code">
#Mol  Natom  #Res Name C [SOLVENT]
</pre>
<div class="Unindented">
where <tt>#Mol</tt> is the molecule number, <tt>Natom</tt> is the number of atoms in the molecule, <tt>#Res</tt> and <tt>Name</tt> are the residue number and residue name of the first residue in the molecule respectively, and C is the chain ID of the first residue. <tt>SOLVENT</tt> will be printed if the molecule is currently considered a solvent molecule.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.11">9.11</a> parm
</h2>
<pre class="LyX-Code">
parm<a class="IndexReference" name="entry-parm-0" href="#index-parm">↓</a> &lt;filename&gt; [{[TAG] | name &lt;setname&gt;}] 
     [{ nobondsearch |
        [bondsearch &lt;offset&gt;] [searchtype {grid|pairlist}]\begin_inset Separator latexpar\end_inset
      }]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>&lt;filename&gt;</b></span> Parameter file to read in; format is auto-detected.
</div>
<div class="Description">
<span class="Description-entry"><b>’[TAG]’</b></span> Optional tag (bounded in brackets) which can be referred to in place of the topology file name in order to simplify references to it (see <a class="Reference" href="#subsec:Parameter/Reference-Tagging">3.4 on page 1↑</a> for examples of how to use tags).
</div>
<div class="Description">
<span class="Description-entry">[name &lt;setname&gt;]</span> Optional name that can be used to refer to the topology in place of the file name.
</div>
<div class="Description">
<span class="Description-entry"><b>[nobondsearch]</b></span> Optional; if specified do not search for bonds via geometry if Topology does not include bond information. May cause some Actions to fail.
</div>
<div class="Description">
<span class="Description-entry"><b>[bondsearch</b></span> <b>&lt;offset&gt;]</b> Optional; when searching for bonds via geometry search (default for Topologies without bond information) add &lt;offset&gt; to distances (default 0.2 Å). Increase this if your system includes unusually long bonds.
</div>
<div class="Description">
<span class="Description-entry">[searchtype {grid|pairlist}]</span> Change search algorithm from the default search between residues algorithm:
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">grid</span> Uses a grid when searching for bonds between residues. This can find bonds between residues that are not sequential (e.g. disulfide bonds).
</div>
<div class="Description">
<span class="Description-entry">pairlist</span> Uses a pair list to search for bonds between atoms. This can potentially find bonds across periodic boundaries, but is the more experimental of the two.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Read in parameter file. The file format will be auto-detected. Current formats recognized by cpptraj are listed <a class="Reference" href="#tab:cpptraj-Topology-formats"> on page 1↑</a>. If the file does not contain bond information, cpptraj will attempt to assign bonds based on a simple distance search of atoms within and between residues. The distance cutoff for determining bonds between atoms depends on the elements of the two atoms in question, augmented by &lt;offset&gt;. Molecule information is then determined from bond information.
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-9.11.1">9.11.1</a> <i>PDB format:</i>
</h3>
<pre class="LyX-Code">
[pqr] [readbox] [noconect]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[pqr]</span> Read charge and radius information from the occupancy and B-factor columns.
</div>
<div class="Description">
<span class="Description-entry">[readbox]</span> Read unit cell information from CRYST1 record if present.
</div>
<div class="Description">
<span class="Description-entry">[noconect]</span> Do not read in CONECT records from PDB file.
</div>

</li>

</ul>
<div class="Paragraph-">
<a class="toc" name="toc-Paragraph--1"></a><b>IMPORTANT NOTES FOR PDB FILES</b>
</div>
<div class="Unindented">
Sometimes PDB files can contain alternate coordinates for the same atom in a residue, e.g.:
</div>
<pre class="LyX-Code">
ATOM    806  CA ACYS A 105       6.460 -34.012 -21.801  0.49 32.23 
ATOM    807  CB ACYS A 105       6.054 -33.502 -20.415  0.49 35.28
ATOM    808  CA BCYS A 105       6.468 -34.015 -21.815  0.51 32.42
ATOM    809  CB BCYS A 105       6.025 -33.499 -20.452  0.51 35.38
</pre>
<div class="Unindented">
If this is the case <i>cpptraj</i> will print a warning about duplicate atom names but will take no other action. Both residues are considered ’CYS’ and the mask ’:CYS@CA’ would select both atom 806 and 809. Residue insertion codes are read in but also not used by the mask parser.
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-9.11.2">9.11.2</a> Charmm PSF:
</h3>
<pre class="LyX-Code">
[param &lt;file&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[param &lt;file&gt;]</span> Read CHARMM parameters from given file. Can do multiple times.
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.12">9.12</a> parmbox
</h2>
<pre class="LyX-Code">
parmbox<a class="IndexReference" name="entry-parmbox-0" href="#index-parmbox">↓</a> [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [nobox] [truncoct]
    	[x &lt;xval&gt;] [y &lt;yval&gt;] [z &lt;zval&gt;] [alpha &lt;a&gt;] [beta &lt;b&gt;] [gamma &lt;g&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Name/tag or index of topology to modify. Default is first loaded topology.
</div>
<div class="Description">
<span class="Description-entry">[nobox]</span> Remove box information.
</div>
<div class="Description">
<span class="Description-entry">[truncoct]</span> Set truncated octahedon angles with lengths equal to &lt;xval&gt;.
</div>
<div class="Description">
<span class="Description-entry">[x &lt;xval&gt;]</span> Box X length.
</div>
<div class="Description">
<span class="Description-entry">[y &lt;yval&gt;]</span> Box Y length.
</div>
<div class="Description">
<span class="Description-entry">[z &lt;zval&gt;]</span> Box Z length.
</div>
<div class="Description">
<span class="Description-entry">[alpha &lt;a&gt;]</span> Box alpha angle.
</div>
<div class="Description">
<span class="Description-entry">[beta &lt;b&gt;]</span> Box beta angle.
</div>
<div class="Description">
<span class="Description-entry">[gamma &lt;g&gt;]</span> Box gamma angle.
</div>

</li>

</ul>
<div class="Unindented">
Modify the box information for specified topology. Overwrites any box information if present with specified values; any that are not specified will remain unchanged. Note that unlike the <b><i>box</i></b> action this command affect box information immediately. This can be useful for e.g. removing box information from a parm when stripping solvent:
</div>
<pre class="LyX-Code">
parm mol.water.parm7
parmstrip :WAT
parmbox nobox
parmwrite out strip.mol.nobox.parm7
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.13">9.13</a> parminfo 
</h2>
<pre class="LyX-Code">
parminfo<a class="IndexReference" name="entry-parminfo-0" href="#index-parminfo">↓</a> [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask&gt;]
</pre>
<div class="Unindented">
Print a summary of information contained in the specified topology (first loaded topology by default) .
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.14">9.14</a> parmstrip
</h2>
<pre class="LyX-Code">
parmstrip<a class="IndexReference" name="entry-parmstrip-0" href="#index-parmstrip">↓</a> &lt;mask&gt; [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]
</pre>
<div class="Unindented">
Strip atoms in <b>&lt;mask&gt;</b> from specified topology (by default the first topology loaded). Note that unlike the <b><i>strip</i></b> Action, this permanently modifies the topology for as long as <i>cpptraj</i> is running, so this should not be used if the topology is being used to read or write a trajectory via <b><i>trajin</i></b>/<b><i>trajout</i></b>. This command can be used to quickly created stripped Amber topology files. For example, to strip all residues name WAT from a topology and write a new topology:
</div>
<pre class="LyX-Code">
parm mol.water.parm7
parmstrip :WAT
parmwrite out strip.mol.parm7
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.15">9.15</a> parmwrite
</h2>
<pre class="LyX-Code">
parmwrite<a class="IndexReference" name="entry-parmwrite-0" href="#index-parmwrite">↓</a> out &lt;filename&gt; [{parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt; | crdset &lt;setname&gt;}]
          [&lt;fmt&gt;] [nochamber]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;filename&gt;</span> File to write to.
</div>
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Topology to write out.
</div>
<div class="Description">
<span class="Description-entry">[crdset &lt;setname&gt;]</span> Write topology from specified COORDS data set.
</div>
<div class="Description">
<span class="Description-entry">[&lt;fmt&gt;]</span> Format keyword. If not specified the file name extension will be used. Default is Amber Topology.
</div>
<div class="Description">
<span class="Description-entry">[nochamber]</span> (Amber topology only) Remove any CHAMBER information from the topology.
</div>

</li>

</ul>
<div class="Unindented">
Write out specified topology (first topology loaded by default) to <b>&lt;filename&gt;</b> with format <b>&lt;fmt&gt;</b> (Amber topology if not specified). Note that the Amber topology format is the only fully supported format for topology writes.
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-9.15.1">9.15.1</a> Amber Topology
</h3>
<pre class="LyX-Code">
[nochamber] [writeempty]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[nochamber]</span> Do not write CHAMBER information to topology (useful for e.g. using topology for visualization with older versions of VMD).
</div>
<div class="Description">
<span class="Description-entry">[writeempty]</span> Write Amber tree, join, and rotate info even if not present.
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-9.15.2">9.15.2</a> Charmm PSF
</h3>
<pre class="LyX-Code">
[oldpsf] [ext]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[oldpsf]</span> Write atom type indices instead of type names (not recommended).
</div>
<div class="Description">
<span class="Description-entry">[ext]</span> Use extended format.
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.16">9.16</a> printub | ubinfo
</h2>
<pre class="LyX-Code">
printub [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask1&gt;] [&lt;mask2&gt;] [out &lt;file&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Name/tag or index of topology. Default is first loaded topology.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask1&gt;]</span> Atoms to print UB info for.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask2&gt;]</span> If specified, UB info must match both masks.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to print to (default STDOUT).
</div>

</li>

</ul>
<div class="Unindented">
For specified topology (first by default) either print CHARMM Urey-Bradley info for all atoms in &lt;mask1&gt;, or print info for bonds with first atom in &lt;mask1&gt; and second atom in &lt;mask2&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.17">9.17</a> resinfo
</h2>
<pre class="LyX-Code">
resinfo<a class="IndexReference" name="entry-parmresinfo-0" href="#index-parmresinfo">↓</a> [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] [&lt;mask&gt;] [short [maxwidth &lt;#res&gt;]]
        [out &lt;file&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;]</span> Name/tag or index of topology. Default is first loaded topology.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Mask selecting residues to print info for.
</div>
<div class="Description">
<span class="Description-entry">[short]</span> Use a short 1 character residue name format
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[maxwidth &lt;#res&gt;]</span> Max # of residues to print in one line (default 50).
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to print to (default STDOUT).
</div>

</li>

</ul>
<div class="Unindented">
Print residue information for atoms in &lt;mask&gt; for selected topology (first loaded topology by default) with format:
</div>
<pre class="LyX-Code">
#Res  Name First  Last Natom #Orig #Mol C
</pre>
<div class="Unindented">
where <tt>#Res</tt> is the residue number, <tt>Name</tt> is the residue name, <tt>First</tt> and <tt>Last</tt> are the first and last atom numbers of the residue, <tt>Natom</tt> is the total number of atoms in the residue, <tt>#Orig</tt> is the original residue number (in PDB files), #Mol is the molecule number, and C is the chain ID.
</div>
<div class="Indented">
If <b>short</b> is specified then residues will be printed out in a condensed format. Each residue name will be shortened to 1 character, and residues are printed out in groups of 10, 5 groups to a line, with each line beginning with a residue number, e.g.
</div>
<pre class="LyX-Code">
&gt; resinfo short 4 
1        MGFLAGKKIL ITGLLSNKSI AYGIAKAMHR EGAELAFTYV GQFKDRVEKL 
51       CAEFNPAAVL PCDVISDQEI KDLFVELGKV WDGLDAIVHS IAFAPRDQLE 
</pre>
<div class="Unindented">
If the 1 character name for a residue is unknown it will be shown as the first letter of the residue name in lower-case.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.18">9.18</a> scaledihedralk
</h2>
<pre class="LyX-Code">
scaledihedralk [parm &lt;name&gt; | parmindex &lt;#&gt;] &lt;scale factor&gt; [&lt;mask&gt; [useall]]
</pre>
<div class="Unindented">
Scale dihedral force constants for dihderals selected by &lt;mask&gt; for specified topology. If <b>useall</b> is specified all atoms in &lt;mask&gt; must be present to select a dihedral, otherwise any atom in &lt;mask&gt; will selected a dihedral.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.19">9.19</a> solvent<a class="Label" name="subsec:cpptraj_solvent"> </a>
</h2>
<pre class="LyX-Code">
solvent [parm &lt;name&gt; | parmindex &lt;#&gt; | &lt;#&gt;] { &lt;mask&gt; | none }
</pre>
<div class="Unindented">
Set solvent for selected topology (first loaded topology by default) based on <b>&lt;mask&gt;</b>, or set nothing as solvent if <b>none</b> is specified.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-9.20">9.20</a> updateparameters
</h2>
<pre class="LyX-Code">
parm &lt;name&gt; | parmindex &lt;#&gt; setname &lt;parm set&gt;
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;name&gt; | parmindex &lt;#&gt;</span> Topology to update.
</div>
<div class="Description">
<span class="Description-entry">setname &lt;parm set&gt;</span> Topology or parameter data set containing parameters to use.
</div>

</li>

</ul>
<div class="Unindented">
<i>NOTE: This command is provided for convenience only. For editing topology files, ParmEd is a much better alternative.</i>
</div>
<div class="Indented">
Update parameters in specified topology with those from &lt;parm set&gt;. &lt;parm set&gt; can either be a parameter set or a topology. If a parameter from &lt;parm set&gt; does not exist in the topology it will be added.
</div>
<div class="Indented">
For example, to modify parameters in a topology file named lys.parm7 with those from parameter file kcx.str:
</div>
<pre class="LyX-Code">
# Read Topology to modify
parm lys.parm7
# Read CHARMM parameters
readdata kcx.str as charmmrtfprm name MyParm
# Update parameters in Topology with those from kcx.str
updateparameters parmindex 0 setname MyParm
# Write out the updated Topology
parmwrite out lys.kcx.parm7 
</pre>
<h1 class="Section">
<a class="toc" name="toc-Section-10">10</a> Trajectory File Commands
</h1>
<div class="Unindented">
These commands control the reading and writing of trajectory files. There are three trajectory types in <i>cpptraj</i>: input, output, and reference. In <i>cpptraj</i>, trajectories are always associated with a topology file. If a topology file is not specified, a trajectory file will be associated with the first topology file loaded by default (this is true for both input and output trajectories.
</div>
<div class="Indented">
Cpptraj currently understands the following trajectory file formats:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top" style="width: 25%;">
<b>Format</b>
</td>
<td align="center" valign="top" style="width: 25%;">
<b>Keyword(s)</b>
</td>
<td align="center" valign="top" style="width: 25%;">
<b>Extension</b>
</td>
<td align="center" valign="top" style="width: 25%;">
<b>Notes</b>
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Amber Trajectory
</td>
<td align="center" valign="top" style="width: 25%;">
crd
</td>
<td align="center" valign="top" style="width: 25%;">
.crd
</td>
<td align="center" valign="top" style="width: 25%;">
Default format if keywords/extensions not recognized.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Amber NetCDF
</td>
<td align="center" valign="top" style="width: 25%;">
cdf, netcdf
</td>
<td align="center" valign="top" style="width: 25%;">
.nc
</td>
<td align="center" valign="top" style="width: 25%;">
No compression.
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Amber Restart
</td>
<td align="center" valign="top" style="width: 25%;">
restart
</td>
<td align="center" valign="top" style="width: 25%;">
.rst7, .rst
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Amber NetCDF Restart
</td>
<td align="center" valign="top" style="width: 25%;">
ncrestart, restartnc
</td>
<td align="center" valign="top" style="width: 25%;">
.ncrst
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Charmm &ldquo;DCD&rdquo; Trajectory
</td>
<td align="center" valign="top" style="width: 25%;">
dcd, charmm
</td>
<td align="center" valign="top" style="width: 25%;">
.dcd
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Charmm COORdinates
</td>
<td align="center" valign="top" style="width: 25%;">
cor
</td>
<td align="center" valign="top" style="width: 25%;">
.cor
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Charmm Restart
</td>
<td align="center" valign="top" style="width: 25%;">
charmmres
</td>
<td align="center" valign="top" style="width: 25%;">
.res
</td>
<td align="center" valign="top" style="width: 25%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
PDB
</td>
<td align="center" valign="top" style="width: 25%;">
pdb
</td>
<td align="center" valign="top" style="width: 25%;">
.pdb
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Mol2
</td>
<td align="center" valign="top" style="width: 25%;">
mol2
</td>
<td align="center" valign="top" style="width: 25%;">
.mol2
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Scripps Binpos
</td>
<td align="center" valign="top" style="width: 25%;">
binpos
</td>
<td align="center" valign="top" style="width: 25%;">
.binpos
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Gromacs TRR
</td>
<td align="center" valign="top" style="width: 25%;">
trr
</td>
<td align="center" valign="top" style="width: 25%;">
.trr
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Gromacs GRO
</td>
<td align="center" valign="top" style="width: 25%;">
gro
</td>
<td align="center" valign="top" style="width: 25%;">
.gro
</td>
<td align="center" valign="top" style="width: 25%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Gromacs XTC
</td>
<td align="center" valign="top" style="width: 25%;">
xtc
</td>
<td align="center" valign="top" style="width: 25%;">
.xtc
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Gromacs TNG
</td>
<td align="center" valign="top" style="width: 25%;">
tng
</td>
<td align="center" valign="top" style="width: 25%;">
.tng
</td>
<td align="center" valign="top" style="width: 25%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
CIF
</td>
<td align="center" valign="top" style="width: 25%;">
cif
</td>
<td align="center" valign="top" style="width: 25%;">
.cif
</td>
<td align="center" valign="top" style="width: 25%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Tinker ARC
</td>
<td align="center" valign="top" style="width: 25%;">
arc
</td>
<td align="center" valign="top" style="width: 25%;">
.arc
</td>
<td align="center" valign="top" style="width: 25%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
SQM Input
</td>
<td align="center" valign="top" style="width: 25%;">
sqm
</td>
<td align="center" valign="top" style="width: 25%;">
.sqm
</td>
<td align="center" valign="top" style="width: 25%;">
Write Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
SDF
</td>
<td align="center" valign="top" style="width: 25%;">
sdf
</td>
<td align="center" valign="top" style="width: 25%;">
.sdf
</td>
<td align="center" valign="top" style="width: 25%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
XYZ
</td>
<td align="center" valign="top" style="width: 25%;">
xyz
</td>
<td align="center" valign="top" style="width: 25%;">
.xyz
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
Desmond DTR (Anton)
</td>
<td align="center" valign="top" style="width: 25%;">
dtr
</td>
<td align="center" valign="top" style="width: 25%;">
.dtr
</td>
<td align="center" valign="top" style="width: 25%;">
Read Only
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
LMOD Conflib
</td>
<td align="center" valign="top" style="width: 25%;">
conflib
</td>
<td align="center" valign="top" style="width: 25%;">
.conflib
</td>
<td align="center" valign="top" style="width: 25%;">
Read Only, Detection by extension
</td>

</tr>

</table>
<a class="Label" name="tab:cpptraj-trajformats"> </a>
</div>

</div>
<div class="Indented">
The following trajectory-related commands are available:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
<b>Command</b>
</td>
<td align="center" valign="top">
<b>Description</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
ensemble
</td>
<td align="center" valign="top">
Set up a trajectory ensemble for reading during a run.
</td>

</tr>
<tr>
<td align="center" valign="top">
ensemblesize
</td>
<td align="center" valign="top">
(MPI only) specify number of members expected in subsequent <b><i>ensemble</i></b> commands.
</td>

</tr>
<tr>
<td align="center" valign="top">
reference
</td>
<td align="center" valign="top">
Read in a reference structure.
</td>

</tr>
<tr>
<td align="center" valign="top">
trajin
</td>
<td align="center" valign="top">
Set up a trajectory for reading during a Run.
</td>

</tr>
<tr>
<td align="center" valign="top">
trajout
</td>
<td align="center" valign="top">
Set up an output trajectory or ensemble for writing during a Run.
</td>

</tr>

</table>

</div>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-10.1">10.1</a> ensemble
</h2>
<pre class="LyX-Code">
ensemble &lt;file0&gt; {[&lt;start&gt;] [&lt;stop&gt; | last] [offset]} | lastframe
         [parm &lt;parmfile / tag&gt; | parmindex &lt;#&gt;]
         [trajnames &lt;file1&gt;,&lt;file2&gt;,...,&lt;fileN&gt;
         [remlog &lt;remlogfile&gt; [nstlim &lt;nstlim&gt; ntwx &lt;ntwx&gt;]]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;file0&gt;</span> Lowest replica filename.
</div>
<div class="Description">
<span class="Description-entry">[&lt;start&gt;]</span> Frame to begin reading ensemble at (default 1).
</div>
<div class="Description">
<span class="Description-entry">[&lt;stop&gt; | last]</span> Frame to stop reading ensemble at; if not specified or ’last’ specified, end of trajectories.
</div>
<div class="Description">
<span class="Description-entry">[&lt;offset&gt;]</span> Offset for reading in trajectory frames (default 1).
</div>
<div class="Description">
<span class="Description-entry">[lastframe]</span> Select only the final frame of the trajectories.
</div>
<div class="Description">
<span class="Description-entry">[parm &lt;parmfile&gt;]</span> Topology filename/tag to associate with trajectories (default first topology).
</div>
<div class="Description">
<span class="Description-entry">[parmindex &lt;#&gt;]</span> Index of Topology to associate with trajectories (default 0, first topology).
</div>
<div class="Description">
<span class="Description-entry">[trajnames &lt;file1&gt;,...,&lt;fileN&gt;]</span> Do not automatically search for additional replica trajectories; use comma-separated list of trajectory names.
</div>
<div class="Description">
<span class="Description-entry">[remlog &lt;remlogfile&gt;]</span> For H-REMD trajectories only, use specified REMD log file to sort trajectories by coordinate index (instead of by Hamiltonian).\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[nstlim &lt;nstlim&gt; ntwx &lt;ntwx&gt;]</span> If trajectory and REMD log were not written at the same rate, these are the values for nstlim (steps between each exchange) and ntwx (steps between trajectory write) used in the REMD simulation.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Read in and process trajectories as an ensemble. Similar to ’<b><i>trajin</i></b> remdtraj’, except instead of processing one frame at a target temperature, process all frames. This means that action and trajout commands apply to the entire ensemble; note however that not all actions currently function in <b><i>’ensemble’</i></b> mode. For example, to read in a replica ensemble, convert it to temperature trajectories, and calculate a distance at each temperature:
</div>
<pre class="LyX-Code">
parm ala2.99sb.mbondi2.parm7
ensemble rem.crd.000 trajnames rem.crd.001,rem.crd.002,rem.crd.003  
trajout temp.crd
distance d1 out d1.ensemble.dat @1 @21
</pre>
<div class="Unindented">
This will output 4 temperature trajectories named ’temp.crd.X’, where X ranges from 0 to 3 with 0 corresponding to the lowest temperature, and ’d1.ensemble.dat’ containing 4 columns, each corresponding to a temperature. If run with MPI, data will be written to separate files named ’d1.ensemble.dat.X’, similar to the output trajectories.
</div>
<div class="Indented">
Note that in parallel (i.e. MPI) users should specify the <b><i>ensemblesize</i></b> command prior to <b><i>ensemble</i></b> in order to improve set up efficiency.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-10.2">10.2</a> ensemblesize
</h2>
<pre class="LyX-Code">
ensemblesize &lt;#&gt;
</pre>
<div class="Unindented">
This command is MPI only. It is used to set the expected number of members in any subsequent <b><i>ensemble</i></b> command, which dramatically improves set up efficiency.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-10.3">10.3</a> reference<a class="Label" name="subsec:cpptraj_reference"> </a>
</h2>
<pre class="LyX-Code">
reference<a class="IndexReference" name="entry-reference-0" href="#index-reference">↓</a> &lt;name&gt; [&lt;frame#&gt;] [&lt;mask&gt;] ([tag]) [lastframe] [crdset]
          [parm &lt;parmfile / tag&gt; | parmindex &lt;#&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> File name (or COORDS set name if ’crdset’ specified) to read in as reference; any trajectory recognized by ’trajin’ can be used.
</div>
<div class="Description">
<span class="Description-entry">[&lt;frame#&gt;]</span> Frame number to use (default 1).
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Only load atoms corresponding to &lt;mask&gt; from reference.
</div>
<div class="Description">
<span class="Description-entry">([tag])</span> Tag to give this reference file, e.g. &ldquo;[MyRef]&rdquo;; BRACKETS MUST BE INCLUDED.
</div>
<div class="Description">
<span class="Description-entry">[lastframe]</span> Use last frame of reference.
</div>
<div class="Description">
<span class="Description-entry">[crdset]</span> Use for COORDS data set named &lt;name&gt; instead of file.
</div>
<div class="Description">
<span class="Description-entry">[parm &lt;parmfile/tag&gt;]</span> Topology filename/tag to associate with reference (default first topology).
</div>
<div class="Description">
<span class="Description-entry">[parmindex &lt;#&gt;]</span> Index of Topology to associate with reference (default 0, first topology).
</div>

</li>

</ul>
<div class="Unindented">
Use specified trajectory as reference coordinates. For trajectories with multiple frames, the first frame is used if a specific frame is not specified. An optional tag can be given (bounded in brackets) which can then be used in place of the name (see <a class="Reference" href="#subsec:Parameter/Reference-Tagging">3.4 on page 1↑</a> for examples of how to use tags). If desired, an atom mask can be used to read in only specified atoms from a reference.
</div>
<div class="Indented">
Reference coordinates are now considered COORDS data sets and can be used anywhere a COORDS data set could, which allows reference structures to be manipulated once they are loaded. For example, a reference structure could be centered on the origin like so:
</div>
<pre class="LyX-Code">
reference tz2.rst7 [MyRef]
crdaction [MyRef] center origin
</pre>
<div class="Unindented">
Note that the ’average’ keyword has been deprecated for reference. If desired, an averaged reference COORDS data set can be created from a trajectory using the ’average’ command like so:
</div>
<pre class="LyX-Code">
parm myparm.parm7
trajin mytraj.nc
rms first :1-12
average crdset RefAvg
run
rms ToAvg reference :1-12 out ToAvg.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-10.4">10.4</a> trajin <a class="Label" name="subsec:cpptraj_trajin"> </a>
</h2>
<pre class="LyX-Code">
trajin<a class="IndexReference" name="entry-trajin-0" href="#index-trajin">↓</a> &lt;filename&gt; {[&lt;start&gt; [&lt;stop&gt; | last] [&lt;offset&gt;]]} | lastframe
       [parm &lt;parmfile / tag&gt; | parmindex &lt;#&gt;]
       [mdvel &lt;velocities&gt;] [mdfrc &lt;forces&gt;]
       [as &lt;format keyword&gt;] [ &lt;Format Options&gt; ]
       [ remdtraj {remdtrajtemp &lt;Temperature&gt; | remdtrajidx &lt;idx1,idx2,...&gt; 
                   | remdtrajvalues &lt;value1,value2,...&gt;}
                  [trajnames &lt;file1&gt;,&lt;file2&gt;,...,&lt;fileN&gt;] ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;filename&gt;</span> Trajectory file to read in.
</div>
<div class="Description">
<span class="Description-entry">[&lt;start&gt;]</span> Frame to begin reading at (default 1). If a negative value is given it means &ldquo;&lt;start&gt; frames before &lt;stop&gt;&rdquo;.
</div>
<div class="Description">
<span class="Description-entry">[&lt;stop&gt; | last]</span> Frame to stop reading at; if not specified or ’last’ specified, end of trajectory.
</div>
<div class="Description">
<span class="Description-entry">[&lt;offset&gt;]</span> Offset for reading in trajectory frames (default 1).
</div>
<div class="Description">
<span class="Description-entry">[lastframe]</span> Select only the final frame of the trajectory.
</div>
<div class="Description">
<span class="Description-entry">[parm &lt;parmfile/tag&gt;]</span> Topology filename/tag to associate with trajectory (default first topology).
</div>
<div class="Description">
<span class="Description-entry">[parmindex &lt;#&gt;]</span> Index of Topology to associate with trajectory (default 0, first topology).
</div>
<div class="Description">
<span class="Description-entry">[mdvel &lt;velocities&gt;]</span> Use velocities from specified file.
</div>
<div class="Description">
<span class="Description-entry">[mdfrc &lt;forces&gt;]</span> Use forces from specified file.
</div>
<div class="Description">
<span class="Description-entry">[as &lt;format keyword&gt;]</span> Force file to be read as specified format; overrides file autodetection.
</div>
<div class="Description">
<span class="Description-entry">[&lt;Format Options&gt;]</span> See below.
</div>
<div class="Description">
<span class="Description-entry">[remdtraj]</span> Read &lt;filename&gt; as the first replica in a group of replica trajectories.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">remdtrajtemp &lt;Temperature&gt; | remdtrajidx &lt;idx1,idx2,...&gt;</span> Use frames at &lt;Temperature&gt; (for temperature replica trajectories) or index &lt;idx1,idx2,...&gt; (for Hamiltonian replica trajectories); For Multidimensional REMD simulations, multiple values are comma-separated.
</div>
<div class="Description">
<span class="Description-entry">remdtrajvalues &lt;value1,value2,...&gt;</span> Use frames at &lt;value1,value2,...&gt; (for Multidimensional REMD trajectories). Each value may correspond to either temperature, pH, Redox Potential or Hamiltonian index. The values need to be entered in the same order as the dimensions in the Multidimensional REMD simulation. For example, for T,pH-REMD value1 would correspond to a temperature and value2 to a pH. In the command, the values are comma-separated.
</div>
<div class="Description">
<span class="Description-entry">[trajnames &lt;file1&gt;,...,&lt;fileN&gt;]</span> Do not automatically search for additional replica trajectories; use comma-separated list of trajectory names.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Read in trajectory specified by filename. See page <a class="Reference" href="#tab:cpptraj-trajformats">1↑</a> for currently recognized trajectory file formats. If just the &lt;start&gt; argument is given, all frames from &lt;start&gt; to the last frame of the trajectory will be read. To read in a trajectory with offsets where the last frame # is not known, specify the <b>last</b> keyword instead of a &lt;stop&gt; argument, e.g.
</div>
<pre class="LyX-Code">
trajin Test1.crd 10 last 2
</pre>
<div class="Unindented">
This will process Test1.crd from frame 10 to the last frame, skipping by 2 frames. To explicitly select only the last frame, specify the <b>lastframe</b> keyword:
</div>
<pre class="LyX-Code">
trajin Test1.crd lastframe
</pre>
<div class="Unindented">
Here is an example of loading in multiple trajectories which have difference topology files:
</div>
<pre class="LyX-Code">
parm top0.parm7
parm top1.parm7
parm top2.parm7 [top2]
parm top3.parm7
trajin Test0.crd
trajin Test1.crd parm top1.parm7
trajin Test2.crd parm [top2]
trajin Test3.crd parmindex 3
</pre>
<div class="Unindented">
Test0.crd is associated with top0.parm7; since no parm was specified it defaulted to the first parm read in. Test1.crd was associated with top1.parm7 by filename, Test2.crd was associated with top2.parm7 by its tag, and finally Test3.crd was associated with top3.parm7 by its index (based on the order it was read in).
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--14"></a>Replica Trajectory Processing
</h3>
<div class="Unindented">
If the <b>remdtraj</b> keyword is specified the trajectory is treated as belonging to the lowest # replica of a group of REMD trajectories. The remaining replicas can be either automatically detected by following a naming convention of &lt;REMDFILENAME&gt;.X, where X is the replica number, or explicitly specified in a comma-separated list following the <b>trajnames</b> keyword. All trajectories will be processed at the same time, but only frames with a temperature matching the one specified by <b>remdtrajtemp</b> or <b>remdtrajidx</b> will be processed. For example, to process replica trajectories rem.001, rem.002, rem.003, and rem.004, grabbing only the frames at temperature 300.0 (assuming that this is a temperature in the ensemble):
</div>
<pre class="LyX-Code">
trajin rem.001 remdtraj remdtrajtemp 300
</pre>
<div class="Unindented">
or
</div>
<pre class="LyX-Code">
trajin rem.001 remdtraj remdtrajtemp 300 trajnames rem.002,rem.003,rem.004
</pre>
<div class="Unindented">
Note that the <b>remdout</b> keyword is deprecated. For this functionality see the <b>ensemble</b> keyword.
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.4.1">10.4.1</a> <i>Options for Amber NetCDF, Amber NC Restart, Amber Restart:</i>
</h3>
<pre class="LyX-Code">
  [usevelascoords] [usefrcascoords]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">usevelascoords</span> Read in velocities in place of coordinates if present.
</div>
<div class="Description">
<span class="Description-entry">usefrcascoords</span> Read in forces in place of coordinates if present.
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.4.2">10.4.2</a> <i>Options for CHARMM DCD:</i>
</h3>
<pre class="LyX-Code">
[{ucell | shape}]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">ucell</span> Force reading of box information as unit cell (for e.g. NAMD DCD trajectories).
</div>
<div class="Description">
<span class="Description-entry">shape</span> Force reading of box information as shape matrix.
</div>

</li>

</ul>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-10.5">10.5</a> trajout<a class="Label" name="subsec:cpptraj-trajout"> </a>
</h2>
<pre class="LyX-Code">
trajout<a class="IndexReference" name="entry-trajout-0" href="#index-trajout">↓</a> &lt;filename&gt; [&lt;format&gt;] [append] [nobox] [novelocity]
        [notemperature] [notime] [noforce] [noreplicadim]
        [parm &lt;parmfile&gt; | parmindex &lt;#&gt;] [onlyframes &lt;range&gt;] [title &lt;title&gt;]
        [onlymembers &lt;memberlist&gt;]
        [start &lt;start&gt;] [stop &lt;stop&gt;] [offset &lt;offset&gt;]
        [ &lt;Format Options&gt; ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;filename&gt;</span> Trajectory file to write to.
</div>
<div class="Description">
<span class="Description-entry">[&lt;format&gt;]</span> Keyword specifying output format (see Table <a class="Reference" href="#tab:cpptraj-trajformats"> on page 1↑</a>). If not specified format will be determined from extension, otherwise default to Amber trajectory.
</div>
<div class="Description">
<span class="Description-entry">[append]</span> If &lt;filename&gt; exists, frames will be appended to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[nobox]</span> Do not write box coordinates to trajectory.
</div>
<div class="Description">
<span class="Description-entry">[novelocity]</span> Do not write velocities to trajectory.
</div>
<div class="Description">
<span class="Description-entry">[notemperature]</span> Do not write temperature to trajectory.
</div>
<div class="Description">
<span class="Description-entry">[notime]</span> Do not write time to trajectory.
</div>
<div class="Description">
<span class="Description-entry">[noreplicadim]</span> Do not write replica dimensions to trajectory.
</div>
<div class="Description">
<span class="Description-entry">[parm &lt;parmfile&gt;]</span> Topology filename/tag to associate with trajectory (default first topology).
</div>
<div class="Description">
<span class="Description-entry">[parmindex &lt;#&gt;]</span> Index of Topology to associate with trajectory (default 0, first topology).
</div>
<div class="Description">
<span class="Description-entry">[onlyframes &lt;range&gt;]</span> Write only the specified input frames to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[title &lt;title&gt;]</span> Output trajectory title.
</div>
<div class="Description">
<span class="Description-entry">[onlymembers &lt;memberlist&gt;]</span> Ensemble processing only; only write from specified members (starting from 0).
</div>
<div class="Description">
<span class="Description-entry">[start &lt;start&gt;]</span> Begin output at frame &lt;start&gt; (1 by default).
</div>
<div class="Description">
<span class="Description-entry">[stop &lt;stop&gt;]</span> End output at frame &lt;stop&gt; (last frame by default).
</div>
<div class="Description">
<span class="Description-entry">[offset &lt;offset&gt;]</span> Skip &lt;offset&gt; frames between each output (1 by default).
</div>

</li>

</ul>
<div class="Unindented">
During a run, write frames to trajectory specified by filename in specified file format (Amber trajectory if none specified) after all Action processing has occurred. To write out trajectories within the Action queue see the outtraj Action (<a class="Reference" href="#subsec:cpptraj_outtraj">11.54 on page 1↓</a>). See page <a class="Reference" href="#tab:cpptraj-trajformats">1↑</a> for currently recognized output trajectory formats and their associated keyword(s). Note that now the file type can be determined from the output extension if not specified by a keyword. Multiple output trajectories of any format can be specified. 
</div>
<div class="Indented">
<b>Frames will be written to the output trajectory when the parameter file being processed matches the parameter file the output trajectory was set up with.</b> So given the input:
</div>
<pre class="LyX-Code">
parm top0.parm7
parm top1.parm7 [top1]
trajin input0.crd
trajin input1.crd parm [top1]
trajout output.crd parm [top1]
</pre>
<div class="Unindented">
only frames read in from input1.crd (which is associated with top1.parm7) will be written to output.crd. The trajectory input0.crd is associated with top0.parm7; since no output trajectory is associated with top0.parm7 no frames will be written when processing top0.parm7/input0.crd.
</div>
<div class="Indented">
If <b>onlyframes</b> is specified, only input frames matching the specified range will be written out. For example, given the input:
</div>
<pre class="LyX-Code">
trajin input.crd 1 10
trajout output.crd onlyframes 2,5-7
</pre>
<div class="Unindented">
only frames 2, 5, 6, and 7 from input.crd will be written to output.crd.
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.1">10.5.1</a> <i>Options for pdb format</i>
</h3>
<pre class="LyX-Code">
  [dumpq | parse | vdw] [pdbres] [pdbatom]
  [pdbv3] [teradvance] [terbyres | pdbter | noter] 
  [model | multi] [chainid &lt;ID&gt;] [sg &lt;group&gt;]
  [include_ep] [conect] [keepext] [usecol21]
  [bfacdefault &lt;#&gt;] [occdefault &lt;#&gt;]
  [bfacdata &lt;set&gt;] [occdata &lt;set&gt;] [bfacbyres] [occbyres]
  [bfacscale] [occscale] [bfacmax &lt;max&gt;] [occmax &lt;max&gt;]
  [adpdata &lt;set&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">dumpq</span> PQR format; write charges (in units of e-) and GB radii to occupancy and B-factor columns respectively.
</div>
<div class="Description">
<span class="Description-entry">parse</span> PQR format; write charges and PARSE radii to occupancy/B-factor columns.
</div>
<div class="Description">
<span class="Description-entry">vdw</span> PQR format; write charges and vdW radii to occupancy/B-factor columns.
</div>
<div class="Description">
<span class="Description-entry">pdbres</span> Use PDB V3 residue names. Will write a default chain ID (’Z’) for each residue if the corresponding topology does not have chain ID information.
</div>
<div class="Description">
<span class="Description-entry">pdbatom</span> Use PDB V3 atom names.
</div>
<div class="Description">
<span class="Description-entry">pdbv3</span> Use PDB V3 residue/atom names. Same as specifying ’pdbres’ and ’pdbatom’.
</div>
<div class="Description">
<span class="Description-entry">topresnum</span> Use topology residue numbers; otherwise use original residue numbers.
</div>
<div class="Description">
<span class="Description-entry">teradvance</span> Increment record (atom) number for TER records (not done by default).
</div>
<div class="Description">
<span class="Description-entry">terbyres</span> Print TER cards based on residue sequence instead of molecules.
</div>
<div class="Description">
<span class="Description-entry">pdbter</span> Print TER cards according to original PDB TER (if available).
</div>
<div class="Description">
<span class="Description-entry">noter</span> Do not write TER cards.
</div>
<div class="Description">
<span class="Description-entry">model</span> (Default) Frames will be written to a single PDB file separated by MODEL/ENDMDL keywords.
</div>
<div class="Description">
<span class="Description-entry">multi</span> Each frame will be written to a separate file with the frame # appended to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">chainid &lt;ID&gt;</span> Write PDB file with chain ID &lt;ID&gt;.
</div>
<div class="Description">
<span class="Description-entry">sg &lt;group&gt;</span> Space group for CRYST1 record; only used if box coordinates written.
</div>
<div class="Description">
<span class="Description-entry">include_ep</span> Include extra points.
</div>
<div class="Description">
<span class="Description-entry">conect</span> Write CONECT records for all bonds.
</div>
<div class="Description">
<span class="Description-entry">keepext</span> Keep filename extension; write ’&lt;name&gt;.&lt;num&gt;.&lt;ext&gt;’ instead (implies ’multi’).
</div>
<div class="Description">
<span class="Description-entry">usecol21</span> Use column 21 for 4-letter residue names.
</div>
<div class="Description">
<span class="Description-entry">bfacdefault &lt;#&gt;</span> Default value to use in B-factor column (default 0.0).
</div>
<div class="Description">
<span class="Description-entry">occdefault &lt;#&gt;</span> Default value to use in occupancy column (default 1.0).
</div>
<div class="Description">
<span class="Description-entry">bfacdata &lt;set&gt;</span> Use data in &lt;set&gt; for B-factor column.
</div>
<div class="Description">
<span class="Description-entry">occdata &lt;set&gt;</span> Use data in &lt;set&gt; for occupancy column.
</div>
<div class="Description">
<span class="Description-entry">bfacbyres</span> If specified assume X values in B-factor data set are residue numbers.
</div>
<div class="Description">
<span class="Description-entry">occbyres</span> If specified assume X values in occupancy data set are residue numbers.
</div>
<div class="Description">
<span class="Description-entry">bfacscale</span> If specified scale values in B-factor column between 0 and &lt;bfacmax&gt;.
</div>
<div class="Description">
<span class="Description-entry">occscale</span> If specified scale values in occupancy column between 0 and &lt;occmax&gt;.
</div>
<div class="Description">
<span class="Description-entry">bfacmax &lt;max&gt;</span> Max value for bfacscale.
</div>
<div class="Description">
<span class="Description-entry">occmax &lt;max&gt;</span> Max value for occscale.
</div>
<div class="Description">
<span class="Description-entry">adpdata &lt;set&gt;</span> Use data in &lt;set&gt; (e.g. from the <b><i>atomicfluct</i></b> command, <a class="Reference" href="#subsec:cpptraj_atomicfluct"> on page 1↓</a>) for anisotropic B-factors.
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.2">10.5.2</a> <i>Options for Amber ASCII format:</i>
</h3>
<pre class="LyX-Code">
  [remdtraj] [highprecision] [mdvel|mdfrc]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">remdtraj</span> Write REMD header to trajectory that includes temperature: ’REMD &lt;Replica&gt; &lt;Step&gt; &lt;Total_Steps&gt; &lt;Temperature&gt;’. Since <i>cpptraj</i> has no concept of replica number, 0 is printed for &lt;Replica&gt;. &lt;Step&gt; and &lt;Total_Steps&gt; are set to the current frame #.
</div>
<div class="Description">
<span class="Description-entry">highprecision</span> (EXPERT USE ONLY) Write with 8.6 precision instead of 8.3. Note that since the width does not change, the precision of large coords may be lower than 6.
</div>
<div class="Description">
<span class="Description-entry">mdvel</span> Write velocities instead of coordinates.
</div>
<div class="Description">
<span class="Description-entry">mdfrc</span> Write forces instead of coordinates.
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.3">10.5.3</a> <i>Options for Amber NetCDF format:</i>
</h3>
<pre class="LyX-Code">
  [remdtraj] [mdvel] [mdfrc] [mdcrd]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">remdtraj</span> Write replica temperature to trajectory.
</div>
<div class="Description">
<span class="Description-entry">mdvel</span> Write only velocity information in trajectory.
</div>
<div class="Description">
<span class="Description-entry">mdfrc</span> Write only force information in trajectory.
</div>
<div class="Description">
<span class="Description-entry">mdcrd</span> Write coordinates to trajectory (only required with mdvel/mdfrc).
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.4">10.5.4</a> <i>Options for Amber Restart/NetCDF Restart format:</i>
</h3>
<pre class="LyX-Code">
  [remdtraj] [novelocity] [notime] [time0 &lt;initial time&gt;] [dt &lt;timestep&gt;] [keepext]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">remdtraj</span> Write replica temperature to restart. Note that this will automatically include time in the restart file (see the <b>time0<tt></tt></b> keyword).
</div>
<div class="Description">
<span class="Description-entry">time0 &lt;initial time&gt;</span> Time for first frame (default 1.0).
</div>
<div class="Description">
<span class="Description-entry">dt &lt;timestep&gt;</span> Time step between frames (default 1.0). Time is calculated as t=(time0+frame)*dt.
</div>
<div class="Description">
<span class="Description-entry">keepext</span> Keep filename extension; write ’&lt;name&gt;.&lt;num&gt;.&lt;ext&gt;’ instead.
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.5">10.5.5</a> <i>Options for CHARMM COORdinates:</i>
</h3>
<pre class="LyX-Code">
[keepext] [ext] [segid &lt;segid&gt;] [segmask &lt;mask&gt; &lt;segid&gt; ...]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">keepext</span> Keep filename extension; write ’&lt;name&gt;.&lt;num&gt;.&lt;ext&gt;’
</div>
<div class="Description">
<span class="Description-entry">ext</span> Use ’extended’ format (default when &gt; 99999 atoms).
</div>
<div class="Description">
<span class="Description-entry">segid &lt;segid&gt;</span> Use &lt;segid&gt; as segment ID for all atoms.
</div>
<div class="Description">
<span class="Description-entry">segmask &lt;mask&gt; &lt;segid&gt;</span> Use &lt;segid&gt; as segment ID for atoms selected by &lt;mask&gt;. Can be specified more than once.
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.6">10.5.6</a> <i>Options for CHARMM DCD:</i>
</h3>
<pre class="LyX-Code">
[x64] [ucell] [veltraj]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">x64</span> Use 8 byte block size (default 4 bytes).
</div>
<div class="Description">
<span class="Description-entry">ucell</span> Write older (v21) format trajectory that stores unit cell params instead of shape matrix.
</div>
<div class="Description">
<span class="Description-entry">veltraj</span> Write velocity trajectory instead of coordinates.
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.7">10.5.7</a> <i>Options for GROMACS TRX/XTC format:</i>
</h3>
<pre class="LyX-Code">
[dt &lt;time step&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">dt</span> Time step tp multiply set numbers by (default 1.0). Ignored if time already present.
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.8">10.5.8</a> <i>Options for mol2 format:</i>
</h3>
<pre class="LyX-Code">
  [single | multi] [sybyltype] [sybylatom &lt;file&gt;] [sybylbond &lt;file&gt;] [keepext]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">single</span> (Default) Frames will be written to a single Mol2 file separated by MOLECULE keywords.
</div>
<div class="Description">
<span class="Description-entry">multi</span> Each frame will be written to a separate file with the frame # appended to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">sybyltype</span> Convert Amber atom types (if present) to SYBYL types. Requires $AMBERHOME is set.
</div>
<div class="Description">
<span class="Description-entry">sybylatom</span> File containing Amber to SYBYL atom type correspondance (optional).
</div>
<div class="Description">
<span class="Description-entry">sybylbond</span> File containing Amber to SYBYL bond type correspondance (optional).
</div>
<div class="Description">
<span class="Description-entry">keepext</span> Keep filename extension; write ’&lt;name&gt;.&lt;num&gt;.&lt;ext&gt;’ instead (implies ’multi’).
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.9">10.5.9</a> Options for SQM input format:
</h3>
<pre class="LyX-Code">
  [charge &lt;c&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">charge &lt;c&gt;</span> Set total integer charge. If not specified it will be calculated from atomic charges.
</div>

</li>

</ul>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-10.5.10">10.5.10</a> Options for XYZ format:
</h3>
<pre class="LyX-Code">
ftype {atomxyz|xyz} titletype {none|single|perframe} width &lt;#&gt; prec &lt;#&gt;
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">ftype {atomxyz|xyz}</span> Choose either ’ATOM X Y Z’ (default) or ’X Y Z’ output format.
</div>
<div class="Description">
<span class="Description-entry">titletype {none|single|perframe}</span> No title, one title (default), or title before every frame.
</div>
<div class="Description">
<span class="Description-entry">width &lt;#&gt;</span> Output format width.
</div>
<div class="Description">
<span class="Description-entry">prec &lt;#&gt;</span> Output format precision.
</div>

</li>

</ul>
<h1 class="Section">
<a class="toc" name="toc-Section-11">11</a> Action Commands<a class="Label" name="sec:cpptraj_actions"> </a>
</h1>
<div class="Unindented">
Actions in <i>cpptraj</i> operate on frames read in by the <b><i>trajin</i></b> or <b><i>ensemble</i></b> commands one at a time and extract derived data, modify the coordinates/topology in some way, or both. Most Actions in <i>cpptraj</i> function exactly the way they do in <i>ptraj</i> and are backwards-compatible. Some Action commands in <i>cpptraj</i> have extra functionality compared to <i>ptraj</i> (such as the per-residue RMSD function of the <b><i>rmsd</i></b> Action, or the ability to write out stripped topologies for visualization in the <b><i>strip</i></b> Action), while other Actions produce slightly different output (like the <b><i>hbond</i></b>/<b><i>secstruct</i></b> Actions).
</div>
<div class="Indented">
Unlike some other command types, when an Action command is issued it is by default added to the Action queue and is not executed until trajectory processing is started (e.g. by a <b><i>run</i></b> or <b><i>go</i></b> command). However, Actions can be executed immediately on COORDS data sets via the <b><i>crdaction</i></b> command (<a class="Reference" href="#subsec:cpptraj_crdaction">7.3 on page 1↑</a>).
</div>
<div class="Indented">
When a frame is modified by an Action, it is modified for every Action that follows them during trajectory processing. For example, given a solvated system with water residues named WAT and the following Action commands:
</div>
<pre class="LyX-Code">
rmsd R1 first :WAT out water-rmsd.dat
strip :WAT
rmsd R2 first :WAT out water-rmsd-2.dat
</pre>
<div class="Unindented">
the first <b><i>rms</i></b> command will be valid, but the second <b><i>rms</i></b> command will not since all residues named WAT are removed from the state by the <b><i>strip</i></b> command.
</div>
<div class="Indented">
Note that for commands which can use a reference mask as well as a target mask (e.g. <b><i>rms</i></b>, <b><i>drmsd</i></b>, <b><i>symmrmsd</i></b>, etc.) there must be a 1 to 1 correspondence between the atoms in each mask, i.e. the <i>number of atoms and the ordering of selected atoms must be the same</i>.
</div>
<div class="Indented">
The following Actions are available. If an Action may modify coordinate/topology information for subsequent Actions it is denoted with an X in the <tt><b>Mod</b></tt> column.\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top" style="width: 25%;">
<b>Command</b>
</td>
<td align="center" valign="top" style="width: 65%;">
<b>Description</b>
</td>
<td align="center" valign="top" style="width: 10%;">
<b>Mod</b>
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
align
</td>
<td align="center" valign="top" style="width: 65%;">
Align structure to a reference.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
angle
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate the angle between three points.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
areapermol
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate area per molecule for molecules in a specified plane.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
atomiccorr
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate average correlation between motions of specified atoms.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
atomicfluct, rmsf
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate root mean square fluctuation of specified atoms/residues.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
atommap
</td>
<td align="center" valign="top" style="width: 65%;">
Attempt to create a map between atoms in molecules with different atom ordering.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
autoimage
</td>
<td align="center" valign="top" style="width: 65%;">
Automatically re-image coordinates.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
average
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate average structure.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
bounds
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate the min/max coordinates for specified atoms. Can be used to create grid data sets.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
box
</td>
<td align="center" valign="top" style="width: 65%;">
Set or overwrite box information for frames.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
center
</td>
<td align="center" valign="top" style="width: 65%;">
Center specified coordinates to box center or onto reference structure.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
check, checkoverlap,
</td>
<td align="center" valign="top" style="width: 65%;">
Check for bad atomic overlaps or bond lengths. 
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
 checkstructure
</td>
<td align="center" valign="top" style="width: 65%;">
Can be used to skip corrupted frames.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
checkchirality
</td>
<td align="center" valign="top" style="width: 65%;">
Report chirality around alpha carbons in amino acids (L, D).
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
closest, closestwaters
</td>
<td align="center" valign="top" style="width: 65%;">
Retain only the specified number of solvent molecules closest to specified solute.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
clusterdihedral
</td>
<td align="center" valign="top" style="width: 65%;">
Assign frames into clusters based on binning of backbone dihedral angles in amino acids.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
contacts
</td>
<td align="center" valign="top" style="width: 65%;">
Older version of <b><i>nativecontacts</i></b>, retained for backwards compatibility.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
createcrd
</td>
<td align="center" valign="top" style="width: 65%;">
Create a COORDS data set from input frames.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
createreservoir
</td>
<td align="center" valign="top" style="width: 65%;">
Create a structure reservoir for use with reservoir REMD simulations.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
density
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate density along a coordinate.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
diffusion
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate translational diffusion of molecules.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
dihedral
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate the dihedral angle using four points.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
<div class="PlainVisible">

</div>
<div class="PlainVisible">
dihrms<br/>
dihedralrms
</div>

</td>
<td align="center" valign="top" style="width: 65%;">
Calculate the RMSD of dihedrals to dihedrals in a reference structure.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
dipole
</td>
<td align="center" valign="top" style="width: 65%;">
Bin dipoles of solvent molecules in 3D grid. Not well tested, may be obsolete.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
distance
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate the distance between two points.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
drms, drmsd
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate the RMSD of distance pairs within selected atoms.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
dssp, secstruct
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate secondary structure content using the DSSP algorithm
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
energy
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate simple bond, angle, dihedral, and non-bonded energy terms (no PME).
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
esander
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate energies using via SANDER; requires compilation with the SANDER API.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
filter
</td>
<td align="center" valign="top" style="width: 65%;">
Filter frames for subsequent Actions using data sets and user defined criteria.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
fixatomorder
</td>
<td align="center" valign="top" style="width: 65%;">
Fix atom ordering so that all atoms in molecules are sequential.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
fiximagedbonds
</td>
<td align="center" valign="top" style="width: 65%;">
Fix bonds which have been split across periodic boundaries by imaging.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
gist
</td>
<td align="center" valign="top" style="width: 65%;">
Perform grid inhomogenous solvation theory.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
grid
</td>
<td align="center" valign="top" style="width: 65%;">
Bin selected atoms on a 3D grid.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
hbond
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate hydrogen bonds using geometric criteria.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
image
</td>
<td align="center" valign="top" style="width: 65%;">
Re-image coordinates. The <b><i>autoimage</i></b> command typically provides better results.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
jcoupling
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate J-coupling values from specified dihedral angles.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
lessplit
</td>
<td align="center" valign="top" style="width: 65%;">
Split/average frames from LES trajectories.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
lie
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate linear interaction energy between user-specified ligand and surroundings.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
lipidorder
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate order parameters for lipids in planar membranes.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
lipidscd
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate lipid order parameters SCD (|&lt;P2&gt;|) for lipid chains. Automatically identifies lipids.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
makestructure
</td>
<td align="center" valign="top" style="width: 65%;">
Modify structure by applying dihedral values to specified residues.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
mask
</td>
<td align="center" valign="top" style="width: 65%;">
Print the results of selection by specified atom mask. Good for distnace-based masks.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
matrix
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate a matrix of the specified type from input coordinates.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
minimage
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate minimum non-self imaged distance between atoms in specified masks.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
molsurf
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate Connolly surface area of specified atoms. Cannot do partial surface areas.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
multidihedral
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate multiple dihedral angles of specified/given types.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
multivector
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate multiple vectors between specified atoms.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
nastruct
</td>
<td align="center" valign="top" style="width: 65%;">
Perform nucelic acid structure analysis.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
nativecontacts
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate native contacts within a region or between two regions using a given reference. 
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">

</td>
<td align="center" valign="top" style="width: 65%;">
Can also be used to get min/max distances between groups of atoms.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
outtraj
</td>
<td align="center" valign="top" style="width: 65%;">
Write frames to a trajectory file within a list of Actions.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
pairdist
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate pair distribution function.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
pairwise
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate pair-wise non-bonded energies.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
principal
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate and optionally align system along principal axes.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
projection
</td>
<td align="center" valign="top" style="width: 65%;">
Project coordinates along given eigenvectors.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
pucker
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate ring pucker using five or six points.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
radgyr, rog
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate radius of gyration (and optionally tensor) for specified atoms.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
radial, rdf
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate radial distribution function.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
randomizeions
</td>
<td align="center" valign="top" style="width: 65%;">
Swap specified ions with randomly selected solvent molecules.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
replicatecell
</td>
<td align="center" valign="top" style="width: 65%;">
Replicate unit cell in specified (or all) directions for specfied atoms and write to trajectory.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
rms, rmsd
</td>
<td align="center" valign="top" style="width: 65%;">
Perform best fit of coordinates to reference and calculate coordinate RMSD.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">

</td>
<td align="center" valign="top" style="width: 65%;">
 Fitting can be disabled.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
rotate
</td>
<td align="center" valign="top" style="width: 65%;">
Rotate the system around X/Y/Z axes, a specified axis, or via given rotation matrices.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
runavg, 
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate the running average of coordinates over specified window size.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
runningaverage
</td>
<td align="center" valign="top" style="width: 65%;">

</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
scale
</td>
<td align="center" valign="top" style="width: 65%;">
Scale coordinates in X/Y/Z directions by specified factors.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
setvelocity
</td>
<td align="center" valign="top" style="width: 65%;">
\strikeout off\uuline off\uwave offSet velocities for specified atoms using Maxwellian distribution<span class="default">\uuline default\uwave default based on given temperature.</span>
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
spam
</td>
<td align="center" valign="top" style="width: 65%;">
SPAM method for estimating relative free energies of waters in hydration shell around proteins.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
stfcdiffusion
</td>
<td align="center" valign="top" style="width: 65%;">
Alternative translational diffusion calculation which can calculate diffusion in specified regions.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
strip
</td>
<td align="center" valign="top" style="width: 65%;">
Remove specified atoms from the system.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
surf
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate the LCPO surface area of specified atoms. Can do partial surface areas.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
symmrmsd
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate symmetry-corrected RMSD.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
temperature
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate system temperature using velocities of specified atoms.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
time
</td>
<td align="center" valign="top" style="width: 65%;">
Add/remove/modify time information in frames.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
trans, translate
</td>
<td align="center" valign="top" style="width: 65%;">
Translate specified atoms by specified amounts in X/Y/Z directions.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
unstrip
</td>
<td align="center" valign="top" style="width: 65%;">
Undo all previous <b><i>strip</i></b> Action commands.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
unwrap
</td>
<td align="center" valign="top" style="width: 65%;">
Reverse of <b><i>image</i></b>; unwrap selected atoms so they have continuous trajectories.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
vector
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate various types of vector quantities.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
velocityautocorr
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate velocity autocorrelation function.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
volmap
</td>
<td align="center" valign="top" style="width: 65%;">
Create volumetric map for specified coordinates; similar to <b><i>grid</i></b> but takes
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">

</td>
<td align="center" valign="top" style="width: 65%;">
 into account atomic radii. Similar to VMD <b><i>volmap</i></b>.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
volume
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate unit cell volume.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
watershell
</td>
<td align="center" valign="top" style="width: 65%;">
Calculate the number of waters in the first and second solvation shells based on distance critera.
</td>
<td align="center" valign="top" style="width: 10%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
xtalsymm
</td>
<td align="center" valign="top" style="width: 65%;">
Re-image coordinates based on crystal space group symmetry operations and asymmetric unit volume.
</td>
<td align="center" valign="top" style="width: 10%;">
X
</td>

</tr>

</table>

</div>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.1">11.1</a> align
</h2>
<pre class="LyX-Code">
align &lt;mask&gt; [&lt;refmask&gt;] [move &lt;mask&gt;] [mass]
  	[ first | reference | ref &lt;name&gt; | refindex &lt;#&gt; | previous |
        reftraj &lt;name&gt; [parm &lt;name&gt; | parmindex &lt;#&gt;] ]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Target atoms to fit.
</div>
<div class="Description">
<span class="Description-entry">[&lt;refmask&gt;]</span> Reference atoms to fit (default is target mask).
</div>
<div class="Description">
<span class="Description-entry">[move &lt;mask&gt;]</span> Atoms to move when aligning (default is target mask).
</div>
<div class="Description">
<span class="Description-entry">[mass]</span> Mass-weight the fit.
</div>
<div class="Unindented">
Reference keywords:
</div>
<div class="Description">
<span class="Description-entry"><b>first</b></span> Use the first trajectory frame processed as reference.
</div>
<div class="Description">
<span class="Description-entry"><b>reference</b></span> Use the first previously read in reference structure (refindex 0).
</div>
<div class="Description">
<span class="Description-entry"><b>ref &lt;name&gt;</b></span> Use previously read in reference structure specified by filename/tag.
</div>
<div class="Description">
<span class="Description-entry"><b>refindex &lt;#&gt;</b></span> Use previously read in reference structure specified by &lt;#&gt; (based on order read in).
</div>
<div class="Description">
<span class="Description-entry">previous</span> Use frame prior to current frame as reference.
</div>
<div class="Description">
<span class="Description-entry"><b>reftraj</b> &lt;name&gt;</span> Use frames from COORDS set &lt;name&gt; or read in from trajectory file &lt;name&gt; as references. Each frame from &lt;name&gt; is used in turn, so that frame 1 is compared to frame 1 from &lt;name&gt;, frame 2 is compared to frame 2 from &lt;name&gt; and so on. If &lt;trajname&gt; runs out of frames before processing is complete, the last frame of &lt;trajname&gt; continues to be used as the reference.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;parmname&gt; | parmindex &lt;#&gt;</span> If <b>reftraj</b> specifies a trajectory file, associate it with specified topology; if not specified the first topology is used.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Align structure using specified &lt;mask&gt; onto reference. If ’move’ is specified, only move atoms in the move mask.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.2">11.2</a> angle
</h2>
<pre class="LyX-Code">
angle<a class="IndexReference" name="entry-angle-0" href="#index-angle">↓</a> [&lt;dataset name&gt;] &lt;mask1&gt; &lt;mask2&gt; &lt;mask3&gt; [out &lt;filename&gt;] [mass]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;dataset name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;maskX&gt;</span> Three atom masks selecting atom(s) to calculate angle for.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file name.
</div>
<div class="Description">
<span class="Description-entry">[mass]</span> Use center of mass of atoms in &lt;maskX&gt; instead of geometric center.
</div>

</li>

</ul>
<div class="Unindented">
Calculate angle (in degrees) between atoms in &lt;mask1&gt;, &lt;mask2&gt;, and &lt;mask3&gt;. For example, to calculate the angle between the first three atoms in the system:
</div>
<pre class="LyX-Code">
angle A123 @1 @2 @3 out A123.agr
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.3">11.3</a> areapermol
</h2>
<pre class="LyX-Code">
areapermol [&lt;name&gt;] {[&lt;mask1&gt;] [nlayers &lt;#&gt;] | nmols &lt;#&gt;} [out &lt;filename&gt;]
           [{xy | xz | yz}]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask1&gt;]</span> Atom mask for selecting molecules. If any atom in a molecule is selected the whole molecule is selected.
</div>
<div class="Description">
<span class="Description-entry">[nlayers &lt;#&gt;]</span> Number of layers of molecules. Total number of molecules used will be # molecules divided by # layers.
</div>
<div class="Description">
<span class="Description-entry">[nmols &lt;#&gt;]</span> If &lt;mask1&gt; is not specified, the number of molecules to use when calculating area per molecule.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file name.
</div>
<div class="Description">
<span class="Description-entry">[{xy|xz|yz}]</span> Cross-section of box to calculate area of. Default is X-Y.
</div>

</li>

</ul>
<div class="Unindented">
Calculate area per molecule as Area / # molecules. The area is determined from the specified cross-section of the box (X-Y by default). Currently the calculation is only guaranteed to work properly with orthorhombic unit cells. For example, to get the area per molecule of residues named &ldquo;OL&rdquo; which are arranged in 2 layers:
</div>
<pre class="LyX-Code">
areapermol OL_area :OL nlayers 2 out apm.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.4">11.4</a> atomiccorr
</h2>
<pre class="LyX-Code">
atomiccorr [&lt;mask&gt;] out &lt;filename&gt; [cut &lt;cutoff&gt;] [min &lt;min spacing&gt;]
           [byatom | byres]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Atoms to calculate motion vectors for.
</div>
<div class="Description">
<span class="Description-entry">out &lt;filename&gt;</span> File to write results to.
</div>
<div class="Description">
<span class="Description-entry">cut &lt;cutoff&gt;</span> Only print correlations with absolute value greater than &lt;cutoff&gt;.
</div>
<div class="Description">
<span class="Description-entry">min &lt;min spacing&gt;</span> Only calculate correlations for motion vectors spaced &lt;min spacing&gt; apart.
</div>
<div class="Description">
<span class="Description-entry">byatom</span> Default; calculate atomic motion vectors.
</div>
<div class="Description">
<span class="Description-entry">byres</span> Calculate motion vectors for entire residues (selected atoms in residues only).
</div>

</li>

</ul>
<div class="Unindented">
Calculate average correlations between the motion of atoms in &lt;mask&gt;. For each frame, a motion vector is calculated for each selected atom from its previous position to its current position. For each pair of motion vectors <i>Va</i> and <i>Vb</i>, the average correlation between those vectors is calculated as the average of the dot product of those vectors over all <i>N</i> frames. \begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<span class="formula"><i>AvgCorr</i>(<i>a</i>, <i>b</i>) = <span class="fraction"><span class="ignored">(</span><span class="numerator"><span class="limits"><span class="limit">∑</span></span><i>V</i><sub><i>a</i></sub>(<i>i</i>)⋅<i>V</i><sub><i>b</i></sub>(<i>i</i>)</span><span class="ignored">)/(</span><span class="denominator"><i>N</i></span><span class="ignored">)</span></span></span>
</div>

</div>
<div class="Indented">
The value of AvgCorr can range from 1.0 (correlated) to 0.0 (no correlation) to -1.0 (anti-correlated). For example, to calculate the correlation of motion vectors between residues 1 to 13, writing to a Gnuplot-readable formatted file:
</div>
<pre class="LyX-Code">
atomiccorr :1-13 out acorr.gnu byres
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.5">11.5</a> atomicfluct | rmsf<a class="Label" name="subsec:cpptraj_atomicfluct"> </a>
</h2>
<pre class="LyX-Code">
atomicfluct<a class="IndexReference" name="entry-atomicfluct-0" href="#index-atomicfluct">↓</a> [&lt;name&gt;] [out &lt;filename&gt;] [&lt;mask&gt;] [byres | byatom | bymask]
            [bfactor] [calcadp [adpout &lt;file&gt;]]
            [start &lt;start&gt;] [stop &lt;stop&gt;] [offset &lt;offset&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">out &lt;filename&gt;</span> Write data to file named &lt;filename&gt;
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Calculate fluctuations for atoms in &lt;mask&gt; (all if not specified).
</div>
<div class="Description">
<span class="Description-entry">byres</span> Output the average (mass-weighted) fluctuation by residue.
</div>
<div class="Description">
<span class="Description-entry">bymask</span> Output the average (mass-weighted) fluctuation for all atoms in &lt;mask&gt;.
</div>
<div class="Description">
<span class="Description-entry">byatom (default)</span> Output the fluctuation by atom.
</div>
<div class="Description">
<span class="Description-entry">[bfactor]</span> Calculate atomic positional fluctuations squared and weight by <span class="formula"><span class="fraction"><span class="ignored">(</span><span class="numerator">8</span><span class="ignored">)/(</span><span class="denominator">3</span><span class="ignored">)</span></span><i>π</i><sup>2</sup></span>; this is similar but not necessarily equivalent to the calculation of crystallographic B-factors.
</div>
<div class="Description">
<span class="Description-entry">[calcadp [adpout &lt;file&gt;]]</span> Calculate anisotropic displacement parameters and optionally output them to &lt;file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[&lt;start&gt;]</span> Frame to begin calculation at (default 1).
</div>
<div class="Description">
<span class="Description-entry">[&lt;stop&gt;]</span> Frame to end calculation at (default last).
</div>
<div class="Description">
<span class="Description-entry">[&lt;offset&gt;]</span> Frames to skip between calculations (default 1).
</div>
<div class="Unindented">
DataSets created
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Hold atomic fluctuations.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[ADP]</span> Hold anisotropic displacement parameters if ’calcadp’ specified.
</div>

</li>

</ul>
<div class="Unindented">
Compute the atomic positional fluctuations (also referred to as root-mean-square fluctuations, RMSF) for atoms specified in the <b>&lt;mask&gt;</b>. The RMSF of a given atom <i>i</i> is calculated as:
</div>
<div class="Indented">
<div class="center">
<span class="formula"><i>RMSF</i><sub><i>i</i></sub> = <span class="sqrt"><span class="radical">√</span><span class="ignored">(</span><span class="root"><span class="symbol">⟨</span><span class="symbol">(</span><i>x</i><sub><i>i</i></sub> − <span class="symbol">⟨</span><i>x</i><sub><i>i</i></sub><span class="symbol">⟩</span><span class="symbol">)</span><sup>2</sup><span class="symbol">⟩</span></span><span class="ignored">)</span></span></span>
</div>

</div>
<div class="Indented">
where <i>x</i> denotes atomic positions and the averages are over all input frames.
</div>
<div class="Indented">
Note that RMS fitting is not done implicitly. If you want fluctuations without rotations or translations (for example to the average structure), perform an RMS fit to the average structure (best) or the first structure (see <b><i>rmsd</i></b>) prior to this calculation. The units are (Å) for RMSF or Å<span class="formula"><sup><span class="text">2</span></sup></span> × <span class="formula"><span class="fraction"><span class="ignored">(</span><span class="numerator">8</span><span class="ignored">)/(</span><span class="denominator">3</span><span class="ignored">)</span></span></span><span class="formula"><i>π</i><sup>2</sup></span> if <b>bfactor</b> is specified.
</div>
<div class="Indented">
If <b>byres</b> or <b>bymask</b> are specified, the mass-weighted average of atomic fluctuations of each atom for either each residue or the entire mask will be calculated respectively:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<span class="formula"><span class="symbol">⟨</span><i>Fluct</i><span class="symbol">⟩</span> = <span class="fraction"><span class="ignored">(</span><span class="numerator"><span class="limits"><span class="limit">∑</span></span><i>AtomFluct</i><sub><i>i</i></sub>*<i>Mass</i><sub><i>i</i></sub></span><span class="ignored">)/(</span><span class="denominator"><span class="limits"><span class="limit">∑</span></span><i>Mass</i><sub><i>i</i></sub></span><span class="ignored">)</span></span></span>
</div>

</div>
<div class="Indented">
If <b>calcadp</b> is specified, anisotropic displacement factors for atoms will be calculated and written to the file specified by <b>adpout</b> (or STDOUT if not specified) using PDB ANISOU record format. The displacement factors will be saved to a data set. Note that <b>calcadp</b> automatically implies <b>bfactor</b>.
</div>
<div class="Indented">
With <i>cpptraj</i> it is possible to perform coordinate averaging, the fit to average coordinates, and the atomic fluctuation calculation in a single execution like so:
</div>
<pre class="LyX-Code">
parm myparm.parm7
trajin mytrajectory.crd
rms first
average crdset MyAvg
run
rms ref MyAvg
atomicfluct out fluct.agr
</pre>
<div class="Unindented">
To write the mass-weighted B-factors for the protein backbone atoms C, CA, and N, averaged by residue use the command:
</div>
<pre class="LyX-Code">
atomicfluct out back.agr @C,CA,N byres bfactor
</pre>
<div class="Unindented">
To write the RMSF or atomic positional fluctuations of the same atoms, use the command:
</div>
<pre class="LyX-Code">
atomicfluct out backbone-atoms.agr @C,CA,N
</pre>
<div class="Unindented">
To write a PDB of averaged coordinates (after fitting to the first frame) with both B-factors and anisotropic temperature factors:
</div>
<pre class="LyX-Code">
parm myparm.parm7
trajin mytraj.nc
rms first
average crdset MyAvg
atomicfluct MyFluct calcadp
run
crdout MyAvg mypdb.pdb adpdata MyFluct[ADP] bfacdata MyFluct
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.6">11.6</a> atommap
</h2>
<pre class="LyX-Code">
atommap<a class="IndexReference" name="entry-atommap-0" href="#index-atommap">↓</a> &lt;target&gt; &lt;reference&gt; [mapout &lt;filename&gt;] [maponly]
        [rmsfit [ rmsout &lt;rmsout&gt; ]]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;target&gt;</span> Reference structure whose atoms will be remapped.
</div>
<div class="Description">
<span class="Description-entry">&lt;reference&gt;</span> Reference structure that <b>&lt;target&gt;</b> should be mapped to.
</div>
<div class="Description">
<span class="Description-entry">mapout &lt;filename&gt;</span> Write atom map to &lt;filename&gt; with format:<br/>
TargetAtomNumber TargetAtomName ReferenceAtomNumber ReferenceAtomName<br/>
Target atoms that cannot be mapped to a reference atom are denoted &ldquo;—&rdquo;.
</div>
<div class="Description">
<span class="Description-entry">maponly</span> Write atom map but do not reorder atoms.
</div>
<div class="Description">
<span class="Description-entry">rmsfit</span> Any input frames using the same topology as &lt;target&gt; will be RMS fit to &lt;reference&gt; using whatever atoms could be mapped.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">rmsout &lt;rmsout&gt;</span> If rmsfit specified, write resulting RMSDs to &lt;rmsout&gt;.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Attempt to map the atoms of &lt;target&gt; to those of &lt;reference&gt; based on structural similarity. This is useful e.g. when there are two files containing the same structure but with different atom names or atom ordering. Both &lt;target&gt; and &lt;reference&gt; need to have been read in with a previous <b><i>reference</i></b> command. The state will then be modified so that any trajectory read in with the same parameter file as &lt;target&gt; will have its atoms mapped (i.e. reordered) to match those of &lt;reference&gt;. If the number of atoms that can be mapped in &lt;target&gt; are less than those in &lt;reference&gt;, the reference structure specified by &lt;reference&gt; will be modified to include only mapped atoms; this is useful if for example the reference structure is protonated with respect to the target. The <b>rmsfit</b> keyword is useful in cases where the atom mapping will not be complete (e.g. two ligands with the same scaffold but different substituents).
</div>
<div class="Indented">
For example, say you have the same ligand structure in two files, Ref.mol2 and Lig.mol2, but the atom ordering in each file is different. To map the atoms in Lig.mol2 onto those of Ref.mol2 so that Lig.mol2 has the same ordering as Ref.mol2:
</div>
<pre class="LyX-Code">
parm Lig.mol2 
reference Lig.mol2
parm Ref.mol2 
reference Ref.mol2 parmindex 1
atommap Lig.mol2 Ref.mol2 mapout atommap.dat
trajin Lig.mol2 
trajout Lig.reordered.mol2 mol2
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.7">11.7</a> autoimage
</h2>
<pre class="LyX-Code">
autoimage [&lt;mask&gt; | anchor &lt;mask&gt; [fixed &lt;mask&gt;] [mobile &lt;mask&gt;]]
          [origin] [firstatom] [familiar | triclinic]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;mask&gt; | anchor &lt;mask&gt;]</span> Atoms to image around; this is the region that will be centered. Default is the entire first molecule.
</div>
<div class="Description">
<span class="Description-entry">[fixed &lt;mask&gt;]</span> Molecules that should remain ’fixed’ to the anchor region; default is all non-ion/non-solvent molecules.
</div>
<div class="Description">
<span class="Description-entry">[mobile &lt;mask&gt;]</span> Molecules that can be freely imaged; default is all ion/solvent molecules.
</div>
<div class="Description">
<span class="Description-entry">[origin]</span> Center anchor region at the origin; if not specified, center at box center.
</div>
<div class="Description">
<span class="Description-entry">[firstatom]</span> Image based on molecule first atom; default is to image by molecule center of mass.
</div>
<div class="Description">
<span class="Description-entry">[familiar]</span> Image to familiar truncated-octahedral shape; this is on by default if the original cell is truncated octahedron.
</div>
<div class="Description">
<span class="Description-entry">[triclinic]</span> Force general triclinic imaging.
</div>

</li>

</ul>
<div class="Unindented">
Automatically center and image (by molecule) a trajectory with periodic boundaries. For most cases just specifying <b><i>’autoimage’</i></b> alone is sufficient. The atoms of the <b>’anchor’</b> region (default the entire first molecule) will be centered; all <b>’fixed’</b> molecules will be imaged only if imaging brings them closer to the <b>’anchor’</b> molecule (default for <b>’fixed’</b> molecules is all non-solvent non-ion molecules). All other molecules (referred to as <b>’mobile’</b>) will be imaged freely.
</div>
<div class="Indented">
The autoimage command works for the majority of systems; however, for very densely packed systems the default anchor (entire first molecule) may not be appropriate. In these cases, it is recommended to choose as the anchor a small region which should lie near the center of your system. For example, in a protein dimer system one could choose a single residue that is near the center of the interface between the two monomers.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.8">11.8</a> average<a class="Label" name="subsec:cpptraj-average"> </a>
</h2>
<pre class="LyX-Code">
average<a class="IndexReference" name="entry-average-0" href="#index-average">↓</a> {crdset &lt;set name&gt; | &lt;filename&gt;} [&lt;mask&gt;]
        [start &lt;start&gt;] [stop &lt;stop&gt;] [offset &lt;offset&gt;]
        [Trajout Args]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;filename&gt;</span> If specified, write averaged coordinates to &lt;filename&gt; (not compatible with crdset).
</div>
<div class="Description">
<span class="Description-entry">crdset &lt;set name&gt;</span> If specified, save averaged coordinates to COORDS set &lt;set name&gt; (not compatible with &lt;filename&gt;).
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Average coordinates in &lt;mask&gt; (all atoms if not specified).
</div>
<div class="Description">
<span class="Description-entry">[&lt;start&gt;]</span> Frame to begin calculation at (default 1).
</div>
<div class="Description">
<span class="Description-entry">[&lt;stop&gt;]</span> Frame to end calculation at (default last).
</div>
<div class="Description">
<span class="Description-entry">[&lt;offset&gt;]</span> Frames to skip between calculations (default 1).
</div>
<div class="Description">
<span class="Description-entry">[Trajout args]</span> Output trajectory format argument(s) (default Amber Trajectory).
</div>

</li>

</ul>
<div class="Unindented">
Calculate the average of input coordinates and write out to file named <b>&lt;filename&gt;</b> or save to COORDS set named <b>&lt;set name&gt;</b> in any trajectory format <i>cpptraj</i> recognizes (Amber Trajectory if not specified). If the number of atoms in <b>&lt;mask&gt;</b> are less than the total number of atoms, the topology will be stripped to match <b>&lt;mask&gt;</b>. 
</div>
<div class="Indented">
Note that since coordinates are being averaged over many frames, resulting structures may appear distorted. For example, if one averages the coordinates of a freely rotating methyl group the average position of the hydrogen atoms will be close to the center of rotation. Also note that typically one will want to remove global rotational and translation movement prior to this command by using e.g. the <b><i>rms</i></b> (<a class="Reference" href="#subsec:cpptraj-rmsd">11.64 on page 1↓</a>) command.
</div>
<div class="Indented">
Any arguments that are valid for the <b><i>trajout</i></b> command (<a class="Reference" href="#subsec:cpptraj-trajout">10.5 on page 1↑</a>) can be passed to this command in order to control the format of the output coordinates. For example, to write out a PDB file containing the averaged coordinates over all frames:
</div>
<pre class="LyX-Code">
average test.pdb pdb
</pre>
<div class="Unindented">
To write out a mol2 file containing only the averaged coordinates of residues 1 to 10 for frames 1 to 100:
</div>
<pre class="LyX-Code">
average test.mol2 mol2 start 1 stop 100 :1-10
</pre>
<div class="Unindented">
To create an average structure of atoms named CA and then use it as a reference for an rms command in a subsequent run:
</div>
<pre class="LyX-Code">
trajin Input.nc
average crdset MyAvg @CA
run
rms ref MyAvg @CA out RmsToAvg.dat
run
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.9">11.9</a> avgcoord
</h2>
<div class="Unindented">
This command is deprecated. Use ’vector center’ (optionally with keyword ’magnitude’) instead.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.10">11.10</a> bounds<a class="Label" name="subsec:cpptraj-bounds"> </a>
</h2>
<pre class="LyX-Code">
bounds [&lt;mask&gt;] [out &lt;filename&gt;]
       [dx &lt;dx&gt; [dy &lt;dy&gt;] [dz &lt;dz&gt;] name &lt;gridname&gt; [offset &lt;bin offset&gt;]]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Mask of atoms to determine bounds of.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write bounds to (default STDOUT if not specified).
</div>
<div class="Description">
<span class="Description-entry">[dx &lt;dx&gt; [dy &lt;dy&gt;] [dz &lt;dz&gt;]]</span> Triggers creation of a grid data set from bounds. Spacings of generated grid in the X, Y and Z directions. If only dx is specified &lt;dx&gt; will be used for &lt;dy&gt; and &lt;dz&gt; as well.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;gridname&gt;]</span> Name of generated data sets.
</div>
<div class="Description">
<span class="Description-entry">[offset &lt;bin offset&gt;]</span> Number of bins to add/subtract in each direction to generated grid.
</div>
<div class="Unindented">
DataSets Generated
</div>
<div class="Description">
<span class="Description-entry">&lt;gridname&gt;</span> The 3D grid (only if ’dx’ etc specified).
</div>
<div class="Description">
<span class="Description-entry">&lt;gridname&gt;[xmin]</span> The minimum x coordinate encountered.
</div>
<div class="Description">
<span class="Description-entry">&lt;gridname&gt;[xmax]</span> The maximum x coordinate encountered.
</div>
<div class="Description">
<span class="Description-entry">&lt;gridname&gt;[ymin]</span> The minimum y coordinate encountered.
</div>
<div class="Description">
<span class="Description-entry">&lt;gridname&gt;[ymax]</span> The maximum y coordinate encountered.
</div>
<div class="Description">
<span class="Description-entry">&lt;gridname&gt;[zmin]</span> The minimum z coordinate encountered.
</div>
<div class="Description">
<span class="Description-entry">&lt;gridname&gt;[zmax]</span> The maximum z coordinate encountered.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the boundaries (i.e. the max/min X/Y/Z coordinates) of atoms in <b>&lt;mask&gt;</b> and write to <b>&lt;filename&gt;</b> (STDOUT if not specified). Useful for determining dimensions for the <b><i>grid</i></b> command, and can be used to generate a grid data set that can be used by <b><i>grid</i></b> (see <a class="Reference" href="#subsec:cpptraj_grid">11.36 on page 1↓</a>).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.11">11.11</a> box
</h2>
<pre class="LyX-Code">
box [x &lt;xval&gt;] [y &lt;yval&gt;] [z &lt;zval&gt;] [alpha &lt;a&gt;] [beta &lt;b&gt;] [gamma &lt;g&gt;]
    [nobox] [truncoct]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[x &lt;xval&gt;] [y &lt;yval&gt;] [z &lt;zval&gt;]<b></b></span> <b></b>Change box length(s) to specified value(s).
</div>
<div class="Description">
<span class="Description-entry">[alpha &lt;a&gt;] [beta &lt;b&gt;] [gamma &lt;g&gt;]</span> Change box angle(s) to specified value(s).
</div>
<div class="Description">
<span class="Description-entry">[nobox]</span> Remove any existing box information.
</div>
<div class="Description">
<span class="Description-entry">[truncoct]</span> Set box angles to truncated octahedron.
</div>

</li>

</ul>
<div class="Unindented">
Modify box information during trajectory processing. Note that this will permanently modify the box information for topology files during trajectory processing as well. It is possible to modify any number of the box parameters (e.g. only the Z length can be modified if desired while leaving all other parameters intact).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.12">11.12</a> center
</h2>
<pre class="LyX-Code">
center<a class="IndexReference" name="entry-center-0" href="#index-center">↓</a> [&lt;mask&gt;] [origin] [mass] 
       [ reference | ref &lt;name&gt; | refindex &lt;#&gt; [&lt;refmask&gt;]]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Center based on atoms in mask; default is all atoms.
</div>
<div class="Description">
<span class="Description-entry">[origin]</span> Center to origin (0, 0, 0); default is center to box center (X/2, Y/2, Z/2).
</div>
<div class="Description">
<span class="Description-entry">[mass]</span> Use center of mass instead of geometric center.
</div>
<div class="Description">
<span class="Description-entry">[reference | ref &lt;name&gt; | refindex &lt;#&gt; [&lt;refmask]]</span> Center using coordinates in specified reference structure selected by &lt;refmask&gt; (&lt;mask&gt; if not specified.
</div>

</li>

</ul>
<div class="Unindented">
Move all atoms so that the center of the atoms in <b>&lt;mask&gt;</b> is centered at the specified location: box center (default), coordinate origin, or reference coordinates.
</div>
<div class="Indented">
For example, to move all coordinates so that the center of mass of residue 1 is at the center of the box:
</div>
<pre class="LyX-Code">
center :1 mass
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.13">11.13</a> check | checkoverlap | checkstructure
</h2>
<pre class="LyX-Code">
check<a class="IndexReference" name="entry-check-0" href="#index-check">↓</a> [&lt;mask&gt;] [around &lt;mask2&gt;] [reportfile &lt;report&gt;] [noimage] [skipbadframes]
      [offset &lt;offset&gt;] [cut &lt;cut&gt;] [nobondcheck] [silent]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Check structure of atoms in &lt;mask&gt; (all if not specified).
</div>
<div class="Description">
<span class="Description-entry">[around &lt;mask2&gt;]</span> If specified, only check for problems between atoms in &lt;mask&gt; and atoms in &lt;mask2&gt;.
</div>
<div class="Description">
<span class="Description-entry">[reportfile &lt;report&gt;]</span> Write any problems found to &lt;report&gt; (STDOUT if not specified).
</div>
<div class="Description">
<span class="Description-entry">[noimage]</span> Do not image distances.
</div>
<div class="Description">
<span class="Description-entry">[skipbadframes]</span> If errors are encountered for a frame, subsequent actions/trajectory output will be skipped.
</div>
<div class="Description">
<span class="Description-entry">[offset &lt;offset&gt;]</span> Report bond lengths greater than the equilibrium value plus &lt;offset&gt; (default 1.0 Å)
</div>
<div class="Description">
<span class="Description-entry">[cut &lt;cut&gt;]</span> Report atoms closer than &lt;cut&gt; (default 0.8 Å).
</div>
<div class="Description">
<span class="Description-entry">[nobondcheck]</span> Check overlaps only.
</div>
<div class="Description">
<span class="Description-entry">[silent]</span> Do not print information for bad frames - useful in conjunction with the <b>skipbadframes</b> option.
</div>

</li>

</ul>
<div class="Unindented">
Check the structure and report problems related to atomic overlap/unusual bond length. Problems are reported when any two atoms in the mask are closer than <b>&lt;cut&gt;</b>. If bonds are being checked then bond lengths greater than their equilibrium value + <b>&lt;offset&gt;</b> are reported as well. This command can also be used to skip corrupted frames in a trajectory during processing. For example, if this message is encountered:
</div>
<pre class="LyX-Code">
Warning: Frame 10 coords 1 &amp; 2 overlap at origin; may be corrupt.
</pre>
<div class="Unindented">
One could use <b><i>check</i></b> so that e.g. a subsequent <b><i>distance</i></b> command is not processed for bad frames:
</div>
<pre class="LyX-Code">
check @1,2 skipbadframes silent
distance d1 :1 :10
</pre>
<div class="Unindented">
Usually frame corruption can be detected using only a few atoms, but this may not catch all types of corruption. The more atoms that are used the better the corruption detection will be, but the slower it will be to process the command. Typically a good procedure to follow when corruption is suspected is to run <b><i>check</i></b> using all important atoms (e.g. all solute heavy atoms) with the <b>skipbadframes</b> keyword followed by a <b><i>trajout</i></b> command to write all non-corrupt frames, for example:
</div>
<pre class="LyX-Code">
trajin corrupted.crd
check :1-13 skipbadframes silent
trajout fixed.corrupted.nc
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.14">11.14</a> checkchirality
</h2>
<pre class="LyX-Code">
checkchirality [&lt;name&gt;] [&lt;mask&gt;] [out &lt;filename&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atoms to check.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write results to.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[L]</span> Number of frames ’L’ for each residue.
</div>
<div class="Description">
<span class="Description-entry">[D]</span> Number of frames ’D’ for each residue.
</div>

</li>

</ul>
<div class="Unindented">
Check the chirality around the alpha carbon in amino acid residues selected by &lt;mask&gt;. Note that cpptraj expects atom names to correspond to the PDB V3 standard: N, CA, C, CB. For each residue, the number of frames in which the amino acid is ’L’ or ’D’ will be recorded. For example, to check the chirality of all amino acids in a system and write to a file named chiral.dat with data set name DPDP:
</div>
<pre class="LyX-Code">
checkchirality DPDP out chiral.dat
</pre>
<div class="Unindented">
Output will have format similar to:
</div>
<pre class="LyX-Code">
#Res      DPDP[L]  DPDP[D]
   2.000      100        0
</pre>
<div class="Unindented">
So in this example residue 2 was ’L’ for 100 frames and ’D’ for 0 frames.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.15">11.15</a> closest | closestwaters
</h2>
<pre class="LyX-Code">
closest<a class="IndexReference" name="entry-closest-0" href="#index-closest">↓</a> &lt;# to keep&gt; &lt;mask&gt; [solventmask &lt;solvent mask&gt;] [noimage]
    	[first | oxygen] [center] [closestout &lt;filename&gt; [name &lt;setname&gt;]]
    	[outprefix &lt;parmprefix&gt;] [parmout &lt;file&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;# to keep&gt;</span> Number of solvent molecules to keep around &lt;mask&gt;
</div>
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Mask of atoms to search for closest waters around.
</div>
<div class="Description">
<span class="Description-entry">[solventmask &lt;solvent mask&gt;]</span> Optional mask for selecting solvent atoms. If not specified, atoms in all molecules marked as &ldquo;solvent&rdquo; will be used.
</div>
<div class="Description">
<span class="Description-entry">[noimage]</span> Do not perform imaging; only recommended if trajectory has previously been imaged.
</div>
<div class="Description">
<span class="Description-entry">[first | oxygen]</span> Calculate distances between all atoms in <b>&lt;mask&gt;</b> and the first atom of solvent only (recommended for standard water models as it will increase speed of calculation).
</div>
<div class="Description">
<span class="Description-entry">[center]</span> Search for waters closest to geometric center of <b>&lt;mask&gt;</b> instead of each atom in <b>&lt;mask&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[closestout &lt;filename&gt;]</span> Write information on the closest solvent molecules to <b>&lt;filename&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[outprefix &lt;prefix&gt;]</span> Write corresponding topology to file with name prefix <b>&lt;prefix&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[parmout &lt;file&gt;]</span> Write corresponding topology to file with name &lt;file&gt;.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[Frame]</span> Frame number.
</div>
<div class="Description">
<span class="Description-entry">[Mol]</span> Original solvent molecule number.
</div>
<div class="Description">
<span class="Description-entry">[Dist]</span> Solvent molecule distance in Å.
</div>
<div class="Description">
<span class="Description-entry">[FirstAtm]</span> First atom number of original solvent molecule.
</div>

</li>

</ul>
<div class="Unindented">
Similar to the <b><i>strip</i></b> command, but modify coordinate frame and topology by keeping only the specified number of closest solvent molecules to the region specified by the given mask. Solvent molecules can be determined automatically by <i>cpptraj</i> (by default residues named WAT, HOH, or TIP3), can be specified prior via the <b><i>solvent</i></b> command (<a class="Reference" href="#subsec:cpptraj_solvent">9.19 on page 1↑</a>), or can be selected by <b>solventmask</b>.
</div>
<div class="Indented">
The format of the <b>closestout</b> file is:
</div>
<pre class="LyX-Code">
Frame    Molecule    Distance    FirstAtom#
</pre>
<div class="Unindented">
For example, to obtain the 10 closest waters to residues 1-268 by distance to the first atom of the waters, write out which waters were closest for each frame to a file called &ldquo;closestmols.dat&rdquo;, and write out the stripped topology with prefix &ldquo;closest&rdquo; containing only the solute and 10 waters:
</div>
<pre class="LyX-Code">
closest 10 :1-268 first closestout closestmols.dat outprefix closest
</pre>
<div class="Unindented">
As of version 17 this command is CUDA-enabled in CUDA versions of CPPTRAJ.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.16">11.16</a> cluster
</h2>
<div class="Unindented">
Although the <b>’cluster’</b> command can still be specified as an action, it is now considered an analysis. See <a class="Reference" href="#subsec:cpptraj-cluster-analysis">12.4 on page 1↓</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.17">11.17</a> clusterdihedral
</h2>
<pre class="LyX-Code">
clusterdihedral<a class="IndexReference" name="entry-clusterdihedral-0" href="#index-clusterdihedral">↓</a> [phibins &lt;N&gt;] [psibins &lt;M&gt;] [out &lt;outfile&gt;] 
                [dihedralfile &lt;dfile&gt; | &lt;mask&gt;]
                [framefile &lt;framefile&gt;] [clusterinfo &lt;infofile&gt;]
                [clustervtime &lt;cvtfile&gt;] [cut &lt;CUT&gt;]
</pre>
<div class="Unindented">
Cluster frames in a trajectory using dihedral angles. To define which dihedral angles will be used for clustering either an atom mask or an input file specified by the <b>dihedralfile</b> keyword should be used. If dihedral file is used, each line in the file should contain a dihedral to be binned with format:
</div>
<pre class="LyX-Code">
ATOM#1 ATOM#2 ATOM#3 ATOM#4 #BINS
</pre>
<div class="Unindented">
where the ATOM arguments are the atom numbers (starting from 1) defining the dihedral and #BINS is the number of bins to be used (so if #BINS=10 the width of each bin will be 36º). If an atom mask is specified, only protein backbone dihedrals (Phi and Psi defined using atom names C-N-CA-C and N-CA-C-N) within the mask will be used, with the bin sizes specified by the phibins and psibins keywords (default for each is 10 bins).
</div>
<div class="Indented">
Output will either be written to STDOUT or the file specified by the <b>out</b> keyword. First, information about which dihedrals were clustered will be printed. Then the number of clusters will be printed, followed by detailed information of each cluster. The clusters are sorted from most populated to least populated. Each cluster line has format
</div>
<pre class="LyX-Code">
Cluster CLUSTERNUM CLUSTERPOP [ dihedral1bin, dihedral2bin ... dihedralNbin ]
</pre>
<div class="Unindented">
followed by a list of frame numbers that belong to that cluster. If a cutoff is specified by <b>cut</b>, only clusters with population greater than CUT will be printed.
</div>
<div class="Indented">
If specified by the <b>clustervtime</b> keyword, the number of clusters for each frame will be printed to &lt;cvtfile&gt;. If specified by the <b>framefile</b> keyword, a file containing cluster information for each frame will be written with format
</div>
<pre class="LyX-Code">
Frame CLUSTERNUM CLUSTERSIZE DIHEDRALBINID
</pre>
<div class="Unindented">
where DIHEDRALBINID is a number that identifies the unique combination of dihedral bins this cluster belongs to (specifically it is a 3*number-of-dihedral-characters long number composed of the individual dihedral bins).
</div>
<div class="Indented">
If specified by the <b>clusterinfo</b> keyword, a file containing information on each dihedral and each cluster will be printed. This file can be read by SANDER for use with REMD with a structure reservoir (-rremd=3). The file, which is essentially a simplified version of the main output file, has the following format:
</div>
<pre class="LyX-Code">
#DIHEDRALS
dihedral1_atom1 dihedral1_atom2 dihedral1_atom3 dihedral1_atom4
...
#CLUSTERS
CLUSTERNUM1 CLUSTERSIZE1 DIHEDRALBINID1
...
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.18">11.18</a> contacts
</h2>
<pre class="LyX-Code">
contacts [ first | reference | ref &lt;ref&gt; | refindex &lt;#&gt; ] [byresidue]
         [out &lt;filename&gt;] [time &lt;interval&gt;] [distance &lt;cutoff&gt;] [&lt;mask&gt;]
</pre>
<div class="Unindented">
NOTE: Users are encouraged to try the <b><i>nativecontacts</i></b> command (<a class="Reference" href="#subsec:cpptraj-nativecontacts"> on page 1↓</a>), an update version of this command.
</div>
<div class="Indented">
For each atom given in <i>mask</i>, calculate the number of other atoms (contacts) within the distance <i>cutoff</i>. The default cutoff is 7.0 A. Only atoms in <i>mask</i> are potential interaction partners (e.g., a mask @CA will evaluate only contacts between CA atoms). The results are dumped to <i>filename</i> if the keyword &ldquo;<tt>out</tt>&rdquo; is specified. Thereby, the time between snapshots is taken to be <i>interval</i>. In addition to the number of overall contacts, the number of native contacts is also determined. Native contacts are those that have been found either in the first snapshot of the trajectory (if the keyword &ldquo;<tt>first</tt>&rdquo; is specified) or in a reference structure (if the keyword &ldquo;<tt>reference</tt>&rdquo; is specified). Finally, if the keyword &ldquo;<tt>byresidue</tt>&rdquo; is provided, results are output on a per-residue basis for each snapshot, whereby the number of native contacts is written to <i>filename</i>.native.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.19">11.19</a> createcrd<a class="Label" name="subsec:cpptraj_createcrd"> </a>
</h2>
<pre class="LyX-Code">
createcrd [&lt;name&gt;] [ parm &lt;name&gt; | parmindex &lt;#&gt; ]
</pre>
<div class="Unindented">
This command creates a COORDS data set named &lt;name&gt; using trajectory frames that are associated with the specified topology.
</div>
<div class="Indented">
For example, to save frames that have been previously RMS-fit to a reference structure into a COORDS set named MyCrd you would use the input:
</div>
<pre class="LyX-Code">
rms reference :1-12@CA
createcrd MyCrd
strip :6-8
</pre>
<div class="Unindented">
Note that here the <b><i>strip</i></b> command will have no effect on the coordinates saved in MyCrd since it occurs after the <b><i>createcrd</i></b> command.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.20">11.20</a> createreservoir
</h2>
<pre class="LyX-Code">
createreservoir &lt;filename&gt; ene &lt;energy data set&gt; [bin &lt;cluster bin data set&gt;]
                temp0 &lt;temp0&gt; iseed &lt;iseed&gt; [velocity]
                [parm &lt;parmfile&gt; | parmindex &lt;#&gt;] [title &lt;title&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;filename&gt;</span> File name of the reservoir to create.
</div>
<div class="Description">
<span class="Description-entry">ene &lt;energy data set&gt;</span> Data set with energies corresponding to frames.
</div>
<div class="Description">
<span class="Description-entry">[bin &lt;cluster bin data set&gt;]</span> Data set with bin numbers (for RREMD=3).
</div>
<div class="Description">
<span class="Description-entry">temp0 &lt;temp0&gt;</span> Reservoir temperature.
</div>
<div class="Description">
<span class="Description-entry">iseed &lt;iseed&gt;</span> Reservoir random number seed.
</div>
<div class="Description">
<span class="Description-entry">[velocity]</span> Include velocities in the reservoir.
</div>
<div class="Description">
<span class="Description-entry">[parm &lt;parmfile&gt; | parmindex &lt;#&gt;]</span> Associated topology.
</div>
<div class="Description">
<span class="Description-entry">[title &lt;title&gt;]</span> Reservoir title.
</div>

</li>

</ul>
<div class="Unindented">
Create structure reservoir for use with reservoir REMD simulations using energies in &lt;energy data set&gt;, temperature &lt;temp0&gt; and random seed &lt;iseed&gt; Include velocities if [velocity] is specified. If &lt;cluster bin data set&gt; is specified from e.g. a previous ’clusterdihedral’ command, the reservoir can be used for non-Boltzmann reservoir REMD (rremd==3).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.21">11.21</a> density
</h2>
<pre class="LyX-Code">
density [out &lt;filename&gt;] 
        [ &lt;mask1&gt; ... &lt;maskN&gt; [name &lt;set name&gt;] [delta &lt;resolution&gt;] [{x|y|z}]
          [{number|mass|charge|electron}] [{bincenter|binedge}] ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file for histogram (relative distances vs. densities for each mask) or total density.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;set name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask1&gt; ... &lt;maskN&gt;</span> Arbitrary number of masks for atom selection; a dataset is created and the output will contain entries for each mask.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[delta &lt;resolution&gt;]</span> Resolution, i.e. determines number of slices (i.e. histogram bins). (default 0.25 Å)
</div>
<div class="Description">
<span class="Description-entry">[{x|y|z}]</span> Coordinate for density calculation. (default z)
</div>
<div class="Description">
<span class="Description-entry">[{number|mass|charge|electron}]</span> Number, mass, partial charge (q) or electron (Ne - q) density. Electron density will be converted to e-/Å3 by dividing the average area spanned by the other two dimensions. (default number)
</div>
<div class="Description">
<span class="Description-entry">[{bincenter|binedge}]</span> Determine whether histogram bin coordinates will be based on bin center (default) or bin edges.
</div>

</li>

</ul>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[avg]</span> Average density over coordinate.
</div>
<div class="Description">
<span class="Description-entry">[sd]</span> Standard deviation of density over coordinate.
</div>

</li>

</ul>
<div class="Unindented">
If no arguments are specified, calculate the total system density. Otherwise, calculate specified density along the given axis for atoms in specified mask(s). Defaults are shown in parentheses above. The format of the file is as follows. Comments are lines starting with ’#’ or empty lines. All other lines must contain the atom type followed by an integer number for the electron number. Entries must be separated by spaces or ’=’. Example input:
</div>
<pre class="LyX-Code">
density out number_density.dat number delta 0.25 ":POPC@P1" ":POPC@N" \
        ":POPC@C2" ":POPC"
density out mass_density.dat mass delta 0.25 ":POPC@P1" ":POPC@N" \
        ":POPC@C2" ":POPC"
density out charge_density.dat charge delta 0.25 ":POPC@P1" ":POPC@N" \
        ":POPC@C2" ":POPC"
density out electron_density.dat electron delta 0.25 efile Nelec.in \
        ":POPC@P1" ":POPC@N" ":POPC@C2" ":POPC" ":TIP3" \
        ":POPC | :TIP3" "*"
density out ion_density.dat number delta 0.25 ":SOD" ":CLA"
</pre>
<div class="Unindented">
See also $AMBERHOME/AmberTools/test/cpptraj/Test_Density.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.22">11.22</a> diffusion
</h2>
<div class="Unindented">
<i>Note that although the syntax for <b>diffusion</b> has changed as of version 16, the old syntax is still supported.</i>
</div>
<pre class="LyX-Code">
diffusion [{out &lt;filename&gt; | separateout &lt;suffix&gt;}] [time &lt;time per frame&gt;] [noimage]
      	[&lt;mask&gt;] [&lt;set name&gt;] [individual] [diffout &lt;filename&gt;] [nocalc]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write mean-square displacement (MSD) data set output to file specified by &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[separateout &lt;suffix&gt;]</span> Write each MSD data set type to files with suffix &lt;suffix&gt;; see description below.
</div>
<div class="Description">
<span class="Description-entry">[time &lt;time_per_frame&gt;]</span> Time in-between each coordinate frame in ps; default is 1.0.
</div>
<div class="Description">
<span class="Description-entry">[noimage]</span> If specified do not perform imaging. If this is specified coordinates should be unwrapped prior to this command.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Mask of atoms to calculate diffusion for; default all atoms.
</div>
<div class="Description">
<span class="Description-entry">[&lt;set name&gt;]</span> MSD data set name.
</div>
<div class="Description">
<span class="Description-entry">[individual]</span> Write diffusion for each individual atom as well as average diffusion for atoms in mask.
</div>
<div class="Description">
<span class="Description-entry">[diffout &lt;filename&gt;]</span> Write diffusion contants calculated from fits of MSD data sets to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[nocalc]</span> Do not calculate diffusion constants.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[X]</span> MSD(s) in the X direction.
</div>
<div class="Description">
<span class="Description-entry">[Y]</span> MSD(s) in the Y direction.
</div>
<div class="Description">
<span class="Description-entry">[Z]</span> MSD(s) in the Z direction.
</div>
<div class="Description">
<span class="Description-entry">[R]</span> Overall MSD(s).
</div>
<div class="Description">
<span class="Description-entry">[A]</span> Overall displacement(s).
</div>
<div class="Description">
<span class="Description-entry">[D]</span> Diffusion constants.
</div>
<div class="Description">
<span class="Description-entry">[Label]</span> Diffusion constant lablels.
</div>
<div class="Description">
<span class="Description-entry">[Slope]</span> Linear regression slopes.
</div>
<div class="Description">
<span class="Description-entry">[Intercept]</span> Linear refression Y-intercepts.
</div>
<div class="Description">
<span class="Description-entry">[Corr]</span> Linear regression correlation coefficients.
</div>

</li>

</ul>
<div class="Unindented">
Compute mean square displacement (MSD) plots (using distance traveled from initial position) for the atoms in <b>&lt;mask&gt;</b>. By default only the diffusion averaged over all atoms in <b>&lt;mask&gt;</b> is calculated; if <b>individual</b> is specified diffusion for individual atoms is calculated as well. 
</div>
<div class="Indented">
In order to correctly calculate diffusion molecules should take continuous paths, so imaging of atoms is autoimatically performed. If the trajectory is already unwrapped (or the unwrap command is used prior to this command) the <b>noimage</b> keyword can be used. 
</div>
<div class="Indented">
The following types of displacements are calculated. If <b>separateout</b> is specfied the following files will be created:
</div>
<div class="Description">
<span class="Description-entry">x_&lt;suffix&gt;</span> Mean square displacement(s) in the X direction (in Å<span class="formula"><sup><span class="text">2</span></sup></span>/ps).
</div>
<div class="Description">
<span class="Description-entry">y_&lt;suffix&gt;</span> Mean square displacement(s) in the Y direction (in Å<span class="formula"><sup><span class="text">2</span></sup></span>/ps).
</div>
<div class="Description">
<span class="Description-entry">z_&lt;suffix&gt;</span> Mean square displacement(s) in the Z direction (in Å<span class="formula"><sup><span class="text">2</span></sup></span>/ps).
</div>
<div class="Description">
<span class="Description-entry">r_&lt;suffix&gt;</span> Overall mean square displacement(s) (in Å<span class="formula"><sup><span class="text">2</span></sup></span>/ps).
</div>
<div class="Description">
<span class="Description-entry">a_&lt;suffix&gt;</span> Total distance traveled (in Å/ps).
</div>
<div class="Unindented">
The diffusion coefficient <b><i>D</i></b> can be calculated using the Einstein relation:
</div>
<div class="Indented">
<div class="formula">
2<i>nD</i> = <span class="limits"><sup class="limit"> </sup><span class="limit">lim</span><sub class="limit"><i>t</i> → ∞</sub></span><span class="fraction"><span class="ignored">(</span><span class="numerator"><i>MSD</i></span><span class="ignored">)/(</span><span class="denominator"><i>t</i></span><span class="ignored">)</span></span>
</div>

</div>
<div class="Indented">
Where <b><i>n</i></b> is the number of dimensions; for overall MSD n = 3, for single dimension MSD (e.g. X) n = 1, etc. Unless <b>nocalc</b> is specified, the diffusion constant is calculated automatically from MSD data sets (and written to the file specified by <b>diffout</b>) in the following manner. The slope the plot of MSD versus time is obtained via linear regression. To convert from units of Å<span class="formula"><sup><span class="text">2</span></sup></span>/ps to 1x10<sup>-5</sup> cm<sup>2</sup>/s, the slope is multiplied by 10.0/(2n). Both the calculated diffusion constants as well as the results of the fit are reported.
</div>
<div class="Indented">
Due to the fact that diffusion is currently calculated from initial positions only, diffusion calculated for small numbers of atoms will be inherently stochastic, so the results are most sensible when averaged over many atoms; for example, the diffusion of water should be calculated using all waters in the system.
</div>
<div class="Indented">
For example, to calculate the diffusion of water in a system:
</div>
<pre class="LyX-Code">
diffusion :WAT@O out WAT_O.agr WAT_O diffout DC.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.23">11.23</a> dihedral 
</h2>
<pre class="LyX-Code">
dihedral<a class="IndexReference" name="entry-dihedral-0" href="#index-dihedral">↓</a> [&lt;name&gt;] &lt;mask1&gt; &lt;mask2&gt; &lt;mask3&gt; &lt;mask4&gt; [out &lt;filename&gt;] [mass]
         [type {alpha|beta|gamma|delta|epsilon|zeta|chi|c2p|h1p|phi|psi|omega|pchi}]
         [range360]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;maskX&gt;</span> Four atom masks selecting atom(s) to calculate dihedral for.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file name.
</div>
<div class="Description">
<span class="Description-entry">[mass]</span> Use center of mass of atoms in &lt;maskX&gt;; default is geometric center.
</div>
<div class="Description">
<span class="Description-entry">[range360]</span> Output dihedral angle values from 0 to 360 degrees instead of -180 to 180 degrees.
</div>
<div class="Description">
<span class="Description-entry">[type &lt;type&gt;]</span> Label dihedral as &lt;type&gt; for use with <b><i>statistics</i></b> analysis; note ’chi’ is nucleic acid chi and ’pchi’ is protein chi.
</div>

</li>

</ul>
<div class="Unindented">
Calculate dihedral angle (in degrees) between the planes defined by atoms in &lt;mask1&gt;, &lt;mask2&gt;, &lt;mask3&gt; and &lt;mask2&gt;, &lt;mask3&gt;, &lt;mask4&gt;. To calculate multiple dihedral angles see the <b><i>multidihedral</i></b> command on page <a class="Reference" href="#subsec:cpptraj_multidihedral">1↓</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.24">11.24</a> dihedralrms | dihrms
</h2>
<pre class="LyX-Code">
	dihedralrms [&lt;name&gt;] &lt;dihedral types&gt; [out &lt;file&gt;]
            	[ first | reference | ref &lt;name&gt; | refindex &lt;#&gt; | previous |
            	  reftraj &lt;name&gt; [parm &lt;name&gt; | parmindex &lt;#&gt;] ]
            	[dihtype &lt;name&gt;:&lt;a0&gt;:&lt;a1&gt;:&lt;a2&gt;:&lt;a3&gt;[:&lt;offset&gt;] ...]
                [tgtrange &lt;range&gt; [refrange &lt;range&gt;]]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;dihedral types&gt;</span> Dihedral types to look for. Note that chip is ’protein chi’, chin is ’nucleic chi’.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file name.
</div>
<div class="Description">
<span class="Description-entry">[dihtype &lt;name&gt;:&lt;a0&gt;:&lt;a1&gt;:&lt;a2&gt;:&lt;a3&gt;[:&lt;offset&gt;]</span> Search for a custom dihedral type called &lt;name&gt; using atom names &lt;a0&gt;, &lt;a1&gt;, &lt;a2&gt;, and &lt;a3&gt;.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Unindented">
Offset: -2=&lt;a0&gt;&lt;a1&gt; in previous res, -1=&lt;a0&gt; in previous res, 0=All &lt;aX&gt; in single res, 1=&lt;a3&gt; in next res, 2=&lt;a2&gt;&lt;a3&gt; in next res.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[tgtrange &lt;range&gt;]</span> Residue range to look for target dihedrals in. Default is all solute residues.
</div>
<div class="Description">
<span class="Description-entry">[refrange &lt;range&gt;]</span> Residues range to look for reference dihedrals in. If not specified, use target range.
</div>

</li>

</ul>
<div class="Unindented">
Calculate RMSD of selected dihedrals to dihedrals in a reference structure. See the <b><i>multidihedral</i></b> command syntax on page <a class="Reference" href="#subsec:cpptraj_multidihedral">1↓</a> for a list of all available dihedral types.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.25">11.25</a> dihedralscan
</h2>
<div class="Unindented">
This command has been replaced by <b><i>permutedihedrals</i></b>; see <a class="Reference" href="#subsec:cpptraj_permutedihedrals">7.8 on page 1↑</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.26">11.26</a> dipole
</h2>
<pre class="LyX-Code">
dipole &lt;filename&gt; {data &lt;dsname&gt; | &lt;nx&gt; &lt;dx&gt; &lt;ny&gt; &lt;dy&gt; &lt;nz&gt; &lt;dz&gt; 
       [gridcenter &lt;cx&gt; &lt;cy&gt; &lt;cz&gt;]}
   	[box|origin|center &lt;mask&gt;] [negative] [name &lt;gridname&gt;]
       &lt;mask1&gt; {origin | box} [max &lt;max_percent&gt;]
</pre>
<div class="Unindented">
NOTE: This command is not well-tested and may be obsolete.
</div>
<div class="Indented">
Same as <span class="sans"><b>grid</b></span> (see <a class="Reference" href="#subsec:cpptraj_grid">11.36 on page 1↓</a> below) except that dipoles of the solvent molecules are binned. The output file format is for Chris Bayly’s discern delegate program that comes with Midas/Plus. Consult the code in Action_Dipole.cpp for more information.<i> </i>
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.27">11.27</a> distance 
</h2>
<pre class="LyX-Code">
distance<a class="IndexReference" name="entry-distance-0" href="#index-distance">↓</a> [&lt;name&gt;] &lt;mask1&gt; [&lt;mask2&gt;] [point &lt;X&gt; &lt;Y&gt; &lt;Z&gt;]
         [ reference | ref &lt;name&gt; | refindex &lt;#&gt; ]
     	[out &lt;filename&gt;] [geom] [noimage] [type noe]
	Options for ’type noe’:
	  [bound &lt;lower&gt; bound &lt;upper&gt;] [rexp &lt;expected&gt;] [noe_strong] [noe_medium] [noe_weak]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Output data set name
</div>
<div class="Description">
<span class="Description-entry">&lt;mask1&gt;</span> Atom mask selecting atom(s) to calculate distance between.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask2&gt;</span> If specified, second atom mask selection atom(s) to calculate distance from &lt;mask1&gt;.
</div>
<div class="Description">
<span class="Description-entry">point &lt;X&gt; &lt;Y&gt; &lt;Z&gt;</span> If specified instead of second mask, calculate distance between &lt;mask1&gt; and specified XYZ coordinates.
</div>
<div class="Description">
<span class="Description-entry">reference | ref &lt;name&gt; | refindex &lt;#&gt;</span> If specified, calculate distance between &lt;mask1&gt; in each input frame and &lt;mask2&gt; in the specified reference.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output filename.
</div>
<div class="Description">
<span class="Description-entry">[geom]</span> Use geometric center of atoms in &lt;mask1&gt;/&lt;mask2&gt;; default is to use center of mass.
</div>
<div class="Description">
<span class="Description-entry">[noimage]</span> Do not image distances across periodic boundaries.
</div>
<div class="Description">
<span class="Description-entry">[type noe]</span> Mark distance as ’noe’ for use with <b><i>statistics</i></b> analysis.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[bound &lt;lower&gt; bound &lt;upper&gt;]</span> Lower and upper bounds for NOE (in Angstroms); must specify both.
</div>
<div class="Description">
<span class="Description-entry">[rexp &lt;expected&gt;]</span> Expected value for NOE (in Angstroms); if not given ’(&lt;lower&gt; + &lt;upper&gt;)’ / 2.0 is used.
</div>
<div class="Description">
<span class="Description-entry">[noe_strong]</span> Set lower and upper bounds to 1.8 and 2.9 Å respectively.
</div>
<div class="Description">
<span class="Description-entry">[noe_medium]</span> Set lower and upper bounds to 2.9 and 3.5 Å respectively.
</div>
<div class="Description">
<span class="Description-entry">[noe_weak]</span> Set lower and upper bounds to 3.5 and 5.0 Å respectively.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Calculate distance between the center of mass of atoms in &lt;mask1&gt; to atoms in &lt;mask2&gt;, between atoms in &lt;mask1&gt; from each input frame and atoms in &lt;mask2&gt; in specified reference, or atoms in &lt;mask1&gt; and the specified point. If <b>geom</b> is specified use the geometric center instead. For periodic systems imaging is turned on by default; the <b>noimage</b> keyword disables imaging.
</div>
<div class="Indented">
A distance can be labeled using ’<tt>type noe</tt>’ for further analysis as an NOE using the <b><i>’statistics’</i></b> analysis command (<a class="Reference" href="#subsec:cpptraj_statistics">12.34 on page 1↓</a>).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.28">11.28</a> drms | drmsd (distance RMSD)
</h2>
<pre class="LyX-Code">
drmsd<a class="IndexReference" name="entry-drmsd-0" href="#index-drmsd">↓</a> [&lt;dataset name&gt;] [&lt;mask&gt; [&lt;refmask&gt;]] [out &lt;filename&gt;]
      [ first | ref &lt;refname&gt; | refindex &lt;#&gt; | 
        reftraj &lt;trajname&gt; [parm &lt;trajparm&gt; | parmindex &lt;parm#&gt;] ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;dataset name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atoms to calculate DRMSD for.
</div>
<div class="Description">
<span class="Description-entry">[&lt;refmask&gt;]</span> Mask corresponding to atoms in reference; if not specified, &lt;mask&gt; is used.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file name.
</div>
<div class="Description">
<span class="Description-entry">[first]</span> Use the first trajectory frame processed as reference.
</div>
<div class="Description">
<span class="Description-entry"><b>[reference]</b></span> Use the first previously read in reference structure.
</div>
<div class="Description">
<span class="Description-entry"><b>[ref</b> &lt;refname&gt;]</span> Use previously read in reference structure specified by &lt;refname&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>[refindex</b> &lt;#&gt;]</span> Use previously read in reference structure specified by &lt;#&gt; (based on order read in).
</div>
<div class="Description">
<span class="Description-entry">previous</span> Use frame prior to current frame as reference.
</div>
<div class="Description">
<span class="Description-entry"><b>reftraj</b> &lt;name&gt;</span> Use frames from COORDS set &lt;name&gt; or read in from trajectory file &lt;name&gt; as references. Each frame from &lt;name&gt; is used in turn, so that frame 1 is compared to frame 1 from &lt;name&gt;, frame 2 is compared to frame 2 from &lt;name&gt; and so on. If &lt;trajname&gt; runs out of frames before processing is complete, the last frame of &lt;trajname&gt; continues to be used as the reference.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;parmname&gt; | parmindex &lt;#&gt;</span> If <b>reftraj</b> specifies a file associate trajectory &lt;name&gt; with specified topology; if not specified the first topology is used.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Calculate the distance RMSD (i.e. the RMSD of all pairs of internal distances) between atoms in the frame defined by <b>&lt;mask&gt;</b> (all if no <b>&lt;mask&gt;</b> specified) to atoms in a reference defined by <b>&lt;refmask&gt;</b> (<b>&lt;mask&gt;</b> if <b>&lt;refmask&gt;</b> not specified). Both <b>&lt;mask&gt;</b> and <b>&lt;refmask&gt;</b> must specify the same number of atoms, otherwise an error will occur. 
</div>
<div class="Indented">
Because this method compares pairs of internal distances and not absolute coordinates, it is not sensitive to translations and rotations the way that a no-fit RMSD calculation is. It can be more time consuming however, as (N<sup>2</sup>-N)/2 distances must be calculated and compared for both the target and reference structures. 
</div>
<div class="Indented">
For example, to get the DRMSD of a residue named LIG to its structure in the first frame read in:
</div>
<pre class="LyX-Code">
drmsd :LIG first out drmsd.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.29">11.29</a> dssp
</h2>
<div class="Unindented">
See <a class="Reference" href="#subsec:cpptraj-secstruct">11.72 on page 1↓</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.30">11.30</a> energy
</h2>
<pre class="LyX-Code">
energy [&lt;name&gt;] [&lt;mask1&gt;] [out &lt;filename&gt;]
   	[bond] [angle] [dihedral] {[nb14] | [e14] | [v14]}
   	{[nonbond] | [elec] [vdw]} [kinetic [ketype {vel|vv}] [dt &lt;dt&gt;]]
   	[ etype { simple |
   	          directsum [npoints &lt;N&gt;] |
   	          ewald [cut &lt;cutoff&gt;] [dsumtol &lt;dtol&gt;] [rsumtol &lt;rtol&gt;]
   	                [ewcoeff &lt;coeff&gt;] [maxexp &lt;max&gt;] [skinnb &lt;skinnb&gt;]
   	                [mlimits &lt;X&gt;,&lt;Y&gt;,&lt;Z&gt;] [erfcdx &lt;dx&gt;]
   	          pme [cut &lt;cutoff&gt;] [dsumtol &lt;dtol&gt;] [order &lt;order&gt;] [ljswidth &lt;width&gt;]
   	              [ewcoeff &lt;coeff&gt;] [ljpme] [ewcoefflj] [skinnb &lt;skinnb&gt;]
   	              [nfft &lt;nfft1&gt;,&lt;nfft2&gt;,&lt;nfft3&gt;] [erfcdx &lt;dx&gt;]
   	        } ] \begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask1&gt;]</span> Mask of atoms to calculate energy for.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write results to.
</div>
<div class="Description">
<span class="Description-entry">[bond]</span> Calculate bond energy.
</div>
<div class="Description">
<span class="Description-entry">[angle]</span> Calculate angle energy.
</div>
<div class="Description">
<span class="Description-entry">[dihedral]</span> Calculate dihedral energy.
</div>
<div class="Description">
<span class="Description-entry">[nb14]</span> Calculate nonbonded 1-4 energy.
</div>
<div class="Description">
<span class="Description-entry">[e14]</span> Calculate 1-4 electrostatics.
</div>
<div class="Description">
<span class="Description-entry">[v14]</span> Calculate 1-4 van der Waals.
</div>
<div class="Description">
<span class="Description-entry">[nonbond]</span> Calculate nonbonded energy (electrostatics and van der Waals).
</div>
<div class="Description">
<span class="Description-entry">[elec]</span> Calculate electrostatic energy (Coulomb potential).
</div>
<div class="Description">
<span class="Description-entry">[vdw]</span> Calculate van der Waals energy (Lennard-Jones 6-12 potential).
</div>
<div class="Description">
<span class="Description-entry">[etype &lt;type&gt;]</span> Calculate electrostatics via specified type.
</div>
<div class="Description">
<span class="Description-entry">[simple]</span> Use simple Coulomb term for electrostatics, no cutoff.
</div>
<div class="Description">
<span class="Description-entry">[directsum]</span> Use direct summation method for electrostatics.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[npoints &lt;N&gt;]</span> Number of cells in each direction to calculate the direct sum.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[ewald]</span> Use Ewald summation for electrostatics. If van der Waals energy will be calculated a long-range correction for periodicity will be applied.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">cut &lt;cutoff&gt;</span> Direct space cutoff in Angstroms (default 8.0).
</div>
<div class="Description">
<span class="Description-entry">dsumtol &lt;dtol&gt;</span> Direct sum tolerance (default 0.00001). Used to determine Ewald coefficient.
</div>
<div class="Description">
<span class="Description-entry">rsumtol &lt;rtol&gt;</span> Reciprocal sum tolerance (default 0.00005). Used to determine number of reciprocal space vectors.
</div>
<div class="Description">
<span class="Description-entry">ewcoeff &lt;coeff&gt;</span> Ewald coefficient in 1/Ang.
</div>
<div class="Description">
<span class="Description-entry">skinnb</span> Used to determine pairlist atoms (added to cut, so pairlist cutoff is <b>cut</b> + <b>skinnb</b>); included in order to maintain consistency with results from sander.
</div>
<div class="Description">
<span class="Description-entry">mlimits &lt;X&gt;,&lt;Y&gt;,&lt;Z&gt;</span> Explicitly set the number of reciprocal space vectors in each dimension. Will be determined automatically if not specified.
</div>
<div class="Description">
<span class="Description-entry">erfcdx &lt;dx&gt;</span> Spacing to use for the ERFC splines (default 0.0002 Ang.).
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[pme]</span> Use particle mesh Ewald for electrostatics. If van der Waals energy will be calculated a long-range correction for periodicity will be applied.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">cut &lt;cutoff&gt;</span> Direct space cutoff in Angstroms (default 8.0).
</div>
<div class="Description">
<span class="Description-entry">dsumtol &lt;dtol&gt;</span> Direct sum tolerance (default 0.00001). Used to determine Ewald coefficient.
</div>
<div class="Description">
<span class="Description-entry">order &lt;order&gt;</span> Spline order for charges.
</div>
<div class="Description">
<span class="Description-entry">ljswidth &lt;width&gt;</span> If specified, use a force-switching form for the Lennard-Jones calculation from &lt;cutoff&gt;-&lt;width&gt; to &lt;cutoff&gt;.
</div>
<div class="Description">
<span class="Description-entry">ewcoeff &lt;coeff&gt;</span> Ewald coefficient in 1/Ang.
</div>
<div class="Description">
<span class="Description-entry">ljpme</span> If specified use particle mesh Ewald for calculating Lennard-Jones interactions.
</div>
<div class="Description">
<span class="Description-entry">ewcoefflj</span> Ewald coefficient for Lennard-Jones PME.
</div>
<div class="Description">
<span class="Description-entry">skinnb</span> Used to determine pairlist atoms (added to cut, so pairlist cutoff is <b>cut</b> + <b>skinnb</b>); included in order to maintain consistency with results from sander.
</div>
<div class="Description">
<span class="Description-entry">nfft &lt;nfft1&gt;,&lt;nfft2&gt;,&lt;nfft3&gt;</span> Explicitly set the number of FFT grid points in each dimension. Will be determined automatically if not specified.
</div>
<div class="Description">
<span class="Description-entry">erfcdx &lt;dx&gt;</span> Spacing to use for the ERFC splines (default 0.0002 Ang.).
</div>

</li>

</ul>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[bond]</span> Bond energy.
</div>
<div class="Description">
<span class="Description-entry">[angle]</span> Angle energy.
</div>
<div class="Description">
<span class="Description-entry">[dih]</span> Dihedral energy.
</div>
<div class="Description">
<span class="Description-entry">[vdw14]</span> 1-4 van der Waals energy.
</div>
<div class="Description">
<span class="Description-entry">[elec14]</span> 1-4 electrostatic energy.
</div>
<div class="Description">
<span class="Description-entry">[vdw]</span> van der Waals energy.
</div>
<div class="Description">
<span class="Description-entry">[elec]</span> Electrostatic energy.
</div>
<div class="Description">
<span class="Description-entry">[total]</span> Total energy.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the energy for atoms in &lt;mask&gt;. If no terms are specified, all terms are calculated. Note that the non-bonded energy terms for <b>’simple’</b> do not take into account periodicity and there is no distance cut-off. Electrostatics can also be determined via the direct sum, Ewald, or particle-mesh Ewald summation procedures. The particle mesh Ewald functionality requires that CPPTRAJ be compiled with FFTW and a C++11 compliant compiler.
</div>
<div class="Indented">
Calculation of energy terms requires that the associated topology file have parameters for any of the calculated terms, so for example angle calculations are not possible when using a PDB file as a topology, etc. All nonbonded calculations methods other than <b>simple</b> require unit cell parameters.
</div>
<div class="Indented">
For example, to calculate all energy terms and write to a Grace-format file:
</div>
<pre class="LyX-Code">
parm DPDP.parm7 
trajin DPDP.nc 
energy DPDP out ene.agr 
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.31">11.31</a> esander
</h2>
<pre class="LyX-Code">
esander [&lt;name&gt;] [out &lt;filename&gt;] [saveforces] [parmname &lt;file&gt;] [keepfiles]
        [&lt;namelist vars&gt; ...]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write results to.
</div>
<div class="Description">
<span class="Description-entry">[saveforces]</span> If specified, save forces to frames. Requires writing frames in NetCDF format.
</div>
<div class="Description">
<span class="Description-entry">[parmname &lt;file&gt;]</span> Name of temporary topology file (default: ’CpptrajEsander.parm7’).
</div>
<div class="Description">
<span class="Description-entry">[keepfiles]</span> Keep temporary topology file after program execution.
</div>
<div class="Description">
<span class="Description-entry">[&lt;namelist vars&gt;]</span> Namelist variables supported by the sander API in format ’var &lt;value&gt;’; see below.
</div>

</li>

</ul>
<div class="Unindented">
Calculate energies for input frames using the sander API. It requires compilation with the SANDER API (sanderlib). This can be considered as a faster alternative to energy post-processing with sander (imin = 5). Currently the following sander namelist variables are supported: <b>extidel</b>, <b>intdiel</b>, <b>rgbmax</b>, <b>saltcon</b>, <b>cut</b>, <b>dielc</b>, <b>igb</b>, <b>alpb</b>, <b>gbsa</b>, <b>lj1264</b>, <b>ipb</b>, <b>inp</b>, <b>vdwmeth</b>, <b>ew_type</b>, <b>ntb</b>, <b>ntf</b>, <b>ntc</b>. See <a class="Reference" href="#chap:sander"> on page 1↓</a> for details.
</div>
<div class="Indented">
If ntb/cut/igb are not specified cpptraj will attempt to pick reasonable values based on the input system. The defaults for a non-periodic system are ntb=0, cut=9999.0, igb=1. The defaults for a periodic system are ntb=1, cut=8.0, igb=0. This currently requires writing a temporary Amber topology, the name of which can be set by <b>parmname</b>. If <b>keepfiles</b> is specified this temporary topology will not be deleted after execution.
</div>
<div class="Indented">
For example, to calculate energies for a non-periodic system using igb=1 (the default) with GB surface area turned on (gbsa=1):
</div>
<pre class="LyX-Code">
parm DPDP.parm7
trajin DPDP.nc
esander DPDP out Edpdp.dat gbsa 1 
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.32">11.32</a> filter<a class="Label" name="subsec:cpptraj-filter"> </a>
</h2>
<pre class="LyX-Code">
filter &lt;dataset1 arg&gt; min &lt;min1&gt; max &lt;max1&gt;
       [&lt;dataset2 arg&gt; min &lt;min2&gt; max &lt;max2&gt; ...]
       [out &lt;file&gt; [name &lt;setname&gt;]]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;datasetX arg&gt;</span> Data set name(s) to use for filtering
</div>
<div class="Description">
<span class="Description-entry">min &lt;minX&gt;</span> Allow values greater than &lt;min&gt; in dataset X.
</div>
<div class="Description">
<span class="Description-entry">max &lt;maxX&gt;</span> Allow values greater than &lt;max&gt; in dataset X.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File containing 1 for frames that were allowed, 0 for frames that were filtered.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;setname&gt;]</span> Filtered data set name containing 1 for allowed frames, 0 for filtered frames.
</div>

</li>

</ul>
<div class="Unindented">
For all following actions, only include frames that are between <b>&lt;min&gt;</b> and <b>&lt;max&gt;</b> of data sets in <b>&lt;dataset arg&gt;</b>. There must be at least one <b>&lt;min&gt;</b> and <b>&lt;max&gt;</b> argument, and there must be as many <b>&lt;min&gt;</b>/<b>&lt;max&gt;</b> arguments as there are specified data sets. For example, to write only frames in-between an RMSD of 0.7-0.8 Angstroms for a given input trajectory:
</div>
<pre class="LyX-Code">
trajin ../tz2.truncoct.nc
rms R1 first :2-11
filter R1 min 0.7 max 0.8 out filter.dat
outtraj maxmin.crd
</pre>
<div class="Unindented">
The output trajectory will only contain frames that meet the RMSD requirement, and the <i>filter.dat</i> file can be used to see which frames those were that were output.
</div>
<div class="Indented">
A similar command that can be used with data that already exists (e.g. it has been read in with <b><i>readdata</i></b>) is <b><i>datafilter</i></b> (see page <a class="Reference" href="#subsec:cpptraj-datafilter">1↑</a>).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.33">11.33</a> fixatomorder
</h2>
<pre class="LyX-Code">
fixatomorder [outprefix &lt;name&gt;]
</pre>
<div class="Unindented">
Cpptraj (and most of Amber) expects that atom indices in molecules to increase monotonically. However, occasionally atom indices in molecules can become disordered or non-sequential, in which case cpptraj will print an error message such as the following:
</div>
<pre class="LyX-Code">
Error: Atom 45 was assigned a lower molecule # (1) than previous atom (2).
</pre>
<div class="Unindented">
and:
</div>
<pre class="LyX-Code">
Error: Could not determine molecule information for &lt;topology file&gt;.
</pre>
<div class="Unindented">
. This command fixes atom ordering so that all atoms in molecules are sequential. The <b>outprefix</b> keyword will write out the re-ordered topology with name <b>&lt;name&gt;.&lt;original name&gt;</b>.
</div>
<div class="Indented">
For example, given an out of order topology named ’outoforder.parm7’ and a corresponding trajectory ’min1.crd’, the following will produce a reordered topology named ’reorder.outoforder.parm7’ and a reordered trajectory named ’reorder.mdcrd’:
</div>
<pre class="LyX-Code">
parm outoforder.parm7
trajin min1.crd 1 10
fixatomorder outprefix reorder
trajout reorder.mdcrd 
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.34">11.34</a> fiximagedbonds
</h2>
<pre class="LyX-Code">
fiximagedbonds [&lt;mask&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Mask expression of atoms to check.
</div>

</li>

</ul>
<div class="Unindented">
Fix bonds that have been split across periodic boundary conditions by imaging. It may be desirable to reimage the coordinates after this with <b><i>autoimage</i></b>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.35">11.35</a> gist (Grid Inhomogeneous Solvation Theory)
</h2>
<pre class="LyX-Code">
gist<a class="IndexReference" name="entry-gist-0" href="#index-gist">↓</a> [doorder] [doeij] [skipS] [skipE] [refdens &lt;rdval&gt;] [temp &lt;tval&gt;]
     [noimage] [gridcntr &lt;xval&gt; &lt;yval&gt; &lt;zval&gt;] [excludeions]
     [griddim &lt;xval&gt; &lt;yval&gt; &lt;zval&gt;] [gridspacn &lt;spaceval&gt;]
     [prefix &lt;filename prefix&gt;] [ext &lt;grid extension&gt;] [out &lt;output&gt;]
     [info &lt;info&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>[doorder]</b></span> Calculate the water order parameter <span class="bibcites">[<a class="bibliocite" name="cite-8" href="#biblio-8"><span class="bib-index">8</span></a>]</span> for each voxel.
</div>
<div class="Description">
<span class="Description-entry"><b>[doeij]</b></span> Calculate the triangular matrix representing the water-water interactions between pairs of voxels (see below).
</div>
<div class="Description">
<span class="Description-entry">[skipE]</span> Skip all energy calculations (cannot be specified with <b>’doeij’</b>).
</div>
<div class="Description">
<span class="Description-entry">[skipS]</span> Skip all entropy calculations.
</div>
<div class="Description">
<span class="Description-entry"><b>[refdens rdval&gt;]</b></span> Reference density of bulk water, used in computing<b> g_O</b>,<b> g_H</b>, and the translational entropy. Default is 0.0334 molecules/Å<span class="formula"><sup>3</sup></span>.
</div>
<div class="Description">
<span class="Description-entry">[temp &lt;tval&gt;]</span> Temperature of the input trajectory.
</div>
<div class="Description">
<span class="Description-entry">[noimage]</span> Disable distance imaging in energy calculation.
</div>
<div class="Description">
<span class="Description-entry">[excludeions]</span> If specified, exclude any ions from the calculation.
</div>
<div class="Description">
<span class="Description-entry"><b>[gridcntr &lt;xval&gt; &lt;yval&gt; &lt;zval&gt;]</b></span> Coordinates (Å) of the center of the grid (default 0.0, 0.0, 0.0).
</div>
<div class="Description">
<span class="Description-entry"><b>[griddim &lt;xval&gt; &lt;yval&gt; &lt;zval&gt;]</b></span> Grid dimensions along each coordinate axis (default 40, 40, 40).
</div>
<div class="Description">
<span class="Description-entry"><b>[gridspacn &lt;spaceval&gt;]</b></span> Grid spacing (linear dimension of each voxel) in Angstroms. Values greater than 0.75 Å are not recommended (default 0.5 Å).
</div>
<div class="Description">
<span class="Description-entry">[prefix &lt;filename prefix&gt;]</span> Output file name prefix (default &ldquo;gist&rdquo;).
</div>
<div class="Description">
<span class="Description-entry">[ext &lt;grid extension&gt;]</span> Output grid file name extension (default &ldquo;.dx&rdquo;).
</div>
<div class="Description">
<span class="Description-entry"><b>[out &lt;output&gt;]</b></span> Name of the main GIST output file. If not specified set to ’&lt;prefix&gt;-output.dat’.
</div>
<div class="Description">
<span class="Description-entry">[info &lt;info&gt;]</span> Name of main GIST info file. If not specified info is written to standard output.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[gO]</span> Number density of oxygen centers found in the voxel, in units of the bulk density.
</div>
<div class="Description">
<span class="Description-entry">[gH]</span> Number density of hydrogen centers found in the voxel in units of the reference bulk density.
</div>
<div class="Description">
<span class="Description-entry">[Esw]</span> Mean solute-water interaction energy density.
</div>
<div class="Description">
<span class="Description-entry">[Eww]</span> Mean water-water interaction energy density.
</div>
<div class="Description">
<span class="Description-entry">[dTStrans]</span> First order translational entropy density.
</div>
<div class="Description">
<span class="Description-entry">[dTSorient]</span> First order orientational entropy density .
</div>
<div class="Description">
<span class="Description-entry">[neighbor]</span> Mean number of waters neighboring the water molecules found in this voxel multiplied by the voxel number density.
</div>
<div class="Description">
<span class="Description-entry">[dipole]</span> Magnitude of mean dipole moment (polarization).
</div>
<div class="Description">
<span class="Description-entry">[order]</span> Average Tetrahedral Order Parameter.
</div>
<div class="Description">
<span class="Description-entry">[dipolex]</span> x-component of the mean water dipole moment density
</div>
<div class="Description">
<span class="Description-entry">[dipoley]</span> y-component of the mean water dipole moment density
</div>
<div class="Description">
<span class="Description-entry">[dipolez]</span> z-component of the mean water dipole moment density
</div>
<div class="Description">
<span class="Description-entry">[Eij]</span> Water-water interaction matrix.
</div>

</li>

</ul>
<div class="Unindented">
Grid Inhomogeneous Solvation Theory <span class="bibcites">[<a class="bibliocite" name="cite-21" href="#biblio-21"><span class="bib-index">21</span></a>, <a class="bibliocite" name="cite-25" href="#biblio-25"><span class="bib-index">25</span></a>]</span> (GIST) is a method for analyzing the structure and thermodynamics of solvent in the vicinity of a solute molecule. The current implementation works for only water, but the method can be generalized to other solvents whose molecules are rigid like water, such as chloroform or dimethylsulfoxide (DMSO). GIST post-processes explicit solvent simulation data to create a three-dimensional mapping of water density and thermodynamic properties within a region of interest, which is defined by a user-specified 3D rectangular grid. The small grid boxes are referred to as voxels, and each voxel is associated with solvent properties. (See Fig. <a class="Reference" href="#gist-1-1">1↓</a>.) The GIST implementation incorporated into AmberTools <i>cpptraj</i> also calculates a number of other local water properties, as listed below. GIST works for the nonpolarizable water models currently supported by AMBER.
</div>
<div class="Indented">
In order to carry out a GIST calculation, you must have a trajectory file generated with explicit water, as well as the corresponding topology file. To generate the most readily interpretable results, it is recommended that the solute (e.g., a protein) be restrained into essentially one conformation. GIST will then provide information about the structure and thermodynamics of the solvent for that conformation. For a room-temperature simulation of a solvent-exposed binding site, and a grid-spacing of 0.5 Å, it is recommended that the simulation be at least 10-20 ns in duration, and it is also a good idea to check for convergence of the GIST properties you are interested in by loading and then processing successively more frames of your trajectory file. Because GIST assumes that the solute of interest comprises all molecules in the simulation that are not waters, it is a good idea to remove all counterions and cosolutes with cpptraj’s strip command before running GIST. A sample series of cpptraj commands for running GIST is provided below.
</div>
<div class="Indented">
<div class="float">
<a class="Label" name="Figure-1"> </a><div class="figure" style="max-width: 70%;">
span class="align-center"<img class="figure" src="gist.png" alt="figure gist.png" style="max-width: 887px; max-height: 351px;"/>
 <div class="caption">
Figure 1 Diagram, in 2D, of GIST’s gridded water properties in a binding site.
</div>
<a class="Label" name="gist-1-1"> </a>
</div>

</div>

</div>
<div class="Indented">
Although it is not mandatory to supply values of <b>gridcntr</b>, <b>griddim</b> and <b>gridspcn</b>, these parameters should be carefully chosen, because they determine the region to be analyzed (<b>gridcntr</b> and <b>griddim</b>) and the spatial resolution and convergence properties of the results (<b>gridspcn</b>). In particular, although smaller grid spacings will give finer spatial resolution, longer simulation times will be needed to converge the properties in the smaller voxels that result. A larger grid spacing will allow earlier convergence, but will smooth the spatial distributions and hence can reduce accuracy. 
</div>
<div class="Indented">
The reference density of water (<b>rdval</b>) is taken by default to be the experimental number density of pure water at 300 K and 1 atm. However, different water models may yield slightly different bulk densities under these conditions, and the density also depends on T and P. If you know that the bulk density of the water model you are using, at the T and P of your simulation, deviates significantly from 0.0334 water molecules/Å<span class="formula"><sup>3</sup></span>, it would be advisable to supply the actual value with the <b>refdens</b> keyword, instead of allowing GIST to supply the default value.
</div>
<div class="Indented">
For GIST, a GPU accelerated version is available, in which the interaction energy is calculated using CUDA. When using the GPU accelerated version of GIST, the <b>doeij </b>keyword is not available. It is recommended to use a grid covering the entire box, when using the GPU implementation. You may also choose a smaller grid, but all interaction energies, i.e., each atom with each atom, will always be calculated independent of the chosen grid. This ensures optimum performance when calculating the interaction energies. Thus, the additional time required to calculate the order parameters (<b>doorder</b>) is negligible.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--15"></a><b>GIST Output</b> 
</h3>
<div class="Unindented">
GIST generates a main output file and a collection of grid data files that by default are in Data Explorer format (.dx); this can be changed via the <b>ext</b> keyword. These grid files enable visualization of the various gridded quantities, such as with the program VMD <span class="bibcites">[<a class="bibliocite" name="cite-18" href="#biblio-18"><span class="bib-index">18</span></a>]</span>. If the <b>doeij</b> keyword is provided, GIST also writes out a matrix of water-water interactions between pairs of voxels. In addition, run details are written to stdout, which can be redirected into a log file. 
</div>
<div class="Indented">
Note that a number of quantities are written out as both densities and normalized quantities. For example, the output file includes both the solute-water energy density and the normalized (per water) solute-water energy. In all cases, the normalized quantity at voxel <span class="formula"><i>i</i></span>, <span class="formula"><i>X</i><sub><i>i</i>, <i>norm</i></sub></span> is related to the corresponding density, <span class="formula"><i>X</i><sub><i>i</i>, <i>dens</i></sub></span>, by the relationship <span class="formula"><i>X</i><sub><i>i</i>, <i>norm</i></sub> = <i>ρ</i><sub><i>i</i></sub><i>X</i><sub><i>i</i>, <i>dens</i></sub></span>, where <span class="formula"><i>ρ</i><sub><i>i</i></sub></span> is the number density of water in the voxel. The normalized quantity provides information regarding the nature of the water found in the voxel. The density has the property that, if the grid extended over the entire simulation volume, the total system quantity would be given by <span class="formula"><i>X</i><sub><i>tot</i></sub> = <i>V</i><sub><i>voxel</i></sub><span class="limits"><span class="limit">∑</span></span><sub><i>i</i></sub><i>X</i><sub><i>i</i>, <i>dens</i></sub></span>, where <span class="formula"><i>V</i><sub><i>voxel</i></sub></span> is the volume of one grid voxel. 
</div>
<div class="Indented">
The main output file takes the form of a space-delimited-variable file, where each row corresponds to one voxel of the grid. This file can easily be opened with and manipulated with spreadsheet programs like Excel and LibreOffice Calc. The columns are as follows. 
</div>
<ul>
<li>
<b>index</b> - A unique, sequential integer assigned to each voxel
</li>
<li>
<b>xcoord</b> - x coordinate of the center of the voxel (Å)
</li>
<li>
<b>ycoord</b> - y coordinate of the center of the voxel (Å)
</li>
<li>
<b>zcoord</b> - z coordinate of the center of the voxel (Å)
</li>
<li>
<b>population </b>- Number of water molecule, <span class="formula"><i>n</i><sub><i>i</i></sub></span>, found in the voxel over the entire simulation. A water molecule is deemed to populate a voxel if its oxygen coordinates are inside the voxel. The expectation value of this quantity increases in proportion to the length of the simulation.
</li>
<li>
<b>g_O</b> - Number density of oxygen centers found in the voxel, in units of the bulk density (rdval). Thus, the expectation value of <b>g_O</b> for a neat water system is unity.
</li>
<li>
<b>g_H</b> - Number density of hydrogen centers found in the voxel in units of the reference bulk density (2<span class="formula"> × </span>rdval). Thus, the expectation value of <b>g_H</b> for a neat water system would be unity.
</li>
<li>
<b>dTStrans-dens</b> - First order translational entropy density (kcal/mole/Å<span class="formula"><sup>3</sup></span>), referenced to the translational entropy of bulk water, based on the value rdval.
</li>
<li>
<b>dTStrans-norm</b> - First order translational entropy per water molecule (kcal/mole/molecule), referenced to the translational entropy of bulk water, based on the value rdval. The quantity<b> dTStrans-norm</b> equals <b>dTStrans-dens</b> divided by the number density of the voxel.
</li>
<li>
<b>dTSorient-dens</b> - First order orientational entropy density (kcal/mole/Å<span class="formula"><sup>3</sup></span>), referenced to bulk solvent (see below).
</li>
<li>
<b>dTSorient-norm</b> - First order orientational entropy per water molecule (kcal/mole/water), referenced to bulk solvent (see below). This quantity equals<b> dTSorient-dens</b> divided by the number density of the voxel.
</li>
<li>
<b>Esw-dens</b> - Mean solute-water interaction energy density (kcal/mole/Å<span class="formula"><sup>3</sup></span>). This is the interaction of the solvent in a given voxel with the entire solute. Both Lennard-Jones and electrostatic interactions are computed without any cutoff, within the minimum image convention but without Ewald summation. This quantity is referenced to bulk, in the trivial sense that the solute-solvent interaction energy is zero in bulk.
</li>
<li>
<b>Esw-norm</b> - Mean solute-water interaction energy per water molecule. This equals<b> Esw-dens</b> divided by the number density of the voxel (kcal/mole/molecule).
</li>
<li>
<b>Eww-dens</b> - Mean water-water interaction energy density, scaled by ½ to prevent double-counting, and not referenced to the corresponding bulk value of this quantity (see below). This quantity is one half of the mean interaction energy of the water in a given voxel with all other waters in the system, both on and off the GIST grid, divided by the volume of the voxel (kcal/mole/Å<span class="formula"><sup>3</sup></span>). Again, both Lennard-Jones and electrostatic interactions are computed without any cutoff, within the minimum image convention.
</li>
<li>
<b>Eww-norm</b> - Mean water-water interaction energy, normalized to the mean number of water molecules in the voxel (kcal/mole/water). See prior column definition for details.
</li>
<li>
<b>Dipole_x-dens</b> - x-component of the mean water dipole moment density (Debye/Å<span class="formula"><sup>3</sup></span>).
</li>
<li>
<b>Dipole_y-dens</b> - y-component of the mean water dipole moment density (Debye/Å<span class="formula"><sup>3</sup></span>).
</li>
<li>
<b>Dipole_z-dens</b> - z-component of the mean water dipole moment density (Debye/Å<span class="formula"><sup>3</sup></span>).
</li>
<li>
<b>Dipole-dens</b> - Magnitude of mean dipole moment (polarization) (Debye/Å<span class="formula"><sup>3</sup></span>).
</li>
<li>
<b>Neighbor-dens</b> - Mean number of waters neighboring the water molecules found in this voxel multiplied by the voxel number density. Two waters are considered neighbors if their oxygens are within 3.5 angstroms of each other. For any given frame, the contribution to the average is set to zero if no water is found in the voxel (units of number/Å<span class="formula"><sup>3</sup></span>).
</li>
<li>
<b>Neighbor-norm</b> - Mean number of neighboring water molecules, per water molecule found in the voxel (units of number per water).
</li>
<li>
<b>Order-norm</b> - Average Tetrahedral Order Parameter <span class="bibcites">[<a class="bibliocite" name="cite-8" href="#biblio-8"><span class="bib-index">8</span></a>]</span>, <span class="formula"><i>q</i><sub><i>tet</i></sub></span>, for water molecules found in the voxel, normalized by the number of waters in the voxel. The order parameter for water <span class="formula"><i>i</i></span> in a given frame is given by: <span class="formula"><i>q</i><sub><i>tet</i></sub>(<i>i</i>) = 1 − <span class="fraction"><span class="ignored">(</span><span class="numerator">3</span><span class="ignored">)/(</span><span class="denominator">8</span><span class="ignored">)</span></span><span class="limits"><span class="limit">∑</span></span><span class="scripts"><sup class="script">3</sup><sub class="script"><i>j</i> = 1</sub></span><span class="limits"><span class="limit">∑</span></span><span class="scripts"><sup class="script">4</sup><sub class="script"><i>k</i> = <i>j</i> + 1</sub></span>(<i>cos</i><i>φ</i><sub><i>ijk</i></sub> + <span class="fraction"><span class="ignored">(</span><span class="numerator">1</span><span class="ignored">)/(</span><span class="denominator">3</span><span class="ignored">)</span></span>)<sup>2</sup></span> where <span class="formula"><i>j</i></span> and <span class="formula"><i>k</i></span> index the 4 closest water neighbors to water <span class="formula"><i>i</i></span>, and <span class="formula"><i>φ</i><sub><i>ijk</i></sub></span> is the angle formed by water <span class="formula"><i>i</i></span>, <span class="formula"><i>j</i></span>, and <span class="formula"><i>k</i></span>. If the doorder keyword is not provided or is set to FALSE, then this calculation will not be done, and the entries in this column will be set to zero.
</li>

</ul>
<div class="Unindented">
Grid files are provided for all computed quantities listed above, except that the normalized quantities are not included. The filenames are as follows: gist-gO.dx, gist-gH.dx, gist-dTStrans-dens.dx, gist-dTSorient-dens.dx, gist-Esw-dens.dx, gist-Eww-dens.dx, gist-dipolex-dens.dx, gist-dipoley-dens.dx, gist-dipolez-dens.dx, gist-dipole-dens.dx, gist-neighbor-dens.dx, gist-neighbor-norm.dx, gist-order-norm.dx. If the <b>doorder</b> keyword is not provided, then the data in gist-order-norm.dx will all be zeroes. Note that the file of voxel water densities, gist-gO.dx, can be used as input to the program Placevent <span class="bibcites">[<a class="bibliocite" name="cite-34" href="#biblio-34"><span class="bib-index">34</span></a>]</span>, in order to define spherical hydration sites based on the density distribution.
</div>
<div class="Indented">
Similar grid files with other computed quantities can be generated by reading the gist.out file into a spreadsheet program, processing the numbers to generate a new column of voxel data of interest, and writing this column to an ascii text file. Then the Perl script write_dx_file.pl, which should be available on the GIST tutorial web-site, may be used to read in the column of data and create the corresponding dx file. The input format, and an example, are as follows: 
</div>
<pre class="LyX-Code">
./write_dx_file.pl [filename] [x-dimension y-dimension z-dimension]
[x-origin y-origin z-origin] [grid spacing]
./write_dx_file.pl file.dat 40 40 40 13.0 13.0 13.0 0.75
</pre>
<div class="Unindented">
If the doeij keyword is provided, GIST also writes a large file, Eww_ij.dat, containing the mean water-water interaction energies between pairs of voxels, scaled by ½. (See below.) This file has three columns. The first two columns are voxel indexes, <span class="formula"><i>i</i></span>, <span class="formula"><i>j</i></span>, where <span class="formula"><i>j</i> &gt; <i>i</i></span>, so that no pair appears more than once, and the third column is the mean interaction energy (kcal/mole) of water in voxels <span class="formula"><i>i</i></span> and<span class="formula"><i>j</i></span>, scaled by ½. If the occupancy of either voxel is 0, such as for voxels covered by solute atoms, then the interaction energy is zero. In order to save space, such interactions are omitted from the file. 
</div>
<div class="Indented">
<b>Sample cpptraj input file to run GIST </b>
</div>
<div class="Indented">
The following input file, gist.in, causes cpptraj to read a parameter file named topology.top; read in the first 5000 frames of the trajectory file named trajectoryfile.mdcrd; strip out all Na and Cl ions; and carry out a GIST run which computes order parameters, uses a 41x41x45 grid centered at (25.0, 31.0, 30.0) with a spacing of 0.5 Å, uses the default bulk water density of 0.0334 molecules/Å<span class="formula"><sup>3</sup></span>, and generates the main output file gist.out. 
</div>
<pre class="LyX-Code">
parm topology.top
trajin trajectoryfile.mdcrd 1 5000
strip @Na
strip @Cl
gist doorder doeij gridcntr 25.0 31.0 30.0 griddim 41 41 45 
    gridspacn 0.50 out gist.out
go
</pre>
<div class="Unindented">
To execute this run in the background, use 
</div>
<pre class="LyX-Code">
cpptraj&lt;gist.in&gt;gist.log&amp; or cpptraj –i gist.in&gt;gist.log&amp; 
</pre>
<div class="Unindented">
<b>Referencing GIST results to unperturbed (bulk) water</b> 
</div>
<div class="Indented">
Inhomogeneous fluid solvation theory, which is the basis of GIST, is designed to provide information on how water structure and thermodynamics around a solute molecule, such as a protein, are changed relative to the structure and thermodynamics of unperturbed (bulk) water. Accordingly, the quantities reported by GIST are most informative when the results are referenced to the corresponding bulk water properties. For the orientational entropy, the reference value is the same regardless of water model or conditions, because the first order orientational distribution of water in the bulk is always uniform. Therefore, the GIST results for orientational entropies are already referenced to bulk. However, cpptraj reports unreferenced values for those GIST quantities whose reference values depend upon the water model and the simulation conditions; i.e., the energies. The translational entropy as well as the number densities will be referenced to bulk using the input referenced density or the default density value of 0.0334. The table below provides useful reference values for these quantities, computed for various water models at P=1atm, T=300K, using GIST in order to ensure a consistent minimum image treatment of periodic boundary conditions. 
</div>
<div class="Indented">
<div class="float">
<a class="Label" name="tab:cpptraj-gist-1"> </a><div class="table">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
Water Model
</td>
<td align="center" valign="top">
 Mean Energy (Eww-norm) (kcal/mol/water) 
</td>
<td align="center" valign="top">
Number Density (Å<span class="formula"><sup> − 3</sup></span>)
</td>

</tr>
<tr>
<td align="center" valign="top">
TIP3P
</td>
<td align="center" valign="top">
-9.533
</td>
<td align="center" valign="top">
0.0329
</td>

</tr>
<tr>
<td align="center" valign="top">
TIP4PEW
</td>
<td align="center" valign="top">
-11.036
</td>
<td align="center" valign="top">
0.0332
</td>

</tr>
<tr>
<td align="center" valign="top">
TIP4P
</td>
<td align="center" valign="top">
-9.856
</td>
<td align="center" valign="top">
0.0332
</td>

</tr>
<tr>
<td align="center" valign="top">
TIP5P
</td>
<td align="center" valign="top">
-9.596
</td>
<td align="center" valign="top">
0.0329
</td>

</tr>
<tr>
<td align="center" valign="top">
Tip3PFW
</td>
<td align="center" valign="top">
-11.369
</td>
<td align="center" valign="top">
0.0334
</td>

</tr>
<tr>
<td align="center" valign="top">
SPCE
</td>
<td align="center" valign="top">
-11.123
</td>
<td align="center" valign="top">
0.0333
</td>

</tr>
<tr>
<td align="center" valign="top">
SPCFW
</td>
<td align="center" valign="top">
-11.873
</td>
<td align="center" valign="top">
0.0329
</td>

</tr>

</table>

</div>
<div class="caption">
Table 2 Water model energy and density.
</div>

</div>

</div>

</div>
<div class="Indented">
Users running calculations under significantly different conditions, or with different water models, should consider generating their own reference quantities by applying GIST to a simulation of pure water under their conditions of interest. The quantities of interest can then be obtained in their most precise available form by averaging over voxels, for the pure water simulation. If the quantity of interest is <span class="formula"><i>Q</i></span>, then its average reference value is <span class="formula"><i>Q</i><sub><i>reference</i></sub> = <span class="fraction"><span class="ignored">(</span><span class="numerator"><span class="limits"><span class="limit">∑</span></span><i>n</i><sub><i>i</i></sub><i>Q</i><sub><i>i</i></sub></span><span class="ignored">)/(</span><span class="denominator"><span class="limits"><span class="limit">∑</span></span><i>n</i><sub><i>i</i></sub></span><span class="ignored">)</span></span></span>, where <span class="formula"><i>Q</i><sub><i>i</i></sub></span> and <span class="formula"><i>n</i><sub><i>i</i></sub></span> are, respectively, GIST’s reported values of the quantity and the population in voxel <span class="formula"><i>i</i></span>. The densities, <span class="formula"><i>ρ</i><sub><i>i</i></sub></span>, are referenced to the corresponding bulk densities, <span class="formula"><i>ρ</i><sup><i>o</i></sup></span>, as <span class="formula"><i>g</i><sub><i>i</i></sub> = <i>ρ</i><sub><i>i</i></sub> ⁄ <i>ρ</i><sup><i>o</i></sup></span>, while the energy and entropy terms are referenced by subtracting their bulk values.
</div>
<div class="Indented">
<b>Interpreting GIST results</b> 
</div>
<div class="Indented">
GIST provides access to the first order entropies and the first- and second-order energies of inhomogeneous fluid solvation theory. Non-zero higher-order entropies exist but are not yet computationally accessible. However, for a pairwise additive force-field, such as those listed in the Table above, the energy is fully described at the second order provided by GIST. 
</div>
<div class="Indented">
GIST is a research tool, and its applications (to, for example, protein-ligand binding and protein function) are still being explored. The following general comments may be helpful to users studying GIST results. 
</div>
<div class="Indented">
1. The water in voxels near a solute (e.g., a protein) almost always has unfavorable water-water interaction energies, relative to bulk, simply because the solute displaces water, resulting in fewer proximal water-water interactions. 
</div>
<div class="Indented">
2. The unfavorable water-water energies mentioned in <span class="bibcites">[<a class="bibliocite" name="cite-21" href="#biblio-21"><span class="bib-index">21</span></a>]</span> may be balanced by favorable water-solute interactions. If they are not, as may occur especially for voxels in small, hydrophobic pockets, then the net energy of the water in the voxel may be unfavorable relative to bulk, in which case a ligand which displaces water from the voxel into bulk may get a boost in affinity. 
</div>
<div class="Indented">
3. Because the first order orientational distribution of bulk water is uniform, and a nonuniform distribution always has lower entropy than a uniform one, the solute can only lower the orientational entropy of water, relative to bulk. Thus, this term always opposes solvation, and displacing oriented water into the bulk is always favorable from the standpoint of orientational entropy. 
</div>
<div class="Indented">
4. Localized water, which corresponds to voxels with high water density, has a low first order translational entropy, and the translational entropy around a solute is lower than that in bulk, as a nonuniform translational distribution takes the place of the uniform translational distribution of bulk water. 
</div>
<div class="Indented">
5. The displacement of highly oriented (low orientational entropy) and localized (low translational entropy) water into bulk leads to a favorable increase in these entropy terms. 
</div>
<div class="Indented">
6. However, highly oriented and localized water is often the consequence of strongly favorable polar interactions, such as hydrogen-bonding, between water and the solute. As a consequence, the net favorability of displacing such water is frequently a balance between favorable entropic consequences and unfavorable energetic consequences. 
</div>
<div class="Indented">
7. The water-water energy associated with a given voxel accounts for the interactions of the waters in this voxel with all other waters in the system, including waters in other voxels. This quantity is multiplied by ½, so that, in a pure-water system where the GIST grid covers the entire simulation box, the sum over all voxels equals the correct mean water-water interaction energy. Note that Reference <span class="bibcites">[<a class="bibliocite" name="cite-25" href="#biblio-25"><span class="bib-index">25</span></a>]</span> does not include this factor of ½. 
</div>
<div class="Indented">
8. For a typical GIST application, in which the grid occupies only part of the simulation box, the energy bookkeeping can become complicated, as discussed in Section II.B.3 (page 044101-6) of Reference <span class="bibcites">[<a class="bibliocite" name="cite-25" href="#biblio-25"><span class="bib-index">25</span></a>]</span>. That section also explains how one can compute the water-water energy associated with a region <span class="formula"><i>R</i></span> defined by a set of voxels, <span class="formula"><i>E</i><span class="scripts"><sup class="script"><i>R</i></sup><sub class="script"><i>WW</i></sub></span></span>. The regional water-water energy, on a normalized (per water) basis, is given by <span class="formula"><i>E</i><span class="scripts"><sup class="script"><i>R</i></sup><sub class="script"><i>WW</i></sub></span> = 2(<span class="limits"><span class="limit">∑</span></span><sub><i>i</i> ∈ <i>R</i></sub><i>E</i><sub><i>i</i>, <i>WW</i></sub> − <span class="limits"><span class="limit">∑</span></span><sub><i>i</i> ∈ <i>R</i></sub><span class="limits"><span class="limit">∑</span></span><sub><i>j</i> ∈ <i>R</i>, <i>j</i> &gt; <i>i</i></sub><i>E</i><sub><i>i</i>, <i>j</i>, <i>WW</i></sub>)</span> where <span class="formula"><i>i</i> ∈ <i>R</i></span> means that voxel <span class="formula"><i>i</i></span> is in region <span class="formula"><i>R</i></span>, <span class="formula"><i>E</i><sub><i>i</i>, <i>WW</i></sub></span> is the value of Eww-norm for voxel <span class="formula"><i>i</i></span>, and <span class="formula"><i>E</i><sub><i>i</i>, <i>j</i>, <i>WW</i></sub></span> is the value of the water-water interaction energy between voxels <span class="formula"><i>i</i></span> and <span class="formula"><i>j</i></span>, taken from the file Eww_ij.dat. The extra factor of 2 in the present formula, relative to that in the paper, results from application of an extra factor of ½ to the reported water-water interaction energies here. 
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.36">11.36</a> grid<a class="Label" name="subsec:cpptraj_grid"> </a>
</h2>
<pre class="LyX-Code">
grid &lt;filename&gt;
 	{ data &lt;dsname&gt; | boxref &lt;ref name/tag&gt; &lt;nx&gt; &lt;ny&gt; &lt;nz&gt; |
 	  &lt;nx&gt; &lt;dx&gt; &lt;ny&gt; &lt;dy&gt; &lt;nz&gt; &lt;dz&gt; [gridcenter &lt;cx&gt; &lt;cy&gt; &lt;cz&gt;] }
 	[box|origin|center &lt;mask&gt;] [negative] [name &lt;gridname&gt;]
 	&lt;mask&gt; [normframe | normdensity [density &lt;density&gt;]]
	 [pdb &lt;pdbout&gt; [max &lt;fraction&gt;]] [{byres|mymol}]
 	[[smoothdensity &lt;value&gt;] [invert]] [madura &lt;madura&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;filename&gt;</span> File to write out grid to. Use &ldquo;.grid&rdquo; or &ldquo;.xplor&rdquo; extension for XPLOR format, &ldquo;.dx&rdquo; for OpenDX format.
</div>
<div class="Unindented">
Options for setting up grid:
</div>
<div class="Description">
<span class="Description-entry">data &lt;dsname&gt;</span> Use previously calculated/loaded grid data set named &lt;dsname&gt;. When using this option there is no need to specify grid bins/spacing/center.
</div>
<div class="Description">
<span class="Description-entry">boxref &lt;ref name/tag&gt; &lt;nx&gt; &lt;ny&gt; &lt;nz&gt;</span> Set up grid using box information from a previously loaded reference structure. Currently the only way to set up non-orthogonal grids.
</div>
<div class="Description">
<span class="Description-entry">&lt;nx&gt; &lt;dx&gt; &lt;ny&gt; &lt;dy&gt; &lt;nz&gt; &lt;dz&gt;</span> Number of grid bins and spacing in the X/Y/Z directions.
</div>
<div class="Description">
<span class="Description-entry">[gridcenter &lt;cx&gt; &lt;cy&gt; &lt;cz&gt;]</span> Location of grid center, default is origin (0.0, 0.0, 0.0).
</div>
<div class="Unindented">
Options for offset during grid binning (must center grid at origin):
</div>
<div class="Description">
<span class="Description-entry">[box]</span> Offset each point by location of box center prior to gridding. Cannot be used with ’gridcenter’.
</div>
<div class="Description">
<span class="Description-entry">[origin]</span> No offset (default)
</div>
<div class="Description">
<span class="Description-entry">[center &lt;mask&gt;]</span> Offset each point by center of atoms in &lt;mask&gt; prior to gridding. Cannot be used with ’gridcenter’.
</div>
<div class="Unindented">
Other options:
</div>
<div class="Description">
<span class="Description-entry">[negative]</span> Grid negative density instead of positive density.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;gridname&gt;]</span> Grid data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Mask of atoms to grid.
</div>
<div class="Description">
<span class="Description-entry">[normframe]</span> Normalize grid bins by the number of frames.
</div>
<div class="Description">
<span class="Description-entry">[normdensity [density &lt;density&gt;]]</span> Normalize grid bins by density: GridBin = GridBin / (Nframes * BinVolume * density). Default particle density (molecules/Ang^3) for water based on 1.0 g/mL.
</div>
<div class="Description">
<span class="Description-entry">[pdb &lt;pdbout&gt; [max &lt;fraction&gt;]]</span> Write a pseudo-PDB of grid points that have density greater than &lt;fraction&gt; (default 0.80) of the grid max value.
</div>
<div class="Description">
<span class="Description-entry">[{byres|bymol}]</span> Grid the centers of mass of residues or molecules selected by &lt;mask&gt;.
</div>
<div class="Unindented">
Less common options:
</div>
<div class="Description">
<span class="Description-entry">[smoothdensity &lt;smooth&gt;]</span> Used to smooth density. The smoothing takes the form of GridBin = 0 if GridBin &lt; smooth, otherwise GridBin = GridBin - (GridBin * exp[-(GridBin - smooth)^2 / (0.2 * smooth^2)]).
</div>
<div class="Description">
<span class="Description-entry">[invert]</span> (Only used if smoothdensity also used) Do inverse smoothing (i.e. if GridBin &gt; smooth).
</div>
<div class="Description">
<span class="Description-entry">[madura &lt;madura&gt;]</span> Grid values lower than &lt;madura&gt; become flipped in sign, exposes low density.
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;</span> Grid data set.
</div>

</li>

</ul>
<div class="Unindented">
Create a grid representing the histogram of atoms in <i>mask1</i> on the 3D grid that is "<i>nx</i> * <i>x_spacing</i> by <i>ny</i> * <i>y_spacing</i> by <i>nz</i> * <i>z_spacing</i> angstroms (cubed). By default the grid is centered at the origin unless <b>gridcenter</b> is specified. Grid points can be offset by either the box center (using <b>box</b>) or the center of specified atoms (using <b>center &lt;mask&gt;</b>); if either of these options are used the grid must be centered at the origin. Note that the <b><i>bounds</i></b> command (<a class="Reference" href="#subsec:cpptraj-bounds"> on page 1↑</a>) can be very useful for determining grid dimensions.
</div>
<div class="Indented">
Note that when calculating grid densities for things like solvent/ions, the solute of interest (about which the atomic densities are binned) should be rms fit, centered and imaged prior to the <span class="sans"><b><i>grid</i></b></span> call in order to provide any meaningful representation of the density. If the optional keyword <tt><b>negative</b></tt> is also specified, then these density will be stored as negative numbers. Output can be in the XPLOR or OpenDX data formats.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--16"></a>Examples
</h3>
<div class="Unindented">
Grid water density around a solute.
</div>
<pre class="LyX-Code">
trajin tz2.truncoct.nc
autoimage origin
rms first :1-13
# Create average of solute to view with grid.
average avg.mol2 :1-13 
grid out.dx 20 0.5 20 0.5 20 0.5 :WAT@O 
</pre>
<div class="Unindented">
Generate grid from bounds command.
</div>
<pre class="LyX-Code">
trajin tz2.ortho.nc 
autoimage 
rms first :1-13&amp;!@H= mass
bounds :1-13 dx .5 name MyGrid out bounds.dat 
average bounds.mol2 :1-13
# Save coordinates for second pass.
createcrd MyCoords
run
# Grid using grid data set from bounds command.
crdaction MyCoords grid bounds.xplor data MyGrid :WAT@O
</pre>
<div class="Unindented">
Create non-orthogonal grid:
</div>
<pre class="LyX-Code">
trajin tz2.truncoct.nc
reference ../tz2.truncoct.nc [REF]
autoimage triclinic
grid nonortho.dx boxref [REF] 50 50 50 :WAT@O pdb nonortho.pdb
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.37">11.37</a> hbond<a class="Label" name="subsec:cpptraj-hbond"> </a>
</h2>
<pre class="LyX-Code">
hbond<a class="IndexReference" name="entry-hbond-0" href="#index-hbond">↓</a> [&lt;dsname&gt;] [out &lt;filename&gt;] [&lt;mask&gt;] [angle &lt;acut&gt;] [dist &lt;dcut&gt;]
      [donormask &lt;dmask&gt; [donorhmask &lt;dhmask&gt;]] [acceptormask &lt;amask&gt;]
      [avgout &lt;filename&gt;] [printatomnum] [nointramol] [image]
      [solventdonor &lt;sdmask&gt;] [solventacceptor &lt;samask&gt;]
      [solvout &lt;filename&gt;] [bridgeout &lt;filename&gt;] [bridgebyatom]
      [series [uuseries &lt;filename&gt;] [uvseries &lt;filename&gt;]]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;dsname&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write # of solute-solute hydrogen bonds (aspect [UU]) vs time to this file. If searching for solute-solvent hydrogen bonds, write # of solute-solvent hydrogen bonds (aspect [UV]) and # of bridging solvent molecules (aspect [Bridge]), as well as the residue # of the bridging solvent and the solute residues being bridged with format ’&lt;solvent resnum&gt;(&lt;solute res1&gt;+&lt;solute res2&gt;+...+),...’ (aspect [ID]).
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atoms to search for solute hydrogen bond donors/acceptors.
</div>
<div class="Description">
<span class="Description-entry">[angle &lt;acut&gt;]</span> Angle cutoff for hydrogen bonds (default 135°). Can be disabled by specifying -1.
</div>
<div class="Description">
<span class="Description-entry">[dist &lt;dcut&gt;]</span> Distance cutoff for hydrogen bonds (acceptor to donor heavy atom, default 3.0 Å).
</div>
<div class="Description">
<span class="Description-entry">[donormask &lt;dmask&gt;]</span> Use atoms in &lt;dmask&gt; as solute donor heavy atoms. If ’donorhmask’ not specified only atoms bonded to hydrogen will be considered donors.
</div>
<div class="Description">
<span class="Description-entry">[donorhmask &lt;dhmask&gt;]</span> Use atoms in &lt;dmask&gt; as solute donor hydrogen atoms. Should only be specified if ’donormask’ is. Should be a 1 to 1 correspondence between donormask and donorhmask.
</div>
<div class="Description">
<span class="Description-entry">[acceptormask &lt;amask&gt;]</span> Use atoms in &lt;amask&gt; as solute acceptor atoms.
</div>
<div class="Description">
<span class="Description-entry">[avgout &lt;filename&gt;]</span> Write solute-solute hydrogen bond averages to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[printatomnum]</span> Add atom numbers to the output, in addtion to residue name, residue number and atom name.
</div>
<div class="Description">
<span class="Description-entry">[nointramol]</span> Ignore intramolecular hydrogen bonds.
</div>
<div class="Description">
<span class="Description-entry">[image]</span> Turn on imaging of distances/angles.
</div>
<div class="Description">
<span class="Description-entry">[solventdonor &lt;sdmask&gt;]</span> Use atoms in &lt;sdmask&gt; as solvent donors. Can specify ions as well.
</div>
<div class="Description">
<span class="Description-entry">[solventacceptor &lt;samask&gt;]</span> Use atoms in &lt;samask&gt; as solvent acceptors. Can specify ions as well.
</div>
<div class="Description">
<span class="Description-entry">[solvout &lt;filename&gt;]</span> Write solute-solvent hydrogen bond averages to &lt;filename&gt;. If not specified and ’avgout’ is, solute-solvent hydrogen bonds averages will be written to that file.
</div>
<div class="Description">
<span class="Description-entry">[bridgeout &lt;filename&gt;]</span> Write information on detected solvent bridges to &lt;filename&gt;. If not specified, will be written to same place as ’solvout’.
</div>
<div class="Description">
<span class="Description-entry">[bridgebyatom]</span> Report bridging results by atom instead of by residue.
</div>
<div class="Description">
<span class="Description-entry">[series]</span> Save hydrogen bond formed (1.0) or not formed (0.0) per frame for any detected hydrogen bond. Solute-solute hydrogen bonds are saved with aspect [solutehb], solute-solvent hydrogen bonds are saved with aspect [solventhb].\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[uuseries &lt;filename&gt;]</span> File to write solute-solute hbond time series data to.
</div>
<div class="Description">
<span class="Description-entry">[uvseries &lt;filename&gt;]</span> File to write solute-solvent hbond time series data to.
</div>

</li>

</ul>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[UU]</span> Number of solute-solute hydrogen bonds.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[UV]</span> (only for <b>solventdonor</b>/<b>solventacceptor</b>) Number of solute-solvent hydrogen bonds.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[Bridge]</span> (only for <b>solventdonor</b>/<b>solventacceptor</b>) Number of bridging solvent molecules.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[ID]</span> (only for <b>solventdonor</b>/<b>solventacceptor</b>) String identifying bridging solvent residues and the solute residues they bridge.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[solutehb]</span> (<b>series</b> only) Time series for solute-solute hydrogen bonds; 1 for present, 0 for not present.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[solventhb]</span> (<b>series</b> only) Time series for solute-solvent hydrogen bonds; 1 for present, 0 for not present.
</div>

</li>

</ul>
<div class="Unindented">
<i>Note that <b>series</b> data sets are not generated until hydrogen bonds are actually determined (i.e. <b>run</b> is called).</i>
</div>
<div class="Indented">
Determine hydrogen bonds in each coordinate frame using simple geometric criteria. A hydrogen bond is defined as being between an acceptor heavy atom A, a donor hydrogen atom H, and a donor heavy atom D. If the A to D distance is less than the distance cutoff and the A-H-D angle is greater than the angle cutoff a hydrogen bond is considered formed. Imaging of distances/angles is not performed by default, but can be turned on using the <b>image</b> keyword.
</div>
<div class="Indented">
Potential hydrogen bond donor/acceptor atoms are searched for as follows:
</div>
<ol>
<li>
If just <b>&lt;mask&gt;</b> is specified donors and acceptors will be automatically determined from <b>&lt;mask&gt;</b>.
</li>
<li>
If <b>donormask</b> is specified donors will be determined from <b>&lt;dmask&gt;</b> (only atoms bonded to hydrogen will be considered valid). Optionally, <b>donorhmask</b> can be used in conjunction with <b>donormask</b> to explicitly specify the hydrogen atoms bonded to donor atoms. Acceptors will be automatically determined from <b>&lt;mask&gt;</b>.
</li>
<li>
If <b>acceptormask</b> is specified acceptors will be determined from <b>&lt;amask&gt;</b>. Donors will be automatically determined from <b>&lt;mask&gt;</b>.
</li>
<li>
If both <b>acceptormask</b> and <b>donormask</b> are specified only <b>&lt;amask&gt;</b> and <b>&lt;dmask&gt;</b> will be used; no searching will occur in <b>&lt;mask&gt;</b>.
</li>

</ol>
<div class="Unindented">
Automatic determination of hydrogen bond donors/acceptors uses the simplistic criterion that &ldquo;hydrogen bonds are FON&rdquo;, i.e., hydrogens bonded to F, O, and N atoms are considered donors, and F, O, and N atoms are considered acceptors. Intra-molecular hydrogen bonds can be ignored using the <b>nointramol</b> keyword.
</div>
<div class="Indented">
The number of hydrogen bonds present at each frame will be determined and written to the file specified by <b>out</b>. If the <b>series</b> keyword is specified the time series for each hydrogen bond (1 for present, 0 for not present) will also be saved for subsequent analysis (e.g. with <b><i>lifetime</i></b>, see <a class="Reference" href="#subsec:cpptraj-lifetime"> on page 1↓</a>); solute-solute hydrogen bonds will be saved to ’&lt;dataset name&gt;[solutehb]’ and solute-solvent hydrogen bonds will be saved to ’&lt;dataset name&gt;[solventhb]’. The data set legends are set with the residues and atoms involved in the hydrogen bonds. In the case of solute to non-specific solvent hydrogen bonds, a V is used in place of solvent.
</div>
<div class="Indented">
If <b>avgout</b> is specified the average of each solute-solute hydrogen bond (sorted by population) formed over the course of the trajectory is printed with the format:
</div>
<pre class="LyX-Code">
Acceptor   DonorH   Donor   Frames   Frac   AvgDist   AvgAng
</pre>
<div class="Unindented">
where <i>Acceptor</i>, <i>DonorH</i>, and <i>Donor</i> are the residue and atom name of the atoms involved in the hydrogen bond, <i>Frames</i> is the number of frames the bond is present, <i>Frac</i> is the fraction of frames the bond is present, <i>AvgDist</i> is the average distance of the bond when present, and <i>AvgAng</i> is the average angle of the bond when present. The <b>printatomnum</b> keyword can be used to print atom numbers as well.
</div>
<div class="Indented">
Solute to non-specific solvent hydrogen bonds can be tracked by using the <b>solventdonor</b> and/or <b>solventacceptor</b> keywords. The number of solute-solvent hydrogen bonds and number of &ldquo;bridging&rdquo; solvent molecules (i.e. solvent that is hydrogen bonded to two or more different solute residues at the same time) will also be written to the file specified by <b>out</b>. These keywords can also be used to track non-specific interactions with ions. If <b>avgout</b> or <b>solvavg</b> is specified the average of each solute solvent hydrogen bond will be printed with the format:
</div>
<pre class="LyX-Code">
Acceptor   DonorH   Donor   Count   Frac   AvgDist   AvgAng
</pre>
<div class="Unindented">
where <i>Acceptor</i>, <i>DonorH</i>, and <i>Donor</i> are either the residue and atom name of the solute atoms or &ldquo;SolventAcc&rdquo;/&rdquo;SolventH&rdquo;/&rdquo;SolventDnr&rdquo; representing solvent, <i>Count</i> is the total number of interactions between solute and solvent (note this can be greater than the total number of frames since for any given frame more than one solvent molecule can hydrogen bond to the same place on solute and vice versa), <i>AvgDist</i> is the average distance of the bond when present, and <i>AvgAng</i> is the average angle of the bond when present. If <b>avgout</b> or <b>bridgeout</b> is specified information on residues that were bridged by a solvent molecule over the course of the trajectory will be written to &lt;bfilename&gt; with format:
</div>
<pre class="LyX-Code">
Bridge Res &lt;N0:RES0&gt; &lt;N1:RES1&gt; ... , &lt;X&gt; frames.
</pre>
<div class="Unindented">
where ’&lt;N0:RES0&gt; ...’ is a list of residues that were bridged (residue # followed by residue name) and &lt;X&gt; is the number of frames the residues were bridged.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--17"></a>hbond Examples
</h3>
<div class="Unindented">
To search for all hydrogen bonds within residues 1-22, writing the number of hydrogen bonds per frame to &ldquo;nhb.dat&rdquo; and information on each hydrogen bond found to &ldquo;avghb.dat&rdquo;:
</div>
<pre class="LyX-Code">
hbond :1-22 out nhb.dat avgout avghb.dat
</pre>
<div class="Unindented">
To search for all hydrogen bonds formed between donors in residue 1 and acceptors in residue 2:
</div>
<pre class="LyX-Code">
hbond donormask :1 acceptormask :2 out nhb.dat avgout avghb.dat
</pre>
<div class="Unindented">
To search for all intermolecular hydrogen bonds only and solute-solvent hydrogen bonds, saving time series data to HB:
</div>
<pre class="LyX-Code">
hbond HB out nhb.dat avgout solute_avg.dat \
  solventacceptor :WAT@O solventdonor :WAT \
  solvout solvent_avg.dat bridgeout bridge.dat \
  series uuseries uuhbonds.agr uvseries uvhbonds.agr
</pre>
<div class="Unindented">
To search for non-specific hydrogen bonds between solute and ions named Na+:
</div>
<pre class="LyX-Code">
hbond HB-Ion out nhb.agr avgout ion_avg.dat \
  solventacceptor :Na+ solventdonor :Na+
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.38">11.38</a> image
</h2>
<pre class="LyX-Code">
image<a class="IndexReference" name="entry-image-0" href="#index-image">↓</a> [origin] [center] [triclinic | familiar [com &lt;commask&gt;]] [&lt;mask&gt;]
      [ bymol | byres | byatom ] [xoffset &lt;x&gt;] [yoffset &lt;y&gt;] [zoffset &lt;z&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[origin]</span> Image to coordinate origin (0.0, 0.0, 0.0); default is to image to box center.
</div>
<div class="Description">
<span class="Description-entry">[center]</span> For bymol/byres, image by center of mass; default is to image by first atom position.
</div>
<div class="Description">
<span class="Description-entry">[triclinic]</span> Force imaging with triclinic code. This is the default for non-orthorhombic cells.
</div>
<div class="Description">
<span class="Description-entry">[familiar [com &lt;commask&gt;]]</span> Image to truncated octahedron shape. If ’com &lt;commask&gt;’ is given, image with respect to the center of mass of atoms in &lt;commask&gt;.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Image atoms/residues/molecules in mask.
</div>
<div class="Description">
<span class="Description-entry">[bymol]</span> Image by molecule (default).
</div>
<div class="Description">
<span class="Description-entry">[byres]</span> Image by residue.
</div>
<div class="Description">
<span class="Description-entry">[byatom]</span> Image by atom.
</div>
<div class="Description">
<span class="Description-entry">[xoffset &lt;x&gt;]</span> Shift atoms by a factor of &lt;x&gt; in the X-direction.
</div>
<div class="Description">
<span class="Description-entry">[yoffset &lt;y&gt;]</span> Shift atoms by a factor of &lt;y&gt; in the Y-direction.
</div>
<div class="Description">
<span class="Description-entry">[zoffset &lt;z&gt;]</span> Shift atoms by a factor of &lt;z&gt; in the Z-direction.
</div>

</li>

</ul>
<div class="Unindented">
Note this command is intended for advanced use; for most cases the <b><i>autoimage</i></b> command should be sufficient.
</div>
<div class="Indented">
For periodic systems only, image molecules/residues/atoms that are outside of the box back into the box. Currently both orthorhombic and non-orthorhombic boxes are supported. A typical use of <b><i>image</i></b> is to move molecules back into the box after performing <b><i>center</i></b>. For example, the following commands move all atoms so that the center of residue 1 is at the center of the box, then image so that all molecules that are outside the box after centering are wrapped back inside:
</div>
<pre class="LyX-Code">
center :1
image
</pre>
<div class="Unindented">
The xoffset etc. keywords can be used to shift the entire unit cell in a certain direction by the given factor, which can be useful for visualizing trajectories with periodic boundary conditions. For example, to generate a trajectory that is offset by 1.0 box length in the X direction, one could use:
</div>
<pre class="LyX-Code">
image xoffset 1.0
trajout traj.offsetx1.nc
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.39">11.39</a> jcoupling
</h2>
<pre class="LyX-Code">
jcoupling<a class="IndexReference" name="entry-jcoupling-0" href="#index-jcoupling">↓</a> &lt;mask&gt; [outfile &lt;filename&gt;] [kfile &lt;param file&gt;] [out &lt;filename&gt;]
          [name &lt;dsname&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Atom mask in which to search for dihedrals within.
</div>
<div class="Description">
<span class="Description-entry">[outfile &lt;filename&gt;]</span> File to write j-coupling values to with fixed format.
</div>
<div class="Description">
<span class="Description-entry">[kfile &lt;param file&gt;]</span> File containing Karplus parameters (default is $AMBERHOME/dat/Karplus.txt).
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write data set output to.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;dsname&gt;]</span> Data set name.
</div>

</li>

</ul>
<div class="Unindented">
<i>Note data sets are not generated until <b>run</b> is called.</i>
</div>
<div class="Indented">
Calculate J-coupling values for all dihedrals found within <b>&lt;mask&gt;</b> (all atoms if no mask given). In order to use this function, Karplus parameters for all dihedrals which will be calculated must be loaded. By default <i>cpptraj</i> will use the data found in $AMBERHOME/dat/Karplus.txt; if this is not found <i>cpptraj</i> will look for the file specified by the $KARPLUS environment variable. 
</div>
<div class="Indented">
In the Karplus parameter file each parameter set consists of two lines for each dihedral with the format:
</div>
<pre class="LyX-Code">
[&lt;Type&gt;]&lt;Name1&gt;&lt;Name2&gt;&lt;Name3&gt;&lt;Name4&gt;&lt;A&gt;&lt;B&gt;&lt;C&gt;[&lt;D&gt;]
&lt;Resname1&gt;[&lt;Resname2&gt;...]
</pre>
<div class="Unindented">
The first line defines the parameter set for a dihedral. &lt;Type&gt; is optional; if not given the form for calculating the J-coupling will be as described by Chou et al.<span class="bibcites">[<a class="bibliocite" name="cite-9" href="#biblio-9"><span class="bib-index">9</span></a>]</span>; if ’C’ the form will be as described by Perez et al.<span class="bibcites">[<a class="bibliocite" name="cite-27" href="#biblio-27"><span class="bib-index">27</span></a>]</span>. The &lt;NameX&gt; parameters define the four atoms involved in the dihedral. Each &lt;NameX&gt; parameter is 5 characters wide, starting with a plus ’+’, minus ’-’ or space ’ ’ character indicating the atom belongs to the next, previous, or current residue. The remaining 4 characters are the atom name. The parameters &lt;A&gt;, &lt;B&gt;, &lt;C&gt;, and &lt;D&gt; are floating point values 6 characters wide describing the Karplus parameters. For the ’C’ form A, B, and C correspond to C0, C1, and C2; D is unused and should not be specified. The second line is a list of residue names (4 characters each) to which the dihedral applies. For example:
</div>
<pre class="LyX-Code">
C HA   CA   CB   HB    5.40 -1.37  3.61
ILE VAL 
</pre>
<div class="Unindented">
Describes a dihedral between atoms HA-CA-CB-HB using the Perez et al. form with constants C0=5.40, C1=-1.37, C2=3.61 applied to ILE and VAL residues.
</div>
<div class="Indented">
Output can be in both a fixed format (<b>outfile &lt;filename&gt;</b>) and using cpptraj data set/data file formatting (<b>out &lt;filename&gt;</b>). The fixed format has each dihedral that is defined from <b>&lt;mask1&gt;</b> printed along with its calculated J-coupling value for each frame, e.g.:
</div>
<pre class="LyX-Code">
#Frame 1 
1 SER HA CA CB HB2 45.334742 4.024759 
1 SER HA CA CB HB3 -69.437134 1.829510
...
</pre>
<div class="Unindented">
First the frame number is printed, then for each dihedral: Residue number, residue name, atom names 1-4 in the dihedral, the value of the dihedral, the J-coupling value.
</div>
<div class="Indented">
In <tt><i>cpptraj</i></tt> format, only the J-coupling value is written.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.40">11.40</a> lessplit
</h2>
<pre class="LyX-Code">
lessplit [out &lt;filename prefix&gt;] [average &lt;avg filename&gt;] &lt;trajout args&gt;\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[out &lt;filename prefix&gt;]</span> Write split LES trajectories to &lt;filename prefix&gt;.X, where X is an integer.
</div>
<div class="Description">
<span class="Description-entry">[average &lt;avg filename&gt;]</span> Write trajectory of averaged LES regions to &lt;avg filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">&lt;trajout args&gt;</span> Arguments for output trajectories.
</div>

</li>

</ul>
<div class="Unindented">
Split and/or average LES trajectory. At least one of <b>’out’</b> or <b>’average’</b> must be specified. If both are specified they share &lt;trajout args&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.41">11.41</a> lie
</h2>
<pre class="LyX-Code">
lie<a class="IndexReference" name="entry-lie-0" href="#index-lie">↓</a> [&lt;name&gt;] &lt;Ligand mask&gt; [&lt;Surroundings mask&gt;] [out &lt;filename&gt;] [nopbc]
    [noelec] [novdw] [cutvdw &lt;cutoff&gt;] [cutelec &lt;cutoff&gt;] [diel &lt;dielc&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[EELEC]</span> Electrostatic energy (kcal/mol).
</div>
<div class="Description">
<span class="Description-entry">[EVDW]</span> van der Waals energy (kcal/mol).
</div>

</li>

</ul>
<div class="Unindented">
For each frame, calculate the non-bonded interactions between all atoms in &lt;Ligand mask&gt; with all atoms in &lt;Surroundings mask&gt;. Electrostatic and van der Waals interactions will be calculated for all atom pairs. A separate electrostatic and van der Waals cutoff can be applied, the default is 12.0 Angstroms for both. &lt;dielc&gt; is an optional dielectric constant. Either the electrostatic or van der Waals calculations can be suppressed via the keywords noelec and novdw, respectively. Periodic boundary conditions (and the minimum image convention) can be abandoned with the &ldquo;nopbc&rdquo; keyword. Note, however, that no prior imaging is performed if the frames contain periodic boundaries. This may be useful for instances when you are simulating a microscopic droplets.
</div>
<div class="Indented">
The electrostatic interactions are calculated according to a simple shifting function shown below. The data file will contain two data sets—one for electrostatic interactions and one for van der Waals interactions. Periodic topologies and trajectories are required (i.e., explicit solvent is necessary). The minimum image convention is followed.
</div>
<div class="Indented">
<div class="formula">
<i>E</i><sub><i>elec</i></sub> = <i>k</i><span class="fraction"><span class="ignored">(</span><span class="numerator"><i>q</i><sub><i>i</i></sub><i>q</i><sub><i>j</i></sub></span><span class="ignored">)/(</span><span class="denominator"><i>r</i><sub><i>ij</i></sub></span><span class="ignored">)</span></span><span class="array"><span class="arrayrow"><span class="bracket align-left">⎛</span></span><span class="arrayrow"><span class="bracket align-left">⎝</span></span></span>1 − <span class="fraction"><span class="ignored">(</span><span class="numerator"><i>r</i><span class="scripts"><sup class="script">2</sup><sub class="script"><i>ij</i></sub></span></span><span class="ignored">)/(</span><span class="denominator"><i>r</i><span class="scripts"><sup class="script">2</sup><sub class="script"><i>cut</i></sub></span></span><span class="ignored">)</span></span><span class="array"><span class="arrayrow"><span class="bracket align-right">⎞</span></span><span class="arrayrow"><span class="bracket align-right">⎠</span></span></span><sup>2</sup>
</div>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.42">11.42</a> lipidorder
</h2>
<pre class="LyX-Code">
order out &lt;filename&gt; [x|y|z] [scd] [unsat &lt;mask&gt;] 
      [taildist &lt;filename&gt; [delta &lt;resolution&gt;] tailstart &lt;mask&gt;
       tailend &lt;mask&gt;] &lt;mask0&gt; ... &lt;maskN&gt; \begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">out</span> Output file for order parameters: Sx, Sy, Sz (each succeeded by the standard deviation), and two estimates for the deuterium-order parameter |SCD| = 0.5Sz and |SCD| = -(2Sx + Sy)/3. If scd is set then the order parameter directly computed from the C-H vectors is output.
</div>
<div class="Description">
<span class="Description-entry">x|y|z</span> Reference axis. (z) 
</div>
<div class="Description">
<span class="Description-entry">unsat</span> Mask for unsaturated bonds. Sz is calculated for vector Cn-Cn+1. This is only relevant if scd (below) is not set, i.e. order parameters are calculated from carbon position only. 
</div>
<div class="Description">
<span class="Description-entry">scd</span> Calculate the deuterium-order parameter |SCD| directly from the C-H vectors (masks must contain C-H-H triplets, see below). Otherwise the order parameter is estimated from carbon positions only (masks must contain only relevant carbons). (false) 
</div>
<div class="Description">
<span class="Description-entry">taildist</span> Optional output file for end-to-end distances. 
</div>
<div class="Description">
<span class="Description-entry">delta</span> Optional resolution for taildist. (0.1) 
</div>
<div class="Description">
<span class="Description-entry">tailstart</span> Mask for the start of the tail. Must be given if taildist. 
</div>
<div class="Description">
<span class="Description-entry">tailend</span> Mask for the end of the tail. Must be given if taildist. 
</div>
<div class="Description">
<span class="Description-entry">mask0</span> ... maskN Masks for each group in the lipid chain.
</div>

</li>

</ul>
<div class="Unindented">
The order parameters <span class="formula"><i>S</i><sub><i>x</i></sub></span>, <span class="formula"><i>S</i><sub><i>y</i></sub></span>, <span class="formula"><i>S</i><sub><i>z</i></sub></span> and |SCD| are calculated. Carbons must be given in bonding order. If <b>scd</b> the masks must be made up of C-H-H triples, hence hydrogens to double bonds must be enumerated twice while methyl groups require an additional mask which will also create two entries in the output.
</div>
<div class="Indented">
<span class="formula"><i>S</i><sub><i>z</i></sub></span>is the vector joining carbons <span class="formula"><i>C</i><sub><i>n</i> − 1</sub></span> and <span class="formula"><i>C</i><sub><i>n</i> + 1</sub></span>, <span class="formula"><i>S</i><sub><i>x</i></sub></span> the vector normal to the <span class="formula"><i>C</i><sub><i>n</i> − 1</sub> − <i>C</i><sub><i>n</i></sub></span> and <span class="formula"><i>C</i><sub><i>n</i></sub> − <i>C</i><sub><i>n</i> + 1</sub></span> plane and <span class="formula"><i>S</i><sub><i>y</i></sub></span> is the third axis in the molecular coordinate system. The order parameter is then calculated from <span class="formula"><i>Sc</i> = 0.5 &lt; 3cos(2<i>θ</i>) &gt;  − 1</span>, where <span class="formula"><i>θ</i></span> is the angle to the chosen reference axis. See example input file.
</div>
<div class="Indented">
Example input (all atom names according to CHARMM27 force field for POPC). 
</div>
<div class="Indented">
sn1 chain: order parameters <span class="formula"><i>S</i><sub><i>x</i></sub></span>, <span class="formula"><i>S</i><sub><i>y</i></sub></span>, <span class="formula"><i>S</i><sub><i>z</i></sub></span> and <span class="formula">|<i>SCD</i>| = 0.5 × <i>S</i><sub><i>z</i></sub></span> and <span class="formula">|<i>SCD</i>| =  − (2<i>S</i><sub><i>x</i></sub> + <i>S</i><sub><i>y</i></sub>) ⁄ 3</span>
</div>
<pre class="LyX-Code">
lipidorder out sn1.dat z taildist e2e_sn1.dat delta 0.1 \
           tailstart ":POPC@C32" tailend ":POPC@C316" \
           ":POPC@C32" ":POPC@C33" ":POPC@C34" ":POPC@C35" \
           ":POPC@C36" ":POPC@C37" ":POPC@C38" ":POPC@C39" \
           ":POPC@C310" ":POPC@C311" ":POPC@C312" ":POPC@C313" \
           ":POPC@C314" ":POPC@C315" ":POPC@C316"
</pre>
<div class="Unindented">
See also <tt>$AMBERHOME/AmberTools/test/cpptraj/Test_LipidOrder</tt>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.43">11.43</a> lipidscd
</h2>
<pre class="LyX-Code">
lipidscd [&lt;name&gt;] [&lt;mask&gt;] [{x|y|z}] [out &lt;file&gt;] [p2]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Atom mask specifying where to search for lipids.
</div>
<div class="Description">
<span class="Description-entry">x|y|z</span> Axis to calculate order parameters with respect to (default z).
</div>
<div class="Description">
<span class="Description-entry">out &lt;file&gt;</span> File to write order parameters to.
</div>
<div class="Description">
<span class="Description-entry">p2</span> If specified, report raw &lt;P2&gt; values.
</div>
<div class="Unindented">
DataSets Generated:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[H1]:&lt;idx&gt;</span> Hold lipid order parameters for each C-H1. Each lipid type will have a different &lt;idx&gt; starting from 0.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[H2]:&lt;idx&gt;</span> Hold lipid order parameters for each C-H2. If no H2, the C-H1 value will be used.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[H3]:&lt;idx&gt;</span> Hold lipid order parameters for each C-H3. If no H3, the C-H2/C-H1 value will be used.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[SDHX]:&lt;idx&gt;</span> Hold standard deviation of lipid order parameters for each C-HX.
</div>

</li>

</ul>
<div class="Unindented">
Calculate lipid order parameters SCD (|&lt;P2&gt;|) for lipid chains in mask   &lt;mask&gt;. Lipid chains are identified by carboxyl groups, i.e.   O-(C=O)-C1-...-CN, where C1 is the first carbon in the acyl chain   and CN is the last. Order parameters will be determined for each   hydrogen bonded to each carbon. If ’p2’ is specified the raw &lt;P2&gt;   values will be reported.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.44">11.44</a> makestructure
</h2>
<pre class="LyX-Code">
makestructure &lt;List of Args&gt;
</pre>
<div class="Unindented">
Apply dihedrals to specified residues using arguments found in &lt;List of Args&gt;, where an argument is 1 or more of the following arg types: 
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--18"></a>&lt;sstype keyword&gt;:&lt;res range&gt;
</h3>
<div class="Unindented">
Apply secondary structure type (via phi/psi backbone angles) to residues in given range. If the secondary structure type is a turn, the residue range must correspond to a multiple of 2 residues.
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
<b>Keyword</b>
</td>
<td align="center" valign="top">
<b>phi, psi (deg.)</b>
</td>
<td align="center" valign="top">
<b># residues</b>
</td>

</tr>
<tr>
<td align="center" valign="top">
alpha
</td>
<td align="center" valign="top">
-57.8, -47.0
</td>
<td align="center" valign="top">
1
</td>

</tr>
<tr>
<td align="center" valign="top">
left
</td>
<td align="center" valign="top">
-57.8, 47.0
</td>
<td align="center" valign="top">
1
</td>

</tr>
<tr>
<td align="center" valign="top">
pp2
</td>
<td align="center" valign="top">
-75.0, 145.0
</td>
<td align="center" valign="top">
1
</td>

</tr>
<tr>
<td align="center" valign="top">
hairpin
</td>
<td align="center" valign="top">
-100.0, 130.0
</td>
<td align="center" valign="top">
1
</td>

</tr>
<tr>
<td align="center" valign="top">
extended
</td>
<td align="center" valign="top">
-150.0, 155.0
</td>
<td align="center" valign="top">
1
</td>

</tr>
<tr>
<td align="center" valign="top">
typeI
</td>
<td align="center" valign="top">
-60.0, -30.0 | -90.0, 0.0
</td>
<td align="center" valign="top">
2
</td>

</tr>
<tr>
<td align="center" valign="top">
typeII
</td>
<td align="center" valign="top">
-60.0, 120.0 | 80.0, 0.0
</td>
<td align="center" valign="top">
2
</td>

</tr>
<tr>
<td align="center" valign="top">
typeVIII
</td>
<td align="center" valign="top">
-60.0, -30.0 | -120.0, 120.0
</td>
<td align="center" valign="top">
2
</td>

</tr>
<tr>
<td align="center" valign="top">
typeI’
</td>
<td align="center" valign="top">
60.0, 30.0 | 90.0, 0.0
</td>
<td align="center" valign="top">
2
</td>

</tr>
<tr>
<td align="center" valign="top">
typeII
</td>
<td align="center" valign="top">
60.0, -120.0 | -80.0, 0.0
</td>
<td align="center" valign="top">
2
</td>

</tr>
<tr>
<td align="center" valign="top">
typeVIa1
</td>
<td align="center" valign="top">
-60.0, 120.0 | -90.0, 0.0
</td>
<td align="center" valign="top">
2
</td>

</tr>
<tr>
<td align="center" valign="top">
typeVIa2
</td>
<td align="center" valign="top">
-120.0, 120.0 | -60.0, 0.0
</td>
<td align="center" valign="top">
2
</td>

</tr>
<tr>
<td align="center" valign="top">
typeVIb
</td>
<td align="center" valign="top">
-135.0, 135.0 | -75.0, 160.0
</td>
<td align="center" valign="top">
2
</td>

</tr>

</table>

</div>

</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--19"></a>&lt;custom ss name&gt;:&lt;res range&gt;[:&lt;phi&gt;:&lt;psi&gt;]
</h3>
<div class="Unindented">
If &lt;phi&gt; and &lt;psi&gt; are given, define a custom secondary structure conformation named &lt;custom_ss&gt; and apply to residues in range. If &lt;custom_ss&gt; has been previously defined then apply it to residues in range.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--20"></a>&lt;custom turn name&gt;:&lt;res range&gt;[:&lt;phi1&gt;:&lt;psi1&gt;:&lt;phi2&gt;:&lt;psi2&gt;]
</h3>
<div class="Unindented">
If &lt;phi1&gt;, &lt;psi1&gt;, &lt;phi2&gt;, and &lt;psi2&gt; are given, defined a custom turn conformation named &lt;custom_turn&gt; and apply to residues in range (range must correspond to a multiple of 2 residues). If &lt;custom_turn&gt; has been previously defined then apply it to residues in range.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--21"></a>&lt;custom dih name&gt;:&lt;res range&gt;[:&lt;dih type&gt;:&lt;angle&gt;]
</h3>
<pre class="LyX-Code">
&lt;dih type&gt; = alpha beta gamma delta epsilon zeta nu0 nu1 nu2 nu3 nu4 
             h1p c2p chin phi psi chip omega chi2 chi3 chi4 chi5
</pre>
<div class="Unindented">
If &lt;dih type&gt; and &lt;angle&gt; are given, apply &lt;angle&gt; to selected dihedrals of type in range. If &lt;custom dih&gt; has been previously defined then apply it to residues in range.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--22"></a>&lt;custom dih name&gt;:&lt;res range&gt;[:&lt;at0&gt;:&lt;at1&gt;:&lt;at2&gt;:&lt;at3&gt;:&lt;angle&gt;[:&lt;offset&gt;]]
</h3>
<div class="Unindented">
Apply &lt;angle&gt; to dihedral defined by atoms &lt;at1&gt;, &lt;at2&gt;, &lt;at3&gt;, and &lt;at4&gt;, or use previously defined &lt;custom_dih&gt;.
</div>
<div class="Indented">
<table>
<tr>
<td align="center" valign="top">
&lt;offset&gt;
</td>
<td align="center" valign="top">
Description
</td>

</tr>
<tr>
<td align="center" valign="top">
-2
</td>
<td align="center" valign="top">
&lt;at0&gt; and &lt;at1&gt; in previous residue.
</td>

</tr>
<tr>
<td align="center" valign="top">
-1
</td>
<td align="center" valign="top">
&lt;at0&gt; in previous residue.
</td>

</tr>
<tr>
<td align="center" valign="top">
0
</td>
<td align="center" valign="top">
All atoms in single residue.
</td>

</tr>
<tr>
<td align="center" valign="top">
1
</td>
<td align="center" valign="top">
&lt;at3&gt; in next residue.
</td>

</tr>
<tr>
<td align="center" valign="top">
2
</td>
<td align="center" valign="top">
&lt;at2&gt; and &lt;at3&gt; in next residue.
</td>

</tr>

</table>

</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--23"></a>ref:&lt;range&gt;:&lt;refname&gt;[:&lt;ref range&gt;[:&lt;dih types&gt;]] [refvalsout &lt;file&gt;] [founddihout &lt;file&gt;]
</h3>
<div class="Unindented">
Apply dihedrals from residues &lt;ref_range&gt; in previously loaded reference structure &lt;refname&gt; to dihedrals in &lt;range&gt;. If &lt;ref range&gt; is specified, use those residues from reference. The dihedral types to be used (see <b>&lt;dih_type&gt;</b> above) can be specified in a comma-separated list; default is phi/psi. Note that in order to specify &lt;dih types&gt;, &lt;ref range&gt; must be specified. The ’refvalsout’ and ’founddihout’ keywords can be used to print dihedrals found in the reference and target structures respectively to files.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--24"></a>Examples
</h3>
<div class="Unindented">
Assign polyproline II structure to residues 1 through 13:
</div>
<pre class="LyX-Code">
makestructure pp2:1-13
</pre>
<div class="Unindented">
Make residues 1 and 12 ’extended’, residues 6 and 7 a type I’ turn, and two custom assignments, one (custom1) for residues 2-5, the other (custom2) for residues 8-11:
</div>
<pre class="LyX-Code">
makestructure extended:1,12 \
              custom1:2-5:-80.0:130.0:-130.0:140.0 \
              typeI’:6-7 \
              custom2:8-11:-140.0:170.0:-100.0:140.0
</pre>
<div class="Unindented">
Assign residue 5 phi 90 degrees, residues 6 and 7 phi=-70 and psi=60 degrees:
</div>
<pre class="LyX-Code">
makestructure customdih:5:phi:90 custom:6,7:-70:60
</pre>
<div class="Unindented">
Create a new dihedral named chi1 and assign it a value of 35 degrees in residue 8:
</div>
<pre class="LyX-Code">
makestructure chi1:8:N:CA:CB:CG:35
</pre>
<div class="Unindented">
Assign ’extended’ structure to residues 1 and 12, a custom turn to residues 2-5 and 8-11, and a typeI’ turn to residues 6-7:
</div>
<pre class="LyX-Code">
makestructure extended:1,12 \
              custom1:2-5:-80.0:130.0:-130.0:140.0 \
              typeI’:6-7 \
              custom1:8-11
</pre>
<div class="Unindented">
Assign secondary structure from reference structure:
</div>
<pre class="LyX-Code">
parm ../tz2.parm7
reference ../tz2.rst7
trajin pp2.rst7.save
makestructure "ref:1-13:tz2.rst7" rmsd reference
trajout fromref.pdb multi 
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.45">11.45</a> mask
</h2>
<pre class="LyX-Code">
mask<a class="IndexReference" name="entry-mask-0" href="#index-mask">↓</a> &lt;mask&gt; [maskout &lt;filename&gt;]
     [ {maskpdb &lt;filename&gt; | maskmol2 &lt;filename&gt;}
       [trajargs &lt;comma-separated args&gt;] ]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Atom mask to process.
</div>
<div class="Description">
<span class="Description-entry">maskout &lt;filename&gt;</span> Write information on atoms in &lt;mask&gt; to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">maskpdb &lt;filename&gt;</span> Write PDB of atoms in &lt;mask&gt; to &lt;name&gt;.X.
</div>
<div class="Description">
<span class="Description-entry">maskmol2 &lt;filename&gt;</span> Write Mol2 of atoms in &lt;mask&gt; to &lt;name&gt;.X.
</div>
<div class="Description">
<span class="Description-entry">trajargs &lt;comma-separated args&gt;</span> When writing output PDB/Mol2, additional trajectory arguments to pass to the output trajectory.
</div>

</li>

</ul>
<div class="Unindented">
For each frame determine all atoms that correspond to <b>&lt;mask&gt;</b>. This is most useful when using distance-based masks, since the atoms in the mask are updated for every frame read in. If <b>maskout</b> is specified information on all atoms in &lt;mask&gt; will be written to &lt;filename&gt; with format:
</div>
<pre class="LyX-Code">
#Frame AtomNum Atom ResNum Res MolNum
</pre>
<div class="Unindented">
where <tt>#Frame</tt> is the frame number, <tt>AtomNum</tt> is the number of the selected atom, <tt>Atom</tt> is the name of the selected atom, <tt>ResNum</tt> is the residue number of the selected atom, <tt>Res</tt> is the residue name, and <tt>MolNum</tt> is the molecule number of the selected atom.
</div>
<div class="Indented">
If <b>maskpdb</b> or <b>maskmol2</b> are specified a PDB/Mol2 file corresponding to &lt;mask&gt; will be written out every frame with name &ldquo;<i>&lt;name&gt;.frame#</i>&rdquo;.
</div>
<div class="Indented">
For example, to write out all residues within 3.0 Angstroms of residue 195 that are named WAT to &ldquo;Res195WAT.dat&rdquo;, as well as write out corresponding PDB files:
</div>
<pre class="LyX-Code">
mask &ldquo;(:195&lt;:3.0)&amp;:WAT&rdquo; maskout Res195WAT.dat maskpdb Res195WAT.pdb
</pre>
<div class="Unindented">
To write all out atoms outside of 5.0 Angstroms of residues named ARG to PDB files with a chain ID of ’B’:
</div>
<pre class="LyX-Code">
mask :ARG&gt;@5.0 maskpdb Outside5Arg.pdb trajargs &ldquo;chainid ’B’&rdquo;
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.46">11.46</a> matrix
</h2>
<pre class="LyX-Code">
matrix [out &lt;filename&gt;] [start &lt;#&gt;] [stop|end &lt;#&gt;] [offset &lt;#&gt;]
       [name &lt;name&gt;] [ byatom | byres [mass] | bymask [mass] ]
       [ ired [order &lt;#&gt;] ]
       [ {distcovar | idea} &lt;mask1&gt; ]
       [ {dist | correl | covar | mwcovar} &lt;mask1&gt; [&lt;mask2&gt;] ]
       [ dihcovar dihedrals &lt;dataset arg&gt; ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>[out &lt;filename&gt;]</b></span> If specified, write matrix to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>[start &lt;#&gt;] [stop|end &lt;#&gt;] [offset &lt;#&gt;]</b></span> Start, stop, and offset frames to use (as a subset of all frames read in).
</div>
<div class="Description">
<span class="Description-entry"><b>[name &lt;name&gt;]</b></span> Name of the matrix dataset (for referral in subsequent analysis).
</div>
<div class="Description">
<span class="Description-entry"><b>byatom</b></span> Write results by atom (default). This is the sole option for <tt>covar</tt>, <tt>mwcovar</tt>, and <tt>ired</tt>.
</div>
<div class="Description">
<span class="Description-entry"><b>byres</b></span> Write results by calculating an average for each residue (mass weighted if <b>mass</b> is specified).
</div>
<div class="Description">
<span class="Description-entry"><b>bymask</b></span> Write average over &lt;mask1&gt;, and if &lt;mask2&gt; is specified &lt;mask1&gt; x &lt;mask2&gt; and &lt;mask2&gt; as well (mass weighted if <b>mass</b> is specified).
</div>

</li>

</ul>
<div class="Unindented">
Calculate matrix of the specified type from input coordinate frames:
</div>
<div class="Description">
<span class="Description-entry">dist &lt;mask1&gt; [&lt;mask2&gt;]</span> Distance matrix (default).
</div>
<div class="Description">
<span class="Description-entry">correl &lt;mask1&gt; [&lt;mask2&gt;]</span> Correlation matrix (aka dynamic cross correlation<span class="bibcites">[<a class="bibliocite" name="cite-19" href="#biblio-19"><span class="bib-index">19</span></a>]</span>).
</div>
<div class="Description">
<span class="Description-entry">covar &lt;mask1&gt; [&lt;mask2&gt;]</span> Coordinate covariance matrix. 
</div>
<div class="Description">
<span class="Description-entry">mwcovar &lt;mask1&gt; [&lt;mask2&gt;]</span> Mass-weighted coordinate covariance matrix.
</div>
<div class="Description">
<span class="Description-entry">distcovar &lt;mask1&gt;</span> Distance covariance matrix.
</div>
<div class="Description">
<span class="Description-entry">idea &lt;mask1&gt;</span> Isotropically Distributed Ensemble Analysis matrix.<span class="bibcites">[<a class="bibliocite" name="cite-28" href="#biblio-28"><span class="bib-index">28</span></a>]</span>
</div>
<div class="Description">
<span class="Description-entry">ired [order &lt;#&gt;]</span> Isotropic Reorientational Eigenmode Dynamics matrix<span class="bibcites">[<a class="bibliocite" name="cite-29" href="#biblio-29"><span class="bib-index">29</span></a>]</span> with Legendre polynomials of specified order (default 1). IRED vectors must have been specified previously with ’<b><i>vector</i></b> <b>ired</b>’ (see <a class="Reference" href="#subsec:cpptraj-vector">11.84 on page 1↓</a>).
</div>
<div class="Description">
<span class="Description-entry">dihcovar dihedrals &lt;dataset arg&gt;</span> Dihedral covariance matrix. Dihedral data sets must have been previously defined with e.g. <b><i>dihedral</i></b> or <b><i>multidihedral</i></b> commands or read in externally with <b><i>readdata</i></b> and marked as dihedrals.
</div>
<div class="Unindented">
Matrix dimensions will be of the order of N x M for <b>dist</b>, <b>correl</b>, <b>idea</b>, and <b>ired</b>, 2N x 2N for <b>dihcovar</b>, 3N x 3M for <b>covar</b> and <b>mwcovar</b>, and N(N-1) x N(N-1) / 4 for <b>distcovar</b> (with N being the number of data sets in the case of <b>ired</b> and <b>dihcovar</b> and the number of atoms in <b><i><i>&lt;mask1&gt;</i></i></b> otherwise, and M being the number of atoms in <b>&lt;mask2&gt;</b> if specified or <b><i><i>&lt;mask1&gt;</i></i></b> otherwise). No mask is required for <b>ired</b>; the matrix will be made up of previously defined IRED vectors (see the <b><i>vector</i></b> command <a class="Reference" href="#subsec:cpptraj-vector"> on page 1↓</a>). Similarly no mask is required for dihcovar; dihedral data sets must have been previously defined. Only one mask can be used with <b>distcovar</b> and <b>idea</b> matrices (i.e. they can be symmetric only), otherwise one or two masks can be used (for symmetric and full matrices respectively). If two masks are specified the number of atoms covered by <i>mask1</i> must be greater than or equal to the number of atoms covered by <i>mask2,</i> and on output <b>&lt;mask1&gt;</b><i> </i>corresponds to columns while <b>&lt;mask2&gt;</b> corresponds to rows.
</div>
<div class="Indented">
Note that for backwards compatibility, output files written with ’<b>out &lt;filename&gt;</b>’ will have the options ’<b>noheader noxcol square2d</b>’ applied to them (see <a class="Reference" href="#sec:Datafile-Commands">6 on page 1↑</a> for more details). To prevent any of these from taking effect, simply specify ’<b>header</b>’, ’<b>xcol</b>’, and/or ’<b>nosquare2d</b>’ after ’<b>out &lt;filename&gt;</b>’.
</div>
<div class="Indented">
As a simple example, a distance matrix of all CA atoms is generated and output to ’distmat.dat’.
</div>
<pre class="LyX-Code">
matrix dist @CA out distmat.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.47">11.47</a> mindist
</h2>
<div class="Unindented">
This functionality is now part of the <b><i>nativecontacts</i></b> command; see <a class="Reference" href="#subsec:cpptraj-nativecontacts">11.53 on page 1↓</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.48">11.48</a> minimage
</h2>
<pre class="LyX-Code">
minimage [&lt;name&gt;] &lt;mask1&gt; &lt;mask2&gt; [out &lt;filename&gt;] [geom] [maskcenter]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask1&gt;</span> First atom mask.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask2&gt;</span> Second atom mask.
</div>
<div class="Description">
<span class="Description-entry">out &lt;filename&gt;</span> File to write to.
</div>
<div class="Description">
<span class="Description-entry">geom</span> (<b>maskcenter</b> only) If specified, use geometric center instead of center of mass.
</div>
<div class="Description">
<span class="Description-entry">maskcenter</span> Calculate distance from center of masks instead of between each atom.
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Minimum distance to an image in Ang.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[A1]</span> Atom number in mask 1 involved in minimum distance.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[A2]</span> Atom number in mask 2 involved in minimum distance.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the shortest distance to an image, i.e. the distance to a neighboring unit cell, as well as the numbers of the atoms involved in the distance. By default the distance between each atom in &lt;mask1&gt; and &lt;mask2&gt; is considered; if <b>maskcenter</b> is specified the center of the masks is used. By convention, the lower atom number is saved as A1 and the higher is saved as A2.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.49">11.49</a> molsurf
</h2>
<pre class="LyX-Code">
molsurf<a class="IndexReference" name="entry-molsurf-0" href="#index-molsurf">↓</a> [&lt;name&gt;] [&lt;mask&gt;] [out filename] [probe &lt;probe_rad&gt;]
    	[radii {gb | parse | vdw}] [offset &lt;rad_offset&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Name of surface area data set.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atoms to calculate surface area of.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write values to.
</div>
<div class="Description">
<span class="Description-entry">[probe &lt;probe_rad&gt;]</span> Probe radius (default 1.4 Angstrom).
</div>
<div class="Description">
<span class="Description-entry">[offset &lt;rad_offset&gt;]</span> Add &lt;rad_offset&gt; to each atom radius (default 0.0).
</div>
<div class="Description">
<span class="Description-entry">[radii {gb|parse|vdw}]</span> Specify radii to use:\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">gb</span> GB radii (default).
</div>
<div class="Description">
<span class="Description-entry">parse</span> PARSE radii.
</div>
<div class="Description">
<span class="Description-entry">vdw</span> van der Waals radii.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Calculate the Connolly surface area<span class="bibcites">[<a class="bibliocite" name="cite-10" href="#biblio-10"><span class="bib-index">10</span></a>]</span> of atoms in &lt;mask&gt; (default all atoms if no mask specified) using routines from molsurf (originally developed by Paul Beroza) using the probe radius specified by <b>probe</b> (1.4 Å if not specified). Note that if GB/VDW radii are not present in the topology file (e.g. for PDB files), then PARSE radii can be used. Also note that this routine only calculate absolute surface areas, i.e. it cannot be used to get the contribution of a subset of atoms to overall surface area; if such functionality is needed try the <b><i>surf</i></b> command (<a class="Reference" href="#subsec:cpptraj_surf">11.77 on page 1↓</a>).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.50">11.50</a> multidihedral<a class="Label" name="subsec:cpptraj_multidihedral"> </a>
</h2>
<pre class="LyX-Code">
multidihedral [&lt;name&gt;] &lt;dihedral types&gt; [resrange &lt;range&gt;] [out &lt;filename&gt;] [range360]
              [dihtype &lt;name&gt;:&lt;a0&gt;:&lt;a1&gt;:&lt;a2&gt;:&lt;a3&gt;[:&lt;offset&gt;] ...]
                  Offset -2=&lt;at0&gt;&lt;at1&gt; in previous res, -1=&lt;at0&gt; in previous res,
                          0=All &lt;atX&gt; in single res,
                          1=&lt;at3&gt; in next res, 2=&lt;at2&gt;&lt;at3&gt; in next res.
              &lt;dihedral types&gt; =  alpha beta gamma delta epsilon zeta 
                                  nu0 nu1 nu2 nu3 nu4 h1p c2p chin 
                                  phi psi chip omega chi2 chi3 chi4 chi5\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;dihedral types&gt;</span> Dihedral types to look for. Note that chip is ’protein chi’, chin is ’nucleic chi’.
</div>
<div class="Description">
<span class="Description-entry">[resrange &lt;range&gt;]</span> Residue range to look for dihedrals in. Default is all solute residues.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file name.
</div>
<div class="Description">
<span class="Description-entry">[range360]</span> Wrap torsion values from 0.0 to 360.0 (default is -180.0 to 180.0).
</div>
<div class="Description">
<span class="Description-entry">[dihtype &lt;name&gt;:&lt;a0&gt;:&lt;a1&gt;:&lt;a2&gt;:&lt;a3&gt;[:&lt;offset&gt;]</span> Search for a custom dihedral type called &lt;name&gt; using atom names &lt;a0&gt;, &lt;a1&gt;, &lt;a2&gt;, and &lt;a3&gt;.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Unindented">
Offset: -2=&lt;a0&gt;&lt;a1&gt; in previous res, -1=&lt;a0&gt; in previous res, 0=All &lt;aX&gt; in single res, 1=&lt;a3&gt; in next res, 2=&lt;a2&gt;&lt;a3&gt; in next res.
</div>

</li>

</ul>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[&lt;dihedral type&gt;]:&lt;#&gt;</span> Aspect corresponds to the dihedral type name (e.g. [phi], [psi], etc). The index is the residue number.
</div>

</li>

</ul>
<div class="Unindented">
<i>Note data sets are not generated until <b>run</b> is called.</i>
</div>
<div class="Indented">
Calculate specified dihedral angle types for residues in given range. By default, dihedral angles are identified based on standard Amber atom names. The resulting data sets will have aspect equal to [&lt;dihedral type&gt;] and index equal to residue #. To differentiate the chi angle, chip is used for proteins and chin for nucleic acids. For example, to calculate all phi/psi dihedrals for residues 6 to 9:
</div>
<pre class="LyX-Code">
multidihedral MyTorsions phi psi resrange 6-9 out PhiPsi_6-9.dat
</pre>
<div class="Unindented">
This will generate data sets named MyTorsions[phi]:6, MyTorsions[psi]:6, MyTorsions[phi]:7, etc. Dihedrals other than those defined in <b>&lt;dihedral types&gt;</b> can be searched for using <b>dihtype</b>. For example to create a custom dihedral type called chi1 using atoms N, CA, CB, and CG (all in the same residue), then search for and calculate the dihedral in all residues:
</div>
<pre class="LyX-Code">
multidihedral dihtype chi1:N:CA:CB:CG out custom.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.51">11.51</a> multivector
</h2>
<pre class="LyX-Code">
multivector [&lt;name&gt;] [resrange &lt;range&gt;] name1 &lt;name1&gt; name2 &lt;name2&gt; [out &lt;filename&gt;]
            [ired]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[resrange &lt;range&gt;]</span> Range of residues to look for vectors in.
</div>
<div class="Description">
<span class="Description-entry">name1 &lt;name1&gt;</span> Name of first atom in each residue.
</div>
<div class="Description">
<span class="Description-entry">name2 &lt;name2&gt;</span> Name of second atom in each residue.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write results to.
</div>

</li>

</ul>
<div class="Unindented">
Search for and calculate atomic vectors between atoms named &lt;name1&gt; and &lt;name2&gt; in residues specified by the given &lt;range&gt;; each one is equivalent to the command ’<tt><b>vector</b> &lt;name1&gt; &lt;name2&gt;</tt>’. For example, to calculate all vectors between atoms named ’N’ and atoms named ’H’ in residues 5-20, storing the results in data sets named NH and writing to NH.dat:
</div>
<pre class="LyX-Code">
multivector NH name1 N name2 H ired out NH.dat resrange 5-20
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.52">11.52</a> nastruct
</h2>
<pre class="LyX-Code">
nastruct<a class="IndexReference" name="entry-nastruct-0" href="#index-nastruct">↓</a> [&lt;dataset name&gt;] [resrange &lt;range&gt;] [naout &lt;suffix&gt;]
     	[noheader] [resmap &lt;ResName&gt;:{A,C,G,T,U} ...] [calcnohb]
         [noframespaces] [baseref &lt;file&gt;] ...
     	[hbcut &lt;hbcut&gt;] [origincut &lt;origincut&gt;] [altona | cremer]
     	[zcut &lt;zcut&gt;] [zanglecut &lt;zanglecut&gt;] [groovecalc {simple | 3dna}]
     	[{ first | reference | ref &lt;name&gt; | refindex &lt;#&gt; | allframes | guessbp}]\begin_inset Separator latexpar\end_inset
         [bptype {anti | para} ...]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;dataset name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[resrange &lt;range&gt;]</span> Residue range to search for nucleic acids in (default all).
</div>
<div class="Description">
<span class="Description-entry">[naout &lt;suffix&gt;]</span> File name suffix for output files; BP.&lt;suffix&gt; for base pair parameters, BPstep.&lt;suffix&gt; for base pair step parameters, and Helix.&lt;suffix&gt; for base pair step helical parameters.
</div>
<div class="Description">
<span class="Description-entry">[noheader]</span> Do not print header to <b>naout</b> file.
</div>
<div class="Description">
<span class="Description-entry">[resmap &lt;ResName&gt;:{A,C,G,T,U}]</span> Attempt to treat residues named &lt;ResName&gt; as if it were A, C, G, T, or U; useful for residues with modifications or non-standard residue names. This will only work if enough reference atoms are present in &lt;ResName&gt;.
</div>
<div class="Description">
<span class="Description-entry">[calcnohb]</span> Calculate parameters between bases in base pairs even if no hydrogen bonds present between them.
</div>
<div class="Description">
<span class="Description-entry">[noframespaces]</span> If specified there will be no spaces between frames in the <b>naout</b> files.
</div>
<div class="Description">
<span class="Description-entry">[baseref &lt;file&gt;]</span> Specify a custom nucleic acid base reference. One file per custom residue; multiple ’baseref’ keywords may be present. See below for details.
</div>
<div class="Description">
<span class="Description-entry">[hbcut &lt;hbcut&gt;]</span> Distance cutoff (in Angstroms) for determining hydrogen bonds between bases (default 3.5).
</div>
<div class="Description">
<span class="Description-entry">[origincut &lt;origincut&gt;]</span> Distance cutoff (in Angstroms) between base pair axis origins for determining which bases are eligible for base-pairing (default 2.5).
</div>
<div class="Description">
<span class="Description-entry">[altona]</span> Use method of Altona &amp; Sundaralingam to calculate sugar pucker (default, see <b><i>pucker</i></b> command).
</div>
<div class="Description">
<span class="Description-entry">[cremer]</span> Use method of Cremer and Pople to calculate sugar pucker (see <b><i>pucker</i></b> command).
</div>
<div class="Description">
<span class="Description-entry">[zcut]</span> Distance cutoff (in Angstroms) between base reference axes along the Z axis (i.e. stagger) for determining base pairing (default 2).
</div>
<div class="Description">
<span class="Description-entry">[zanglecut]</span> Angle cutoff (in degrees) between base reference Z axes for determining base pairing (default 65).
</div>
<div class="Description">
<span class="Description-entry">[groovecalc]</span> Groove width calculation method:\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">simple</span> Use P-P distance for major groove, O4-O4 distance for minor groove. Output to ’BP.&lt;suffix&gt;’.
</div>
<div class="Description">
<span class="Description-entry">3dna</span> Use groove width calculation of El Hassan and Calladine<span class="bibcites">[<a class="bibliocite" name="cite-23" href="#biblio-23"><span class="bib-index">23</span></a>]</span>. Output to ’BPstep.&lt;suffix&gt;’.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[first]</span> Use first frame to determine base pairing (default).
</div>
<div class="Description">
<span class="Description-entry">[reference | refindex &lt;#&gt; | ref &lt;name&gt;]</span> Reference structure to use to determine base pairing.
</div>
<div class="Description">
<span class="Description-entry">[allframes]</span> If specified determine base pairing each frame.
</div>
<div class="Description">
<span class="Description-entry">[guessbp [bptype{anti|para}]]</span> If specified base pairing will be determined based on selected NA strands. It is assumed that consecutive strands will be base-paired and that they are arranged 5’ to 3’. The specific type of base pairing between strands can be specified with one or more ’bptype’ arguments.
</div>
<div class="Unindented">
DataSets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[pucker]:X</span> Base X (residue number) sugar pucker.
</div>
<div class="Unindented">
Base pairs:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[shear]:X</span> Base pair X (starting from 1) shear.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[stretch]:X</span> Base pair stretch.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[stagger]:X</span> Base pair stagger.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[buckle]:X</span> Base pair buckle.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[prop]:X</span> Base pair propeller.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[open]:X</span> Base pair opening.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[hb]:X</span> Number of WC hydrogen bonds between bases in base pair.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[bp]:X</span> Contain 1 if bases are base paired, 0 otherwise.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[major]:X</span> (If groovecalc simple) Major groove width calculated between P atoms of each base.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[minor]:X</span> (If groovecalc simple) Minor groove width calculated between O4 atoms of each base.
</div>
<div class="Unindented">
Base pair steps:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[shift]:X</span> Base pair step X (starting from 1) shift.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[slide]:X</span> Base pair step slide.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[rise]:X</span> Base pair step rise.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[title]:X</span> Base pair step tilt.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[roll]:X</span> Base pair step roll.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[twist]:X</span> Base pair step twist.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[zp]:X</span> Base pair step Zp value.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[major]:X</span> (If groovecalc 3dna) Major groove width, El Hassan and Calladine.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[minor]:X</span> (If groovecalc 3dna) Minor groove width, El Hassan and Calladine.
</div>
<div class="Unindented">
Helical steps:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[xdisp]:X</span> Helical step X (starting from 1) X displacement.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[ydisp]:X</span> Helical Y displacement.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[hrise]:X</span> Helical rise.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[incl]:X</span> Helical inclination.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[tip]:X</span> Helical tip.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[htwist]:X</span> Helical twist.
</div>

</li>

</ul>
<div class="Unindented">
<i>Note that data sets are not created until base pairing is determined.</i>
</div>
<div class="Indented">
Calculate basic nucleic acid (NA) structure parameters for all residues in the range specified by <b>resrange</b> (or all NA residues if no range specified). Residue names are recognized with the following priority: standard Amber residue names DA, DG, DC, DT, RA, RG, RC, and RU; 3 letter residue names ADE, GUA, CYT, THY, and URA; and finally 1 letter residue names A, G, C, T, and U. Non-standard/modified NA bases can be recognized by using the <b>resmap</b> keyword. For example, to make <i>cpptraj</i> recognize all 8-oxoguanine residues named ’8OG’ as a guanine-based residue:
</div>
<pre class="LyX-Code">
nastruct naout nastruct.dat resrange 274-305 resmap 8OG:G
</pre>
<div class="Unindented">
The <b>resmap</b> keyword can be specified multiple times, but only one mapping per unique residue name is allowed. Note that <b>resmap</b> may fail if the residue is missing heavy atoms normally present in the specified base type.
</div>
<div class="Indented">
Base pairs are determined either once from the first frame or from a reference structure, or can be determined each frame if <b>allframes</b> is specified. Base pairing is determined first by base reference axis origin distance, then by stagger, then by angle between base Z axes, then finally by hydrogen bonding (at least one hydrogen bond must be present). Base pair parameters will only be written for determined base pairs. Both Watson-Crick and other types of base pairing can be detected. Note that although all possible hydrogen bonds are searched for, only WC hydrogen bonds are reported in the BP.&lt;suffix&gt; file.
</div>
<div class="Indented">
The procedure used to calculate NA structural parameters is the same as 3DNA<span class="bibcites">[<a class="bibliocite" name="cite-22" href="#biblio-22"><span class="bib-index">22</span></a>]</span>, with algorithms adapted from Babcok et al.<span class="bibcites">[<a class="bibliocite" name="cite-4" href="#biblio-4"><span class="bib-index">4</span></a>]</span> and reference frame coordinates from Olson et al.<span class="bibcites">[<a class="bibliocite" name="cite-26" href="#biblio-26"><span class="bib-index">26</span></a>]</span>. Given the same base pairs are determined, <i>cpptraj</i> nastruct gives the exact same numbers as 3DNA. 
</div>
<div class="Indented">
Calculated NA structure parameters are written to three separate files, the suffix of which is specified by <b>naout</b>. Base pair parameters (shear, stretch, stagger, buckle, propeller twist, opening, # WC hydrogen bonds, base pairing, and simple groove widths) are written to BP.&lt;suffix&gt;, along with the number of WC hydrogen bonds detected. Base pair step parameters (shift, slide, rise, tilt, roll, twist, Zp, and El Hassan and Calladine groove widths) are written to BPstep.&lt;suffix&gt;, and helical parameters (X-displacement, Y-displacement, rise, inclination, tip, and twist) are written to Helix.&lt;suffix&gt;. If <b>noheader</b> is specified a header will not be written to the output files. Note that although base puckering is calculated, it is not written to an output file by default. You can output pucker to a file via the create or write/writedata commands after the data has been generated, e.g.:
</div>
<pre class="LyX-Code">
nastruct NA naout nastruct.dat resrange 1-3,28-30
run
writedata NApucker.dat NA[pucker]
</pre>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--25"></a>Custom Nucleic Acid Base References
</h3>
<div class="Unindented">
Users can now specify <b>baseref &lt;file&gt;</b> to load a custom nucleic acid base reference. The base reference files are white-space delimited, begin with the line NASTRUCT REFERENCE, and have the following format:
</div>
<pre class="LyX-Code">
NASTRUCT REFERENCE
&lt;base character&gt; &lt;res name 0&gt; [&lt;res name 1&gt; ...]
&lt;atom name&gt; &lt;X&gt; &lt;Y&gt; &lt;Z&gt; &lt;HB type&gt; &lt;RMS fit&gt;
...
</pre>
<div class="Unindented">
There is a line for each reference atom. Lines beginning with ’#’ are ignored as comments.
</div>
<div class="Description">
<span class="Description-entry">&lt;base character&gt;</span> Used to identify the underlying base type: A G C T or U. If none of these, it will be considered an unknown residue (which just means WC hydrogen bonding will not be identified).
</div>
<div class="Description">
<span class="Description-entry">&lt;res name X&gt;</span> Specifies what residue names this reference corresponds to. There must be at least one residue name. There can be any number of these specified.
</div>
<div class="Description">
<span class="Description-entry">&lt;atom name&gt;</span> A reference atom name.
</div>
<div class="Description">
<span class="Description-entry">&lt;X&gt; &lt;Y&gt; &lt;Z&gt;</span> The X Y and Z coordinates of the reference atom.
</div>
<div class="Description">
<span class="Description-entry">&lt;HB type&gt;</span> Denotes if and how the atom participates in hydrogen bonding. Can be ’d’onor, ’a’cceptor, or ’n’one (or the numbers 1, 2, 0 respectively). Only the first character of the word actually matters.
</div>
<div class="Description">
<span class="Description-entry">&lt;RMS fit&gt;</span> Denotes whether the atom is involved in RMS-fitting.
</div>
<div class="Unindented">
Here is an example for GUA:
</div>
<pre class="LyX-Code">
NASTRUCT REFERENCE
G G G5 G3
# Modified into format readable by cpptraj nastruct
C1’ -2.477   5.399   0.000 0      0
N9  -1.289   4.551   0.000 0      1
C8   0.023   4.962   0.000 0      1
N7   0.870   3.969   0.000 accept 1
C5   0.071   2.833   0.000 0      1
C6   0.424   1.460   0.000 0      1
O6   1.554   0.955   0.000 accept 0
N1  -0.700   0.641   0.000 donor  1
C2  -1.999   1.087   0.000 0      1
N2  -2.949   0.139  -0.001 donor  0
N3  -2.342   2.364   0.001 accept 1
C4  -1.265   3.177   0.000 0      1
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.53">11.53</a> nativecontacts<a class="Label" name="subsec:cpptraj-nativecontacts"> </a>
</h2>
<pre class="LyX-Code">
nativecontacts [&lt;mask1&gt; [&lt;mask2&gt;]] [writecontacts &lt;outfile&gt;] [resout &lt;resfile&gt;]
               [noimage] [distance &lt;cut&gt;] [out &lt;filename&gt;] [includesolvent]
               [ first | reference | ref &lt;name&gt; | refindex &lt;#&gt; ]
               [resoffset &lt;n&gt;] [contactpdb &lt;file&gt;] [pdbcut &lt;cut&gt;] [mindist] [maxdist]
               [name &lt;dsname&gt;] [byresidue] [map [mapout &lt;mapfile&gt;]]
               [series [seriesout &lt;file&gt;]]\begin_inset Separator latexpar\end_inset
               [savenonnative [seriesnnout &lt;file&gt;] [nncontactpdb &lt;file&gt;]]
               [resseries { present | sum } [resseriesout &lt;file&gt;]] [skipnative]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask1&gt;</span> First mask to calculate contacts for.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask2&gt;]</span> (Optional) Second mask to calculate contacts for.
</div>
<div class="Description">
<span class="Description-entry">[writecontacts &lt;outfile&gt;]</span> Write information on native contacts to <b>&lt;outfile&gt;</b> (STDOUT if not specified).
</div>
<div class="Description">
<span class="Description-entry">[resout &lt;resfile&gt;]</span> File to write contact residue pairs to.
</div>
<div class="Description">
<span class="Description-entry">[noimage]</span> Do not image distances.
</div>
<div class="Description">
<span class="Description-entry">[distance &lt;cut&gt;]</span> Distance cutoff for determining native contacts in Angstroms (default 7.0 Ang).
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write number of native contacts and non-native contacts.
</div>
<div class="Description">
<span class="Description-entry">[includesolvent]</span> By default solvent molecules are ignored; this will explicitly include solvent molecules.
</div>
<div class="Description">
<span class="Description-entry">[first | reference | ref &lt;name&gt; | refindex &lt;#&gt;]</span> Reference structure to use for determining native contacts.
</div>
<div class="Description">
<span class="Description-entry">[resoffset &lt;n&gt;]</span> (<b>byresidue</b> only) Ignore contacts between residues spaced less than &lt;n&gt; residues apart in sequence.
</div>
<div class="Description">
<span class="Description-entry">[contactpdb &lt;file&gt;]</span> Write PDB with B-factor column containing relative contact strength for native contacts (strongest is 100.0).
</div>
<div class="Description">
<span class="Description-entry">[pdbcut &lt;cut&gt;]</span> If writing <b>contactpdb</b>, only write contacts with relative contact strength greater than <b>&lt;cut&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[mindist]</span> If specified, determine the minimum distance between any atoms in the mask(s).
</div>
<div class="Description">
<span class="Description-entry">[maxdist]</span> If specified, determine the maximum distance between any atoms in the mask(s).
</div>
<div class="Description">
<span class="Description-entry">[name &lt;dsname&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[byresidue]</span> Write out the contact map by residue instead of by atom.
</div>
<div class="Description">
<span class="Description-entry">[map]</span> Calculate matrices of native contacts ([nativemap]) and non-native contacts ([nonnatmap]). These matrices are normalized by the total number of frames, so that a value of 1.0 means &ldquo;contact always present&rdquo;. If <b>byresidue</b> specified, the values for each individual atom pair are summed over the residues they belong to (this means for <b>byresidue</b> values greater than 1.0 are possible).
</div>
<div class="Description">
<span class="Description-entry">[mapout &lt;mapfile&gt;]</span> Write native/non-native matrices to ’native.&lt;mapfile&gt;’ and ’nonnative.&lt;mapfile&gt;’ respectively.
</div>
<div class="Description">
<span class="Description-entry">[series]</span> Calculate native contact time series data, 1 for contact present and 0 otherwise.
</div>
<div class="Description">
<span class="Description-entry">[seriesout &lt;file&gt;]</span> Write native contact time series data to file.
</div>
<div class="Description">
<span class="Description-entry">[savenonnative]</span> Save non-native contacts; <b>series</b> must also be specified. This is enabled by default if <b>skipnative</b> specified.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[seriesnnout &lt;file&gt;]</span> Write non-native contact time series data to file.
</div>
<div class="Description">
<span class="Description-entry">[nncontactpdb &lt;file&gt;]</span> Write PDB with B-factor column containing relative contact strength for non-native contacts (strongest is 100.0).
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[resseries {present | sum}</span> Create contacts time series by residue; <b>series</b> must also be specified.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">present</span> Record a 1 if <i>any</i> contact is present and 0 if no contact is present for the residue pair.
</div>
<div class="Description">
<span class="Description-entry">sum</span> The sum of all individual contacts is recorded for the residue pair.
</div>
<div class="Description">
<span class="Description-entry">[resseriesout &lt;file&gt;]</span> Write residue time series data to &lt;file&gt;.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[skipnative]</span> If specified, skip native contacts determination, i.e. treat all sonctacts as non-native contacts. Implies <b>savenonnative.</b>
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[native]</span> Number of native contacts.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[nonnative]</span> Number of non-native contacts.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[mindist]</span> (mindist only) Minimum observed distance each frame.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[maxdist]</span> (maxdist only) Maximum observed distance each frame.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[nativemap]</span> (map only) Native contacts matrix (2D).
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[nonnatmap]</span> Non-native contacts matrix (2D).
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[NC]</span> Native contacts time series.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[NN]</span> Non-native contacts time series.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[NCRES]</span> Residue native contacts time series.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[NNRES]</span> Residue non-native contacts time series.
</div>

</li>

</ul>
<div class="Unindented">
Define and track &ldquo;native&rdquo; contacts as determined by a simple distance cut-off, i.e. any atoms which are closer than <b>&lt;cut&gt;</b> in the specified reference frame (the first frame if no reference specified) are considered a native contact. If one mask is provided, contacts are looked for within <b>&lt;mask1&gt;</b>; if two masks are provided, only contacts between atoms in <b>&lt;mask1&gt;</b> and atoms in <b>&lt;mask2&gt;</b> are looked for (useful for determining intermolecular contacts). By default only native contacts are tracked. This can be changed by specifying the <b>savenonnative</b> keyword or by specifying <b>skipnative</b>. The time series for contacts can be saved using the <b>series</b> keyword; these can be further consolidated by residue using the <b>resseries</b> keyword. When using &lt;resseries&gt; the data set index is calculated as (r2 * nres) + r1 so that indices can be matched between native/non-native contact pairs. Non-native residue contact legends have an nn_ prefix.
</div>
<div class="Indented">
Native contacts that are found are written to the file specified by writecontacts (or STDOUT) with format:
</div>
<pre class="LyX-Code">
# Contact Nframes Frac. Avg Stdev 
</pre>
<div class="Unindented">
Where <tt>Contact</tt> takes the form ’:&lt;residue1 num&gt;@&lt;atom name&gt;_:&lt;residue2 num&gt;@&lt;atom name&gt;, <tt>Nframes</tt> is the number of frames the contact is present, <tt>Frac.</tt> is the total fraction of frames the contact is present, <tt>Avg</tt> is the average distance of the contact when present, and <tt>Stdev</tt> is the standard deviation of the contact distance when present. If <b>resout</b> is specified the total fraction of contacts is printed for all residue pairs having native contacts with format:
</div>
<pre class="LyX-Code">
#Res1 #Res2 TotalFrac Contacts 
</pre>
<div class="Unindented">
Where <tt>#Res1</tt> is the first residue number, <tt>#Res2</tt> is the second residue number, <tt>TotalFrac</tt> is the total fraction of contacts for the residue pair, and <tt>Contacts</tt> is the total number of native contacts involved with the residue pair. Since <tt>TotalFrac</tt> is calculated for each pair as the sum of each contact involving that pair divided by the total number of frames, it is possible to have <tt>TotalFrac</tt> values greater than 1 if the residue pair includes more than 1 native contact.
</div>
<div class="Indented">
During trajectory processing, non-native contacts (i.e. any pair satisfying the distance cut-off which is not already a native contact) are also searched for. The time series for native contacts can be saved as well, with 1 for contact present and 0 otherwise (similar to the <b><i>hbond</i></b> command). This data can be subsequently analyzed using e.g.<a class="Reference" href="#subsec:cpptraj-lifetime">12.19 on page 1↓</a>.
</div>
<div class="Indented">
Contact maps (matrices) are generated for native and non-native contacts. If <b>byresidue</b> is specified, contact maps are summed over residues, and contacts between residues spaced <b>&lt;resoffset&gt;</b> residues apart in sequence are ignored.
</div>
<div class="Indented">
If <b>contactpdb</b> is specified a PDB is generated containing relative contact strengths in the B-factor column. The relative contact strength is normalized so that a value of 100 means that atom participated in the most contacts with other atoms.
</div>
<div class="Indented">
Example command looking for contacts between residues 210 to 260 and residue named NDP, using reference structure ’FtuFabI.WT.pdb’ to define native contacts:
</div>
<pre class="LyX-Code">
parm FtuFabI.parm7
trajin FtuFabI.nc
reference FtuFabI.WT.pdb
nativecontacts name NC1 :210-260&amp;!@H= :NDP&amp;!@H= \
    byresidue out nc.all.res.dat mindist maxdist \
    distance 3.0 reference map mapout resmap.gnu \
    contactpdb Loop-NDP.pdb \
    series seriesout native.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.54">11.54</a> outtraj<a class="Label" name="subsec:cpptraj_outtraj"> </a>
</h2>
<pre class="LyX-Code">
outtraj<a class="IndexReference" name="entry-outtraj-0" href="#index-outtraj">↓</a> &lt;filename&gt; [ trajout args ]
        [maxmin &lt;dataset&gt; min &lt;min&gt; max &lt;max&gt;] ...\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;filename&gt;</span> Output trajectory file name.
</div>
<div class="Description">
<span class="Description-entry">[trajout args]</span> Output trajectory arguments (see <a class="Reference" href="#subsec:cpptraj-trajout">10.5 on page 1↑</a>).
</div>
<div class="Description">
<span class="Description-entry">[maxmin &lt;dataset&gt; min &lt;min&gt; max &lt;max&gt;]</span> Only write frames to &lt;filename&gt; if values in &lt;dataset&gt; for those frames are between &lt;min&gt; and &lt;max&gt;. Can be specified for one or more data sets.
</div>

</li>

</ul>
<div class="Unindented">
The outtraj command is similar in function to <b><i>trajout</i></b>, and takes all of the same arguments. However, instead of writing a trajectory frame after all actions are complete outtraj writes the trajectory frame at its position in the Action queue. For example, given the input:
</div>
<pre class="LyX-Code">
trajin mdcrd.crd
trajout output.crd
outtraj BeforeRmsd.crd
rms R1 first :1-20@CA out rmsd.dat
outtraj AfterRmsd.crd
</pre>
<div class="Unindented">
three trajectories will be written: output.crd, BeforeRmsd.crd, and AfterRmsd.crd. The output.crd and AfterRmsd.crd trajectories will be identical, but the BeforeRmsd.crd trajectory will contain the coordinates of mdcrd.crd before they are RMS-fit.
</div>
<div class="Indented">
The maxmin keyword can be used to restrict output using one more more data sets. For example, to only write frames for which the RMSD value is between 0.7 and 0.8:
</div>
<pre class="LyX-Code">
trajin tz2.truncoct.nc
rms R1 first :2-11
outtraj maxmin.crd maxmin R1 min 0.7 max 0.8
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.55">11.55</a> pairdist
</h2>
<pre class="LyX-Code">
pairdist out &lt;filename&gt; mask &lt;mask&gt; [delta &lt;resolution&gt;]
</pre>
<div class="Unindented">
Calculate pair distribution function. In the following, defaults are given in parentheses. The <b>out</b> keyword specifies output file for histogram: distance, P(r), s(P(r)). The <b>mask</b> option specifies atoms for which distances should be computed. The <b>delta</b> option specifies resolution. (0.1 Å)
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.56">11.56</a> pairwise
</h2>
<pre class="LyX-Code">
pairwise [&lt;name&gt;] [&lt;mask&gt;] [out &lt;filename&gt;] [cuteelec &lt;ecut&gt;] [cutevdw &lt;vcut&gt;] 	
         [ reference | ref &lt;name&gt; | refindex &lt;#&gt; ] [cutout &lt;cut mol2 prefix&gt;]
     	[vmapout &lt;vdw map&gt;] [emapout &lt;elec map&gt;] [avgout &lt;avg file&gt;]
     	[eout &lt;eout file&gt;] [pdbout &lt;pdb file&gt;] [scalepdbe] [printmode {only|or|and|}]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Data set name; van der Waals energy will get aspect [EVDW] and electrostatic energy will get aspect [EELEC].
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atoms to calculate energy for.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write total EELEC and EVDW to.
</div>
<div class="Description">
<span class="Description-entry">[eout &lt;eout file&gt;]</span> File to write individual EELEC and EVDW interactions to.
</div>
<div class="Description">
<span class="Description-entry">[reference | ref &lt;name&gt; | refindex &lt;#&gt; ]</span> Specify a reference to compare frames to (i.e. calculate Eref - Eframe).
</div>
<div class="Description">
<span class="Description-entry">[cuteelec &lt;cut&gt;]</span> Only report interaction EELEC (or delta EELEC) if absolute value is greater than &lt;ecut&gt; (default 1.0 kcal/mol).
</div>
<div class="Description">
<span class="Description-entry">[cutevdw &lt;cutv&gt;]</span> Only report interaction EVDW (or delta EVDW) if absolute value is greater than &lt;vcut&gt; (default 1.0 kcal/mol).
</div>
<div class="Description">
<span class="Description-entry">[cutout &lt;cut mol2 prefix&gt;]</span> Write out mol2 containing only atom pairs which satisfy &lt;ecut&gt; and &lt;vcut&gt;.
</div>
<div class="Description">
<span class="Description-entry">[vmapout &lt;vdw map&gt;]</span> Write out interaction EVDW (or delta EVDW) matrix to file &lt;vdw map&gt;.
</div>
<div class="Description">
<span class="Description-entry">[emapout &lt;elec map&gt;]</span> Write out interaction EELEC (or delta EELEC) matrix to file &lt;elec map&gt;.
</div>
<div class="Description">
<span class="Description-entry">[avgout &lt;avg file&gt;]</span> Print average interaction EVDW|EELEC (or average delta EVDW|EELC) to &lt;avg file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[pdbout &lt;pdb file&gt;]</span> Write PDB with EVDW|EELEC in occupancy|B-factor columns to &lt;pdb file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[scalepdbe]</span> Scale energies written to PDB from 0 to 100.
</div>
<div class="Description">
<span class="Description-entry">[printmode {only|or|and}]</span> Control when/how average energies are written
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[EELEC]</span> Electrostatic energy in (kcal/mol).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[EVDW]</span> van der Waals energy in (kcal/mol).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[VMAP]</span> van der Waals energy matrix.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[EMAP]</span> Electrostatic energy matrix.
</div>

</li>

</ul>
<div class="Unindented">
This action has two related functions: 1) Calculate pairwise (i.e. non-bonded) energy (in kcal/mol) for atoms in <b>&lt;mask&gt;</b>, or 2) Compare pairwise energy of frames to a reference frame. This calculation does use an exclusion list but is not periodic.
</div>
<div class="Indented">
When comparing to a reference frame, the <b>eout</b> file will contain the differences for each individual interaction (i.e. Eref - Eframe), otherwise the <b>eout</b> file will contain the absolute value of each individual interaction. The <b>cuteelc</b> and <b>cutevdw</b> keywords can be used to restrict printing of individual interactions to those for which the absolute value is above a cutoff. The VMAP and EMAP matrix elements will contain these values as well (differences for reference, absolute value otherwise) averaged over all frames. The <b>avgout</b> file will contain only these values averaged over all frames that satisfy the cutoffs. The <b>printmode</b> keyword controls when the average energies are written: <b>only</b> means only average energy components that satisfy cutoffs will be printed, <b>or</b> means that both energy components will be printed if either satistfy a cutoff, and <b>and</b> means that both energy components will be written only if both satisfy the cutoffs.
</div>
<div class="Indented">
The <b>cutout</b> keyword can be used to write out MOL2 files each frame named ’&lt;cut mol2 prefix&gt;.evdw.mol2.X’ and ’&lt;cut mol2 prefix&gt;.eelec.mol2.X’ (where X is the frame number) containing only atoms with energies that satisfy the cutoffs. Similarly, the <b>pdbout</b> keyword can be used to write out a PDB file (with 1 MODEL per frame). The occupancy and B-factor columns will contain the total van der Waals and electrostatic energy for each atom if cutoffs are satisfied, or 0.0 otherwise.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.57">11.57</a> principal
</h2>
<pre class="LyX-Code">
principal [&lt;mask&gt;] [dorotation] [out &lt;filename&gt;] [name &lt;dsname&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Mask of atoms used to determine principal axes (default all).
</div>
<div class="Description">
<span class="Description-entry">[dorotation]</span> Align coordinates along principal axes.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write resulting eigenvalues/eigenvectors to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;dsname&gt;]</span> Data set name (3x3 matrices).
</div>
<div class="Unindented">
Data Sets Created (<b>name</b> keyword only):
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[evec]</span> Eigenvectors (3x3 matrix, row-major).
</div>
<div class="Description">
<span class="Description-entry">&lt;dsname&gt;[eval]</span> Eigenvalues (vector).
</div>

</li>

</ul>
<div class="Unindented">
Determine principal axes of each frame determined by diagonalization of the inertial matrix from the coordinates of the specified atoms. At least one of <b>dorotation</b>, <b>out</b>, or <b>name</b> must be specified. The resulting eigenvectors are sorted from largest eigenvalue to smallest, and the corresponding axes labelled using the <i>cpptraj</i> convention of X &gt; Y &gt; Z (similar to ’<b><i>vector</i></b> <b>principal</b>’). If out is specified the eigenvectors and eigenvalues will be written for each frame N with format:
</div>
<pre class="LyX-Code">
&lt;N&gt; EIGENVALUES: &lt;EX&gt; &lt;EY&gt; &lt;EZ&gt;
&lt;N&gt; EIGENVECTOR 0: &lt;Xx&gt; &lt;Xy&gt; &lt;Xz&gt;
&lt;N&gt; EIGENVECTOR 1: &lt;Yx&gt; &lt;Yy&gt; &lt;Yz&gt;
&lt;N&gt; EIGENVECTOR 2: &lt;Zx&gt; &lt;Zy&gt; &lt;Zz&gt;
</pre>
<div class="Unindented">
NOTE: The eigenvector 3x3 matrix data set could subsequently be used e.g. with the <b><i>rotate</i></b> action.
</div>
<div class="Indented">
Example: Align system (residues 1-76) along principle axes:
</div>
<pre class="LyX-Code">
parm myparm.parm7
trajin protein.nc
principal :1-76 dorotation out principal.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.58">11.58</a> projection
</h2>
<pre class="LyX-Code">
projection [&lt;name&gt;] evecs &lt;dataset name&gt; [out &lt;outfile&gt;] [beg &lt;beg&gt;] [end &lt;end&gt;]
           [&lt;mask&gt;] [dihedrals &lt;dataset arg&gt;]
           [start &lt;start&gt;] [stop &lt;stop&gt;] [offset &lt;offset&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">evecs &lt;dataset name&gt;</span> Data set containing eigenvectors (modes).
</div>
<div class="Description">
<span class="Description-entry">[out &lt;outfile&gt;]</span> Write projections to &lt;outfile&gt;.
</div>
<div class="Description">
<span class="Description-entry">[beg &lt;beg&gt;]</span> First eigenvector/mode to use (default 1).
</div>
<div class="Description">
<span class="Description-entry">[end &lt;end&gt;]</span> Final eigenvector/mode to use (default 2).
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> (Not dihedral covariance) Mask of atoms to use in projection; MUST CORRESPOND TO HOW EIGENVECTORS WERE GENERATED.
</div>
<div class="Description">
<span class="Description-entry">[dihedrals &lt;dataset arg&gt;]</span> (Dihedral covariance only) Dihedral data sets to use in projection; MUST CORRESPOND TO HOW EIGENVECTORS WERE GENERATED.
</div>
<div class="Description">
<span class="Description-entry">[start &lt;start&gt;]</span> Frame to start calculating projection.
</div>
<div class="Description">
<span class="Description-entry">[stop &lt;stop&gt;]</span> Frame to stop calculating projection.
</div>
<div class="Description">
<span class="Description-entry">[offset &lt;offset&gt;]</span> Frames to skip between projection calculations.
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Indented">
DataSet indices correspond to mode #.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> (All execpt IDEA) Projection data set.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[X]</span> X component of mode (IDEA modes only).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Y]</span> Y component of mode (IDEA modes only).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Z]</span> Z component of mode (IDEA modes only).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[R]</span> Magnitude of mode (IDEA modes only).
</div>

</li>

</ul>
<div class="Unindented">
Projects snapshots onto eigenvectors obtained by diagonalizing covariance or mass-weighted covariance matrices. Eigenvectors are taken from previously generated (e.g. with <b><i>diagmatrix</i></b>) or previously read-in (e.g. with <b><i>readdata</i></b>) eigenvectors with name &lt;dataset name&gt;. The user has to make sure that the atoms selected by <b>&lt;mask&gt;</b> agree with the ones used to calculate the modes (i.e., if mask = ’@CA’ was used in the &ldquo;<span class="sans"><b>matrix</b></span>&rdquo; command, mask = ’@CA’ needs to be set here as well). See <a class="Reference" href="#sec:cpptraj-Matrix-Vector-Analysis-Examples">13 on page 1↓</a> for examples using the <b><i>projection</i></b> command.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.59">11.59</a> pucker 
</h2>
<pre class="LyX-Code">
pucker<a class="IndexReference" name="entry-pucker-0" href="#index-pucker">↓</a> [&lt;name&gt;] &lt;mask1&gt; &lt;mask2&gt; &lt;mask3&gt; &lt;mask4&gt; &lt;mask5&gt; [&lt;mask6&gt;] [geom]
	   [out &lt;filename&gt;] [altona | cremer] [amplitude] [theta]
   	[range360] [offset &lt;offset&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;maskX&gt;</span> Five (optionally six) atom masks selecting atom(s) to calculate pucker for.
</div>
<div class="Description">
<span class="Description-entry">[geom]</span> Use geometric center of atoms in &lt;maskX&gt; (default is center of mass).
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file name.
</div>
<div class="Description">
<span class="Description-entry">[altona]</span> Use method of Altona &amp; Sundaralingam (5 masks only).
</div>
<div class="Description">
<span class="Description-entry">[cremer]</span> Use method of Cremer and Pople (5 or 6 masks). This is the default when 6 masks are specified.
</div>
<div class="Description">
<span class="Description-entry">[amplitude]</span> Also calculate amplitude.
</div>
<div class="Description">
<span class="Description-entry">[theta]</span> (6 masks only) Also calculate theta.
</div>
<div class="Description">
<span class="Description-entry">[range360]</span> Wrap pucker values from 0.0 to 360.0 (default is -180.0 to 180.0).
</div>
<div class="Description">
<span class="Description-entry">[offset &lt;offset&gt;]</span> Add &lt;offset&gt; to pucker values.
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Pucker in degrees.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Amp]</span> Amplitude (if <b>amplitude</b> was specified).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Theta]</span> Theta (if <b>theta</b> and 6 masks were specified).
</div>

</li>

</ul>
<div class="Unindented">
Calculate the pucker (in degrees) for atoms in &lt;mask1&gt;, &lt;mask2&gt;, &lt;mask3&gt;, &lt;mask4&gt;, &lt;mask5&gt; using the method of Altona &amp; Sundarlingam<span class="bibcites">[<a class="bibliocite" name="cite-3" href="#biblio-3"><span class="bib-index">3</span></a>, <a class="bibliocite" name="cite-16" href="#biblio-16"><span class="bib-index">16</span></a>]</span> (default for 5 masks, or if <b>altona</b> specified), or the method of Cremer &amp; Pople<span class="bibcites">[<a class="bibliocite" name="cite-11" href="#biblio-11"><span class="bib-index">11</span></a>]</span> (default for 6 masks, or if <b>cremer</b> is specified). If the <b>amplitude</b> or <b>theta</b> keywords are given, amplitudes/thetas (also in degrees) will be calculated in addition to pucker. The results from <b><i>pucker</i></b> can be further analyzed with the <b><i>statistics</i></b> analysis.
</div>
<div class="Indented">
By default, pucker values are wrapped to range from -180 to 180 degrees. If the <b>range360</b> keyword is specified values will be wrapped to range from 0 to 360 degrees. Note that the Cremer &amp; Pople convention is offset from Altona &amp; Sundarlingam convention (with nucleic acids) by +90.0 degrees; the <b>offset</b> keyword will add an offset to the final value and so can be used to convert between the two. For example, to convert from Cremer to Altona specify &ldquo;<b>offset</b> 90&rdquo;.
</div>
<div class="Indented">
To calculate nucleic acid pucker specify C1’ first, followed by C2’, C3’, C4’ and O4’. For example, to calculate the sugar pucker for nucleic acid residues 1 and 2 using the method of Altona &amp; Sundarlingam, with final pseudorotation values ranging from 0 to 360:
</div>
<pre class="LyX-Code">
pucker p1 :1@C1’ :1@C2’ :1@C3’ :1@C4’ :1@O4’ range360 out pucker.dat 
pucker p2 :2@C1’ :2@C2’ :2@C3’ :2@C4’ :2@O4’ range360 out pucker.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.60">11.60</a> radgyr | rog
</h2>
<pre class="LyX-Code">
radgyr<a class="IndexReference" name="entry-radgyr-0" href="#index-radgyr">↓</a> [name&gt;] [&lt;mask&gt;] [out &lt;filename&gt;] [mass] [nomax] [tensor]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atoms to calculate radius of gyration for; default all atoms.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write data to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[mass]</span> Mass-weight radius of gyration.
</div>
<div class="Description">
<span class="Description-entry">[nomax]</span> Do not calculate maximum radius of gyration.
</div>
<div class="Description">
<span class="Description-entry">[tensor]</span> Calculate radius of gyration tensor, output format ’XX YY ZZ XY XZ YZ’.
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Radius of gyration in Ang.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Max]</span> Max radius of gyration in Ang.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Tensor]</span> Radius of gyration tensor; format ’XX YY ZZ XY XZ YZ’.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the radius of gyration of specified atoms. For example, to calculate only the mass-weighted radius of gyration (not the maximum) of the non-hydrogen atoms of residues 4 to 10 and print the results to &ldquo;RoG.dat&rdquo;:
</div>
<pre class="LyX-Code">
radgyr :4-10&amp;!(@H=) out RoG.dat mass nomax
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.61">11.61</a> radial | rdf
</h2>
<pre class="LyX-Code">
radial<a class="IndexReference" name="entry-radial-0" href="#index-radial">↓</a> [out &lt;outfilename&gt;] &lt;spacing&gt; &lt;maximum&gt; &lt;solvent mask1&gt;
       [&lt;solute mask2&gt;] [noimage] [density &lt;density&gt; | volume] [&lt;dataset name&gt;]
       [intrdf &lt;file&gt;] [rawrdf &lt;file&gt;]
       [{{center1|center2|nointramol} | [byres1] [byres2] [bymol1] [bymol2]}]]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[out &lt;outfilename&gt;]</span> File to write RDF to.
</div>
<div class="Description">
<span class="Description-entry">&lt;spacing&gt;</span> Bin spacing, required.
</div>
<div class="Description">
<span class="Description-entry">&lt;maximum&gt;</span> Max bin value, required.
</div>
<div class="Description">
<span class="Description-entry">&lt;solvent mask1&gt;</span> Atoms to calculate RDF for, required.
</div>
<div class="Description">
<span class="Description-entry">[&lt;solute mask2&gt;]</span> (Optional) If specified calculate RDF of all atoms in &lt;solvent mask1&gt; to each atom in &lt;solute mask2&gt;.
</div>
<div class="Description">
<span class="Description-entry">[noimage]</span> Do not image distances.
</div>
<div class="Description">
<span class="Description-entry">[density &lt;density&gt;]</span> Use density value of &lt;density&gt; for normalization (default 0.033456 molecules Å<span class="formula"><sup> − <span class="text">3</span></sup></span>).
</div>
<div class="Description">
<span class="Description-entry">[volume]</span> Determine density for normalization from average volume of input frames.
</div>
<div class="Description">
<span class="Description-entry">[&lt;setname&gt;]</span> Name of output data sets.
</div>
<div class="Description">
<span class="Description-entry">[center1]</span> Calculate RDF from geometric center of atoms in &lt;solvent mask1&gt; to all atoms in &lt;solute mask2&gt;.
</div>
<div class="Description">
<span class="Description-entry">[center2]</span> Calculate RDF from geometric center of atoms in &lt;solute mask2&gt; to all atoms in &lt;solvent mask1&gt;.
</div>
<div class="Description">
<span class="Description-entry">[nointramol]</span> Ignore intra-molecular distances.
</div>
<div class="Description">
<span class="Description-entry">[intrdf &lt;file&gt;]</span> Calculate integral of RDF bin values (averaged over # of frames but otherwise not normalized) and write to &lt;file&gt; (can be same as &lt;output_filename&gt;).
</div>
<div class="Description">
<span class="Description-entry">[rawrdf &lt;file&gt;]</span> Write raw (non-normalized) RDF values to &lt;file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[byres1]</span> Calculate using the centers of mass of each residue in the first mask.
</div>
<div class="Description">
<span class="Description-entry">[bymol1]</span> Calculate using the centers of mass of each molecule in the first mask.
</div>
<div class="Description">
<span class="Description-entry">[byres2]</span> Calculate using the centers of mass of each residue in the second mask.
</div>
<div class="Description">
<span class="Description-entry">[bymol2]</span> Calculate using the centers of mass of each molecule in the second mask.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;</span> The radial distribution function.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[int]</span> (<b>intrdf</b> only) Integral of RDF bin values.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[raw]</span> (<b>rawrdf</b> only) Raw (non-normalized) RDF values.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the radial distribution function (RDF, aka pair correlation function) of atoms in <b>&lt;solvent mask1&gt;</b> (note that this mask does not need to be solvent, but this nomenclature is used for clarity). If an optional second mask (<b>&lt;solute mask2&gt;</b>) is given, calculate the RDF of ALL atoms in <b>&lt;solvent mask1&gt;</b> to EACH atom in <b>&lt;solute mask2&gt;</b>. If desired, the geometric center of atoms in <b>&lt;solvent mask1&gt;</b> or <b>&lt;solute mask2&gt;</b> can be used by specifying the <b>center1</b> or <b>center2</b> keywords respectively, or alternatively intra-molecular distances can be ignored by specifying the <b>nointramol</b> keyword.
</div>
<div class="Indented">
The RDF is calculated from the histogram of the number of particles found as a function of distance R, normalized by the expected number of particles at that distance. The normalization is calculated from:\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<span class="formula"><i>Density</i>*<span class="fraction"><span class="ignored">(</span><span class="numerator">4<i>π</i></span><span class="ignored">)/(</span><span class="denominator">3</span><span class="ignored">)</span></span><span class="symbol">(</span><span class="symbol">(</span><i>R</i> + <i>dR</i><span class="symbol">)</span><sup>3</sup> − <i>R</i><sup>3</sup><span class="symbol">)</span></span>
</div>

</div>
<div class="Indented">
where dR is equal to the bin spacing. Some care is required by the user in order to normalize the RDF correctly. The default density value is 0.033456 molecules Å<span class="formula"><sup> − <span class="text">3</span></sup></span>, which corresponds to a density of water approximately equal to 1.0 g mL<span class="formula"><sup> − <span class="text">1</span></sup></span>. To convert a standard density in g mL<span class="formula"><sup> − <span class="text">1</span></sup></span>, multiply the density by <span class="formula"><span class="fraction"><span class="ignored">(</span><span class="numerator">0.6022</span><span class="ignored">)/(</span><span class="denominator"><i>M</i><sub><i>r</i></sub></span><span class="ignored">)</span></span></span>, where <span class="formula"><i>M</i><sub><i>r</i></sub></span> is the mass of the molecule in atomic mass units. Alternatively, if the <b>volume</b> keyword is specified the density is determined from the average volume of the system over all Frames. 
</div>
<div class="Indented">
Note that correct normalization of the RDF depends on the number of atoms in each mask; if multiple topology files are being processed that result in changes in the number of atoms in each mask, the normalization will be off.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.62">11.62</a> randomizeions
</h2>
<pre class="LyX-Code">
randomizeions &lt;mask&gt; [around &lt;mask&gt; by &lt;distance&gt;] [overlap &lt;value&gt;]
              [noimage] [seed &lt;value&gt;]
</pre>
<div class="Unindented">
This can be used to randomly swap the positions of solvent and single atom ions. The &ldquo;<tt>overlap</tt>&rdquo; specifies the minimum distance between ions, and the &ldquo;<tt>around</tt>&rdquo; keyword can be used to specify a solute (or set of atoms) around which the ions can get no closer than the distance specified. The optional keywords &ldquo;<tt>noimage</tt>&rdquo; disable imaging and &ldquo;<tt>seed</tt>&rdquo; update the random number seed. An example usage is
</div>
<pre class="LyX-Code">
randomizeions @NA around :1-20 by 5.0 overlap 3.0
</pre>
<div class="Unindented">
The above will swap Na<span class="formula"><sup><span class="text">+</span></sup></span> ions with water getting no closer than 5.0 Å from residues 1 – 20 and no closer than 3.0 Å from any other Na<span class="formula"><sup><span class="text">+</span></sup></span> ion.<i> </i>
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.63">11.63</a> replicatecell
</h2>
<pre class="LyX-Code">
replicatecell [out &lt;traj filename&gt;] [parmout &lt;parm filename&gt;] [name &lt;dsname&gt;]
              { all | dir &lt;XYZ&gt; [dir &lt;XYZ&gt; ...] } [&lt;mask&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">out &lt;traj filename&gt;</span> Write replicated cell to output trajectory file.
</div>
<div class="Description">
<span class="Description-entry">parmout &lt;parm filename&gt;</span> Write replicated cell topology to topology file. This file will not be viable to use for simulations.
</div>
<div class="Description">
<span class="Description-entry">name &lt;dsname&gt;</span> If specified save replicated cell to COORDS data set.
</div>
<div class="Description">
<span class="Description-entry">all</span> Replicate cell once in all possible directions.
</div>
<div class="Description">
<span class="Description-entry">dir &lt;XYZ&gt;</span> Repicate cell once in specified directions. &lt;XYZ&gt; should consist of 3 numbers with no spaces in between them and are restricted to values of -1, 1, and 0. May be specified more than once.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Mask of atoms to replicate.
</div>

</li>

</ul>
<div class="Unindented">
Create a trajectory where the unit cell is replicated in 1 or more directions (up to 27). The resulting coordinates and topology can be written to a trajectory/topology file. They can also be saved as a COORDS data set for subsequent processing. Currently replication is only allowed 1 axis length in either direction. The <b>all</b> keyword will replicate the cell once in all directions. The <b>dir</b> keyword can be used to restrict replication to specific directions, e.g. ’<tt>dir 10-1</tt>’ would replicate the cell once in the +X, -Z directions.
</div>
<div class="Indented">
For example, to replicate a cell in all directions, writing out to NetCDF trajectory cell.nc:
</div>
<pre class="LyX-Code">
parm ../tz2.truncoct.parm7
trajin ../tz2.truncoct.nc
replicatecell out cell.nc parmout cell.parm7 all
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.64">11.64</a> rms | rmsd<a class="Label" name="subsec:cpptraj-rmsd"> </a>
</h2>
<pre class="LyX-Code">
rmsd<a class="IndexReference" name="entry-rmsd-0" href="#index-rmsd">↓</a> [&lt;name&gt;] &lt;mask&gt; [&lt;refmask&gt;] [out &lt;filename&gt;] [mass]
 	[nofit | norotate | nomod]
 	[savematrices [matricesout &lt;file&gt;]]
 	[savevectors {combined|separate} [vecsout &lt;file&gt;]]
 	[ first | reference | ref &lt;name&gt; | refindex &lt;#&gt; | previous |
 	  reftraj &lt;name&gt; [parm &lt;name&gt; | parmindex &lt;#&gt;] ]
 	[perres perresout &lt;filename&gt; [perresavg &lt;avgfile&gt;]
 	 [range &lt;resRange&gt;] [refrange &lt;refRange&gt;]
 	 [perresmask &lt;additional mask&gt;] [perrescenter] [perresinvert]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Mask of atoms to calculate RMSD for; if not specified, calculate for all atoms.
</div>
<div class="Description">
<span class="Description-entry">[&lt;refmask&gt;]</span> Reference mask; if not specified, use <b>&lt;mask&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output data file name.
</div>
<div class="Description">
<span class="Description-entry">[mass]</span> Mass-weight the RMSD calculation.
</div>
<div class="Description">
<span class="Description-entry">[nofit]</span> Do not perform best-fit RMSD.
</div>
<div class="Description">
<span class="Description-entry">[norotate]</span> If calculating best-fit RMSD, translate but do not rotate coordinates.
</div>
<div class="Description">
<span class="Description-entry">[nomod]</span> If calculating best-fit RMSD, do not modify coordinates.
</div>
<div class="Description">
<span class="Description-entry">[savematrices]</span> If specified save rotation matrices to data set with aspect [RM].
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">matricesout &lt;file&gt;</span> Write rotation matrices to specified file.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[savevectors {combined|separate}]</span> If specified save translation vectors: <b>combined</b> means save target-to-origin plus the origin-to-reference translation vectors, <b>separate</b> means save target-to-origin as Vx, Vy, Vz and save origin-to-reference as Ox Oy Oz in the output vector data set.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">vecsout &lt;file&gt;</span> Output translation vector data set to &lt;file&gt;.
</div>

</li>

</ul>
<div class="Unindented">
Reference keywords:
</div>
<div class="Description">
<span class="Description-entry"><b>first</b></span> Use the first trajectory frame processed as reference.
</div>
<div class="Description">
<span class="Description-entry"><b>reference</b></span> Use the first previously read in reference structure (refindex 0).
</div>
<div class="Description">
<span class="Description-entry"><b>ref &lt;name&gt;</b></span> Use previously read in reference structure specified by filename/tag.
</div>
<div class="Description">
<span class="Description-entry"><b>refindex &lt;#&gt;</b></span> Use previously read in reference structure specified by &lt;#&gt; (based on order read in).
</div>
<div class="Description">
<span class="Description-entry">previous</span> Use frame prior to current frame as reference.
</div>
<div class="Description">
<span class="Description-entry"><b>reftraj</b> &lt;name&gt;</span> Use frames from COORDS set &lt;name&gt; or read in from trajectory file &lt;name&gt; as references. Each frame from &lt;name&gt; is used in turn, so that frame 1 is compared to frame 1 from &lt;name&gt;, frame 2 is compared to frame 2 from &lt;name&gt; and so on. If &lt;trajname&gt; runs out of frames before processing is complete, the last frame of &lt;trajname&gt; continues to be used as the reference.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;parmname&gt; | parmindex &lt;#&gt;</span> If <b>reftraj</b> specifies a trajectory file, associate it with specified topology; if not specified the first topology is used.
</div>

</li>

</ul>
<div class="Unindented">
Per-residue RMSD keywords:
</div>
<div class="Description">
<span class="Description-entry">perres</span> Activate per-residue no-fit RMSD calculation.
</div>
<div class="Description">
<span class="Description-entry">perresout &lt;perresfile&gt;</span> Write per-residue RMSD to &lt;perresfile&gt;.
</div>
<div class="Description">
<span class="Description-entry">perresavg &lt;avgfile&gt;</span> Write average per-residue RMSDs to &lt;avgfile&gt;.
</div>
<div class="Description">
<span class="Description-entry">range &lt;res range&gt;</span> Calculate per-residue RMSDs for residues in &lt;res range&gt; (default all solute residues).
</div>
<div class="Description">
<span class="Description-entry">refrange &lt;ref range&gt;</span> Calculate per-residue RMSDs to reference residues in &lt;ref range&gt; (use &lt;res range&gt; if not specified).
</div>
<div class="Description">
<span class="Description-entry">perresmask &lt;additional mask&gt;</span> By default residues are selected using the mask ’:X’ where X is residue number; this appends &lt;additional mask&gt; to the mask expression.
</div>
<div class="Description">
<span class="Description-entry">perrescenter</span> Translate residues to a common center of mass prior to calculating RMSD.
</div>
<div class="Description">
<span class="Description-entry">perresinvert</span> Make X-axis residue number instead of frame number.
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> RMSD of atoms in mask to reference.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[RM]</span> (savematrices only) Rotation matrices of target to reference.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[TV]</span> (savevectors only) Translation vector.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[res]</span> (<b>perres</b> only) Per-residue RMSDs; index is residue number.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Avg]</span> (<b>perres</b> only) Average per-residue RMSD for each residue.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Stdev]</span> (<b>perres</b> only) Standard deviation of RMSD for each residue.
</div>

</li>

</ul>
<div class="Unindented">
<i>Note that <b>perres</b> data sets are not generated until <b>run</b> is called.</i>
</div>
<div class="Indented">
Calculate the coordinate RMSD of input frames to a reference frame (or reference trajectory). Both &lt;mask&gt; and &lt;refmask&gt; must specify the same number of atoms, otherwise an error will occur. 
</div>
<div class="Indented">
For example, say you have a trajectory and you want to calculate RMSD to two separate reference structures. To calculate the best-fit RMSD of the C, CA, and N atoms of residues 1 to 20 in each frame to the C, CA, and N atoms of residues 3 to 23 in StructX.crd, and then calculate the no-fit RMSD of residue 7 to residue 7 in another structure named Struct-begin.rst7, writing both results to Grace-format file &ldquo;rmsd1.agr&rdquo;:
</div>
<pre class="LyX-Code">
reference StructX.crd [structX]
reference md_begin.rst7 [struct0]
rmsd BB :1-20@C,CA,N ref [structX] :3-23@C,CA,N out rmsd1.agr 
rmsd Res7 :7 ref [struct0] out rmsd1.agr nofit
</pre>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--26"></a>Per-residue RMSD calculation
</h3>
<div class="Unindented">
If the <b><span class="black">perres</span></b> keyword is specified, after the initial RMSD calculation the no-fit RMSD of specified residues is also calculated. So for example:
</div>
<pre class="LyX-Code">
rmsd :10-260 reference perres perresout PRMS.dat range 190-211 perresmask &amp;!(@H=)
</pre>
<div class="Unindented">
will first perform a best-fit RMSD calculation to the first specified reference structure using residues 10 to 260, then calculate the no-fit RMSD of residues 190 to 211 (excluding any hydrogen atoms), writing the results to PRMS.dat. Two additional recommendations for the ’perres’ option: 1) try not including backbone atoms by using the ’perresmask’ keyword, e.g. "perresmask &amp;!@H,N,CA,HA,C,O", and 2) try using the ’perrescenter’ keyword, which centers each residue prior to the ’nofit’ calculation; this is useful for isolating changes in residue conformation.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.65">11.65</a> rms2d | 2drms
</h2>
<div class="Unindented">
Although the <b><i>’rms2d’</i></b> command can still be specified as an action, it is now considered an analysis. See <a class="Reference" href="#subsec:cpptraj-rms2d">12.28 on page 1↓</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.66">11.66</a> rmsavgcorr
</h2>
<div class="Unindented">
Although the <b><i>’rmsavgcorr’</i></b> command can still be specified as an action, it is now considered an analysis. See <a class="Reference" href="#subsec:cpptraj-rmsavgcorr">12.29 on page 1↓</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.67">11.67</a> rmsf | atomicfluct
</h2>
<div class="Unindented">
See <a class="Reference" href="#subsec:cpptraj_atomicfluct">11.5 on page 1↑</a>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.68">11.68</a> rotate
</h2>
<pre class="LyX-Code">
rotate [&lt;mask&gt;] { [x &lt;xdeg&gt;] [y &lt;ydeg&gt;] [z &lt;zdeg&gt;]  |
   	           axis0 &lt;mask0&gt; axis1 &lt;mask1&gt; &lt;deg&gt; |
   	           usedata &lt;set name&gt; [inverse] }\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Rotate atoms in &lt;mask&gt; (default all).
</div>
<div class="Description">
<span class="Description-entry">[x &lt;xdeg]</span> Degrees to rotate around the X axis.
</div>
<div class="Description">
<span class="Description-entry">[y &lt;xdeg]</span> Degrees to rotate around the Y axis.
</div>
<div class="Description">
<span class="Description-entry">[z &lt;xdeg]</span> Degrees to rotate around the Z axis.
</div>
<div class="Description">
<span class="Description-entry">axis0 &lt;mask0&gt;</span> Mask defining the beginning of a user-defined axis.
</div>
<div class="Description">
<span class="Description-entry">axis1 &lt;mask1&gt;</span> Mask defining the end of a user-defined axis.
</div>
<div class="Description">
<span class="Description-entry">&lt;deg&gt;</span> Value in degrees to rotate around user defined axis.
</div>
<div class="Description">
<span class="Description-entry">usedata &lt;set name&gt;</span> If specified, use 3x3 rotation matrices in specified data set to rotate coordinates.
</div>
<div class="Description">
<span class="Description-entry">[inverse]</span> Perform inverse rotation from input rotation matrices.
</div>

</li>

</ul>
<div class="Unindented">
Rotate specified atoms around the X, Y, and/or Z axes by the specified amounts, around a user-defined axis (specified by &lt;mask0&gt; and &lt;mask1&gt;), or use a previously read in or generated data set of 3x3 matrices to perform rotations.
</div>
<div class="Indented">
For example, to rotate the entire system 90 degrees around the X axis:
</div>
<pre class="LyX-Code">
rotate x 90
</pre>
<div class="Unindented">
To rotate residue 270 90 degrees around the axis defined between atoms C1, C2, C3, C4, C5, and C6 in residue 270 and atoms C7, C8, C9, C10, C11, and C12 in residue 270:
</div>
<pre class="LyX-Code">
rotate :270 axis0 :270@C1,C2,C3,C4,C5,C6 axis1 :270@C7,C8,C9,C10,C11,C12 90.0
</pre>
<div class="Unindented">
To rotate the system with rotation matrices read in from rmatrices.dat:
</div>
<pre class="LyX-Code">
trajin tz2.norotate.crd
readdata rmatrices.dat name RM mat3x3
rotate usedata RM 
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.69">11.69</a> rotdif
</h2>
<div class="Unindented">
The <b><i>’rotdif’</i></b> command is now an analysis (see <a class="Reference" href="#subsec:cpptraj-rotdif">12.30 on page 1↓</a>), and requires that rotation matrices be generated via an <b><i>rmsd</i></b> action. For example:
</div>
<pre class="LyX-Code">
reference avgstruct.pdb
trajin tz2.nc
rms R0 reference @CA,C,N,O savematrices 
rotdif rmatrix R0[RM] rseed 1 nvecs 10 dt 0.002 tf 0.190 \
       itmax 500 tol 0.000001 d0 0.03 order 2 rvecout rvecs.dat \
       rmout matrices.dat deffout deffs.dat outfile rotdif.out
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.70">11.70</a> runavg | runningaverage
</h2>
<pre class="LyX-Code">
runavg<a class="IndexReference" name="entry-runavg-0" href="#index-runavg">↓</a> [window &lt;window_size&gt;]
</pre>
<div class="Unindented">
<i>Note that for backwards compatibility with ptraj &ldquo;runningaverage&rdquo; is also accepted.</i>
</div>
<div class="Indented">
Replaces the current frame with a running average over a number of frames specified by <b>window</b> &lt;window_size&gt; (5 if not specified). This means that in order to build up the correct number of frames to calculate the average, the first &lt;window_size&gt; minus one frames will not be processed by subsequent actions. So for example given the input:
</div>
<pre class="LyX-Code">
runavg window 3
rms first out rmsd.dat
</pre>
<div class="Unindented">
the rms command will not take effect until frame 3 since that is the first time 3 frames are available for averaging (1, 2, and 3). The next frame processed would be an average of frames 2, 3, and 4, etc.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.71">11.71</a> scale
</h2>
<pre class="LyX-Code">
scale x &lt;sx&gt; y &lt;sy&gt; z &lt;sz&gt; &lt;mask&gt;
</pre>
<div class="Unindented">
Scale the X|Y|Z coordinates of atoms in &lt;mask&gt; by &lt;sx&gt;|&lt;sy&gt;|&lt;sz&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.72">11.72</a> secstruct<a class="Label" name="subsec:cpptraj-secstruct"> </a>
</h2>
<pre class="LyX-Code">
secstruct<a class="IndexReference" name="entry-secstruct-0" href="#index-secstruct">↓</a> [&lt;name&gt;] [out &lt;filename&gt;] [&lt;mask&gt;] [sumout &lt;filename&gt;]
          [assignout &lt;filename&gt;] [totalout &lt;filename&gt; [ptrajformat]
          [betadetail]
          [namen &lt;N name&gt;] [nameh &lt;H name&gt;] [nameca &lt;CA name&gt;]
          [namec &lt;C name&gt;] [nameo &lt;O name&gt;] [namesg &lt;sulfur name&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file name for secondary structure vs time.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atom mask in which residues should be looked for.
</div>
<div class="Description">
<span class="Description-entry">[sumout &lt;sumfilename&gt;]</span> Write average secondary structure values for each residue to &lt;sumfilename&gt;; if not specified &lt;filename&gt;.sum is used.
</div>
<div class="Description">
<span class="Description-entry">[assignout &lt;filename&gt;]</span> Write overall secondary structure assignment (based on dominant secondary structure type for each residue) to file.
</div>
<div class="Description">
<span class="Description-entry">[ptrajformat]</span> Write secondary structure as a string of characters for each frame, similar to ptraj output.
</div>
<div class="Description">
<span class="Description-entry">[betadetail]</span> Record anti-parallel beta and parallel beta in place of extended and bridge secondary structure. If a residue could be both only anti-parallel is reported.
</div>
<div class="Description">
<span class="Description-entry">[namen &lt;N name&gt;]</span> Backbone amide nitrogen atom name (default ’N’).
</div>
<div class="Description">
<span class="Description-entry">[nameh &lt;H name&gt;]</span> Backbone amide hydrogen atom name (default ’H’).
</div>
<div class="Description">
<span class="Description-entry">[nameca &lt;CA name&gt;]</span> Backbone alpha carbon atom name (default ’CA’).
</div>
<div class="Description">
<span class="Description-entry">[namec &lt;C name&gt;]</span> Backbone carbonyl carbon atom name (default ’C’).
</div>
<div class="Description">
<span class="Description-entry">[nameo &lt;O name&gt;]</span> Backbone carbonyl oxygen atom name (default ’O’).
</div>
<div class="Description">
<span class="Description-entry">[namesg &lt;SG name&gt;]</span> Cysteine sulfur atom name, used to ignore disulfide connectivity (default ’SG’).
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[res]</span> Residue secondary structure per frame; index corresponds to residue number. If <b>ptrajformat</b> specified these will be characters, otherwise integers (see table below).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[avgss]</span> Average of each type of secondary structure; index corresponds to secondary structure type (see table below; no index for &ldquo;None&rdquo;).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[None]</span> Total fraction of residues with no structure vs time.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Para]</span> Total fraction of residues with parallel beta structure vs time.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Anti]</span> Total fraction of residues with anti-parallel beta structure vs time.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[3-10]</span> Total fraction of 3-10 helical structure vs time.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Alpha]</span> Total fraction of alpha helical structure vs time.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Pi]</span> Total fraction of Pi helical structure vs time.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Turn]</span> Total fraction of turn structure vs time.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Bend]</span> Total fraction of bend structure vs time.
</div>

</li>

</ul>
<div class="Unindented">
<b>As of version 4.18.0, this command now produces output that better conforms with the original definitions in Kabsch and Sander 1983; namely that Extended beta (i.e. 2 or more consecutive beta bridges of the same type) and beta Bridge (i.e. an isolated beta bridge) are now reported instead of anti-parallel and parallel beta. To restore the original behavior the ’betadetail’ keyword must be specified.</b>
</div>
<div class="Indented">
<i>Note that the residue and [avgss] data sets are not generated until <b>run</b> is called.</i>
</div>
<div class="Indented">
Calculate secondary structural propensities for residues in <b>&lt;mask&gt;</b> (or all solute residues if no mask given) using the DSSP method of Kabsch and Sander<span class="bibcites">[<a class="bibliocite" name="cite-20" href="#biblio-20"><span class="bib-index">20</span></a>]</span>, which assigns secondary structure types for residues based on backbone amide (N-H) and carbonyl (C=O) atom positions. By default <i>cpptraj</i> assumes these atoms are named &ldquo;N&rdquo;, &ldquo;H&rdquo;, &ldquo;C&rdquo;, and &ldquo;O&rdquo; respectively. If a different naming scheme is used (e.g. amide hydrogens are named &ldquo;HN&rdquo;) the backbone atom names can be customized with the <b>nameX</b> keywords (e.g. ’nameH HN’). Note that it is expected that some residues will not have all of these atoms (such as proline); in this case <i>cpptraj</i> will print an informational message but the calculation will proceed normally. If a residue has no atoms selected it will be skipped. When determining residue connecivity, disulfide bonds will be ignored; <i>cpptraj</i> identifies such bonds based on the <b>namesg</b> atom name (default &ldquo;SG&rdquo;).
</div>
<div class="Indented">
Results will be written to filename specified by <b>out</b> with format:
</div>
<pre class="LyX-Code">
&lt;#Frame&gt;    &lt;ResX SS&gt; &lt;ResX+1 SS&gt; ... &lt;ResN SS&gt;
</pre>
<div class="Unindented">
where &lt;#Frame&gt; is the frame number and &lt;ResX SS&gt; is an integer representing the calculated secondary structure type for residue X. If the keyword <b>ptrajformat</b> is specified, the output format will instead be:
</div>
<pre class="LyX-Code">
&lt;#Frame&gt;    STRING
</pre>
<div class="Unindented">
where STRING is a string of characters (one for each residue) where each character represents a different structural type (this format is similar to what <i>ptraj</i> had outputed and is retained for backwards compatibility). The various secondary structure types and their corresponding integer/character are listed below. If ’betadetail’ is specified what is reported and the characters used change slightly.\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top">
STRING (betadetail)
</td>
<td align="center" valign="top">
Integer
</td>
<td align="center" valign="top">
DSSP
</td>
<td align="center" valign="top">
SS type (betadetail)
</td>

</tr>
<tr>
<td align="center" valign="top">
0
</td>
<td align="center" valign="top">
0
</td>
<td align="center" valign="top">
’ ’
</td>
<td align="center" valign="top">
None
</td>

</tr>
<tr>
<td align="center" valign="top">
E (b)
</td>
<td align="center" valign="top">
1
</td>
<td align="center" valign="top">
’E’
</td>
<td align="center" valign="top">
Extended beta (parallel beta)
</td>

</tr>
<tr>
<td align="center" valign="top">
B
</td>
<td align="center" valign="top">
2
</td>
<td align="center" valign="top">
’B’
</td>
<td align="center" valign="top">
Isolated beta (anti-parallel beta)
</td>

</tr>
<tr>
<td align="center" valign="top">
G
</td>
<td align="center" valign="top">
3
</td>
<td align="center" valign="top">
’G’
</td>
<td align="center" valign="top">
3-10 helix
</td>

</tr>
<tr>
<td align="center" valign="top">
H
</td>
<td align="center" valign="top">
4
</td>
<td align="center" valign="top">
’H’
</td>
<td align="center" valign="top">
Alpha helix
</td>

</tr>
<tr>
<td align="center" valign="top">
I
</td>
<td align="center" valign="top">
5
</td>
<td align="center" valign="top">
’I’
</td>
<td align="center" valign="top">
Pi (3-14) helix
</td>

</tr>
<tr>
<td align="center" valign="top">
T
</td>
<td align="center" valign="top">
6
</td>
<td align="center" valign="top">
’T’
</td>
<td align="center" valign="top">
Turn
</td>

</tr>
<tr>
<td align="center" valign="top">
S
</td>
<td align="center" valign="top">
7
</td>
<td align="center" valign="top">
’S’
</td>
<td align="center" valign="top">
Bend
</td>

</tr>

</table>

</div>

</div>
<div class="Indented">
Average structural propensities over all frames for each residue will be written to the file specified by <b>sumout</b> (or &ldquo;<i>&lt;filename&gt;.sum</i>&rdquo; if <b>sumout</b> is not specified). The total structural propensity over all residues for each secondary structure type will be written to the file specified by <b>totalout</b>. If assignout is specified, the overall secondary structure assignment for each residue will be printed in two line chunks of 50 residues, with the first line containing the residue number the line starts with and one character residue names, and the second line containing secondary structure assignment using DSSP-style characters, like so:
</div>
<pre class="LyX-Code">
1 KCNTATCATQ RLANFLVHSS NNFGAILSST NVGSNTRn
    SSS   TH HHHTTSEEEE TTTEEEE SS     S 
</pre>
<div class="Unindented">
The output of secstruct command is amenable to visualization with gnuplot. To generate a 2D map-style plot of secondary structure vs time, with each residue on the Y axis simply give the output file a &ldquo;.gnu&rdquo; extension. For example, to generate a 2D map of secondary structure vs time, with different colors representing different secondary structure types for residues 1-22:
</div>
<pre class="LyX-Code">
secstruct :1-22 out dssp.gnu
</pre>
<div class="Unindented">
The resulting file can be visualized with gnuplot:
</div>
<pre class="LyX-Code">
gnuplot dssp.gnu
</pre>
<div class="Unindented">
Similarly, the <b>sumout</b> file can be nicely visualized using xmgrace (use &ldquo;.agr&rdquo; extension).
</div>
<pre class="LyX-Code">
secstruct :1-22 out dssp.gnu sumout dssp.agr
xmgrace dssp.agr
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.73">11.73</a> spam
</h2>
<pre class="LyX-Code">
spam &lt;filename&gt; [solv &lt;solvname&gt;] [reorder] [name &lt;name&gt;]
     [purewater] [cut &lt;cut&gt;] [info &lt;infofile&gt;] [summary &lt;summary&gt;]
     [site_size &lt;size&gt;] [sphere] [out &lt;datafile&gt;]
     [dgbulk &lt;dgbulk&gt;] [dhbulk &lt;dhbulk&gt;] [temperature &lt;T&gt;] 
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;filename&gt;</span> File with the peak locations present (XYZ- format) 
</div>
<div class="Description">
<span class="Description-entry">&lt;solvname&gt;</span> Name of the solvent residues 
</div>
<div class="Description">
<span class="Description-entry">&lt;cut&gt;</span> Non-bonded cutoff for energy evaluation
</div>
<div class="Description">
<span class="Description-entry">&lt;dgbulk&gt;</span> SPAM free energy of the bulk solvent in kcal/mol; default is -30.3 kcal/mol (SPC/E water).
</div>
<div class="Description">
<span class="Description-entry">&lt;dhbulk&gt;</span> SPAM enthalpy of the bulk solvent in kcal/mol; default is -22.2 kcal/mol (SPC/E water).
</div>
<div class="Description">
<span class="Description-entry">&lt;T&gt;</span> Temperature at which SPAM calculation was run.
</div>
<div class="Description">
<span class="Description-entry">&lt;infofile&gt;</span> File with stats about which sites are occupied when.
</div>
<div class="Description">
<span class="Description-entry">&lt;size&gt;</span> Size of the water site around each density peak.
</div>
<div class="Description">
<span class="Description-entry">[sphere]</span> Treat each site like a sphere.
</div>
<div class="Description">
<span class="Description-entry">[purewater]</span> The system is pure water---used to parametrize the bulk values.
</div>
<div class="Description">
<span class="Description-entry">[reorder]</span> The solvent should be re-ordered so the same solvent molecule is always in the same site.
</div>
<div class="Description">
<span class="Description-entry">&lt;summary&gt;</span> File with the summary of all SPAM results. If not specified, no SPAM energies will be calculated.
</div>
<div class="Description">
<span class="Description-entry">&lt;datafile&gt;</span> Data file with all SPAM energies for each snapshot.
</div>

</li>

</ul>
<div class="Unindented">
Perform profiling of bound water molecules via SPAM analysis<span class="bibcites">[<a class="bibliocite" name="cite-12" href="#biblio-12"><span class="bib-index">12</span></a>]</span>. Briefly, this method identifies and estimates the free energy profiles of bound waters via calculation of the distribution of interaction energies between the water and it’s environment from explicit solvent MD trajectories. The interaction energies are calculated using a force- and energy-shifted electrostatic term with a hard cutoff.
</div>
<div class="Indented">
Prior to this command, the <b><i>volmap</i></b> command should be run with the <b>peakfile</b> keyword (see <a class="Reference" href="#subsec:cpptraj_volmap">11.86 on page 1↓</a>) to generate the peaks file. If not using peaks from the <b><i>volmap</i></b> command, the peaks file should have one line per peak with format:
</div>
<pre class="LyX-Code">
C &lt;X&gt; &lt;Y&gt; &lt;Z&gt; &lt;Density&gt;
</pre>
<div class="Unindented">
Values of <b>dgbulk</b> and <b>dhbulk</b> for different water models can be calculated from pure water simulations with the <b>purewater</b> keyword.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.74">11.74</a> setvelocity
</h2>
<pre class="LyX-Code">
setvelocity [&lt;mask&gt;]
            [{ tempi &lt;temperature&gt; |
               scale [factor &lt;fac&gt;] [sx &lt;xfac&gt;] [sy &lt;yfac&gt;] [sz &lt;zfac&gt;] |
               none |
               modify}]
            [[ntc &lt;#&gt;]] [[dt &lt;time&gt;] [epsilon &lt;eps&gt;]]
        	[zeromomentum] [ig &lt;random seed&gt;] 
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Mask of atoms to assign velocities to.
</div>
<div class="Description">
<span class="Description-entry">tempi &lt;temperature&gt;</span> Assign velocities at specified temperature (default 300.0 K).
</div>
<div class="Description">
<span class="Description-entry">scale</span> Scale existing velocities
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[factor &lt;fac&gt;]</span> Factor to scale velocities by.
</div>
<div class="Description">
<span class="Description-entry">[sx &lt;xfac&gt;]</span> Factor to scale X component of velocities by.
</div>
<div class="Description">
<span class="Description-entry">[sy &lt;yfac&gt;]</span> Factor to scale Y component of velocities by.
</div>
<div class="Description">
<span class="Description-entry">[sz &lt;zfac&gt;]</span> Factor to scale Z component of velocities by.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">none</span> Remove any velocities.
</div>
<div class="Description">
<span class="Description-entry">modify</span> If specified, do not set, just modify any existing velocities (via ’ntc’ or ’zeromomentum’).
</div>
<div class="Description">
<span class="Description-entry">ig &lt;random seed&gt;</span> Random seed to use to generate velocity distribution.
</div>
<div class="Description">
<span class="Description-entry">ntc &lt;#&gt;</span> Correct set velocities for SHAKE constraints. Numbers match sander/pmemd: 1 = no SHAKE, 2 = SHAKE on hydrogens, 3 = SHAKE on all atoms.
</div>
<div class="Description">
<span class="Description-entry">dt &lt;time&gt;</span> Time step for SHAKE correction.
</div>
<div class="Description">
<span class="Description-entry">epsilon &lt;eps&gt;</span> Epsilon for SHAKE correction
</div>
<div class="Description">
<span class="Description-entry">zeromomentum</span> If specified adjust velocities so the total momentum of atoms in &lt;mask&gt; is zero.
</div>

</li>

</ul>
<div class="Unindented">
Set velocities in frame for atoms in &lt;mask&gt; using Maxwellian distribution based on given temperature.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.75">11.75</a> stfcdiffusion
</h2>
<pre class="LyX-Code">
stfcdiffusion mask &lt;mask&gt; [out &lt;file&gt;] [time &lt;time per frame&gt;]
              [mask2 &lt;mask&gt; [lower &lt;distance&gt;] [upper &lt;distance&gt;]]
              [nwout &lt;file&gt;]) [avout &lt;file&gt;] [distances] [com]
              [x|y|z|xy|xz|yz|xyz]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">mask</span> Atoms for which MSDs will be computed.
</div>
<div class="Description">
<span class="Description-entry">out</span> Output file: time vs. MSD. 
</div>
<div class="Description">
<span class="Description-entry">time</span> Time step in the trajectory. (1.0 ps) 
</div>
<div class="Description">
<span class="Description-entry">mask2</span> Compute MSDs only within the lower and upper limit of mask2. IMPORTANT: may be very slow!!! 
</div>
<div class="Description">
<span class="Description-entry">lower</span> Smaller distance from reference point(s). (0.01 Å) 
</div>
<div class="Description">
<span class="Description-entry">upper</span> Larger distance from reference point(s). (3.5 Å) 
</div>
<div class="Description">
<span class="Description-entry">nwout</span> Output file containing number of water molecules in the chosen region, see mask2. (off) 
</div>
<div class="Description">
<span class="Description-entry">avout</span> Output file containing average distances. (off) 
</div>
<div class="Description">
<span class="Description-entry">x|y|z|xy|xz|yz|xyz</span> Computation of the mean square displacement in the chosen dimension. (xyz) 
</div>
<div class="Description">
<span class="Description-entry">distances</span> Dump un-imaged distances. By default only averages are output. (off) 
</div>
<div class="Description">
<span class="Description-entry">com</span> Calculate MSD for centre of mass. (off)
</div>

</li>

</ul>
<div class="Unindented">
Calculate diffusion for selected atoms using code based on the ’diffusion’ routine developed by Hannes Loeffler at STFC (http://www.stfc.ac.uk/CSE).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.76">11.76</a> strip
</h2>
<pre class="LyX-Code">
strip<a class="IndexReference" name="entry-strip-0" href="#index-strip">↓</a> &lt;mask&gt; [outprefix &lt;name&gt;] [&lt;parmout&gt; file&gt;] [nobox]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Remove atoms specified by mask from the system.
</div>
<div class="Description">
<span class="Description-entry">[outprefix &lt;prefix&gt;]</span> Write out stripped topology file with name ’&lt;prefix&gt;.&lt;Original Topology Name&gt;’.
</div>
<div class="Description">
<span class="Description-entry">[parmout &lt;file&gt;]</span> Write corresponding topology to file with name &lt;file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[nobox]</span> Remove any box information from the stripped topology.
</div>

</li>

</ul>
<div class="Unindented">
Strip all atoms specified by <b>&lt;mask&gt;</b> from the frame and modify the topology to match for any subsequent Actions. The <b>outprefix</b> keyword can be used to write stripped topologies; stripped Amber topologies are fully-functional.
</div>
<div class="Indented">
Note that stripping a system renumbers all atoms and residues, so for example after this command:
</div>
<pre class="LyX-Code">
strip :1
</pre>
<div class="Unindented">
residue 1 will be gone, and the former second residue will now be the first, and so on.
</div>
<div class="Indented">
For example, to strip all residues named WAT from each topology/coordinate frame:
</div>
<pre class="LyX-Code">
strip :WAT
</pre>
<div class="Unindented">
The next example uses a distance-based mask to strip atoms in a single frame. Note that with the exception of the <b><i>mask</i></b> command, distance-based masks do not update on a per-frame basis. To strip all residues outside of 6.0 from any atom in residues 1 to 14 and write out the stripped topology and coordinates, both with no box information:
</div>
<pre class="LyX-Code">
parm parm7 
trajin frame_1000.rst.1
reference frame_1000.rst.1
strip !(:1-14&lt;:6.0) outprefix f1.1 nobox
trajout f1.1.x restart nobox
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.77">11.77</a> surf<a class="Label" name="subsec:cpptraj_surf"> </a>
</h2>
<pre class="LyX-Code">
surf<a class="IndexReference" name="entry-surf-0" href="#index-surf">↓</a> [&lt;name&gt;] [&lt;mask1&gt;] [out &lt;filename&gt;] [solutemask &lt;mask&gt;]
	[offset &lt;offset&gt;] [nbrcut &lt;cut&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;mask1&gt;</span> Atoms to calculate surface area for.
</div>
<div class="Description">
<span class="Description-entry">out &lt;filename&gt;</span> File to write surface area to.
</div>
<div class="Description">
<span class="Description-entry">solutemask &lt;mask&gt;</span> If specified, calculate the contribution of &lt;mask1&gt; to &lt;mask&gt;.
</div>
<div class="Description">
<span class="Description-entry">offset &lt;offset&gt;</span> Increment van der Waals radii by &lt;offset&gt;; 1.4 Ang. is the default (as used by Amber).
</div>
<div class="Description">
<span class="Description-entry">nbrcut &lt;cut&gt;</span> Only atoms with van der Waals radii greater than &lt;cut&gt; are considered to have neighbors (2.5 Ang Amber default).
</div>

</li>

</ul>
<div class="Unindented">
Calculate the surface area in Å<span class="formula"><sup><span class="text">2</span></sup></span> of atoms in &lt;mask&gt; (if no mask specified, all atoms not marked as ’solvent’ that are part of a molecule &gt; 1 atom in size) using the LCPO algorithm of Weiser et al.<span class="bibcites">[<a class="bibliocite" name="cite-35" href="#biblio-35"><span class="bib-index">35</span></a>]</span>. In order for this to work, the topology needs to have bond information and atom type information. 
</div>
<div class="Indented">
Note that even if &lt;mask&gt; does not include all solute atoms, the neighbor list is still calculated for all solute atoms so the surface area calculated reflects the contribution of atoms in &lt;mask&gt; to the overall surface area, not the surface area of &lt;mask&gt; as an isolated system. As a result, it may be possible to obtain a negative surface area if only a small fraction of the solute is selected.
</div>
<div class="Indented">
For example, to calculate the overall surface area of all solute atoms, as well as the contribution of residue 1 to the overall surface area, writing both results to &ldquo;surf.dat&rdquo;:
</div>
<pre class="LyX-Code">
surf out surf.dat
surf :1 out surf.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.78">11.78</a> symmrmsd<a class="Label" name="subsec:cpptraj_symmrmsd"> </a>
</h2>
<pre class="LyX-Code">
symmrmsd  [&lt;name&gt;] [&lt;mask&gt;] [&lt;refmask&gt;] [out &lt;filename&gt;] [nofit] [mass] [remap]
          [ first | reference | ref &lt;name&gt; | refindex &lt;#&gt; | previous |
            reftraj &lt;name&gt; [parm &lt;parmname&gt; | parmindex &lt;#&gt;] ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Mask of atoms to calculate RMSD for; if not specified, calculate for all atoms.
</div>
<div class="Description">
<span class="Description-entry">[&lt;refmask&gt;]</span> Reference mask; if not specified, use <b>&lt;mask&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output data file name.
</div>
<div class="Description">
<span class="Description-entry">[nofit]</span> Do not perform best-fit RMSD (not recommended).
</div>
<div class="Description">
<span class="Description-entry">[mass]</span> Mass-weight the RMSD calculation.
</div>
<div class="Description">
<span class="Description-entry">[remap]</span> Re-arrange atoms according to symmetry. See below for more details.
</div>
<div class="Unindented">
Reference keywords:
</div>
<div class="Description">
<span class="Description-entry"><b>first</b></span> Use the first trajectory frame processed as reference.
</div>
<div class="Description">
<span class="Description-entry"><b>reference</b></span> Use the first previously read in reference structure (refindex 0).
</div>
<div class="Description">
<span class="Description-entry"><b>ref &lt;name&gt;</b></span> Use previously read in reference structure specified by filename/tag.
</div>
<div class="Description">
<span class="Description-entry"><b>refindex &lt;#&gt;</b></span> Use previously read in reference structure specified by &lt;#&gt; (based on order read in).
</div>
<div class="Description">
<span class="Description-entry">previous</span> Use frame prior to current frame as reference.
</div>
<div class="Description">
<span class="Description-entry"><b>reftraj</b> &lt;name&gt;</span> Use frames from COORDS set &lt;name&gt; or read in from trajectory file &lt;name&gt; as references. Each frame from &lt;name&gt; is used in turn, so that frame 1 is compared to frame 1 from &lt;name&gt;, frame 2 is compared to frame 2 from &lt;name&gt; and so on. If &lt;trajname&gt; runs out of frames before processing is complete, the last frame of &lt;trajname&gt; continues to be used as the reference.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">parm &lt;parmname&gt; | parmindex &lt;#&gt;</span> If <b>reftraj</b> specifies a file associate trajectory &lt;name&gt; with specified topology; if not specified the first topology is used.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Perform symmetry-corrected RMSD calculation. This is done by identifying potential symmetric atoms in each residue, performing an initial best-fit, then determining which configuration of symmetric atoms will give the lowest RMSD using atomic distance to reference atoms. 
</div>
<div class="Indented">
<b>Note that when re-mapping, all atoms in the residues of interest should be selected to prevent cases where selected symmetric atoms are swapped but the atoms they are bonded to are not</b>. Also, occasionally larger symmetric structures (e.g. 6 membered rings) may become distorted due to only part of the residue being corrected for symmetry. This appears to happen about 4% of the time but does not overly inflate the RMSD. The <b><i>’check’</i></b> command can be used after <b><i>symmrmsd</i></b> to look for such distortions.
</div>
<div class="Indented">
Warning: the symmetry correction is generally robust enough to account for symmetries in the standard amino and nucleic acid residues, but has not been extensively tested on residues with more extended types of symmetry.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.79">11.79</a> temperature
</h2>
<pre class="LyX-Code">
temperature [&lt;name&gt;] [out &lt;filename&gt;]
            { frame |
              [&lt;mask&gt;] [ntc &lt;#&gt;] [update] [remove {trans|rot|both}]
            }
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write values to.
</div>
<div class="Description">
<span class="Description-entry">frame</span> Do not calculate temperature; use existing frame temperature.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atoms to calculate temperature for.
</div>
<div class="Description">
<span class="Description-entry">[ntc &lt;#&gt;]</span> Value of SHAKE bond constraint: 1 - none, 2 - bonds to H, 3- all bonds (equivalent to SANDER/PMEMD).
</div>
<div class="Description">
<span class="Description-entry">[update]</span> Update temperature in Frames with calculated temperatures.
</div>
<div class="Description">
<span class="Description-entry">[remove {trans|rot|both}]</span> Correct for removed translational, rotational, or both kinds of degrees of freedom.
</div>

</li>

</ul>
<div class="Unindented">
Calculate temperature in frame based on velocity information. If ’<b>update</b>’ is specified, update frame temperature too. If ’<b>frame</b>’ is specified just use frame temperature (e.g. read in from a REMD trajectory).
</div>
<div class="Indented">
The ’ntc’ keyword can be used to correct for lost degrees of freedom due to SHAKE constraints (2 = bonds to hydrogen, 3 = all bonds). The ’remove’ keyword can be used to account for removed translational and/or rotational degrees of freedom.
</div>
<div class="Indented">
For example, if using a trajectory that has been generated with SHAKE on hydrogens, no periodic boundary conditions (i.e. no box), and has had the center of mass periodically removed:
</div>
<pre class="LyX-Code">
temperature T1 ntc 2 remove both out T1.dat
</pre>
<div class="Unindented">
If using a trajectory that has been generated with SHAKE on hydrogens, periodic boundary conditions (i.e. with a box), and has had the center of mass periodically removed:
</div>
<pre class="LyX-Code">
temperature T1 ntc 2 remove trans out T1.dat
</pre>
<div class="Unindented">
If using a trajectory that has been generated with SHAKE on all bonds, periodic boundary conditions, and no center of mass motion removal:
</div>
<pre class="LyX-Code">
temperature T1 ntc 3 out T1.dat
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.80">11.80</a> time
</h2>
<pre class="LyX-Code">
time {time0 &lt;initial time&gt; dt &lt;step&gt; [update] | remove}
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">time0 &lt;initial time&gt;</span> Time of the first frame (ps).
</div>
<div class="Description">
<span class="Description-entry">dt &lt;step&gt;</span> Time step between frames (ps).
</div>
<div class="Description">
<span class="Description-entry">[update]</span> If specified, modify any existing time info.
</div>
<div class="Description">
<span class="Description-entry">remove</span> Remove any time info from frame.
</div>

</li>

</ul>
<div class="Unindented">
Either add time information to frames, modify existing time information in frames, or remove existing time information from frames. Note that currently COORDS data sets do not store time information, so using this command with the <b><i>crdaction</i></b> command will have no effect.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.81">11.81</a> trans | translate
</h2>
<pre class="LyX-Code">
translate [&lt;mask&gt;] [x &lt;dx&gt;] [y &lt;dy&gt;] [z &lt;dz&gt;]
</pre>
<div class="Unindented">
Translate atoms in <b>&lt;mask&gt;</b> (all atoms if no mask specified) <b>&lt;dx&gt;</b> Å in the X direction, <b>&lt;dy&gt;</b> Å in the Y direction, and <b>&lt;dz&gt;</b> Å in the Z direction.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.82">11.82</a> unstrip
</h2>
<pre class="LyX-Code">
unstrip<a class="IndexReference" name="entry-unstrip-0" href="#index-unstrip">↓</a>
</pre>
<div class="Unindented">
Requests that the original topology and frame be used for all following actions. This has the effect of undoing any command that modifies the state (such as strip). For example, the following code takes a solvated complex and uses a combination of strip, unstrip, and outtraj commands to write out separate dry complex, receptor, and ligand files:
</div>
<pre class="LyX-Code">
parm Complex.WAT.pdb
trajin Complex.WAT.pdb
# Remove water, write complex
strip :WAT
outtraj Complex.pdb pdb
# Reset to solvated Complex
unstrip
# Remove water and ligand, write receptor
strip :WAT,LIG
outtraj Receptor.pdb pdb
# Reset to solvated Complex
unstrip
# Remove water and receptor, write ligand
strip :WAT
strip !(:LIG)
outtraj Ligand.pdb pdb
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.83">11.83</a> unwrap
</h2>
<pre class="LyX-Code">
unwrap [center] [{bymol | byres | byatom}]
       [ reference | ref &lt;name&gt; | refindex &lt;#&gt; ] [&lt;mask&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[center]</span> Unwrap by center of mass; otherwise unwrap by first atom position.
</div>
<div class="Description">
<span class="Description-entry">bymol</span> Unwrap by molecule (default).
</div>
<div class="Description">
<span class="Description-entry">byres</span> Unwrap by residue.
</div>
<div class="Description">
<span class="Description-entry">byatom</span> Unwrap by atom.
</div>
<div class="Description">
<span class="Description-entry">[ reference | ref &lt;name&gt; | refindex &lt;#&gt; ]</span> Reference structure to use in unwrapping.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Selection to unwrap.
</div>

</li>

</ul>
<div class="Unindented">
Under periodic boundary conditions, MD trajectories are not continuous if molecules are wrapped(imaged) into the central unit cell. Especially, in sander, with <i>iwrap</i>=1, molecular trajectories become discontinuous when a molecule crosses the boundary of the unit cell. This command, <span class="sans"><b>unwrap</b></span> processes the trajectories to force the <i>mask</i>ed molecules continuous by translating the molecules into the neighboring unit cells. It is the opposite function of <span class="sans"><b>image</b></span>, but this command can also be used to place molecules side by side, for example, two strands of a DNA duplex. However, this command fails when the <i>mask</i>ed molecules travel more than half of the box size within a single frame.
</div>
<div class="Indented">
If the optional argument &ldquo;<tt>reference</tt>&rdquo; is specified, then the first frame is unwrapped according to the reference structure. Otherwise, the first frame is not modified.
</div>
<div class="Indented">
As an example, assume that :1-10 is the first strand of a DNA duplex and :11-20 is the other strand of the duplex. Then the following commands could be used to create system where the two strands are not separated artificially:
</div>
<pre class="LyX-Code">
unwrap :1-20
center :1-20 mass origin
image origin center familiar
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.84">11.84</a> vector<a class="Label" name="subsec:cpptraj-vector"> </a>
</h2>
<pre class="LyX-Code">
vector [&lt;name&gt;] &lt;Type&gt; [out &lt;filename&gt; [ptrajoutput]] [&lt;mask1&gt;] [&lt;mask2&gt;]
   	[magnitude] [ired]
       &lt;Type&gt; = { mask     | minimage  | dipole | center   | corrplane |
                  box      | boxcenter | ucellx | ucelly   | ucellz
   	           momentum | principal [x|y|z]  | velocity | force }\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Vector data set name.
</div>
<div class="Description">
<span class="Description-entry">&lt;Type&gt;</span> Vector type; see below.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write vector data to <b>&lt;filename&gt;</b> with format ’Vx Vy Vz Ox Oy Oz’ where V denotes vector coordinates and ’O’ denotes origin coordinates.
</div>
<div class="Description">
<span class="Description-entry">[ptrajoutput]</span> Write vector data in <i>ptraj</i> style (Vx Vy Vz Ox Oy Oz Vx+Ox Vy+Oy Vz+Oz). This prevents additional formatting of <b>&lt;filename&gt;</b> and is not compatible with <b>’magnitude’</b>.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask1&gt;]</span> Atom mask, required for all types except <b>’box’</b>.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask2&gt;]</span> Second atom mask, only required for type <b>’mask’</b>.
</div>
<div class="Description">
<span class="Description-entry">[magnitude]</span> Store the magnitude of the vector with aspect [Mag].
</div>
<div class="Description">
<span class="Description-entry">[ired]</span> Mark this vector for subsequent IRED analysis with commands ’matrix ired’ and ’ired’.
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Vector data set.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Mag]</span> (<b>magnitude</b> only) Vector magnitude.
</div>

</li>

</ul>
<div class="Unindented">
This command will keep track of a vector value (and its origin) over the trajectory; the data can be referenced for later use based on the <i>name</i> (which must be unique). The types of vectors that can be calculated are:
</div>
<div class="Description">
<span class="Description-entry">mask</span> (Default) Store vector from center of mass of atoms in <b>&lt;mask1&gt;</b> to atoms in <b>&lt;mask2&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">minimage</span> Store minimum-imaged vector from center of mass of atoms in <b>&lt;mask1&gt;</b> to atoms in <b>&lt;mask2&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">dipole</span> Store the dipole and center of mass of the atoms specified in <b>&lt;mask1&gt;</b>. The vector is not converted to appropriate units, nor is the value well-defined if the atoms in the mask are not overall charge neutral.
</div>
<div class="Description">
<span class="Description-entry">center</span> Store the center of mass of atoms in <b>&lt;mask1&gt;</b>. The reference point is the origin (0.0, 0.0, 0.0).
</div>
<div class="Description">
<span class="Description-entry">corrplane</span> This defines a vector perpendicular to the (least-squares best) plane through the atoms in <b>&lt;mask1&gt;</b>. The reference point is the center of mass of atoms in <b>&lt;mask1&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">box</span> (No mask needed) Store the box lengths of the trajectory. The reference point is the origin (0.0, 0.0, 0.0). 
</div>
<div class="Description">
<span class="Description-entry">boxcenter</span> (No mask needed) Store the center of the box as a vector.
</div>
<div class="Description">
<span class="Description-entry">ucell{x|y|z}:</span> (No mask needed) Store specified unit cell (i.e. box) vector.
</div>
<div class="Description">
<span class="Description-entry">momentum</span> Store momentum of atoms selected by &lt;mask1&gt; (requires velocities).
</div>
<div class="Description">
<span class="Description-entry">principal [x|y|z]</span> Store one of the principal axis vectors determined by diagonalization of the inertial matrix from the coordinates of the atoms specified by <b>&lt;mask1&gt;</b>. The eigenvector with the largest eigenvalue is considered &ldquo;<tt>x</tt>&rdquo; (i.e., the hardest axis to rotate around) and the eigenvector with the smallest eigenvalue is considered &ldquo;<tt>z</tt>&rdquo;. If none of <tt>x</tt> or <tt>y</tt> or <tt>z</tt> are specified, then the &ldquo;x&rdquo; principal axis is stored. The reference point is the center of mass of atoms in <b>&lt;mask1&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">velocity</span> Store velocity of atoms in &lt;mask1&gt; (requires velocities).
</div>
<div class="Description">
<span class="Description-entry">force</span> Store force of atoms in &lt;mask1&gt; (requires forces).
</div>
<div class="Unindented">
Cpptraj supports writing out vector data in a pseudo-trajectory format for easy visualization. Once a vector data set has been generated the writedata command can be used with the vectraj keyword (see <a class="Reference" href="#sec:Datafile-Commands">6 on page 1↑</a> for more details) to write a pseudo trajectory consisting of two atoms, one for the vector origin and one for the vector from the origin (i.e. V+O). For example, to create a MOL2 containing a pseudo-trajectory of the minimum-imaged vector from residue 4 to residue 11:
</div>
<pre class="LyX-Code">
trajin tz2.nc
vector v8 minimage out v8.dat :4 :11 
run
writedata v8.mol2 vectraj v8 trajfmt mol2 
</pre>
<div class="Unindented">
Auto-correlation or cross-correlation functions can be calculated subsequently for vectors using either the <b><i>corr</i></b> analysis command or the <b><i>timecorr</i></b> analysis command (to calculate via spherical harmonic theory).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.85">11.85</a> velocityautocorr
</h2>
<pre class="LyX-Code">
velocityautocorr [&lt;set name&gt;] [&lt;mask&gt;] [usevelocity] [out &lt;filename&gt;] [diffout &lt;file&gt;]
                 [maxlag &lt;frames&gt;] [tstep &lt;timestep&gt;] [direct] [norm]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[&lt;set name&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atoms(s) to calculate velocity autocorrelation (VAC) function for.
</div>
<div class="Description">
<span class="Description-entry">[usevelocity]</span> Use velocity information in frame if present. This will only give sensible results if the velocities are recorded close to the order of the simulation time step.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write VAC function to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[diffout &lt;file&gt;]</span> File to write diffusion constants to.
</div>
<div class="Description">
<span class="Description-entry">[maxlag &lt;frames&gt;]</span> Maximum lag in frames to calculate VAC function for. Default is half the total number of frames.
</div>
<div class="Description">
<span class="Description-entry">[tstep &lt;timestep&gt;]</span> Time between frames in ps (default 1.0).
</div>
<div class="Description">
<span class="Description-entry">[direct]</span> Calculate VAC function directly instead of via FFT (will be much slower).
</div>
<div class="Description">
<span class="Description-entry">[norm]</span> Normalize resulting VAC function to 1.0.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[D]</span> Diffusion constant calculated from integral over VAC function in 1x10<sup>-5</sup> cm<sup>2</sup>/s.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the velocity autocorrelation (VAC) function averaged over the atoms in &lt;mask&gt;. Pseudo-velocities are calculated using coordinates and the specified time step. As with all time correlation functions the statistical noise will increase if the maximum lag is greater than half the total number of frames. In addition to calculating the velocity autocorrelation function, the self-diffusion coefficient will be reported in the output, calculated from the integral over the VAC function.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.86">11.86</a> volmap<a class="Label" name="subsec:cpptraj_volmap"> </a>
</h2>
<pre class="LyX-Code">
volmap filename dx dy dz &lt;mask&gt; [radscale &lt;factor&gt;]
   	{ data &lt;existing set&gt; |
   	  name &lt;setname&gt; { size &lt;x,y,z&gt; [center &lt;x,y,z&gt;] |
   	                   centermask &lt;mask&gt; [buffer &lt;buffer&gt;] } }
   	[peakcut &lt;cutoff&gt;] [peakfile &lt;xyzfile&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">filename</span> The name of the output file with the grid density.
</div>
<div class="Description">
<span class="Description-entry">dx, dy, dz</span> The grid spacing (Angstroms) in the X-, Y-, and Z-dimensions, respectively
</div>
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> The atom selection from which to calculate the number density.
</div>
<div class="Description">
<span class="Description-entry">radscale &lt;factor&gt;</span> Factor by which to scale radii (by division). To match the atomic radius of Oxygen used by the VMD volmap tool, a scaling factor of 1.36 should be used. Default 1.0.
</div>
<div class="Description">
<span class="Description-entry">data &lt;setname&gt;</span> Name of existing grid data set to use.
</div>
<div class="Description">
<span class="Description-entry">name &lt;setname&gt;</span> Name of grid set that will be created (<b>size/center</b> or <b>centermask/buffer</b> keywords).
</div>
<div class="Description">
<span class="Description-entry">size &lt;x,y,z&gt;</span> Specify the size of the grid in the X-, Y-, and Z-dimensions. Must be used alongside the center argument.
</div>
<div class="Description">
<span class="Description-entry">center &lt;x,y,z&gt;</span> Specify the grid center explicitly. Note, the size argument must be present in this case. Default is the origin.
</div>
<div class="Description">
<span class="Description-entry">centermask &lt;mask&gt;</span> The mask around which the grid should be centered (via geometric center). If this is omitted and the center and size are not specified, the default &lt;mask&gt; entered (see above) is used in its place.
</div>
<div class="Description">
<span class="Description-entry">buffer &lt;buffer&gt;</span> A buffer distance, in Angstroms, by which the edges of the grid should clear every atom of the centermask (or default mask if centermask is omitted) in every direction. The default value is 3. The buffer is ignored if the center and size are specified (see below).
</div>
<div class="Description">
<span class="Description-entry">peakcut &lt;cutoff&gt;</span> The minimum density required to consider a local maximum a ’density peak’ in the outputted peak file (default 0.05).
</div>
<div class="Description">
<span class="Description-entry">peakfile &lt;xyzfile&gt;</span> A file in XYZ-format that contains a carbon atom centered at the grid point of every local density maximum. This file is necessary input to the spam action command.
</div>

</li>

</ul>
<div class="Unindented">
Grid data as a volumetric map, similar to the ’volmap’ command in VMD. The density is calculated by treating each atom as a 3-dimensional Gaussian function whose standard deviation is equal to the van der Waals radius. The density calculated is the number density averaged over the entire simulation. The grid can be specified in one of three ways:
</div>
<ol>
<li>
An existing grid data set (from e.g. bounds), specified with the <b>data</b> keyword.
</li>
<li>
Via the sizes and center specified by the <b>size</b> and <b>center</b> keywords (comma-separated strings, e.g. ’20,20,20’).
</li>
<li>
Centered on the atoms in the mask given by <b>centermask</b> with an additional buffer in each direction specified by <b>buffer</b>.
</li>

</ol>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.87">11.87</a> volume
</h2>
<pre class="LyX-Code">
volume [&lt;name&gt;] [out &lt;filename&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">out &lt;filename&gt;</span> Output file name.
</div>

</li>

</ul>
<div class="Unindented">
Calculate unit cell volume.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.88">11.88</a> watershell
</h2>
<pre class="LyX-Code">
watershell &lt;solutemask&gt; [out &lt;filename&gt;] [lower &lt;lower cut&gt;] [upper &lt;upper cut&gt;]
           [noimage] [&lt;solventmask&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;solutemask&gt;</span> Atom mask corresponding to solute of interest (required).
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output file name.
</div>
<div class="Description">
<span class="Description-entry">[lower &lt;lower cut&gt;]</span> Cutoff for the first water shell (default 3.4 Angstroms).
</div>
<div class="Description">
<span class="Description-entry">[upper &lt;upper cut&gt;]</span> Cutoff for the second water shell (default 5.0 Angstroms).
</div>
<div class="Description">
<span class="Description-entry">[noimage]</span> Do not image distances.
</div>
<div class="Description">
<span class="Description-entry">[&lt;solventmask&gt;]</span> Optional atom mask corresponding to solvent.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[lower]</span> Number of solvent molecules in first solvent shell.
</div>
<div class="Description">
<span class="Description-entry">[upper]</span> Number of solvent molecules in second solvent shell.
</div>

</li>

</ul>
<div class="Unindented">
This option will count the number of waters within a certain distance of the atoms in the &lt;solutemask&gt; in order to represent the first and second solvation shells. The optional &lt;solventmask&gt; can be used to consider other atoms as the solvent; the default is &ldquo;:WAT&rdquo;.
</div>
<div class="Indented">
This action is often used prior to the <b><i>closest</i></b> command in order to determine how many waters around a solute should be retained to maintain the first and/or second water shells.
</div>
<div class="Indented">
As of version 17 this command is CUDA-enabled in CUDA versions of CPPTRAJ.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-11.89">11.89</a> xtalsymm
</h2>
<pre class="LyX-Code">
xtalsymm &lt;mask&gt; group &lt;space group&gt; [collect [centroid]]
     	[ first | reference | ref &lt;name&gt; | refindex &lt;#&gt; ]
     	[na &lt;na&gt;] [nb &lt;nb&gt;] [nc &lt;nc&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;mask&gt;</span> Atom mask defining the asymmetric unit within the larger system (required).
</div>
<div class="Description">
<span class="Description-entry">group &lt;space group&gt;</span> The space group to which the system belongs. Omit spaces in the name. Example: &ldquo;P22(1)2(1)&rdquo;.
</div>
<div class="Description">
<span class="Description-entry">[collect]</span> Optional flag to have all solvent particles, not just the asymmetric units, re-imaged. This will trigger cpptraj to compute the unit cell volume that constitutes the aymmetric unit and thereby classify all particles for re-imaging.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[centroid]</span> If specified along with <b>collect</b>, re-image solvent molecules by centroids, not individual atom coordinates. This is useful for keeping water molecules intact.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[first | reference | ref &lt;name&gt; | refindex &lt;#&gt;]</span> Reference structure to use for determining crystal symmetry.
</div>
<div class="Description">
<span class="Description-entry">[na &lt;na&gt;] [nb &lt;nb&gt;] [nc &lt;nc&gt;]</span> The number of times the crystal unit cell is replicated along the “a,” “b,” or “c” axes (for orthorhombic unti cells, these are the x, y, and z axes) of the simulation; default is 1. Many crystal unit cells are too small in one or more dimensions for our simulation cutoffs, and replicating the unit cell is an effective way to counter imaging artifacts even for larger unit cells.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the optimal approach for superimposing symmetry-related subunits of the simulation back onto one another. The calculation assumes that the system is a simulation of an X-ray structure in its native crystal lattice, finds all copies of the asymmetric unit among the entire system, and devises plans for re-imagining their coordinates to superimpose them back on the original asymmetric unit. The space group information can be found in a PDB X-ray structure used as the initial coordinates for a simulation. All 230 space groups are supported, and a scan of the PDB was made to ensure that common variants of the names are included (P2(1)22(1) is the same as P22(1)2(1), but with different axis conventions). If your space group is not understood, contact the Amber mailing list. This command is compute intensive, especially for simulations that are “supercells” containing many crystallographic unit cells.
</div>
<div class="Indented">
This command will cause cpptraj to locate all asymmetric units from within the topology, then determine what wrapping, if any, has occurred in order to bring about an optimal re-alignment based on the space group symmetry operations. The user need not worry about wrapping or drift of the simulation over time–the asymmetric units will be re-imaged frame by frame. Coordinate modifications due to this action are permanent and will affect the results of subsequent actions and analyses.
</div>
<h1 class="Section">
<a class="toc" name="toc-Section-12">12</a> Analysis Commands
</h1>
<div class="Unindented">
Analyses in <i>cpptraj</i> operate on data sets which have been generated by Actions in a prior Run or read in with a <b><i>readdata</i></b> command (<a class="Reference" href="#subsec:cpptraj_readdata">8.19 on page 1↑</a>). Unlike <i>ptraj</i>, Analysis commands in <i>cpptraj</i> do not need to be prefaced with <tt>’analysis’</tt>. The exception to this is ’<b><i>analyze matrix</i></b>’ in order to differentiate it from the <b><i>matrix</i></b> Action command; users are encouraged to use the new command <b><i>diagmatrix</i></b> instead.
</div>
<div class="Indented">
Like Actions, when an Analysis command is issued it is by default added to the Analysis queue and is not executed until after trajectory processing is completed; a complete list of data sets available for analysis is shown after trajectory processing (prefaced by ’DATASETS’) or can be shown with the ’<b><i>list dataset</i></b>’ command. Analyses can also be executed immediately via the <b><i>runanalysis</i></b> command (<a class="Reference" href="#subsec:cpptraj_runanalysis">8.24 on page 1↑</a>).
</div>
<div class="Indented">
Note that for Analysis commands that use COORDS data sets, if no COORDS data set is specified then a default one will be automatically created from frames read in by <b><i>trajin</i></b> commands.\begin_inset Separator latexpar\end_inset
</div>
<div class="Indented">
<div class="center">
<table>
<tr>
<td align="center" valign="top" style="width: 25%;">
<b>Command</b>
</td>
<td align="center" valign="top" style="width: 50%;">
<b>Description</b>
</td>
<td align="center" valign="top" style="width: 25%;">
<b>Set Type(s)</b>
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
autocorr
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate autocorrelation function for multiple data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D sclar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
avg
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate average, standard deviation, min, and max for (or over) data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
calcstate
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate states based on given data sets and criteria.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
cluster
</td>
<td align="center" valign="top" style="width: 50%;">
Perform cluster analysis.
</td>
<td align="center" valign="top" style="width: 25%;">
COORDS, N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
corr,
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate auto or cross correlation for 1 or 2 data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
1D scalar, vector
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
 correlationcoe
</td>
<td align="center" valign="top" style="width: 50%;">

</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
cphstats
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate statistics for constant pH data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
pH data sets
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
crank, 
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate crankshaft motion between two data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
2 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
crankshaft
</td>
<td align="center" valign="top" style="width: 50%;">

</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
crdfluct
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate atomic fluctuations (RMSF) for atoms over time blocks.
</td>
<td align="center" valign="top" style="width: 25%;">
COORDS
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
crosscorr
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate a matrix of Pearson product-moment 
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">

</td>
<td align="center" valign="top" style="width: 50%;">
coefficients between given data sets.
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
curvefit
</td>
<td align="center" valign="top" style="width: 50%;">
Perform non-linear curve fitting on given data set.
</td>
<td align="center" valign="top" style="width: 25%;">
1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
diagmatrix
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate eigenvectors and eigenvalues from given symmetric matrix.
</td>
<td align="center" valign="top" style="width: 25%;">
symmetric matrix
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
divergence
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate Kullback-Leibler divergence between two data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
2 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
FFT
</td>
<td align="center" valign="top" style="width: 50%;">
Perform a fast Fourier transform on data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
hausdorff
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate the Hausdorff distance for given matrix data set(s).
</td>
<td align="center" valign="top" style="width: 25%;">
N 2D matrices
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
hist, histogram
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate N-dimensional histogram for N given data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
integrate
</td>
<td align="center" valign="top" style="width: 50%;">
Perform integration on each of the given data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
ired
</td>
<td align="center" valign="top" style="width: 50%;">
Perform isotropic reorientational eigenmode dynamics
</td>
<td align="center" valign="top" style="width: 25%;">
N IRED vectors
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">

</td>
<td align="center" valign="top" style="width: 50%;">
 analysis using given IRED vectors.
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
kde
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate 1D histogram from given data set using a kernel density estimator.
</td>
<td align="center" valign="top" style="width: 25%;">
1 or 2 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">

</td>
<td align="center" valign="top" style="width: 50%;">
Also time-dependent Kullback-Leibler divergence analysis with another set.
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
lifetime
</td>
<td align="center" valign="top" style="width: 50%;">
Perform lifetime analysis on given data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
lowestcurve
</td>
<td align="center" valign="top" style="width: 50%;">
For each given data set, calculate a curve that traces 
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">

</td>
<td align="center" valign="top" style="width: 50%;">
the lowest N points over specified bins.
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
meltcurve
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate a melting curve from given data sets assuming simple 2 state kinetics.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
modes
</td>
<td align="center" valign="top" style="width: 50%;">
Perform various analyses on eigenmodes (from e.g. <b><i>diagmatrix</i></b>).
</td>
<td align="center" valign="top" style="width: 25%;">
eigenmodes
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
multicurve
</td>
<td align="center" valign="top" style="width: 50%;">
Perform non-linear curve fitting for multiple input data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
multihist
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate 1D histograms (optionally with a kernel
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">

</td>
<td align="center" valign="top" style="width: 50%;">
 density estimator) from multiple input data sets.
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
phipsi
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate and plot the average phi and psi values from input dihedral data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N phi/psi dihedrals
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
regress
</td>
<td align="center" valign="top" style="width: 50%;">
Perform linear regression on multiple input data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
remlog
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate various statistics from a replica log data set.
</td>
<td align="center" valign="top" style="width: 25%;">
replica log
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
rms2d, 2drms
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate 2D RMSD between frames in 1 or 2 COORDS data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
1 or 2 COORDS
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
rmsavgcorr
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate RMS average correlation curve for a COORDS data set.
</td>
<td align="center" valign="top" style="width: 25%;">
COORDS
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
rotdif
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate rotational diffusion using given rotation matrices (from e.g. <b><i>rms</i></b>).
</td>
<td align="center" valign="top" style="width: 25%;">
rotation matrices
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
runningavg
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate running average for given data sets using given window size.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
spline
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate cubic splines for given data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
stat, statistics
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate various statistics for given data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
ti
</td>
<td align="center" valign="top" style="width: 50%;">
Peform Gaussian quadrature integration for given DV/DL data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
N 1D scalar
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
timecorr
</td>
<td align="center" valign="top" style="width: 50%;">
Calculate auto/cross-correlation functions for given
</td>
<td align="center" valign="top" style="width: 25%;">
1 or 2 vector
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">

</td>
<td align="center" valign="top" style="width: 50%;">
 vector(s) using spherical harmonics.
</td>
<td align="center" valign="top" style="width: 25%;">

</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
vectormath
</td>
<td align="center" valign="top" style="width: 50%;">
Perform math on given vector data sets.
</td>
<td align="center" valign="top" style="width: 25%;">
2 vector
</td>

</tr>
<tr>
<td align="center" valign="top" style="width: 25%;">
wavelet
</td>
<td align="center" valign="top" style="width: 50%;">
Perform wavelet analysis on coordinates from given COORDS set.
</td>
<td align="center" valign="top" style="width: 25%;">
COORDS
</td>

</tr>

</table>

</div>

</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.1">12.1</a> autocorr
</h2>
<pre class="LyX-Code">
autocorr [name &lt;dsetname&gt;] &lt;dsetarg0&gt; [&lt;dsetarg1&gt; ...] [out &lt;filename&gt;]
         [lagmax &lt;lag&gt;] [nocovar] [direct]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>&lt;dsetarg0&gt; [dsetarg1&gt; ...]</b></span> Argument(s) specifying datasets to be used.
</div>
<div class="Description">
<span class="Description-entry"><b>[name &lt;dsetname&gt;]</b></span> Store results in dataset(s) named &lt;dsetname&gt;:X.
</div>
<div class="Description">
<span class="Description-entry"><b>[out &lt;filename</b>&gt;<b>]</b></span> Write results to file named &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>[lagmax]</b></span> Maximum lag to calculate for. If not specified all frames are used.
</div>
<div class="Description">
<span class="Description-entry"><b>[nocovar]</b></span> Do not calculate covariance.
</div>
<div class="Description">
<span class="Description-entry"><b>[direct]</b></span> Do not use FFTs to calculate correlation; this will be much slower.
</div>

</li>

</ul>
<div class="Unindented">
<i>This is for integer/double/float datasets only; for vectors see the <b>’timecorr’</b> command.</i>
</div>
<div class="Indented">
Calculate auto-correlation (actually auto-covariance by default) function for datasets specified by one or more dataset arguments. The datasets must have the same # of data points.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.2">12.2</a> avg
</h2>
<pre class="LyX-Code">
avg &lt;dset0&gt; [&lt;dset1&gt; ...] [torsion] [out &lt;file&gt;] [oversets]
    [name &lt;name&gt;] [nostdout]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dsetX&gt;</span> Data set(s) to calculate the average for.
</div>
<div class="Description">
<span class="Description-entry">[torsion]</span> If the data sets are not already marked periodic (e.g. if read in via ’readdata’), treat them as periodic torsion.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to write results to.
</div>
<div class="Description">
<span class="Description-entry">[oversets]</span> If specified, calculate the average over all inpout sets instead of each input set.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[nostdout]</span> If ’nostdout’ specified do not write averages to STDOUT when ’out’ not specified.
</div>
<div class="Unindented">
DataSets Created (not <b>oversets</b>):
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[avg]</span> Average of each set.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[sd]</span> Standard deviation of each set.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[ymin]</span> Y minimum of each set.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[ymax]</span> Y maximum of each set.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[yminidx]</span> Index of minimum Y value.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[ymaxidx]</span> Index of maximum Y value.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[names]</span> Name of each set.
</div>
<div class="Unindented">
DataSets Created (oversets)
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Average over all input sets for each frame.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[SD]</span> Standard deviation over all input sets for each frame.
</div>

</li>

</ul>
<pre class="LyX-Code">

</pre>
<div class="Unindented">
Calculate the average, standard deviation, min, and max of given 1D data sets. Alternatively, if <b>oversets</b> is specified the average over each set for each point is calculated; this requires all input sets be the same size.
</div>
<div class="Indented">
For example, to read in data from a file named perres.peptide.dat and calculate the averages etc for all the input sets:
</div>
<pre class="LyX-Code">
readdata perres.peptide.dat 
avg perres.peptide.dat out output.dat name V 
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.3">12.3</a> calcstate
</h2>
<pre class="LyX-Code">
calcstate {state &lt;ID&gt;,&lt;dataset&gt;,&lt;min&gt;,&lt;max&gt;[,&lt;dataset1&gt;,&lt;min1&gt;,&lt;max1&gt;]} ...
      	[out &lt;state v time file&gt;] [name &lt;setname&gt;]
      	[curveout &lt;curve file&gt;] [stateout &lt;states file&gt;]
      	[transout &lt;transitions file&gt;] [countout &lt;count file&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">state &lt;ID&gt;,&lt;dataset&gt;,&lt;min&gt;,&lt;max&gt;</span> Define a state according to given data set and criteria. Multiple states can be given, and each state can have multiple criteria. If multiple criteria are specified, each one must be satisfied in order to assign the state. If the same state is defined multiple times, the state will be assigned if either criteria match.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;ID&gt;</span> Name to give each state index. State indices start at 0. -1 means &ldquo;undefined state&rdquo;.
</div>
<div class="Description">
<span class="Description-entry">&lt;dataset&gt;</span> Data set to use.
</div>
<div class="Description">
<span class="Description-entry">&lt;min&gt;,&lt;max&gt;</span> Frames with data set value above &lt;min&gt; and below &lt;max&gt; will be assigned &lt;ID&gt;.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[out &lt;state v time file&gt;]</span> File to write state index vs frame to.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;setname&gt;]</span> Data set name.
</div>
<div class="Description">
<span class="Description-entry">[curveout &lt;curve file&gt;]</span> File to write state lifetime and transition curves to.
</div>
<div class="Description">
<span class="Description-entry">[stateout &lt;states file&gt;]</span> File to write state lifetime data to.
</div>
<div class="Description">
<span class="Description-entry">[transout &lt;transitions file&gt;]</span> File to write state transition data to.
</div>
<div class="Description">
<span class="Description-entry">[countout &lt;state count file&gt;]</span> File to write state counts (i.e. how many frames each state was observed) to.
</div>
<div class="Unindented">
DataSets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;</span> State index vs frame.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Count]</span> Number of frames each state was observed.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Frac]</span> Fraction of time each state was observed
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Nlifetimes]</span> Number of times each state was reached.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Avglife]</span> Average lifetime length for each state.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Maxlife]</span> Maximum lifetime of each state.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Name]</span> Name (&lt;ID&gt;) of each state.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Xlifetimes]</span> Number of times each state transitioned to each other state.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Xavglife]</span> Average lifetime of each state before transitioning to each other state.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Xmaxlife]</span> Maximum lifetime of each state before transitioning to each other state.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[Xname]</span> Name of each transition, format &ldquo;StateA-&gt;StateB&rdquo;.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[sCurve]:X</span> State curves; lifetime curve for transitions from given state to any other state.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[tCurve]:X</span> Transition curves; lifetime curve for transitions from given state to other specific state.
</div>

</li>

</ul>
<div class="Unindented">
Data for the specified data set(s) that matches the given criteria will be assigned a state index. State indices start from 0 and match the order in which <b>state</b> keywords were given. The -1 state index is reserved for &ldquo;undefined state&rdquo;. For example, the following input:
</div>
<pre class="LyX-Code">
parm DPDP.parm7
trajin DPDP.nc
distance d1 :19@O :12@N
angle a1 :19@O :12@H :12@N
calcstate state D,d1,3.0,4.0 state A,a1,100,120 out state.dat curveout curve.agr \
                stateout States.dat transout States.dat name d1_a1
run
</pre>
<div class="Unindented">
Defines two states. State index 0 is defined as a state named &ldquo;D&rdquo; based on the distance from ’:19@O’ to ’:12@N’ (data set d1) being between 3 and 4 Angstroms. State index 1 is defined as a state named &ldquo;A&rdquo; based on the angle between ’:19@O’, ’:12@H’, and ’:12@N’ (data set a1) being between 100 and 120 degrees. The output in state.dat might look like:
</div>
<pre class="LyX-Code">
#Frame          d1_a1
       1           -1
       2            0
       3            0
       4            0
       5           -1
       6            1
       7           -1
       8           -1
       9            0
      10           -1
</pre>
<div class="Unindented">
where the values in column d1_a1 refer to state index: -1 is undefined, 0 is state &ldquo;D&rdquo;, and 1 is state &ldquo;A&rdquo;.
</div>
<div class="Indented">
To define a state State1 as having a distance named &ldquo;dist&rdquo; between 2.5 and 5.0 Ang. and an angle named &ldquo;ang&rdquo; between 30 and 60 degrees OR having a distance named &ldquo;distA&rdquo; between 0.0 and 3.0 Ang.:
</div>
<pre class="LyX-Code">
calcstate state State1,dist,2.5,5.0,ang,30,60 \
          state State1,distA,0.0,3.0
</pre>
<div class="Unindented">
Lifetime curves (see <a class="Reference" href="#subsec:cpptraj-lifetime">12.19 on page 1↓</a> for further explanation) are calculated for transitions from each state to any other state (aspect [sCurve]) and each state to each other state (aspect [tCurve]). In this case there will be 3 sCurves and 4 tCurves:
</div>
<pre class="LyX-Code">
d1_a1[sCurve]:0 "Undefined" (double), size is 10 
d1_a1[sCurve]:1 "D" (double), size is 3 
d1_a1[sCurve]:2 "A" (double), size is 1 
d1_a1[tCurve]:0 "Undefined-&gt;D" (double), size is 10
d1_a1[tCurve]:1 "D-&gt;Undefined" (double), size is 3
d1_a1[tCurve]:2 "Undefined-&gt;A" (double), size is 1
d1_a1[tCurve]:3 "A-&gt;Undefined" (double), size is 1
</pre>
<div class="Unindented">
Lifetime analysis from each state to any other state is directed to the file specified by <b>stateout</b> and has format:
</div>
<pre class="LyX-Code">
#Index N Average Max State
</pre>
<div class="Unindented">
Where <tt>#Index</tt> is the state index, <tt>N</tt> is the number of lifetimes in that state, <tt>Average</tt> is the average lifetime while in that state (in frames), <tt>Max</tt> is the maximum lifetime while in that state (in frames) and <tt>State</tt> is the name of the state.
</div>
<div class="Indented">
Finally, lifetime analysis of transitions from each state to each other state is directory to the file specified by transout and has format:
</div>
<pre class="LyX-Code">
#N Average Max Transition
</pre>
<div class="Unindented">
Where <tt>#N</tt> is the number of transitions, <tt>Average</tt> is the average lifetime (in frames) in the first state before transitioning to the second state, <tt>Max</tt> is the max lifetime (in frames) before transitioning to the second state, and <tt>Transition</tt> is the name of the transition.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.4">12.4</a> <a class="Label" name="subsec:cpptraj-cluster-analysis"> </a>cluster
</h2>
<pre class="LyX-Code">
cluster<a class="IndexReference" name="entry-cluster-0" href="#index-cluster">↓</a> [crdset &lt;crd set&gt; | nocoords]
  Algorithms:
	[hieragglo [epsilon &lt;e&gt;] [clusters &lt;n&gt;] [linkage|averagelinkage|complete]
      [epsilonplot &lt;file&gt;] [includesieved_cdist]]
	[dbscan minpoints &lt;n&gt; epsilon &lt;e&gt; [sievetoframe] [kdist &lt;k&gt; [kfile &lt;prefix&gt;]]]
    [dpeaks epsilon &lt;e&gt; [noise] [dvdfile &lt;density_vs_dist_file&gt;]
	  [choosepoints {manual | auto}]
	  [distancecut &lt;distcut&gt;] [densitycut &lt;densitycut&gt;]
	  [runavg &lt;runavg_file&gt;] [deltafile &lt;file&gt;] [gauss]]
    [kmeans clusters &lt;n&gt; [randompoint [kseed &lt;seed&gt;]] [maxit &lt;iterations&gt;]
    [{readtxt|readinfo} infofile &lt;file&gt;]
  Distance options:
	{[[rms | srmsd] [&lt;mask&gt;] [mass] [nofit]] | [dme [&lt;mask&gt;]] |
      [data &lt;dset0&gt;[,&lt;dset1&gt;,...]]}
	[sieve &lt;#&gt; [random [sieveseed &lt;#&gt;]]] [loadpairdist] [savepairdist] [pairdist &lt;file&gt;]
    [pairwisecache {mem | none}] [includesieveincalc] [pwrecalc]
  Output options:
	[out &lt;cnumvtime&gt;] [gracecolor] [summary &lt;summaryfile&gt;] [info &lt;infofile&gt;]
	[summarysplit &lt;splitfile&gt;] [splitframe &lt;comma-separated frame list&gt;]
    [bestrep {cumulative|centroid|cumulative_nosieve}] [savenreps &lt;#&gt;]
    [clustersvtime &lt;filename&gt; cvtwindow &lt;window size&gt;]
	[cpopvtime &lt;file&gt; [normpop | normframe] [lifetime]]
    [sil &lt;silhouette file prefix&gt;] [assignrefs [refcut &lt;rms&gt;] [refmask &lt;mask&gt;]]
  Coordinate output options:
	[ clusterout &lt;trajfileprefix&gt; [clusterfmt &lt;trajformat&gt;] ]
	[ singlerepout &lt;trajfilename&gt; [singlerepfmt &lt;trajformat&gt;] ]
	[ repout &lt;repprefix&gt; [repfmt &lt;repfmt&gt;] [repframe] ]
    [ avgout &lt;avgprefix&gt; [avgfmt &lt;avgfmt&gt;] ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[crdset &lt;crd set&gt;]</span> Name of previously generated COORDS data set. If not specified the default COORDS set will be used unless <b>nocoords</b> has been specified.
</div>
<div class="Description">
<span class="Description-entry">[nocoords]</span> Do not use a COORDS data set; distance metrics that require coordinates and coordiante output will be disabled.
</div>
<div class="Unindented">
Algorithms:
</div>
<div class="Description">
<span class="Description-entry">hieragglo</span> (Default) Use hierarchical agglomerative (bottom-up) approach.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[epsilon &lt;e&gt;]</span> Finish clustering when minimum distance between clusters is greater than <b>&lt;e&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[clusters &lt;n&gt;]</span> Finish clustering when <b>&lt;n&gt;</b> clusters remain.
</div>
<div class="Description">
<span class="Description-entry">[<b>linkage]</b></span> Single-linkage; use the shortest distance between members of two clusters.
</div>
<div class="Description">
<span class="Description-entry">[<b>averagelinkage]</b></span> Average-linkage (default); use the average distance between members of two clusters.
</div>
<div class="Description">
<span class="Description-entry"><b>[complete]</b></span> Complete-linkage; use the maximum distance between members of two clusters.
</div>
<div class="Description">
<span class="Description-entry">[epsilonplot &lt;file&gt;]</span> Write number of clusters vs epsilon to <b>&lt;file&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[includesieved_cdist]</span> Include sieved frames in final cluster distance calculation (<u>may be very slow</u>).
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">dbscan</span> Use DBSCAN clustering algorithm of Ester et al.<span class="bibcites">[<a class="bibliocite" name="cite-14" href="#biblio-14"><span class="bib-index">14</span></a>]</span>\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">minpoints &lt;n&gt;</span> Minimum number of points required to form a cluster.
</div>
<div class="Description">
<span class="Description-entry">epsilon &lt;e&gt;</span> Distance cutoff between points for forming a cluster.
</div>
<div class="Description">
<span class="Description-entry">[sievetoframe]</span> When restoring sieved frames, compare frame to every frame in a cluster instead of the centroid; slower but more accurate.
</div>
<div class="Description">
<span class="Description-entry">[kdist &lt;k&gt;]</span> Generate K-dist plot for help in determining DBSCAN parameters (see below).
</div>
<div class="Description">
<span class="Description-entry">[kfile &lt;prefix&gt;]</span> Prefix for K-dist plot file.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">dpeaks</span> Use the density peaks algorithm of Rodriguez and Laio<span class="bibcites">[<a class="bibliocite" name="cite-1" href="#biblio-1"><span class="bib-index">1</span></a>]</span>\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">epsilon &lt;e&gt;</span> Cutoff for determining local density in Angstroms.
</div>
<div class="Description">
<span class="Description-entry">[noise]</span> If specified, treat all points within epsilon of another cluster as noise.
</div>
<div class="Description">
<span class="Description-entry">[dvdfile &lt;density_vs_dist_file&gt;]</span> File to write density versus minimum distance to point with next highest density. This can be used to determine appropriate cutoffs for distance and density in a subsequent step with <b>choosepoints</b> <b>manual</b>.
</div>
<div class="Description">
<span class="Description-entry">[choosepoints {manual | auto}]</span> Specify whether clusters will be chosen based on specified distance/density cutoffs, or automatically. If not specified only the density vs distance file will be written and no clustering will be performed. Currently <b>manual</b> is recommended.
</div>
<div class="Description">
<span class="Description-entry">[distancecut &lt;distcut&gt;] [densitycut &lt;densitycut&gt;]</span> If <b>choosepoints</b> <b>manual</b>, points with minimum distance greather than or equal to &lt;distcut&gt; and density greater than or equal to &lt;densitycut&gt; will be chosen.
</div>
<div class="Description">
<span class="Description-entry">[runavg &lt;runavg file&gt;]</span> If <b>choosepoints</b> <b>automatic</b>, the calculated running average of density versus distance will be written to &lt;runavg file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[deltafile &lt;file&gt;]</span> If <b>choosepoints</b> <b>automatic</b>, distance minus the running average for each point will be written to this file.
</div>
<div class="Description">
<span class="Description-entry">[gauss]</span> Calculate density with Gaussian kernels instead of using discrete density.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">kmeans</span> Use K-means clustering algorithm.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">clusters &lt;n&gt;</span> Finish clustering when number of clusters is <b>&lt;n&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[randompoint]</span> Randomize initial set of points used (recommended).
</div>
<div class="Description">
<span class="Description-entry">[kseed &lt;seed&gt;]</span> Random number generator seed for <b>randompoint</b>.
</div>
<div class="Description">
<span class="Description-entry">[maxit &lt;iteration&gt;]</span> Algorithm will run until frames no longer change clusters of &lt;iteration&gt; iterations are reached (default 100).
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">readtxt|readinfo</span> No clustering - read in previous cluster results.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">infofile &lt;file&gt;</span> Cluster info file to read.
</div>

</li>

</ul>
<div class="Unindented">
Distance Metric Options:
</div>
<div class="Description">
<span class="Description-entry">[rms | srmsd[&lt;mask&gt;]]</span> (Default <b>rms</b>) Distance between frames calculated via best-fit coordinate RMSD using atoms in <b>&lt;mask&gt;</b>. If <b>srmsd</b> specified use symmetry-corrected RMSD (see <a class="Reference" href="#subsec:cpptraj_symmrmsd">11.78 on page 1↑</a>).\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[mass]</span> Mass-weight the RMSD.
</div>
<div class="Description">
<span class="Description-entry">[nofit]</span> Do not fit structures onto each other prior to calculating RMSD.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">dme [&lt;mask&gt;]</span> Distance between frames calculated using distance-RMSD (aka DME, <b><i>distrmsd</i></b>) using atoms in <b>&lt;mask&gt;</b>.
</div>
<div class="Description">
<span class="Description-entry">[data &lt;dset0&gt;[,&lt;dset1&gt;,...]</span> Distance between frames calculated using specified data set(s) (Euclidean distance).
</div>
<div class="Description">
<span class="Description-entry">[sieve &lt;#&gt;]</span> Perform clustering only for every <b>&lt;#&gt;</b> frame. After clustering, all other frames will be added to clusters.
</div>
<div class="Description">
<span class="Description-entry">[random]</span> When <b>sieve</b> is specified, select initial frames to cluster randomly.
</div>
<div class="Description">
<span class="Description-entry">[sieveseed &lt;#&gt;]</span> Seed for random sieving; if not set the wallclock time will be used.
</div>
<div class="Description">
<span class="Description-entry">[pairdist &lt;file&gt;]</span> File to use for loading/saving pairwise distances.
</div>
<div class="Description">
<span class="Description-entry">[loadpairdist]</span> Load pairwise distances from <b>&lt;file&gt;</b> (CpptrajPairDist if pairdist not specified).
</div>
<div class="Description">
<span class="Description-entry">[savepairdist]</span> Save pairwise distances from <b>&lt;file&gt;</b> (CpptrajPairDist if pairdist not specified). NOTE: If sieving was performed only the calculated distances are saved.
</div>
<div class="Description">
<span class="Description-entry">[pairwisecache {mem | disk | none}]</span> Cache pairwise distance data in memory (default), to disk, or disable pairwise caching. No caching will save memory but be extremely slow. Caching to disk will likely be slow unless writing to a fast storage device (e.g. SSD) - data is saved to a file named ’CpptrajPairwiseCache’.
</div>
<div class="Description">
<span class="Description-entry">[includesieveincalc]</span> Include sieved frames when calculating within-cluster average (<u>may be very slow</u>).
</div>
<div class="Description">
<span class="Description-entry">[pwrecalc]</span> If a loaded pairwise distance file does not match the current setup, force recalculation.
</div>
<div class="Unindented">
Output Options:
</div>
<div class="Description">
<span class="Description-entry">[out &lt;cnumvtime&gt;]</span> Write cluster # vs frame to <b>&lt;cnumvtime&gt;</b>. Algorithms that calculate noise (e.g. DBSCAN) will assign noise points a value of -1.
</div>
<div class="Description">
<span class="Description-entry">[gracecolor]</span> Instead of cluster # vs frame, write cluster# + 1 (corresponding to colors used by XMGRACE) vs frame. Cluster #s larger than 15 are given the same color. Algorithms that calculate noise (e.g. DBSCAN) will assign noise points a color of 0 (blank).
</div>
<div class="Description">
<span class="Description-entry">[summary &lt;summaryfile&gt;]</span> Summarize each cluster with format ’#Cluster Frames Frac AvgDist Stdev Centroid AvgCDist’:\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">#Cluster</span> Cluster number starting from 0 (0 is most populated).
</div>
<div class="Description">
<span class="Description-entry">Frames</span> # of frames in cluster.
</div>
<div class="Description">
<span class="Description-entry">Frac</span> Size of cluster as fraction of total trajectory.
</div>
<div class="Description">
<span class="Description-entry">AvgDist</span> Average distance between points in the cluster.
</div>
<div class="Description">
<span class="Description-entry">Stdev</span> Standard deviation of points in the cluster.
</div>
<div class="Description">
<span class="Description-entry">Centroid</span> Frame # of structure in cluster that has the lowest cumulative distance to every other point.
</div>
<div class="Description">
<span class="Description-entry">AvgCDist</span> Average distance of this cluster to every other cluster.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[info &lt;infofile&gt;]</span> Write ptraj-like cluster information to <b>&lt;infofile&gt;</b>. This file has format:<br/>
#Clustering: &lt;X&gt; clusters &lt;N&gt; frames<br/>
#Cluster &lt;I&gt; has average-distance-to-centroid &lt;AVG&gt;<br/>
...<br/>
#DBI: &lt;DBI&gt;<br/>
#pSF: &lt;PSF&gt;<br/>
#Algorithm: &lt;algorithm-specific info&gt;<br/>
&lt;Line for cluster 0&gt;<br/>
...<br/>
#Representative frames: &lt;representative frame list&gt;<br/>
Where &lt;X&gt; is the number of clusters, &lt;N&gt; is the number of frames clustered, &lt;I&gt; ranges from 0 to &lt;X&gt;-1, &lt;AVG&gt; is the average distance of all frames in that cluster to the centroid, &lt;DBI&gt; is the Davies-Bouldin Index, &lt;pSF&gt; is the pseudo-F statistic, and &lt;representative frame list&gt; contains the frame # of the representative frame (i.e. closest to the centroid) for each cluster. Each cluster has a line made up of characters (one for each frame) where ’.’ means ’not in cluster’ and ’X’ means ’in cluster’.
</div>
<div class="Description">
<span class="Description-entry">[summarysplit &lt;splitfile&gt;]</span> Summarize each cluster based on which of its frames fall in portions of the trajectory specified by splitframe with format ’#Cluster Total Frac C# Color NumInX ... FracX ... FirstX’:\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">#Cluster</span> Cluster number starting from 0 (0 is most populated).
</div>
<div class="Description">
<span class="Description-entry">Total</span> # of frames in cluster.
</div>
<div class="Description">
<span class="Description-entry">Frac</span> Size of cluster as a fraction of the total trajectory.
</div>
<div class="Description">
<span class="Description-entry">C#</span> Grace color number.
</div>
<div class="Description">
<span class="Description-entry">Color</span> Text description of the color (based on standard XMGRACE coloring).
</div>
<div class="Description">
<span class="Description-entry">NumInX</span> Number of frames in Xth portion of the trajectory.
</div>
<div class="Description">
<span class="Description-entry">FracX</span> Fraction of frames in Xth portion of the trajectory.
</div>
<div class="Description">
<span class="Description-entry">FirstX</span> Frame in the Xth portion of the trajectory where the cluster is first observed.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[splitframe &lt;frame&gt;]</span> For <b>summarysplit</b>, frame or comma-separated list of frames to split the trajectory at, e.g. ’100,200,300’.
</div>
<div class="Description">
<span class="Description-entry">[bestrep {cumulative|centroid|cumulative_nosieve}]</span> Method for choosing cluster representative frames.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">cumulative</span> Choose by lowest cumulative distance to all other frames in cluster. Default when not sieving.
</div>
<div class="Description">
<span class="Description-entry">centroid</span> Choose by lowest distance to cluster centroid. Default when sieving.
</div>
<div class="Description">
<span class="Description-entry">cumulative_nosieve</span> Choose by lowest cumulative distance to all other frames, ignoring sieved frames.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[savenreps &lt;#&gt;]</span> Number of best representative frames to choose (default 1).
</div>
<div class="Description">
<span class="Description-entry">[clustersvtime &lt;filename&gt;]</span> Write number of unique clusters observed in a given time window to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[cvtwindow &lt;windowsize&gt;]</span> Window size for <b>clustersvtime</b> output.
</div>
<div class="Description">
<span class="Description-entry">[cpopvtime &lt;file&gt; [normpop | normframe]]</span> Write cluster population vs time to <b>&lt;file&gt;</b>; if <b>normpop</b> specified normalize each cluster to 1.0; if <b>normframe</b> specified normalize cluster populations by number of frames.
</div>
<div class="Description">
<span class="Description-entry">[sil &lt;prefix&gt;]</span> Write average cluster silhouette value for each cluster to ’&lt;prefix&gt;.cluster.dat’ and cluster silhouette value for each individual frame to ’&lt;prefix&gt;.frame.dat’.
</div>
<div class="Description">
<span class="Description-entry">assignrefs</span> In <b>summary</b>/<b>summarysplit</b>, assign clusters to loaded representative structures if RMSD to that reference is less than specified cutoff.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[refcut &lt;rms&gt;]</span> RMSD cutoff in Angstroms.
</div>
<div class="Description">
<span class="Description-entry">[refmask &lt;mask&gt;]</span> Mask to use for RMSD calculation. If not specified the default mask is all heavy atoms.
</div>

</li>

</ul>
<div class="Unindented">
Coordinate Output Options:
</div>
<div class="Description">
<span class="Description-entry">clusterout &lt;trajfileprefix&gt;</span> Write frames in each cluster to files named &lt;trajfileprefix&gt;.cX, where X is the cluster number.
</div>
<div class="Description">
<span class="Description-entry">clusterfmt &lt;trajformat&gt;</span> Format keyword for clusterout (default Amber Trajectory).
</div>
<div class="Description">
<span class="Description-entry">singlerepout &lt;trajfilename&gt;</span> Write all representative frames to single trajectory named &lt;trajfilename&gt;.
</div>
<div class="Description">
<span class="Description-entry">singlerepfmt &lt;trajformat&gt;</span> Format keyword for singlerepout (default Amber Trajectory).
</div>
<div class="Description">
<span class="Description-entry">repout &lt;repprefix&gt;</span> Write representative frames to separate files named &lt;repprefix&gt;.X.&lt;ext&gt;, where X is the cluster number and &lt;ext&gt; is a format-specific filename extension.
</div>
<div class="Description">
<span class="Description-entry">repfmt &lt;trajformat&gt;</span> Format keyword for repout (default Amber Trajectory).
</div>
<div class="Description">
<span class="Description-entry">repframe</span> Include representative frame number in <b>repout</b> filename.
</div>
<div class="Description">
<span class="Description-entry">avgout &lt;avgprefix&gt;</span> Write average structure for each cluster to separate files named &lt;avgprefix&gt;.X.&lt;ext&gt;, where X is the cluster number and &lt;ext&gt; is a format-specific filename extension.
</div>
<div class="Description">
<span class="Description-entry">avgfmt &lt;trajformat&gt;</span> Format keyword for avgout.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[Pop]</span> Cluster population vs time; index corresponds to cluster number.
</div>

</li>

</ul>
<div class="Unindented">
<i>Note cluster population vs time data sets are not generated until the analysis has been run.</i>
</div>
<div class="Indented">
Cluster input frames using the specified clustering algorithm and distance metric. In order to speed up clustering of large trajectories, the <b>sieve</b> keyword can be used. In addition, subsequent clustering calculations can be sped up by writing/reading calculated pair distances between each frame to/from a file specified by <b>pairdist</b> (or &ldquo;CpptrajPairDist&rdquo; if <b>pairdist</b> not specified).
</div>
<div class="Indented">
Example: cluster on a specific distance:
</div>
<pre class="LyX-Code">
distance endToEnd :1 :255
cluster data endToEnd clusters 10 epsilon 3.0 summary summary.dat info info.dat
</pre>
<div class="Unindented">
Example: cluster on the CA atoms of residues 2-10 using average-linkage, stopping when either 3 clusters are reached or the minimum distance between clusters is 4.0, writing the cluster number vs time to &ldquo;cnumvtime.dat&rdquo; and a summary of each cluster to &ldquo;avg.summary.dat&rdquo;:
</div>
<pre class="LyX-Code">
cluster C1 :2-10 clusters 3 epsilon 4.0 out cnumvtime.dat summary avg.summary.dat
</pre>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--27"></a>Clustering Metrics
</h3>
<div class="Unindented">
The Davies-Bouldin Index (DBI) measures sum over all clusters of the within cluster scatter to the between cluster separation; <b>the smaller the DBI, the better</b>. The DBI is defined as the average, for all clusters X, of fred, where fred(X) = max, across other clusters Y, of (Cx + Cy)/dXY. Here Cx is the average distance from points in X to the centroid, similarly Cy, and dXY is the distance between cluster centroids. 
</div>
<div class="Indented">
The pseudo-F statistic (pSF) is another measure of clustering goodness. It is intended to capture the ’tightness’ of clusters, and is in essence a ratio of the mean sum of squares between groups to the mean sum of squares within group. <b>High values are good</b>. Generally, one selects a cluster-count that gives a peak in the pseudo-f statistic. Formula: A/B, where A = (T - P)/(G-1), and B = P / (n-G). Here n is the number of points, G is the number of clusters, T is the total distance from the all-data centroid, and P is the sum (for all clusters) of the distances from the cluster centroid. 
</div>
<div class="Indented">
The cluster silhouette is a measure of how well each point fits within a cluster. Values of 1 indicate the point is very similar to other points in the cluster, i.e. it is well-clustered. Values of -1 indicate the point is dissimilar and may fit better in a neighboring cluster. Values of 0 indicate the point is on a border between two clusters. 
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--28"></a>Hints for setting DBSCAN parameters with ’kdist’<a class="Label" name="subsec:cpptraj_dbscan_Hints"> </a>
</h3>
<div class="Unindented">
It is not always obvious what parameters to set for DBSCAN. You can get a rough idea of what to set ’mindist’ and ’epsilon’ to by generating a so-called "K-dist" plot with the ’kidst &lt;k&gt;’ option. The K-dist plot shows for each point (X axis) the Kth farthest distance (Y axis), sorted by decreasing distance. You supply the same distance metric and sieve parameters you want to use for the actual clustering, but nothing else. For example:
</div>
<pre class="LyX-Code">
cluster C0 dbscan kdist 4 rms :1-4@CA sieve 10 loadpairdist pairdist CpptrajPairDist
</pre>
<div class="Unindented">
The K-dist plot will be named &lt;prefix&gt;.&lt;k&gt;.dat, with the default prefix being ’Kdist’ (in this case the file name would be Kdist.4.dat). The K-dist plot usually looks like a curve with an initially steep slope that gradually decreases. Around where the initial part of the curve starts to flatten out (indicating an increas in density) is around where epsilon should be set; minpoints is set to whatever &lt;k&gt; was. It has been suggested that the shape of the K-dist curve doesn’t change too much after Kdist=4, but users are encouraged to experiment.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--29"></a>Using ’dpeaks’ clustering
</h3>
<div class="Unindented">
The ’dpeaks’ (density peaks) algorithm attempts to find clusters by identifying points in high density regions which are far from other points of high density<span class="bibcites">[<a class="bibliocite" name="cite-1" href="#biblio-1"><span class="bib-index">1</span></a>]</span>. There are two ways these points can be chosen. The first and recommended way is manually. In this method, clustering if first run with <b>choosepoints</b> not specified to generate a plot containing density versus minimum distance to point with next highest density (the decision graph). Appropriate cut offs for distance and density can then be chosen based on visual inspection; cutoffs should be chosen so that they select points that have both a high density and a high distance to point with next highest density. Clustering can then be run again with <b>distancecut</b> and <b>densitycut</b> set.
</div>
<div class="Indented">
The second way is automatically; cpptraj will attempt to identify outliers in the density vs distance plot based on distance from the running average. Although this only requires a single pass, this method of choosing points is not well-tested and currently not recommended.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--30"></a>The CpptrajPairDist file format
</h3>
<div class="Unindented">
The CpptrajPairDist file is binary; the exact format depends on what version of cpptraj generated the file (since earlier versions had no concept of ’sieve’). The CpptrajPairDist file starts with a 4 byte header containing the characters ’C’ ’T’ ’M’ followed by the version number. A quick way to figure out the version is to use the linux ’od’ command to output the first 4 bytes as hexadecimal, e.g.:
</div>
<pre class="LyX-Code">
$ od -t x1 -N 4 CpptrajPairDist 0000000 43 54 4d 02
</pre>
<div class="Unindented">
So the CpptrajPairDist file version in the above example is 2.
</div>
<div class="Indented">
The next few numbers describe the matrix size and depend on the version.
</div>
<div class="Description">
<span class="Description-entry">Version 0:</span> Two 4-byte integers: # of rows and # of elements. 
</div>
<div class="Description">
<span class="Description-entry">Version 1:</span> Two 8-byte unsigned integers (equivalent to size_t on most systems): # of rows and # of elements. 
</div>
<div class="Description">
<span class="Description-entry">Version 2:</span> Three 8 byte unsigned integers: original # of rows, actual # of rows, and sieve value.
</div>
<div class="Unindented">
This is followed by the actual matrix data, stored as a single array of floats (4 bytes). For versions 1 and 2 the number of elements is explicitly stored. For version 2, to calculate the number of matrix elements you need to read:
</div>
<pre class="LyX-Code">
Elements = (actual_rows * (actual_rows - 1)) / 2
</pre>
<div class="Unindented">
The cluster pair-distance matrix is an upper-right triangle matrix without the diagonal (in row-major order), so the first element is the distance between elements 0 and 1, the second is between elements 0 and 2, etc.
</div>
<div class="Indented">
In version 2 files, if the sieve value is greater than 1 that means original_rows &gt; actual_rows and there is an additional array of characters original_nrows long, with ’T’ if the row is being ignored (i.e. it was sieved out) and ’F’ if the row is active (i.e. is active in the actual pairwise-distance matrix).
</div>
<div class="Indented">
The code that cpptraj uses to read in CpptrajPairDist files is in ClusterMatrix::LoadFile() (ClusterMatrix.cpp).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.5">12.5</a> cphstats
</h2>
<pre class="LyX-Code">
cphstats &lt;pH sets&gt; [name &lt;name&gt;] [statsout &lt;statsfile&gt;] [deprot]
     	[fracplot [fracplotout &lt;file&gt;]]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;pH sets&gt;</span> Previously read in pH data sets.
</div>
<div class="Description">
<span class="Description-entry">name &lt;name&gt;</span> Output set name.
</div>
<div class="Description">
<span class="Description-entry">statsout &lt;statsfile&gt;</span> Write pH statistics to &lt;statsfile&gt;
</div>
<div class="Description">
<span class="Description-entry">deprot</span> If specified, calculate fraction deprotonated instead of protonated.
</div>
<div class="Description">
<span class="Description-entry">fracplot</span> If specified, calculate fraction protonated/deprotonated vs pH.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">fracplotout &lt;file&gt;</span> File to write fraction plots to.
</div>

</li>

</ul>
<div class="Unindented">
Data Sets Generated
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Frac]:&lt;idx&gt;</span> Fraction protonated/deprotonated for residue &lt;idx&gt;.
</div>

</li>

</ul>
<div class="Unindented">
Calculate statistics for constant pH simulation data previously read in with <b><i>readdata</i></b> (see <a class="Reference" href="#subsec:cpptraj_cphdatafile">6.11 on page 1↑</a>). Statistics are calculated for each residue at each input pH. Output format is as follows:
</div>
<pre class="LyX-Code">
Solvent pH is &lt;pH&gt;
&lt;res name&gt; &lt;res num&gt; : Offset &lt;off&gt; Pred &lt;pred&gt; Frac Prot &lt;frac&gt; Transitions &lt;#trans&gt;
...

Average total molecular protonation: &lt;avg&gt;
</pre>
<div class="Unindented">
Where <b>&lt;off&gt;</b> is offset from predicted, <b>&lt;pred&gt;</b> is predicted pH, and &lt;#trans&gt; is the number of transitions. A line is printed for each residue. This functionality is similar to the <b>cphstats</b> utility that comes with Amber (see <a class="Reference" href="#sec:cphstats"> on page 1↓</a>).
</div>
<div class="Indented">
Note that data from constant pH REMD must be sorted prior to use with <b><i>cphstats</i></b>. See the <b><i>readensembledata</i></b> (<a class="Reference" href="#subsec:cpptraj_readensembledata">8.20 on page 1↑</a>) and <b><i>sortensembledata</i></b> (<a class="Reference" href="#subsec:cpptraj_sortensembledata">8.27 on page 1↑</a>) commands for more details.
</div>
<div class="Indented">
For example, to read in constant pH data from constant pH REMD, sort and analyze:
</div>
<pre class="LyX-Code">
readensembledata ExplicitRemd/cpout.001 cpin ExplicitRemd/cpin name PH
sortensembledata PH
runanalysis cphstats PH[*] statsout stats.dat fracplot fracplotout frac.agr deprot
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.6">12.6</a> corr | correlationcoe
</h2>
<pre class="LyX-Code">
corr<a class="IndexReference" name="entry-corr-0" href="#index-corr">↓</a> out &lt;outfilename&gt; &lt;dataset1&gt; [&lt;dataset2&gt;]
     [lagmax &lt;lag&gt;] [nocovar] [direct]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>out &lt;outfilename&gt;</b></span> Write results to file named &lt;outfilename&gt;. The datasets must have the same # of data points.
</div>
<div class="Description">
<span class="Description-entry"><b>&lt;dataset1&gt; [&lt;dataset2&gt;]</b></span> Data set(s) to calculate correlation for. If one dataset or the same dataset is given twice, the auto-correlation will be calculated, otherwise cross-correlation.
</div>
<div class="Description">
<span class="Description-entry"><b>[lagmax]</b></span> Maximum lag to calculate for. If not specified all frames are used.
</div>
<div class="Description">
<span class="Description-entry"><b>[nocovar]</b></span> Do not calculate covariance.
</div>
<div class="Description">
<span class="Description-entry"><b>[direct]</b></span> Do not use FFTs to calculate correlation; this will be much slower.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[&lt;dataset1&gt;]</span> (Auto-correlation) The aspect will be the name of each of the input data set.
</div>
<div class="Description">
<span class="Description-entry">[&lt;dataset1&gt;-&lt;dataset2&gt;]</span> (Cross-correlation) The aspect will be the names of each of the input data sets joined by a dash (’-’).
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[coeff]</span> Correlation coefficient.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the auto-correlation function for data set named &lt;dataset1&gt; or the cross-correlation function for data sets named &lt;dataset1&gt; and &lt;dataset2&gt; up to &lt;lagmax&gt; frames (all if <b>lagmax</b> not specified), writing the result to file specified by <b>out</b>. The two datasets must have the same # of datapoints. 
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.7">12.7</a> crank | crankshaft
</h2>
<pre class="LyX-Code">
crank {angle | distance} &lt;dsetname1&gt; &lt;dsetname2&gt; info &lt;string&gt;
      [out &lt;filename&gt;] [results &lt;resultsfile&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">angle</span> Analyze angle data sets.
</div>
<div class="Description">
<span class="Description-entry">distance</span> Analyze distance data sets.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsetname1&gt;</span> Data set to analyze.
</div>
<div class="Description">
<span class="Description-entry">&lt;dsetname2&gt;</span> Data set to analyze.
</div>
<div class="Description">
<span class="Description-entry">info &lt;string&gt;</span> Title the analysis &lt;string&gt;.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write frame-vs-bin to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[results &lt;resultsfile&gt;]</span> Write results to &lt;resultsfile&gt;.
</div>

</li>

</ul>
<div class="Unindented">
Calculate crankshaft motion between two data sets.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.8">12.8</a> crdfluct
</h2>
<pre class="LyX-Code">
[crdset &lt;crd set&gt;] [&lt;mask&gt;] [out &lt;filename&gt;] [window &lt;size&gt;] [bfactor]
</pre>
<div class="Unindented">
Calculate atomic positional fluctuations for atoms in <b>&lt;mask&gt;</b> over windows of size <b>&lt;size&gt;</b>. If <b>bfactor</b> is specified, the fluctuations are weighted by <span class="formula"><span class="fraction"><span class="ignored">(</span><span class="numerator">8</span><span class="ignored">)/(</span><span class="denominator">3</span><span class="ignored">)</span></span><i>π</i><sup>2</sup></span>(similar but not necessarily equivalent to crystallographic B-factor calculation). Units are Å, or Å<sup>2</sup>x<span class="formula"><span class="fraction"><span class="ignored">(</span><span class="numerator">8</span><span class="ignored">)/(</span><span class="denominator">3</span><span class="ignored">)</span></span><i>π</i><sup>2</sup></span> if <b>bfactor</b> specified.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.9">12.9</a> crosscorr
</h2>
<pre class="LyX-Code">
crosscorr [name &lt;dsetname&gt;] &lt;dsetarg0&gt; [&lt;dsetarg1&gt; ...] [out &lt;filename&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>[name &lt;dsetname&gt;]</b></span> The resulting upper-triangle matrix is stored with name &lt;dsetname&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>&lt;dsetarg0&gt; [&lt;dsetarg1&gt; ...]</b></span> Argument(s) specifying datasets to be used.
</div>
<div class="Description">
<span class="Description-entry"><b>[out &lt;filename&gt;]</b></span> Write results to file named &lt;filename&gt;.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the Pearson product-moment correlation coefficients between all specified datasets.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.10">12.10</a> curvefit<a class="Label" name="subsec:cpptraj_curvefit"> </a>
</h2>
<pre class="LyX-Code">
curvefit &lt;dset&gt; { &lt;equation&gt;  |
                  name &lt;dsname&gt; {gauss | nexp &lt;m&gt; [form {mexp|mexpk|mexpk_penalty}} }
         [AX=&lt;value&gt; ...] [out &lt;outfile&gt;] [resultsout &lt;results&gt;]
         [maxit &lt;max iterations&gt;] [tol &lt;tolerance&gt;]
         [outxbins &lt;NX&gt; outxmin &lt;xmin&gt; outxmax &lt;xmax&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dset&gt;</span> Data set to fit.
</div>
<div class="Description">
<span class="Description-entry">&lt;equation&gt;</span> Equation to fit of form &lt;Variable&gt; = &lt;Equation&gt;. See <a class="Reference" href="#subsec:cpptraj-Data-Set-Math">5.2 on page 1↑</a> for more details on equations <i>cpptraj</i> understands.
</div>
<div class="Description">
<span class="Description-entry">name &lt;dsname&gt;</span> Final data set name (required if using <b>nexp</b> or <b>gauss</b>).
</div>
<div class="Description">
<span class="Description-entry">gauss</span> Fit to Gaussian of form A0 * exp( -((X - A1)^2) / (2 * A2^2) )
</div>
<div class="Description">
<span class="Description-entry">nexp &lt;m&gt;</span> Fit to specified number of exponentials.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">form &lt;type&gt;</span> Fit to specified exponential form:\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">mexp</span> Multi-exponential, SUM(m)[ An * exp(An+1 * X)]
</div>
<div class="Description">
<span class="Description-entry">mexpk</span> Multi-exponential plus constant, A0 + SUM(m)[An * exp(An+1 * X)]
</div>
<div class="Description">
<span class="Description-entry">mexpk_penalty</span> Same as mexpk except sum of prefactors constrained to 1.0 and exponential constants constrained to &lt; 0.0.
</div>

</li>

</ul>

</li>

</ul>
<div class="Description">
<span class="Description-entry">AX=&lt;value&gt;</span> Value of any constants in specified equation with X starting from 0 (can specify more than one).
</div>
<div class="Description">
<span class="Description-entry">out &lt;outfile&gt;</span> Write resulting fit curve to &lt;outfile&gt;.
</div>
<div class="Description">
<span class="Description-entry">resultsout &lt;results&gt;</span> Write details of the fit to &lt;results&gt; (default STDOUT).
</div>
<div class="Description">
<span class="Description-entry">maxit &lt;max iterations&gt;</span> Number of iterations to run curve fitting algorithm (default 50).
</div>
<div class="Description">
<span class="Description-entry">tol &lt;tolerance&gt;</span> Curve-fitting tolerance (default 1E-4).
</div>
<div class="Description">
<span class="Description-entry">outxbins &lt;NX&gt;</span> Number of points to use when generating final curve (default same number of points as input data set).
</div>
<div class="Description">
<span class="Description-entry">outxmin &lt;xmin&gt;</span> Minimum X value to use for final curve (default same number of points as input data set).
</div>
<div class="Description">
<span class="Description-entry">outxmax &lt;xmax&gt;</span> Maximum X value to use for final curve (default same number of points as input data set).
</div>

</li>

</ul>
<div class="Unindented">
Perform non-linear curve fitting for the specified data set using the Levenberg-Marquardt algorithm. Any equation form that cpptraj understands (see <a class="Reference" href="#subsec:cpptraj-Data-Set-Math">5.2 on page 1↑</a>) can be used, or several preset forms can be used. Similar to Grace (http://plasma-gate.weizmann.ac.il/Grace/), an equation can contain constants for curve fitting termed AX (with X being a numerical digit, one for each constant), and is assigned to a variable which then becomes a data set. For example, to fit a curve to data from a file named Data.dat to a data set named ’FitY’:
</div>
<pre class="LyX-Code">
readdata Data.dat
runanalysis curvefit Data.dat \
  "FitY = (A0 * exp(X * A1)) + (A2 * exp(X * A3))" \
  A0=1 A1=-1 A2=1 A3=-1 \
  out curve.dat tol 0.0001 maxit 50
</pre>
<div class="Unindented">
To perform the same fit but to a multi-exponential curve with two exponentials:
</div>
<pre class="LyX-Code">
readdata Data.dat
runanalysis curvefit Data.dat nexp 2 name FitY \
  A0=1 A1=-1 A2=1 A3=-1 \
  out curve1.dat tol 0.0001 maxit 50
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.11">12.11</a> diagmatrix<a class="Label" name="subsec:diagmatrix-|-analyze"> </a>
</h2>
<pre class="LyX-Code">
diagmatrix &lt;name&gt; [out &lt;filename&gt;] [thermo [outthermo &lt;filename&gt;]]
           [vecs &lt;#&gt;] [name &lt;modesname&gt;] [reduce] 
           [nmwiz [nmwizvecs &lt;#&gt;] [nmwizfile &lt;filename&gt;]]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>&lt;name&gt;</b></span> Name of symmetric matrix to diagonalize.
</div>
<div class="Description">
<span class="Description-entry"><b>[out &lt;filename&gt;]</b></span> Write results to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>[thermo [outthermo &lt;filename&gt;]]</b></span> Mass-weighted covariance (mwcovar) matrix only. Calculate entropy, heat capacity, and internal energy from the structure of a molecule (average coordinates, see above) and its vibrational frequencies using standard statistical mechanical formulas for an ideal gas. Results are written to &lt;filename&gt; if specified, otherwise results are written to STDOUT. <b>Note that this converts the units of the calculated eigenvalues to frequencies (cm<sup>-1</sup>)</b>.
</div>
<div class="Description">
<span class="Description-entry"><b>[vecs &lt;#&gt;]</b></span> Number of eigenvectors to calculate. Default is 0, which is only allowed when ’thermo’ is specified.
</div>
<div class="Description">
<span class="Description-entry"><b>[name &lt;modesname&gt;]</b></span> Store resulting modes data set with name &lt;modesname&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>[reduce]</b></span> Covariance (covar/mwcovar/distcovar) matrices only. For coordinate covariance (covar/mwcovar) matrices, each eigenvector element is reduced via Ei = Eix^2 + Eiy^2 + Eiz^2. For distance covariance (distcovar) the eigenvectors are reduced by taking the sum of the squares of each row. See Abseher &amp; Nilges, JMB 1998, 279, 911-920 for further details. They may be used to compare results from PCA in distance space with those from PCA in cartesian-coordinate space.
</div>
<div class="Description">
<span class="Description-entry">[nmwiz]</span> Generate output in <i>.nmd</i> format file for viewing with NMWiz<span class="bibcites">[<a class="bibliocite" name="cite-5" href="#biblio-5"><span class="bib-index">5</span></a>]</span>. See http://prody.csb.pitt.edu/tutorials/nmwiz_tutorial/ for further details.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[nmwizvecs &lt;#&gt;]</span> Number of vectors to write out for nmwiz output, starting with the lowest frequency mode (default 20).
</div>
<div class="Description">
<span class="Description-entry">[nmwizfile &lt;filename&gt;]</span> Name of nmwiz file to write to (default ’out.nmd’).
</div>
<div class="Description">
<span class="Description-entry">[nmwizmask &lt;mask&gt;]</span> Mask of atoms corresponding to eigenvectors - should be the same one used to generate the matrix.
</div>

</li>

</ul>

</li>

</ul>
<div class="Unindented">
Calculate eigenvectors and eigenvalues for the specified symmetric matrix. This is followed by Principal Component Analysis (in cartesian coordinate space in the case of a covariance matrix or in distance space in the case of a distance-covariance matrix), or Quasiharmonic Analysis (in the case of a mass-weighted covariance matrix). Diagonalization of distance, correlation, idea, and ired matrices are also possible. Eigenvalues are given in cm<span class="formula"><sup> − 1</sup></span> in the case of a mass-weighted covariance matrix and in the units of the matrix elements in all other cases. In the case of a mass-weighted covariance matrix, the eigenvectors are mass-weighted.
</div>
<div class="Indented">
For quasi-harmonic analysis the input must be a mass-weighted covariance matrix. Thermodynamic quantities are calculated based on statistical mechanical formulae that assume the input system is oscillating in a single energy well: see Statistical Thermodynamics by D. A. McQuarrie, particularly chapters 4, 5, and 6 for more details.<span class="bibcites">[<a class="bibliocite" name="cite-13" href="#biblio-13"><span class="bib-index">13</span></a>]</span> For an in-depth discussion of the accuracy of thermodynamic parameters obtained via quasi-harmonic analysis see Chang et al..<span class="bibcites">[<a class="bibliocite" name="cite-7" href="#biblio-7"><span class="bib-index">7</span></a>]</span>
</div>
<div class="Indented">
Note that the maximum number of non-zero eigenvalues obtainable depends on the number of frames used to generate the input matrix; the number of frames should be equal to or greater than the number of columns in the matrix in order to obtain all eigenmodes.
</div>
<div class="Indented">
Results may include average coordinates (in the case of covar, mwcovar, correl), average distances (in the case of distcovar), main diagonal elements (in the case of idea and ired), eigenvalues, and eigenvectors.
</div>
<div class="Indented">
For example, in the following a mass-weighted covariance matrix of all atoms is generated and stored internally with the name mwcvmat; the matrix itself is written to mwcvmat.dat. Subsequently, the first 20 eigenmodes of the matrix are calculated and written to evecs.dat, and quasiharmonic analysis is performed at 300.0 K, with the results written to thermo.dat.
</div>
<pre class="LyX-Code">
matrix mwcovar name mwcvmat out mwcvmat.dat
diagmatrix mwcvmat out evecs.dat vecs 20 \
           thermo outthermo thermo.dat temp 300.0
</pre>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--31"></a>Output Format
</h3>
<div class="Unindented">
The &ldquo;modes&rdquo; or &ldquo;evecs&rdquo; output file is a text file with the following format:
</div>
<pre class="LyX-Code">
 [Reduced] Eigenvector file: &lt;Type&gt; nmodes &lt;#&gt; width &lt;width&gt;
  &lt;# Avg Coords&gt; &lt;Eigenvector Size&gt;
&lt;Average Coordinates&gt;
</pre>
<div class="Unindented">
Where &lt;Type&gt; is a string identifying what kind of matrix the eigenvectors/eigenvalues were determined from, nmodes is how many eigenvectors are in the file, and &lt;Average Coordinates&gt; are in lines 7 columns wide, with each element having width specified by &lt;width&gt;. Then for each eigenvector:
</div>
<pre class="LyX-Code">
 ****
&lt;Eigenvector#&gt; &lt;Eigenvalue&gt;
&lt;Eigenvector Coordinates&gt;
...
</pre>
<div class="Unindented">
Where &lt;Eigenvector Coordinates&gt; are in lines 7 columns wide, with each element having width specified by &lt;width&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.12">12.12</a> divergence
</h2>
<pre class="LyX-Code">
divergence ds1 &lt;ds1&gt; ds2 &lt;ds2&gt;
</pre>
<div class="Unindented">
Calculate Kullback-Leibler divergence between specified data sets.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.13">12.13</a> fft
</h2>
<pre class="LyX-Code">
fft &lt;dset0&gt; [&lt;dset1&gt; ...] [out &lt;outfile&gt;] [name &lt;outsetname&gt;] [dt &lt;samp_int&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>&lt;dset0&gt; [&lt;dset1 ...]</b></span> Argument(s) specifying datasets to be used.
</div>
<div class="Description">
<span class="Description-entry"><b>[out &lt;outfile&gt;]</b></span> Write results to file named &lt;outfile&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>[name &lt;outsetname&gt;]</b></span> The resulting transform will be stored with name &lt;outsetname&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>[dt &lt;samp_int&gt;]</b></span> Set the sampling interval (default is 1.0).
</div>

</li>

</ul>
<div class="Unindented">
Perform fast Fourier transform (FFT) on specified data set(s). If more than 1 data set, they must all have the same size.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.14">12.14</a> hausdorff<a class="Label" name="subsec:cpptraj_hausdorff"> </a>
</h2>
<pre class="LyX-Code">
hausdorff &lt;set arg0&gt; [&lt;set arg1&gt; ...]
          [outtype {basic|trimatrix nrows &lt;#&gt;|fullmatrix nrows &lt;#&gt; [ncols &lt;#&gt;]}]
      	[name &lt;output set name&gt;] [out &lt;file&gt;] [outab &lt;file&gt;] [outba &lt;file&gt;] 
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;set arg0&gt; ...</span> Input matrix data set(s) to calculate Hausdorff distance(s) for.
</div>
<div class="Description">
<span class="Description-entry">[outtype]</span> Specify the output type.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">basic</span> Output the Hausdorff distance for each input matrix as scalar 1D data.
</div>
<div class="Description">
<span class="Description-entry">trimatrix nrows &lt;#&gt;</span> Output Hausdorff distances for each input matrix as a 2D upper-triangular matrix with the given number of rows. Must have (nrows * (nrows-1)) / 2 input sets.
</div>
<div class="Description">
<span class="Description-entry">fullmatrix nrows &lt;#&gt; ncols &lt;#&gt;</span> Output Hausdorff distances for each input matrix as a full matrix with the given number of columns and rows. If <b>ncols</b> is not given, use <b>nrows</b>. Must have nrows * ncols input sets.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">[name &lt;output set name&gt;]</span> Name of output data sets.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to write Hausdorff distances to.
</div>
<div class="Description">
<span class="Description-entry">[outab &lt;file&gt;]</span> File to write directed A-&gt;B Hausdorff distances to.
</div>
<div class="Description">
<span class="Description-entry">[outba &lt;file&gt;]</span> File to write directed B-&gt;A Hausdorff distances to.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the symmetric Hausdorff distance for one or more matrices. The results can be saved as an array or as a full or upper-triangular matrix with the specified dimensions. The Hausdorff distance H is determined from:
</div>
<pre class="LyX-Code">
H = max{dH(A,B), dH(B,A)]
</pre>
<div class="Unindented">
Where dH(A,B) is the directed Hausdorff distance between sets A and B, etc. Colloquially speaking, the directed Hausdorff distance between A and B is determined as follows:
</div>
<ol>
<li>
What is the closest approach (distance) of each point in A to any point in B?
</li>
<li>
Choose the largest distance from among those distances.
</li>

</ol>
<div class="Unindented">
If desired, the output can be formed into a matrix, which can be useful e.g. when doing multiple 2D rms calculations on different regions of a trajectory. For example, the following input divides a 100 frame trajectory into 10 frame chunks, calcultes the 2D RMS matrix for each chunk, then performs Hausdorff analysis on the resulting matrices and forms a full output matrix.
</div>
<pre class="LyX-Code">
parm ../DPDP.parm7
for beg=1;beg&lt;100;beg+=10 end=10;end+=10 i=1;i++
  loadcrd ../DPDP.nc \$beg \$end name Chunk\$i 
done 
# Do the 2drms in chunks 
for i=1;i&lt;11;i++ 
  for j=1;j&lt;11;j++ 
    2drms crdset Chunk\$i reftraj Chunk\$j M\$i.\$j 
  done
done 
hausdorff M* out hausdorff.fullmatrix.gnu title hausdorff.matrix.gnu \
  outtype fullmatrix nrows 10
runanalysis
</pre>
<div class="Unindented">
This type of calculation lends itself well to parallelization. The <b>parallelanalysis</b> command can be used to run all the <b>2drms</b> calculations in parallel with MPI-enabled cpptraj:
</div>
<pre class="LyX-Code">
parm ../DPDP.parm7
for beg=1;beg&lt;100;beg+=10 end=10;end+=10 i=1;i++
  loadcrd ../DPDP.nc \$beg \$end name Chunk\$i 
done 
# Do the 2drms in chunks 
for i=1;i&lt;11;i++ 
  for j=1;j&lt;11;j++ 
    2drms crdset Chunk\$i reftraj Chunk\$j M\$i.\$j 
  done
done
parallelanalysis sync
runanalysis hausdorff M* out hausdorff.fullmatrix.gnu title hausdorff.matrix.gnu \
  outtype fullmatrix nrows 10
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.15">12.15</a> hist | histogram
</h2>
<pre class="LyX-Code">
hist<a class="IndexReference" name="entry-hist-0" href="#index-hist">↓</a> &lt;dataset_name&gt;[,&lt;min&gt;,&lt;max&gt;,&lt;step&gt;,&lt;bins&gt;] ...
	[free &lt;temperature&gt;] [norm | normint] [gnu] [circular] out &lt;filename&gt;
	[amd &lt;amdboost_data&gt;] [name &lt;outputset name&gt;]
    [traj3d &lt;file&gt; [trajfmt &lt;format&gt;] [parmout &lt;file&gt;]]
	[min &lt;min&gt;] [max &lt;max&gt;] [step &lt;step&gt;] [bins &lt;bins&gt;] [nativeout]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>&lt;dataset_name&gt;[,&lt;min&gt;,&lt;max&gt;,&lt;step&gt;,&lt;bins&gt;]</b></span> Dataset(s) to be histogrammed. Optionally, the min, max, step, and/or number of bins can be specified for this dimension after the dataset name separated by commas. It is only necessary to specify the step or number of bins, an asterisk ’*’ indicates the value should be calculated from available data. 
</div>
<div class="Description">
<span class="Description-entry"><b>[free &lt;temperature&gt;]</b></span> If specified, estimate free energy from bin populations using <span class="formula"><i>G</i><sub><i>i</i></sub> =  − <i>k</i><sub><i>B</i></sub><i>T</i>ln<span class="array"><span class="arrayrow"><span class="bracket align-left">⎛</span></span><span class="arrayrow"><span class="bracket align-left">⎝</span></span></span><span class="fraction"><span class="ignored">(</span><span class="numerator"><i>N</i><sub><i>i</i></sub></span><span class="ignored">)/(</span><span class="denominator"><i>N</i><sub><i>Max</i></sub></span><span class="ignored">)</span></span><span class="array"><span class="arrayrow"><span class="bracket align-right">⎞</span></span><span class="arrayrow"><span class="bracket align-right">⎠</span></span></span></span>, where K<sub>B</sub> is Boltzmann’s constant, T is the temperature specified by &lt;temperature&gt;, N<sub>i</sub> is the population of bin i and N<sub>Max</sub> is the population of the most populated bin. Bins with no population are given an artificial barrier equivalent to a population of 0.5.
</div>
<div class="Description">
<span class="Description-entry"><b>[norm]</b></span> If specified, normalize bin populations so the sum over all bins equals 1.0.
</div>
<div class="Description">
<span class="Description-entry">[normint]</span> Normalize bin populations so the integral over them is 1.0.
</div>
<div class="Description">
<span class="Description-entry"><b>[gnu]</b></span> Internal output only; data will be gnuplot-readable, i.e. a space will be printed after the highest order coordinate cycles.
</div>
<div class="Description">
<span class="Description-entry"><b>[circular]</b></span> Internal output only; data will wrap, i.e. an extra bin will be printed before min and after max in each direction. Useful for e.g. dihedral angles.
</div>
<div class="Description">
<span class="Description-entry"><b>out &lt;filename&gt;</b></span> Write results to file named &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[amd &lt;amdboost_data&gt;]</span> Reweight bins using AMD boost energies in data set &lt;amdboost_data&gt; (in KT).
</div>
<div class="Description">
<span class="Description-entry">[name &lt;outputset name&gt;]</span> Output histogram data set name.
</div>
<div class="Description">
<span class="Description-entry">[traj3d &lt;file&gt; [trajfmt &lt;format&gt;]]</span> (3D histograms only) Write a pseudo-trajectory of the 3 data sets (1 atom) to &lt;file&gt; with format &lt;format&gt;.
</div>
<div class="Description">
<span class="Description-entry">[parmout &lt;file&gt;]</span> (3D histograms only) Write a topology corresponding to the pseudo-trajectory to &lt;file&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>[min &lt;min&gt;]</b></span> Default minimum to bin if not specified.
</div>
<div class="Description">
<span class="Description-entry"><b>[max &lt;max&gt;]</b></span> Default max to use if not specified.
</div>
<div class="Description">
<span class="Description-entry"><b>[step &lt;step&gt;]</b></span> Default step size to use if not specified.
</div>
<div class="Description">
<span class="Description-entry"><b>[bins &lt;bins&gt;]</b></span> Default bin size to use if not specified.
</div>
<div class="Description">
<span class="Description-entry">[nativeout]</span> Do not use cpptraj data file framework; only necessary for writing out histograms with &gt; 3 dimensions.
</div>

</li>

</ul>
<div class="Unindented">
Create an N-dimensional histrogram, where N is the number of datasets specified. For 1-dimensional histograms the xmgrace ’.agr’ file format is recommended; for 2-dimensional hisograms the gnuplot ’.gnu’ file format is recommended; for all other dimensions plot formatting is disabled and the routine uses its own internal output format; this is also enabled if <b>gnu</b> or <b>circular</b> is specified.
</div>
<div class="Indented">
For example, to create a two dimensional histogram of two datasets ’phi’ and ’psi’:
</div>
<pre class="LyX-Code">
dihedral phi :2@C :3@N :3@CA :3@C
dihedral psi :3@N :3@CA :3@C :4@N
hist phi,-180,180,*,72 psi,-180,180,*,72 out hist.gnu
</pre>
<div class="Unindented">
In this case the number of bins (72) has been specified for each dimension and ’*’ has been given for the step size, indicating it should be calculated based on min/max/bins. The following ’hist’ command is equivalent:
</div>
<pre class="LyX-Code">
hist phi psi min -180 max 180 bins 72 out hist.gnu
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.16">12.16</a> integrate
</h2>
<pre class="LyX-Code">
integrate &lt;dset0&gt; [&lt;dset1&gt; ...] [out &lt;outfile&gt;] [intout &lt;intfile&gt;]
          [name &lt;name&gt;]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dset0&gt; [&lt;dset1&gt; ...]</span> Data set(s) to integrate.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;outfile&gt;]</span> If specified, write cumulative sum curves to &lt;outfile&gt;.
</div>
<div class="Description">
<span class="Description-entry">[intout &lt;intfile&gt;]</span> If specified, write final integral values to &lt;intfile&gt;.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;name&gt;]</span> Output data set(s) name.
</div>
<div class="Unindented">
DataSets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Final integral values, 1 for each input data set (indexed from 0).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Sum]:&lt;idx&gt;</span> Cumulative sum curves if <b>out</b> was specified, 1 for each input data set (indexed from 0).
</div>

</li>

</ul>
<div class="Unindented">
Integrate specified data set(s) using trapezoid integration. If <b>’out’</b> is specified write cumulative sum curves to &lt;outfile&gt;. If <b>’intout’</b> is specified write final integral values for each set to &lt;intfile&gt;.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.17">12.17</a> ired
</h2>
<pre class="LyX-Code">
ired [relax freq &lt;MHz&gt; [NHdist &lt;distnh&gt;]] [order &lt;order&gt;]
 	tstep &lt;tstep&gt; tcorr &lt;tcorr&gt; out &lt;filename&gt; [norm] [drct]
 	modes &lt;modesname&gt; [name &lt;output sets name&gt;] [ds2matrix &lt;file&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>[relax freq &lt;MHz&gt; [NHdist &lt;distnh&gt;]]</b></span> Should only be used when ired vectors represent N-H bonds; calculate correlation times <span class="formula"><i>τ</i><sub><i>m</i></sub></span><span class="default"><span lang="en"> for each eigenmode and relaxation rates and NOEs for each N-H vector. ’freq &lt;MHz&gt;’ (required) is the Lamor frequency of the measurement. ’NHdist &lt;distnh&gt;’ specifies the length of the NH bond in Angstroms (default is 1.02).</span></span>
</div>
<div class="Description">
<span class="Description-entry">order &lt;order&gt;</span> Order of the Legendre polynomials to use when calculating spherical harmonics (default 2).
</div>
<div class="Description">
<span class="Description-entry"><b>tstep &lt;tstep&gt;</b></span> Time between snapshots in ps (default 1.0).
</div>
<div class="Description">
<span class="Description-entry"><b>tcorr &lt;tcorr&gt;</b></span> Maximum time to calculate correlation functions for in ps (default 10000.0).
</div>
<div class="Description">
<span class="Description-entry"><b>out &lt;filename&gt;</b></span> Name of file to write output to.
</div>
<div class="Description">
<span class="Description-entry"><b>[norm]</b></span> Normalize all correlation functions, i.e., <span class="formula"><i>C</i><sub><i>l</i></sub>(<i>t</i> = 0) = <i>P</i><sub><i>l</i></sub>(<i>t</i> = 0) = 1.0</span>.
</div>
<div class="Description">
<span class="Description-entry"><b>[drct]</b></span> Use the direct method to calculate correlations instead of FFT; this will be much slower.
</div>
<div class="Description">
<span class="Description-entry"><b>modes &lt;modesname&gt;</b></span> Name of previously calculated eigenmodes corresponding to IRED vectors.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[ds2matrix &lt;file&gt;]</span> If specified, write full delta*S^2 matrix (# IRED vector rows by # eigenmodes columns) to &lt;file&gt;.
</div>
<div class="Unindented">
DataSets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[S2]</span> S2 order parameters for each vector.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Plateau]</span> Plateau values for each vector.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[TauM]</span> TauM values for each vector.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[dS2]</span> Full delta*S^2 matrix.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[T1]</span> T1 relaxation values for each vector.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[T2]</span> T2 relaxation values for each vector.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[NOE]</span> NOEs for each vector.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Cm(t)]:X</span> Cm(t) function for vector X.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Cj(t)]:X</span> Cj(t) function for vector X.
</div>

</li>

</ul>
<div class="Unindented">
Peform IRED<span class="bibcites">[<a class="bibliocite" name="cite-29" href="#biblio-29"><span class="bib-index">29</span></a>]</span> analysis on previously defined IRED vectors (see vector ired) using eigenmodes calculated from those vectors with a previous ’diagmatrix’ command. The number of defined IRED vectors should match the number of eigenmodes calculated. Autocorrelation functions for each mode and the corresponding correlation time <span class="formula"><i>τ</i><sub><i>m</i></sub></span><span class="default"><span lang="en"> will be written to <i>filename.cmt</i>. Autocorrelation functions for each vector will be written to <i>filename.cjt</i>. Relaxation rates and NOEs for each N-H vector will be written to &lt;filename&gt; or added to the the end of the standard output. For the calculation of </span></span><span lang="en"><span class="formula"><i>τ</i><sub><i>m</i></sub></span><span class="default"><span lang="en"> the normalized correlation functions and only the first third of the analyzed time steps will be used. For further information on the convergence of correlation functions see [Schneider, Brünger, Nilges, <i>J. Mol. Biol.</i> <b>285</b>, 727 (1999)].</span></span></span>
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--32"></a>Example of IRED in Cpptraj
</h3>
<div class="Unindented">
In <i>cpptraj</i>, IRED analysis<span class="bibcites">[<a class="bibliocite" name="cite-29" href="#biblio-29"><span class="bib-index">29</span></a>]</span> can now be performed in one pass (as opposed to the two passes previously required in <i>ptraj</i>). First, IRED vectors are defined (in this case for N-H bonds) and an IRED matrix is calculated and analyzed. The IRED vectors are then projected onto the calculated IRED eigenvectors in the <b><i>ired</i></b> analysis command to calculate the time correlation functions. If the parameter <i>order</i> is specified, order parameters based on IRED are calculated. By specifying the <i>relax</i> parameter, relaxation rates and NOEs can be obtained for each N-H vector. Note that the order of the IRED matrix should be the same as the one specified for IRED analysis.
</div>
<pre class="LyX-Code">
# Define N-H IRED vectors
vector v0 @5 ired @6
vector v1 @7 ired @8
...
vector v5 @15 ired @16
vector v6 @17 ired @18‘
# Define IRED matrix using all previous IRED vectors
matrix ired name matired order 2
# Diagonalize IRED matrix
diagmatrix matired vecs 6 out ired.vec name ired.vec
# Perform IRED analysis
ired relax NHdist 1.02 freq 500.0 tstep 1.0 tcorr 100.0 out v0.out \
     noefile noe order 2
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.18">12.18</a> kde
</h2>
<pre class="LyX-Code">
kde &lt;dataset&gt; [bandwidth &lt;bw&gt;] [out &lt;file&gt;] [name &lt;dsname&gt;]
    [min &lt;min&gt;] [max &lt;max] [step &lt;step&gt;] [bins &lt;bins&gt;] [free]
    [kldiv &lt;dsname2&gt; [klout &lt;outfile&gt;]] [amd &lt;amdboost_data&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[bandwidth &lt;bw&gt;]</span> Bandwidth to use for KDE; if not specified bandwidth will be estimated using the normal distribution approximation.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> Output file name.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;dsname&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[min &lt;min&gt;]</span> Minimum bin.
</div>
<div class="Description">
<span class="Description-entry">[max &lt;max&gt;]</span> Maximum bin.
</div>
<div class="Description">
<span class="Description-entry">[step &lt;step&gt;]</span> Bin step.
</div>
<div class="Description">
<span class="Description-entry">[bins &lt;bins&gt;]</span> Number of bins.
</div>
<div class="Description">
<span class="Description-entry">[free]</span> Calculate free energy from bin population.
</div>
<div class="Description">
<span class="Description-entry">[kldiv &lt;dsname2&gt; [klout &lt;outfile&gt;]]</span> Calculate Kullback-Leibler divergence over time of &lt;dataset&gt; distribution to &lt;dsname2&gt; distribution. Output to &lt;outfile&gt; if <b>klout</b> specified.
</div>
<div class="Description">
<span class="Description-entry">[amd &lt;amdboost_data&gt;]</span> Reweight histogram using AMD boost data from data set &lt;amdboost_data&gt; (in KT).
</div>

</li>

</ul>
<div class="Unindented">
Histogram 1D data set using a Gaussian kernel density estimator.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.19">12.19</a> lifetime<a class="Label" name="subsec:cpptraj-lifetime"> </a>
</h2>
<pre class="LyX-Code">
lifetime [out &lt;filename&gt;] &lt;dsetarg0&gt; [ &lt;dsetarg1&gt; ... ]
     	[window &lt;windowsize&gt; [name &lt;setname&gt;]] [averageonly]
     	[cumulative] [delta] [cut &lt;cutoff&gt;] [greater | less] [rawcurve]
         [fuzz &lt;fuzzcut&gt;] [nosort]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>[out &lt;filename&gt;]</b></span> Write results to file named &lt;filename&gt;, and lifetime curves to ’crv.&lt;filename&gt;’. If performing windowed lifetime analysis, &lt;filename&gt; contains the fraction present over time windows, and 2 additional files are written: ’max.&lt;filename&gt;’, containing max lifetime over windows, and ’avg.&lt;filename&gt;’, containing average lifetime over windows.
</div>
<div class="Description">
<span class="Description-entry"><b>&lt;dsetarg0&gt; [&lt;dsetarg1&gt; ...]</b></span> Argument(s) specifying datasets to be used.
</div>
<div class="Description">
<span class="Description-entry"><b>[window &lt;windowsize&gt;]</b></span> Size of window (in frames) over which to calculate lifetimes/averages. If not specified lifetime/average will be calculated over all frames.
</div>
<div class="Description">
<span class="Description-entry"><b>[name &lt;setname&gt;]</b></span> Store results in data sets with name &lt;setname&gt;.
</div>
<div class="Description">
<span class="Description-entry"><b>[averageonly]</b></span> <b></b>Just calculate averages (no lifetime analysis).
</div>
<div class="Description">
<span class="Description-entry"><b>[cumulative]</b></span> Calculate cumulative lifetimes/averages over windows.
</div>
<div class="Description">
<span class="Description-entry">[delta]</span> Calculate difference from previous window average.
</div>
<div class="Description">
<span class="Description-entry"><b>[cut &lt;cutoff&gt;]</b></span> Cutoff to use when determining if data is ’present’ (default 0.5).
</div>
<div class="Description">
<span class="Description-entry">[greater]</span> Data is considered present when above the cutoff (default).
</div>
<div class="Description">
<span class="Description-entry">[less]</span> Data is considered present when below the cutoff.
</div>
<div class="Description">
<span class="Description-entry">[rawcurve]</span> Do not normalize lifetime curves to 1.0.
</div>
<div class="Description">
<span class="Description-entry">[fuzz &lt;fuzzcut&gt;]</span> Ignore changes in lifetime state that are less than &lt;fuzzcut&gt; frames.
</div>
<div class="Description">
<span class="Description-entry">[nosort]</span> Do not sort data sets by name.
</div>
<div class="Unindented">
Data Sets Created:
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;</span> Number of lifetimes for each set, or if <b>window</b> specified fraction present over time windows.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[max]</span> Maximum lifetime for each set, or if <b>window</b> specified maximum lifetime over time windows.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[avg]</span> Average lifetime for each set, or if <b>window</b> specified average lifetime over time windows.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[curve]</span> Lifetime curves.
</div>
<div class="Unindented">
The following are created only if <b>window</b> not specified:
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[frames]</span> Total number of frames lifetime present for each set.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[name]</span> Name of each set.
</div>

</li>

</ul>
<div class="Unindented">
Perform lifetime analysis for specified data sets. Lifetime data can either be determined for the entire set, or for time windows of specified size within the set if <b>window</b> specified.
</div>
<div class="Indented">
A &ldquo;lifetime&rdquo; is defined as the length of time something remains ’present’; data is considered present when above or below a certain cutoff (the default is greater than 0.5, useful for analysis of <b><i>hbond</i></b> time series data). For example, in the case of a hydrogen bond <b>’series’</b> data set, if a hydrogen bond is present during a frame the value is 1, otherwise it is 0. Given the hbond time series data set {1 1 1 0 1 0 0 0 1 1}, the overall fraction present is 0.6. However, there are 3 lifetimes of lengths 3, 1, and 2 ({1 1 1}, {1}, and {1 1}). The maximum lifetime is 3 and the average lifetime is 2.0, i.e. (3 + 1 + 2) / 3 lifetimes = 2.0. One can also construct a &ldquo;lifetime curve&rdquo;, which is constructed as the sum of all individual lifetimes. By default these curves are normalized to 1.0, but the raw curve can be obtained using the <b>rawcurve</b> keyword. For the example data set here the raw lifetime curve would be 3 frames long:
</div>
<pre class="LyX-Code">
       1 1 1
       1
       1 1
Curve: 3 2 1
</pre>
<div class="Unindented">
By default data sets are sorted by name unless <b>nosort</b> is specified. The lifetime command can calculate lifetimes over specific time windows by using the <b>window</b> keyword. This can be particularly useful if one wants to get a sense for how lifetimes are changing over the course of very long time series data. In addition, averages can be calculated instead of lifetimes by specifying <b>averageonly</b>. Cumulative averages over windows can be obtained using the <b>cumulative</b> keyword, or the change from the average value in the previous window can be obtained using the <b>delta</b> keyword.
</div>
<div class="Indented">
The <b>fuzz</b> keyword can be used to try and smooth the input data by ignoring changes in state that occur for fewer frames than &lt;fuzzcut&gt;. For example, in the above example hbond time series data set there is a one frame change in state between the first and second lifetimes which could be interpreted as a transient breaking of the hydrogen bond. Using a &lt;fuzzcut&gt; value of 1, this one frame change in state would be ignored, and the data set would effectively appear to lifetime as {1 1 1 1 1 0 0 0 1 1}. The state change between the second and third lifetimes is longer than &lt;fuzzcut&gt; (3 frames) and so it would remain.
</div>
<div class="Indented">
If <b>window</b> is not specified, two files are output: &lt;filename&gt; and crv.&lt;filename&gt;. The file &lt;filename&gt; contains overall lifetime stats for each set with format:
</div>
<pre class="LyX-Code">
#Set &lt;setname&gt; &lt;setname&gt;[max] &lt;setname&gt;[avg] &lt;setname&gt;[frames] &lt;setname&gt;[name]
</pre>
<div class="Unindented">
where <tt>&lt;setname&gt;</tt> denotes the total number of lifetimes, <tt>&lt;setname&gt;[max]</tt> denotes the maximum lifetime, <tt>&lt;setname&gt;[avg] </tt>denotes the average lifetime, <tt>&lt;setname&gt;[frames]</tt> denotes the total number of frames present in all lifetimes, and <tt>&lt;setname&gt;[name] </tt>is the data set name. The file crv.&lt;filename&gt; contains the lifetime curves for each set.
</div>
<div class="Indented">
If <b>window</b> is specified, four files are output: &lt;filename&gt;, max.&lt;filename&gt;, avg.&lt;filename&gt;, and crv.&lt;filename&gt;. &lt;filename&gt; contains the fraction &ldquo;present&rdquo; over each time window for each set, max.&lt;filename&gt; contains the maximum lifetime in each time window for each set, avg.&lt;filename&gt; contains the average lifetime over each window for each set, and crv.&lt;filename&gt; contains the overall lifetime curves for each set. For window output, Gnuplot format is recommended.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--33"></a>Example: hbond lifetime analysis
</h3>
<pre class="LyX-Code">
parm DPDP.parm7
trajin DPDP.nc
hbond HB out hbond.dat @N,H,C,O series uuseries solutehb.agr \
  avgout hbavg.dat printatomnum
# ’run’ is used here to process the trajectory and generate hbond data
run
# Perform lifetime analysis
runanalysis lifetime HB[solutehb] out lifehb.dat
</pre>
<div class="Unindented">
Calculate ion lifetimes from hbond over windows of size 100 frames:
</div>
<pre class="LyX-Code">
hbond ION out ion.dat solventdonor :WAT solventacceptor :WAT@O series
run
lifetime HB[solventhb] out ion.lifetime.100.gnu window 100
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.20">12.20</a> lowestcurve
</h2>
<pre class="LyX-Code">
lowestcurve points &lt;# lowest&gt; [step &lt;stepsize&gt;] &lt;dset0&gt; [&lt;dset1&gt; ...]
                   [out &lt;file&gt;] [name &lt;setname&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;# lowest&gt;</span> Number of lowest points in each bin to average over.
</div>
<div class="Description">
<span class="Description-entry">[step &lt;stepsize&gt;]</span> Bin step size
</div>
<div class="Description">
<span class="Description-entry">&lt;dset0&gt; [&lt;dset1&gt; ...]</span> Data set(s) to use.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> File to write lowest curve to.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;setname&gt;]</span> Output lowest curve set name.
</div>

</li>

</ul>
<div class="Unindented">
Calculate a curve of the average of the # lowest points in bins of stepsize. Essentially each input data set is binned over bins of stepsize, then the lowest &lt;#&gt; points are averaged over for each bin.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.21">12.21</a> meltcurve
</h2>
<pre class="LyX-Code">
meltcurve &lt;dset0&gt; [&lt;dset1&gt; ...] [out &lt;outfile&gt;] [name &lt;outsetname&gt;] cut &lt;cut&gt;
</pre>
<div class="Unindented">
Calculate melting curve from input data sets (i.e. fraction ’folded’ for each data set) assuming a simple 2-state transition model, using data below &lt;cut&gt;as ’folded’ and data above &lt;cut&gt; as ’unfolded’.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.22">12.22</a> modes
</h2>
<pre class="LyX-Code">
modes {fluct|displ|corr|eigenval|trajout|rmsip} name &lt;modesname&gt; [name2 &lt;modesname&gt;]
  	[beg &lt;beg&gt;] [end &lt;end&gt;] [bose] [factor &lt;factor&gt;] [calcall]
  	[out &lt;outfile&gt;] [setname &lt;name&gt;]
    Options for ’trajout’: (Generate pseudo-trajectory)
  	[trajout &lt;name&gt; parm &lt;name&gt; | parmindex &lt;#&gt;
  	  [trajoutfmt &lt;format&gt;] [trajoutmask &lt;mask&gt;]
	    [pcmin &lt;pcmin&gt;] [pcmax &lt;pcmax&gt;] [tmode &lt;mode&gt;]]
    Options for ’corr’: (Calculate dipole correlation)
  	{ maskp &lt;mask1&gt; &lt;mask2&gt; [...] | mask1 &lt;mask&gt; mask2 &lt;mask&gt; }
  	parm &lt;name&gt; | parmindex &lt;#&gt;\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Unindented">
Types of Calculations:
</div>
<div class="Description">
<span class="Description-entry"><b>fluct</b></span> RMS fluctuations (X, Y, Z, and total) for each atom across specified normal modes.
</div>
<div class="Description">
<span class="Description-entry"><b>displ</b></span> Displacement of cartesian coordinates in the X, Y and Z directions for each atom across specified normal modes.
</div>
<div class="Description">
<span class="Description-entry"><b>corr</b></span> Dipole-dipole correlation functions. Must also specify <b>maskp</b> (see below).
</div>
<div class="Description">
<span class="Description-entry">eigenval</span> Calculate eigenvalue fractions.
</div>
<div class="Description">
<span class="Description-entry">trajout</span> Create a pseudo-trajectory along the given mode from the average structure.
</div>
<div class="Description">
<span class="Description-entry">rmsip</span> Calculate the root-mean-square inner product between modes specified by <b>name</b> and <b>name2</b>.
</div>
<div class="Unindented">
Options:
</div>
<div class="Description">
<span class="Description-entry"><b>name &lt;modesname&gt;</b></span> Previously read-in or generated Modes data set name.
</div>
<div class="Description">
<span class="Description-entry"><b>[beg &lt;beg&gt;] [end &lt;end&gt;]</b></span> If modes taken from datafile, beginning and end modes to read. Default for <i>beg</i> is 7 (which skips the first 6 zero-frequency modes in the case of a normal mode analysis); for <i>end</i> it is 50.
</div>
<div class="Description">
<span class="Description-entry"><b>[bose]</b></span> Use quantum (Bose) statistics in populating the modes.
</div>
<div class="Description">
<span class="Description-entry"><b>[factor &lt;factor&gt;]</b></span> multiplicative constant on the amplitude of displacement/pseudo-trajectory, default 1.0.
</div>
<div class="Description">
<span class="Description-entry">[calcall]</span> If specified use all eigenvectors; otherwise eigenvectors associated with zero or negative eigenvalues will be skipped.
</div>
<div class="Description">
<span class="Description-entry"><b>[out &lt;outfile&gt;]</b></span> File to write data results to. If not given results are written to STDOUT.
</div>
<div class="Description">
<span class="Description-entry">[setname &lt;name&gt;]</span> Output data set name.
</div>
<div class="Unindented">
Options for ’trajout’:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Output trajectory file name.
</div>
<div class="Description">
<span class="Description-entry">[parm &lt;parmfile/tag&gt;|parmindex &lt;#&gt;]</span> Topology file to use (default first Topology loaded).
</div>
<div class="Description">
<span class="Description-entry">[trajoutfmt &lt;format&gt;]</span> Output trajectory format.
</div>
<div class="Description">
<span class="Description-entry">[trajoutmask &lt;mask&gt;]</span> Mask of atoms that correspond to how modes were originally generated.
</div>
<div class="Description">
<span class="Description-entry">[pcmin &lt;pcmin&gt;]</span> Lowest principal component projection value to use for output trajectory.
</div>
<div class="Description">
<span class="Description-entry">[pcmax &lt;pcmax&gt;]</span> Highest principal component projection value to use for output trajectory.
</div>
<div class="Description">
<span class="Description-entry">[tmode &lt;mode&gt;]</span> Mode to generate pseudo-trajectory for.
</div>
<div class="Unindented">
Options for ’corr’:
</div>
<div class="Description">
<span class="Description-entry"><b>[maskp &lt;mask1&gt; &lt;mask2&gt; [...]]</b></span> If <b>corr</b>, pairs of atom masks (<i>mask1</i>, <i>mask2</i>; each pair preceded by &ldquo;<tt>maskp</tt>&rdquo; and each mask defining only a single atom) have to be given that specify the atoms for which the correlation functions are desired.
</div>
<div class="Description">
<span class="Description-entry">mask1 &lt;mask&gt; mask2 &lt;mask&gt;</span> Instead of <b>maskp</b>, specify two masks; atoms from the first mask will be paired up with atoms from the second mask.
</div>
<div class="Unindented">
DataSets Created (fluct)
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[rmsX]</span> RMS fluctuations in the X direction.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[rmsY]</span> RMS fluctuations in the Y direction.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[rmsZ]</span> RMS fluctuations in the Z direction.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[rms]</span> Total RMS fluctuations.
</div>
<div class="Unindented">
DataSets Created (displ)
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[displX]</span> Displacement in X direction.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[displY]</span> Displacement in Y direction.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[displZ]</span> Displacement in Z direction.
</div>
<div class="Unindented">
DataSets Created (eigenval)
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Frac]</span> Fraction eigenvalue contributes to overall motion.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Cumulative]</span> Cumulative fraction.
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[Eigenval]</span> Value of eigenvlue.
</div>
<div class="Unindented">
DataSets Created (rmsip)
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Result of RMSIP calculation.
</div>

</li>

</ul>
<div class="Unindented">
Analyze previously calculated eigenmodes obtained from principal component analyses (of covariance matrices) or quasiharmonic analyses (diagmatrix analysis command). Modes are taken from a previously generated data set (i.e. from <b><i>diagmatrix</i></b>) or read in from a data file with <b><i>readdata</i></b>. By default, classical (Boltzmann) statistics are used in populating the modes. A possible series of commands would be &ldquo;<span class="sans"><b>matrix</b></span><tt> covar</tt> <span class="formula">∣</span> <tt>mwcovar</tt> ...&rdquo; to generate the matrix, &ldquo;<tt><b>diagmatrix</b></tt> ...&rdquo; to calculate the modes, and, finally, &ldquo;<tt><b>modes</b></tt> ...&rdquo;. 
</div>
<div class="Indented">
For example, to calculate the RMS fluctuations or displacements of the first 3 eigenmodes caluclated from a mass-weighted covariance matrix:
</div>
<pre class="LyX-Code">
matrix mwcovar name mwcvmat out mwcvmat.dat
diagmatrix mwcvmat name evecs vecs 5
modes fluct out rmsfluct.dat name evecs beg 1 end 3
modes displ out resdispl.dat name evecs beg 1 end 3
</pre>
<div class="Unindented">
Additionally, dipole-dipole correlation functions for modes obtained from principle component analysis or quasiharmonic analysis can be computed.
</div>
<pre class="LyX-Code">
modes corr out cffromvec.dat name evecs beg 1 end 3 \
      maskp @1 @2 maskp @3 @4 maskp @5 @6
</pre>
<div class="Unindented">
or
</div>
<pre class="LyX-Code">
mode corr out cffromvec.dat name evecs beg 1 end 3 mask1 @1,3,5 mask2 @2,4,6
</pre>
<div class="Unindented">
If <b>eigenval</b> is specified, the fraction contribution of each eigenvector to the total motion is calculated and output with format:
</div>
<pre class="LyX-Code">
#Mode Frac. Cumulative Eigenval 
</pre>
<div class="Unindented">
where <tt>#Mode</tt> is the eigenvector number, <tt>Frac.</tt> is the eigenvalue over the sum of all eigenvalues, <tt>Cumulative</tt> is the cumulative sum of <tt>Frac.</tt>, and <tt>Eigenval</tt> is the eigenvalue itself. Note that in order to get an idea for how much each eigenvector contributes to all motion, this is best used when all possible eigenvectors have been determined for a system.
</div>
<div class="Indented">
In order to visualize eigenvectors, pseudo-trajectories along eigenvectors can be created using average coordinates with the <b>trajout</b> keyword. For example, to write a pseudo-trajectory of the first principal component from principal component value of -100 to 100 for a previously calculated Modes data set corresponding to heavy atoms (no hydrogens) for residues 1 to 36:
</div>
<pre class="LyX-Code">
parm ../GAAC.nowat.parm7
readdata evecs.dat
runanalysis modes name evecs.dat trajout test.nc trajoutfmt netcdf \
            trajoutmask :1-36&amp;!@H= pcmin -100 pcmax 100 tmode 1
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.23">12.23</a> multicurve
</h2>
<pre class="LyX-Code">
multicurve set &lt;dset&gt; [set &lt;dset&gt; ...] 
           &lt;dset&gt; { &lt;equation&gt; |
                    name &lt;dsname&gt; nexp &lt;m&gt; [form {mexp|mexpk|mexpk_penalty} }
           [AX=&lt;value&gt; ...] [out &lt;outfile&gt;] [resultsout &lt;results&gt;]
           [maxit &lt;max iterations&gt;] [tol &lt;tolerance&gt;]
           [outxbins &lt;NX&gt; outxmin &lt;xmin&gt; outxmax &lt;xmax&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">set &lt;dset&gt; [set &lt;dset&gt; ...]</span> Data set(s) to fit.
</div>
<div class="Description">
<span class="Description-entry">&lt;equation&gt;</span> Equation to fit of form &lt;Variable&gt; = &lt;Equation&gt;. See <a class="Reference" href="#subsec:cpptraj-Data-Set-Math">5.2 on page 1↑</a> for more details on equations <i>cpptraj</i> understands.
</div>
<div class="Description">
<span class="Description-entry">name &lt;dsname&gt;</span> Name of output data sets (required if using <b>nexp</b>).
</div>
<div class="Description">
<span class="Description-entry">nexp &lt;m&gt;</span> Fit to specified number of exponentials.
</div>
<div class="Description">
<span class="Description-entry">form &lt;type&gt;</span> Fit to specified exponential form:\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">mexp</span> Multi-exponential, SUM(m)[ An * exp(An+1 * X)]
</div>
<div class="Description">
<span class="Description-entry">mexpk</span> Multi-exponential plus constant, A0 + SUM(m)[An * exp(An+1 * X)]
</div>
<div class="Description">
<span class="Description-entry">mexpk_penalty</span> Same as mexpk except sum of prefactors constrained to 1.0 and exponential constants constrained to &lt; 0.0.
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">AX=&lt;value&gt;</span> Value of any constants in specified equation with X starting from 0 (can specify more than one).
</div>
<div class="Description">
<span class="Description-entry">out &lt;outfile&gt;</span> Write resulting fit curve to &lt;outfile&gt;.
</div>
<div class="Description">
<span class="Description-entry">resultsout &lt;results&gt;</span> Write details of the fit to &lt;results&gt; (default STDOUT).
</div>
<div class="Description">
<span class="Description-entry">maxit &lt;max iterations&gt;</span> Number of iterations to run curve fitting algotrithm (default 50).
</div>
<div class="Description">
<span class="Description-entry">tol &lt;tolerance&gt;</span> Curve-fitting tolerance (default 1E-4).
</div>
<div class="Description">
<span class="Description-entry">outxbins &lt;NX&gt;</span> Number of points to use when generating final curve (default same number of points as input data set).
</div>
<div class="Description">
<span class="Description-entry">outxmin &lt;xmin&gt;</span> Minimum X value to use for final curve (default same number of points as input data set).
</div>
<div class="Description">
<span class="Description-entry">outxmax &lt;xmax&gt;</span> Maximum X value to use for final curve (default same number of points as input data set).
</div>

</li>

</ul>
<div class="Unindented">
Fit each input data set &lt;dset&gt; to &lt;equation&gt;. See the <b><i>curvefit</i></b> command <a class="Reference" href="#subsec:cpptraj_curvefit"> on page 1↑</a> for more details.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.24">12.24</a> multihist
</h2>
<pre class="LyX-Code">
multihist [out &lt;filename&gt;] [name &lt;dsname&gt;] [norm | normint] [kde]
          [min &lt;min&gt;] [max &lt;max&gt;] [step &lt;step&gt;] [bins &lt;bins&gt;] [free &lt;T&gt;]
          &lt;dsetarg0&gt; [ &lt;dsetarg1&gt; ... ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">out &lt;filename&gt;</span> Output file.
</div>
<div class="Description">
<span class="Description-entry">name &lt;dsname&gt;</span> Name for resulting histogram data sets.
</div>
<div class="Description">
<span class="Description-entry">norm</span> (Only used if not kde) Normalize so that max bin is 1.0.
</div>
<div class="Description">
<span class="Description-entry">normint</span> (Default for kde) Normalize integral over histogram to 1.0.
</div>
<div class="Description">
<span class="Description-entry">kde</span> Use kernel density estimator to construct histogram.
</div>
<div class="Description">
<span class="Description-entry">min &lt;min&gt;</span> Histogram minimum (default data set minimum).
</div>
<div class="Description">
<span class="Description-entry">max &lt;max&gt;</span> Histogram maximum (default data set maximum).
</div>
<div class="Description">
<span class="Description-entry">step &lt;step&gt;</span> Histogram step.
</div>
<div class="Description">
<span class="Description-entry">bins &lt;bins&gt;</span> Number of histogram bins.
</div>
<div class="Description">
<span class="Description-entry">free &lt;T&gt;</span> Calculate free energy from bin populations as G = -R * &lt;T&gt; * ln( Ni / Nmax ).
</div>
<div class="Description">
<span class="Description-entry">&lt;dsetargX&gt;</span> Data set argument - may specify more than one.
</div>

</li>

</ul>
<div class="Unindented">
Histogram each data set separately in 1D. Must specify at least <b>bins</b> or <b>step</b>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.25">12.25</a> phipsi
</h2>
<pre class="LyX-Code">
phipsi &lt;dsarg0&gt; [&lt;dsarg1&gt; ...] resrange &lt;range&gt; [out &lt;file&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dsargX&gt;</span> Argument selecting data sets. Can specify more than 1.
</div>
<div class="Description">
<span class="Description-entry">resrange &lt;range&gt;</span> Residue range to use (actually uses data set index).
</div>
<div class="Description">
<span class="Description-entry">[out &lt;file&gt;]</span> Output file.
</div>

</li>

</ul>
<div class="Unindented">
Calculate the average and standard deviation of [phi] and [psi] data set pairs, write to &lt;file&gt; with format:
</div>
<pre class="LyX-Code">
#Phi Psi SD(Phi) SD(Psi) Legend
</pre>
<div class="Unindented">
Where Phi is the average value of phi, Psi is the average value of psi, SD(Phi) is the standard deviation of phi, SD(psi) is the standard deviation of psi, and Legend contains text describing the phi and psi data sets used in the calculation. Periodicity is taken into account during averaging. The data sets must have been internally labeled as type ’phi’/’psi’ and must have a data set index set (actions like dihedral and multidihedral do this automatically). For example:
</div>
<pre class="LyX-Code">
parm ../DPDP.parm7
trajin ../DPDP.nc
multidihedral DPDP phi psi
run
phipsi DPDP[phi] DPDP[psi] out phipsi.dat resrange 1-22
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.26">12.26</a> regress
</h2>
<pre class="LyX-Code">
regress &lt;dset0&gt; [&lt;dset1&gt; ...] [name &lt;name&gt;] [nx &lt;nxvals&gt;]
    	[out &lt;filename&gt;] [statsout &lt;filename&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">dsetX</span> Data set(s) to perform linear regression for.
</div>
<div class="Description">
<span class="Description-entry">name &lt;name&gt;</span> Data set name for resulting linear fits.
</div>
<div class="Description">
<span class="Description-entry">nx &lt;nxvals&gt;</span> Number of X values to use in output data set(s) (ranging from input set min to max X). If not specified, input X values used.
</div>
<div class="Description">
<span class="Description-entry">out &lt;filename&gt;</span> File to write fit lines to.
</div>
<div class="Description">
<span class="Description-entry">statsout &lt;filename&gt;</span> File to write fit statistics to.
</div>
<div class="Unindented">
DataSets Generated:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;:&lt;idx&gt;</span> Output fit line(s) (indexed by input set order if more than one input set).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[slope]:&lt;idx&gt;</span> Output fit line slope(s).
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;[intercept]:&lt;idx&gt;</span> Output fit line intercept(s).
</div>

</li>

</ul>
<div class="Unindented">
Perform linear regression on the specified data set(s). The fit line is calculated using either the input X values or <b>&lt;nxvals&gt;</b> values ranging from the input set minimum to maximum X. Statistics for the fit(s) are saved to the file specified by <b>statsout</b> or reported to STDOUT. 
</div>
<div class="Indented">
For example, to fit data read in from a file and then create a set using the fit parameters:
</div>
<pre class="LyX-Code">
readdata esurf_vs_rmsd.dat.txt index 1 name XY
runanalysis regress XY name FitXY statsout statsout.dat
createset "Y = FitXY[slope] * X + FitXY[intercept]" xstep .2 nx 100
writedata Y.dat Y
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.27">12.27</a> remlog<a class="Label" name="subsec:cpptraj-remlog"> </a>
</h2>
<pre class="LyX-Code">
remlog {&lt;remlog dataset&gt; | &lt;remlog filename&gt;} [out &lt;filename&gt;] [crdidx | repidx]
   	[stats [statsout &lt;file&gt;] [printtrips] [reptime &lt;file&gt;]] [lifetime &lt;file&gt;]
   	[reptimeslope &lt;n&gt; reptimeslopeout &lt;file&gt;] [acceptout &lt;file&gt;] [name &lt;setname&gt;]\begin_inset Separator latexpar\end_inset
       [edata [edataout &lt;file&gt;]]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;remlog dataset&gt;</span> Previously read-in REM log data.
</div>
<div class="Description">
<span class="Description-entry">&lt;remlog filename&gt;</span> REM log file name to read in.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write replica/coordinate index versus time to &lt;filename&gt;.\begin_inset Separator latexpar\end_inset
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">crdidx</span> Print coordinate index vs exchange; output sets contain replica indices. 
</div>
<div class="Description">
<span class="Description-entry">repidx</span> Print replica index vs exchange; output sets contain coordinate indices. 
</div>

</li>

</ul>
<div class="Description">
<span class="Description-entry">stats [statsout &lt;file&gt;]</span> Calculate round-trip statistics and optionally write to &lt;file&gt;.
</div>
<div class="Description">
<span class="Description-entry">printtrips</span> Print details of each individual round trip.
</div>
<div class="Description">
<span class="Description-entry">[reptime &lt;file&gt;]</span> Write time spent at each replica to &lt;file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[lifetime &lt;file&gt;]</span> Print lifetime data at each replica to &lt;file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[reptimeslope &lt;n&gt;]</span> Calculate the slope of time spent at each replica every &lt;n&gt; exchanges.
</div>
<div class="Description">
<span class="Description-entry">[reptimeslopeout &lt;file&gt;]</span> File to write <b>reptimeslope</b> output to.
</div>
<div class="Description">
<span class="Description-entry">[acceptout &lt;file&gt;]</span> Write overall exchange acceptances to &lt;file&gt;.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;setname&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[edata [edataout &lt;file&gt;]]</span> Extract energy data from replica log, optionally write to file.
</div>
<div class="Unindented">
DataSets created:
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;:&lt;idx&gt;</span> Replica/coordinate index vs exchange.
</div>
<div class="Description">
<span class="Description-entry">&lt;setname&gt;[E]:&lt;idx&gt;</span> If ’edata’ specified, energy data from replica log.
</div>

</li>

</ul>
<div class="Unindented">
Analyze previously read in (via <b><i>readdata</i></b>) M-REMD/T-REMD/H-REMD replica log data. Statistics calculated include round-trip time, which is the time needed for a coordinate set to travel from the lowest replica to the highest and back, and the number of exchanges each coordinate spent at each replica. For example, to read in REM log data from an Amber M-REMD run and analyze it:
</div>
<pre class="LyX-Code">
readdata rem.log.1.save rem.log.2.save dimfile remd.dim as remlog nosearch 
remlog rem.log.1.save stats reptime mremdreptime.dat 
</pre>
<div class="Unindented">
For an example of <b><i>remlog</i></b> analysis applied to actual REMD data, see Roe et al.<span class="bibcites">[<a class="bibliocite" name="cite-2" href="#biblio-2"><span class="bib-index">2</span></a>]</span>.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.28">12.28</a> rms2d<a class="Label" name="subsec:cpptraj-rms2d"> </a> | 2drms
</h2>
<pre class="LyX-Code">
rms2d<a class="IndexReference" name="entry-rms2d-0" href="#index-rms2d">↓</a> [crdset &lt;crd set&gt;] [&lt;name&gt;] [&lt;mask&gt;] [out &lt;filename&gt;]
      [dme | nofit | srmsd] [mass]
      [reftraj &lt;traj&gt; [parm &lt;parmname&gt; | parmindex &lt;parm#&gt;] [&lt;refmask&gt;]]
      [corr &lt;corrfilename&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[crdset &lt;crd set&gt;]</span> Name of previously generated COORDS DataSet. If not specified the default COORDS set will be used.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Mask of atoms to calculate 2D-RMSD for. Default is all atoms.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write results to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[dme]</span> Calculate distance RMSD instead of coordinate RMSD; this is substantially slower.
</div>
<div class="Description">
<span class="Description-entry">[nofit]</span> Calculate RMSD without fitting.
</div>
<div class="Description">
<span class="Description-entry">[srmsd]</span> Calculate symmetry-corrected RMSD (see <a class="Reference" href="#subsec:cpptraj_symmrmsd">11.78 on page 1↑</a>).
</div>
<div class="Description">
<span class="Description-entry">[mass]</span> Mass-weight RMSD.
</div>
<div class="Description">
<span class="Description-entry">[reftraj &lt;traj&gt;]</span> Calculate 2D RMSD to frames in trajectory &lt;traj&gt; instead (can also be another COORDS set).
</div>
<div class="Description">
<span class="Description-entry">[parm &lt;parmname&gt; | parmindex &lt;#&gt;]</span> Topology to use for &lt;traj&gt;; only useful in conjunction with <b>reftraj</b>.
</div>
<div class="Description">
<span class="Description-entry">[&lt;refmask&gt;]</span> Mask of atoms in reference; only useful in conjunction with <b>reftraj</b>.
</div>
<div class="Description">
<span class="Description-entry">[corr &lt;corrfilename&gt;]</span> Calculate pseudo-auto-correlation <i>C</i> for 2D-RMSD as <span class="formula"><i>C</i>(<i>i</i>) = <span class="fraction"><span class="ignored">(</span><span class="numerator"><span class="limits"><span class="limit">∑</span></span><span class="scripts"><sup class="script"><i>j</i> &lt; <i>N</i> − <i>i</i></sup><sub class="script"><i>j</i> = 0</sub></span>exp( − <i>RMSD</i>(<i>j</i>, <i>j</i> + <i>i</i>))</span><span class="ignored">)/(</span><span class="denominator"><i>N</i> − <i>i</i></span><span class="ignored">)</span></span></span>, where <i>i</i> is the lag, <i>j</i> is the frame #, and <i>N</i> is the total number of frames. An exponential is used to weight the RMSD since 0.0 RMSD is equivalent to correlation of 1.0. This can only be done if <b>reftraj</b> is not used.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[Corr]</span> (<b>corr</b> only) Pseudo-auto-correlation.
</div>

</li>

</ul>
<div class="Unindented">
<i>Note: For backwards compatibility with ptraj the command ’2drms’ will also work.</i>
</div>
<div class="Indented">
Calculate the best-fit RMSD of each frame in &lt;crd set&gt; (the default COORDS set if none specified) to each other frame. This creates an upper-triangle matrix named &lt;name&gt; (or a full matrix if <b>reftraj</b> specified). The output of the rms2d command can be best-viewed using gnuplot; a gnuplot-formatted file can be produced by giving &lt;filename&gt; a ’.gnu’ extension. For example, to calculate the RMSD of non-hydrogen atoms of each frame in trajectory &ldquo;test.nc&rdquo; to each other frame, writing to a gnuplot-viewable file &ldquo;test.2drms.gnu&rdquo;:
</div>
<pre class="LyX-Code">
trajin test.nc
rms2d !(@H=) out test.2drms.gnu
</pre>
<div class="Unindented">
To calculate the RMSD of atoms named CA of each frame in trajectory &ldquo;test.nc&rdquo; to each frame in &ldquo;ref.nc&rdquo; (assuming test.nc and ref.nc are using the default topology file):
</div>
<pre class="LyX-Code">
trajin test.nc
rms2d @CA out test.2drms.gnu reftraj ref.nc
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.29">12.29</a> rmsavgcorr<a class="Label" name="subsec:cpptraj-rmsavgcorr"> </a>
</h2>
<pre class="LyX-Code">
rmsavgcorr<a class="IndexReference" name="entry-rmsavgcorr-0" href="#index-rmsavgcorr">↓</a> [crdset &lt;crd set&gt;] [&lt;name&gt;] [&lt;mask&gt;] [out &lt;filename&gt;] [mass]
       	[stop &lt;maxwindow&gt;] [offset &lt;offset&gt;]
       	{reference &lt;ref file&gt; parm &lt;parmfile&gt; | first}\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[crdset &lt;crd set&gt;]</span> COORDS data set to use (if not specified the default COORDS set will be used).
</div>
<div class="Description">
<span class="Description-entry">[&lt;name&gt;]</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">[&lt;mask&gt;]</span> Atoms to calculate RMS average correlation for.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Output filename.
</div>
<div class="Description">
<span class="Description-entry">[mass]</span> Mass weight the RMSD calculation.
</div>
<div class="Description">
<span class="Description-entry">[stop &lt;maxwindow&gt;]</span> Only calculate RMS average correlation up to &lt;maxwindow&gt;.
</div>
<div class="Description">
<span class="Description-entry">[offset &lt;offset&gt;]</span> Skip every &lt;offset&gt; windows in calculation.
</div>
<div class="Description">
<span class="Description-entry">[first]</span> Use first averaged frame as reference for each window (default).
</div>
<div class="Description">
<span class="Description-entry">[reference &lt;ref file&gt; [parm &lt;parmfile&gt;]</span> Use reference file (with specified parm) as reference for each window.
</div>

</li>

</ul>
<div class="Unindented">
The RMS average correlation<span class="bibcites">[<a class="bibliocite" name="cite-31" href="#biblio-31"><span class="bib-index">31</span></a>]</span> (RAC) is calculated as the average RMSD of running-averaged coordinates over increasing window sizes (or lag). Output has format:
</div>
<pre class="LyX-Code">
&lt;WindowSize&gt; &lt;RAC&gt;
</pre>
<div class="Unindented">
The first entry has a window size of 1, and so is just the average RMSD of all frames to the specified reference structure. The second entry has a window size of two, so it is the average RMSD of all frames averaged over two adjacent windows to the specified reference, and so on. The RAC will be calculated up to the number of frames minus 1 or the value specified by <b>stop</b>, whichever is lower. The offset can be used to speed up the calculation by skipping window sizes. To calculate mass-weighted RMSD specify <b>mass</b>. Note that to reduce memory costs it can be useful to strip all coordinates not involved in the RMS fit from the system prior to specifying ’rmsavgcorr’. For example, to calculate the correlation of C-alpha RMSD of residues 2 to 12:
</div>
<pre class="LyX-Code">
strip !(:2-12@CA)
rmsavgcorr out rmscorr.dat
</pre>
<div class="Unindented">
The curve generated by RAC decays towards zero due to the way RAC is defined. By the time the "lag" is N-1 (where N is the total number of frames) you have only two averaged coordinates: call them Avg1 (averaged over 1 though N-1 frames) and Avg2 (averaged over 2 through N frames). Barring any extraordinary circumstances the RMSD between Avg1 and Avg2 will almost certainly be quite low. 
</div>
<div class="Indented">
The RAC is a way to probe the time scales of interesting events. Any deviation from a smoothly decaying curve is an indication that there are some significant structural differences occurring over that time interval. RAC curves can be particularly useful when comparing independent simulations of the same system.
</div>
<div class="Indented">
One thing to keep in mind that since the underlying metric is RMSD, it can be sensitive to the reference frame you choose. It may be useful to try looking at both RAC from the first frame, as well as an averaged reference frame. For an example of use see Galindo-Murillo et al.<span class="bibcites">[<a class="bibliocite" name="cite-15" href="#biblio-15"><span class="bib-index">15</span></a>]</span>, in particular Figure 2.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.30">12.30</a> rotdif<a class="Label" name="subsec:cpptraj-rotdif"> </a>
</h2>
<pre class="LyX-Code">
rotdif [outfile &lt;outfilename&gt;] [usefft]
  Options for generating random vectors:
	[nvecs &lt;nvecs&gt;] [rvecin &lt;randvecIn&gt;] [rseed &lt;random seed&gt;]
	[rvecout &lt;randvecOut&gt;] [rmatrix &lt;set name&gt; [rmout &lt;rmOut&gt;]]
  Options for calculating vector time correlation functions:
	[order &lt;olegendre&gt;] [ncorr &lt;ncorr&gt;] [corrout &lt;corrOut&gt;]
  *** The options below only apply if ’usefft’ IS NOT specified. ***
  Options for calculating local effective D, small anisotropy:
	[deffout &lt;deffOut&gt;] [itmax &lt;itmax&gt;] [tol &lt;tolerance&gt;] [d0 &lt;d0&gt;]
	[nmesh &lt;NmeshPoints&gt;] dt &lt;tfac&gt; [ti &lt;ti&gt;] tf &lt;tf&gt;
  Options for calculating D with full anisotropy:
	[amoeba_tol &lt;tolerance&gt;] [amoeba_itmax &lt;iterations&gt;]
	[amoeba_nsearch &lt;n&gt;] [scalesimplex &lt;scale&gt;] [gridsearch]
  *** The options below only apply if ’usefft’ IS specified. ***
  Options for curve-fitting:
	[fit_tol &lt;tolerance&gt;] [fit_itmax &lt;max # iterations&gt;]  \begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">outfile &lt;outfilename&gt;</span> File to write all output from rotdif command to.
</div>

</li>

</ul>
<pre class="LyX-Code">
<i>Options for generating random vectors:\begin_inset Separator latexpar</i>\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">nvecs &lt;nvecs&gt;</span> Number of random vectors to generate (default 1000).
</div>
<div class="Description">
<span class="Description-entry">rvecin &lt;randvecIn&gt;</span> File to read random vectors from (format is 1 per line, 4 columns, &lt;#&gt; &lt;VX&gt; &lt;VY&gt; &lt;VZ&gt;).
</div>
<div class="Description">
<span class="Description-entry">rseed &lt;random seed&gt;</span> Seed for random number generator (default 80531). Specify -1 to use wallclock time.
</div>
<div class="Description">
<span class="Description-entry">rvecout &lt;randvecOut&gt;</span> File to write random vectors to (format is 1 per line, 4 columns, &lt;#&gt; &lt;VX&gt; &lt;VY&gt; &lt;VZ&gt;).
</div>
<div class="Description">
<span class="Description-entry">rmatrix &lt;set name&gt;</span> Data set to read rotation matrices from. Rotation matrices will be used to rotate random vectors.
</div>
<div class="Description">
<span class="Description-entry">rmout &lt;rmOut&gt;</span> Write rotation matrices to file, 1 per line, frame # followed by matrix in row-major order.
</div>

</li>

</ul>
<pre class="LyX-Code">
<i>Options for calculating vector time correlation functions:\begin_inset Separator latexpar</i>\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">order &lt;olegendre&gt;</span> The order of Legendre polynomials to use when calculating vector time correlation functions (default 2).
</div>
<div class="Description">
<span class="Description-entry">ncorr &lt;ncorr&gt;</span> Maximum length of time correlation functions in frames. If this is not specified it will be set to (<b>tf</b> - <b>ti</b>) / <b>dt</b> (recommended).
</div>
<div class="Description">
<span class="Description-entry">corrout &lt;corrOut&gt;</span> If specified write vector time correlation functions to &lt;corrOut&gt;.X with format: &lt;Time&gt; &lt;Px&gt;
</div>

</li>

</ul>
<pre class="LyX-Code">
<i>Options for calculating local effective D, small anisotropy:\begin_inset Separator latexpar</i>\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">deffout &lt;deffOut&gt;</span> File to write out local effective diffusion constants determined in the limit of small anisotropy.
</div>
<div class="Description">
<span class="Description-entry">itmax &lt;itmax&gt;</span> Maximum number of iterations to determine each local effective diffusion constant (small anisotropy) assuming fit to single exponential form (default 500).
</div>
<div class="Description">
<span class="Description-entry">tol &lt;tolerance&gt;</span> Tolerance for determining local effective diffusion constant (small anisotropy) assuming fit to single exponential form (default 1E-6).
</div>
<div class="Description">
<span class="Description-entry">d0 &lt;d0&gt;</span> Initial guess for small anisotropy diffusion constant in radians^2/ns (default 0.03).
</div>
<div class="Description">
<span class="Description-entry">nmesh &lt;NmeshPoints&gt;</span> Number of points per frame to use when creating cubic-splined-smoothed forms of vector time correlation curves (default 2).
</div>
<div class="Description">
<span class="Description-entry">dt &lt;tfac&gt;</span> Time interval between frames (used in integrating vector time correlation curves) in ns.
</div>
<div class="Description">
<span class="Description-entry">ti &lt;ti&gt;</span> Initial time value in ns for integrating the time correlation functions (default 0.0).
</div>
<div class="Description">
<span class="Description-entry">tf &lt;tf&gt;</span> Final time value in ns for integrating the time correlation functions. It is recommended this be less than the maximum simulation time since the tails of time correlation functions tend to be noisy.
</div>

</li>

</ul>
<pre class="LyX-Code">
<i>Options for calculating D with full anisotropy:\begin_inset Separator latexpar</i>\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">amoeba_tol &lt;tolerance&gt;</span> Tolerance for downhill-simplex minimizer (default 1E-7).
</div>
<div class="Description">
<span class="Description-entry">amoeba_itmax &lt;iterations&gt;</span> Number of iterations to run downhill-simplex minimizer (default 10000).
</div>
<div class="Description">
<span class="Description-entry">amoeba_nsearch &lt;n&gt;</span> Number of searches to perform with downhill-simplex minimizer (default 1).
</div>
<div class="Description">
<span class="Description-entry">scalesimplex &lt;scale&gt;</span> Factor to use when scaling simplexes (default 0.5).
</div>
<div class="Description">
<span class="Description-entry">gridsearch</span> If specified, perform a brute-force grid search to attempt to find a better solution for diffusion tensor with full anisotropy (may be expensive).
</div>

</li>

</ul>
<div class="Unindented">
Evaluate rotational diffusion properties of a molecule over a trajectory according to an expanded version of the procedure laid out by Wong &amp; Case<span class="bibcites">[<a class="bibliocite" name="cite-36" href="#biblio-36"><span class="bib-index">36</span></a>]</span>. Briefly, random vectors (representing the orientation of the molecule) are rotated according to rotation matrices obtained from an RMS fit to a reference structure (typically an averaged structure). For each random vector the time correlation function of the rotated vector is calculated using Legendre polynomials of the specified order. The integral over this time correlation function (which may be smoothed using cubic splines to improve the integration) is then used to find the effective diffusion constant (D) in the limit of small anisotropy. Then, using each calculated D, the diffusion tensor is determined with full anisotropy. Finally, a downhill simplex minimizer is used to optimize D with full anisotropy; (this last step is not described in the original paper).
</div>
<div class="Indented">
Rotation matrices are generated via an RMS fit to a reference structure (see <a class="Reference" href="#subsec:cpptraj-rmsd">11.64 on page 1↑</a>). It is recommended that the RMS fit be done to an average structure (see <a class="Reference" href="#subsec:cpptraj-average">11.8 on page 1↑</a>). These rotation matrices are used to rotate each random vector M times (where M is the total number of frames), which creates a time series for each random vector. The time correlation functions are calculated for each random vector time series using Legendre polynomials of the specified order (default 2). The maximum length of the correlation function (or lag) can be specified by <b>ncorr</b> (in frames). If <b>ncorr</b> is not specified it will be set internally based on the specified values of <b>ti</b>, <b>tf</b>, and <b>dt</b>; this is recommended. Note that if <b>ncorr</b> is specified it should be set to a number less than the total number of frames since noise in time correlation functions increases as <b>ncorr</b> approaches the # of frames. The integration over the correlation function is from <b>ti </b>(in whatever units are used of <b>dt</b>, generally ns; 0.0 ns if not specified) to <b>tf</b> (same units as <b>ti</b>), with the time between frames specified by <b>dt</b>; the final time should be less than the total simulation time (see example below). The relative size of the mesh used with cubic spline interpolation for integration is controlled by <b>nmesh</b> (size of the mesh is <b>ncorr</b> points * <b>nmesh</b>); <b>nmesh</b> = 1 means no interpolation, default is 2. Note that if the integral of the correlation function for a vector is negative, that vector will be skipped in subsequent calculations (since it would imply a negative value for effective diffusion).
</div>
<div class="Indented">
The iterative solver for effective value of the diffusion constant from the correlation functions is controlled by <b>itmax</b>, <b>tol</b>, and <b>d0</b>, where <b>itmax</b> specifies the number of iterations to perform (default 500), <b>tol</b> specifies the tolerance (default 1E-6), and <b>d0</b> specifies the initial guess for the diffusion constant in radians^2 / ns (default 0.03). Effective diffusion constants for each random vector can be written out to a file specified by <b>deffout</b>. Results are printed to the file specified by <b>outfile</b>. Details on the Q and D tensors are given, as well as observed and calculated tau for each random vector. First, results are printed for analysis in the limit of small anisotropy. Next, results are printed for analysis with full anisotropy. The results of the full anisotropic calculation are first given using results from the small anisotropic analysis as an initial guess, followed by the final results after minimization using the downhill simplex (amoeba) minimizer.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--34"></a>Example
</h3>
<div class="Unindented">
There are two important things to keep in mind when using rotdif analysis:
</div>
<ol>
<li>
When calculating any kind of diffusive property it is best to simulate in the microcanonical (NVE) ensemble with a shorter time step and increased SHAKE tolerance; thermostats and barostats will effect diffusion calculations.
</li>
<li>
Time correlation functions become noisier as the length of the function approaches the maximum. Therefore in general one should choose parameters for the time correlation function that are much shorter than the total simulation length. 
</li>

</ol>
<div class="Unindented">
For example, given a trajectory ’mdcrd.nc’ containing 10000 frames with a total simulation time of 200 ns (so the time between frames is 0.02 ns), to calculate rotational diffusion using 100 vectors using rotation matrices generated via an RMS fit to ’avgstruct.pdb’, computing and integrating the time correlation function for each vector from 0 to 5 ns (1/40th of the simulation), and writing out the effective diffusion constants and results to ’deffs.dat’ and ’rotdif.out’ respectively:
</div>
<pre class="LyX-Code">
reference avgstruct.pdb [avg]
rms R0 @CA,C,N,O ref [avg] savematrices
trajin mdcrd.nc
rotdif nvecs 100 rmatrix R0[RM] \
       ti 0.0 tf 5.0 dt 0.02 deffout deffs.dat \
       outfile rotdif.out
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.31">12.31</a> runningavg
</h2>
<pre class="LyX-Code">
runningavg &lt;dset1&gt; [&lt;dset2&gt; ...] [name &lt;dsetname&gt;] [out &lt;filename&gt;]
                   [ [cumulative] | [window &lt;window&gt;] ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dset1&gt; [&lt;dset2&gt; ...]</span> Data set(s) to calculate running average for.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;dsetname&gt;]</span> Output running average data set name.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> File to write results to.
</div>
<div class="Description">
<span class="Description-entry">[cumulative]</span> Calculate cumulative running average instead.
</div>
<div class="Description">
<span class="Description-entry">[window &lt;window&gt;]</span> Size in frames of window over which to calculate running average.
</div>

</li>

</ul>
<div class="Unindented">
Calculate running average over windows of given size for data in selected data set(s).
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.32">12.32</a> slope
</h2>
<pre class="LyX-Code">
slope &lt;dset0&gt; [&lt;dset1&gt; ...] [out &lt;outfile&gt;] [name &lt;name&gt;]
  	[type {forward|backward|central}]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dset0&gt; [&lt;dset1&gt; ...]</span> Data set(s) to calculate finite difference for.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;outfile&gt;]</span> File to write finite difference curves to.
</div>
<div class="Description">
<span class="Description-entry">[name &lt;name&gt;]</span> Output data set(s) name.
</div>
<div class="Description">
<span class="Description-entry">[type {forward|backward|central}]</span> Specify type of finite difference to calculate (default forward).
</div>
<div class="Unindented">
DataSets generated:
</div>
<div class="Description">
<span class="Description-entry">&lt;name&gt;:&lt;idx&gt;</span> Output finite difference curves for each input data set (indexed from 0).
</div>

</li>

</ul>
<div class="Unindented">
Calculate finite differences for each input data set.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.33">12.33</a> spline
</h2>
<pre class="LyX-Code">
spline &lt;dset0&gt; [&lt;dset1&gt; ...] [out &lt;outfile&gt;] [meshsize &lt;n&gt; | meshfactor &lt;x&gt;]
[meshmin &lt;mmin&gt;] [meshmax &lt;mmax&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dsetX&gt;</span> Data set(s) to perform splining on.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;outfile&gt;]</span> Write splined data to &lt;outfile&gt;.
</div>
<div class="Description">
<span class="Description-entry">[meshsize&lt;n&gt;]</span> Size of the mesh to use for splining.
</div>
<div class="Description">
<span class="Description-entry">[meshfactor &lt;x&gt;]</span> If meshsize is not given, use a mesh of data set size * &lt;x&gt;.
</div>
<div class="Description">
<span class="Description-entry">[meshmin &lt;mmin&gt;]</span> Mesh X minimum value.
</div>
<div class="Description">
<span class="Description-entry">[meshmax &lt;mmax&gt;]</span> Mesh X maximum value.
</div>

</li>

</ul>
<div class="Unindented">
Apply cubic splines to the given input data sets to create new data sets.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.34">12.34</a> statistics | stat<a class="Label" name="subsec:cpptraj_statistics"> </a>
</h2>
<pre class="LyX-Code">
stat {&lt;name&gt; | ALL} [shift &lt;value&gt;] [out &lt;filename&gt;] [noeout &lt;filename&gt;]
	 [ignorenv] [name &lt;noe setname&gt;]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;name&gt;</span> Name of data set to analyze.
</div>
<div class="Description">
<span class="Description-entry">ALL</span> analyze all data sets.
</div>
<div class="Description">
<span class="Description-entry">shift &lt;value&gt;</span> Subtract &lt;value&gt; from all elements in each data set.
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write analysis results to &lt;filename&gt; (STDOUT if not specified).
</div>
<div class="Description">
<span class="Description-entry">[noeout &lt;filename&gt;]</span> (Type ’noe’ only) Write summary of NOE results to &lt;filename&gt;.
</div>
<div class="Description">
<span class="Description-entry">[ignorenv]</span> (Type ’noe’ only) Ignore negative NOE violations (i.e. shorter-than-expected distances).
</div>
<div class="Description">
<span class="Description-entry">[name &lt;noe setname&gt;]</span> (Type ’noe’ only) Name for output NOE data sets.
</div>
<div class="Unindented">
DataSet Aspects for type ’noe’ output:
</div>
<div class="Description">
<span class="Description-entry">[R6]</span> Averaged <span class="formula">1 ⁄ <i>r</i><sup>6</sup></span>distance for each set.
</div>
<div class="Description">
<span class="Description-entry">[NViolations]</span> Number of violations based on given bounds for each set.
</div>
<div class="Description">
<span class="Description-entry">[AvgViolation]</span> <span class="formula">1 ⁄ <i>r</i><sup>6</sup></span> averaged distance minus expected distance for each set.
</div>
<div class="Description">
<span class="Description-entry">[NOEnames]</span> Name of each set.
</div>

</li>

</ul>
<div class="Unindented">
Analyze angles, dihedrals, distances, and/or puckers and calculate various properties. More specific analyses can be obtained by labelling distances/dihedrals/puckers (from e.g. the <b><i>distance</i></b>, <b><i>dihedral</i></b>, <b><i>pucker</i></b> commands or with the <b><i>dataset</i></b> command) with the ’type &lt;label&gt;’ keyword:
</div>
<div class="Description">
<span class="Description-entry">dihedral type labels:</span> alpha, beta, gamma, delta, epsilon, zeta, chi, c2p h1p, phi, psi, omega, pchi
</div>
<div class="Description">
<span class="Description-entry">distance type labels:</span> noe
</div>
<div class="Description">
<span class="Description-entry">pucker type labels:</span> pucker
</div>
<div class="Unindented">
For each input data set, the average, standard deviation, initial and final values will be reported. The cyclic nature of dihedral/pucker data sets is taken into consideration when averaging.
</div>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-12.34.1">12.34.1</a> Torsion Analysis
</h3>
<div class="Unindented">
A table will be written in ASCII format showing the distribution of torsion values for each data set. More specific information may be printed based on the set type. Values in the output marked SNB are from those defined by Schneider, Neidle, and Berman.<span class="bibcites">[<a class="bibliocite" name="cite-33" href="#biblio-33"><span class="bib-index">33</span></a>]</span> For more information on nucleic acid torsion as pertains to RNA see further work by Schneider et al..<span class="bibcites">[<a class="bibliocite" name="cite-32" href="#biblio-32"><span class="bib-index">32</span></a>]</span> 
</div>
<div class="Indented">
For example, to perform in-depth analysis on some nucleic acid dihedral angles:
</div>
<pre class="LyX-Code">
dihedral g0 out dihedrals.dat :1@O5’ :1@C5’ :1@C4’ :1@C3’ type gamma 
dihedral d0 out dihedrals.dat :1@C5’ :1@C4’ :1@C3’ :1@O3’ type delta 
dihedral c0 out dihedrals.dat :1@O4’ :1@C1’ :1@N9 :1@C4 type chi
analyze statistics all out stat.dat
</pre>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-12.34.2">12.34.2</a> Distance Analysis
</h3>
<div class="Unindented">
A table will be written in ASCII format showing the distribution of distance values &lt; 6.5. If a distance is labled as ’type noe’ a compact time series will be printed in ASCII format showing the NOE as strong, medium, or weak. In addition the &lt;r^-6&gt;^(-1/6) averaged value will be reported, as well as the number of upper/lower bound violations. If <b>’noeout’</b> is specified, a summary of these results will be written with format:
</div>
<pre class="LyX-Code">
&lt;#NOE&gt; &lt;R6&gt; &lt;Nviolation&gt; &lt;AvgViolation&gt; &lt;Name&gt;
</pre>
<div class="Unindented">
Where &lt;#NOE&gt; is an index, &lt;R6&gt; is the &lt;r^-6&gt;^(-1/6) averaged distance, &lt;Nviolation&gt; is the total number of bounds violations, &lt;AvgViolation&gt; is the average difference from expected distance <i>Rexp</i> when the distance is violated (note that if not explicitly set, <i>Rexp</i> is set to the upper bound when the lower bound is 0.0, or the average of upper and lower bounds otherwise), and &lt;Name&gt; is the data set legend.
</div>
<div class="Indented">
For example, the following input could be used to check certain distances for NOE violations:
</div>
<pre class="LyX-Code">
distance :3@HB= :10@HG= type noe noe_medium 
distance :3@HE= :10@HG= type noe noe_strong 
distance :3@HA :12@HA type noe noe_medium 
distance :3@HD= :12@HG= type noe noe_medium 
distance :3@HE= :12@HA type noe noe_strong 
analyze statistics all out dpdp.noe.dat noeout noe_graph.dat name Res3_NOE
</pre>
<h3 class="Subsubsection">
<a class="toc" name="toc-Subsubsection-12.34.3">12.34.3</a> Pucker Analysis
</h3>
<div class="Unindented">
A table will be written in ASCII format showing the distribution of pucker phases for each data set.
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.35">12.35</a> ti
</h2>
<pre class="LyX-Code">
ti &lt;dset0&gt; [&lt;dset1&gt; ...] {nq &lt;n quad pts&gt; | xvals &lt;x values&gt;}
	[name &lt;set name&gt;] [out &lt;file&gt;] [curveout &lt;ti curve file&gt;]
	[nskip &lt;#s to skip&gt;]
	[avgincrement &lt;#&gt; [avgmax &lt;#&gt;] [avgskip &lt;#&gt;]]
	[bs_samples &lt;samples&gt; [bs_points &lt;points&gt;] [bs_seed &lt;#&gt;]
	 [bs_fac &lt;factor&gt;]]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">&lt;dset0&gt; [&lt;dset1&gt; ...]</span> Data set arguments specifying input DV/DL values.
</div>
<div class="Description">
<span class="Description-entry">nq &lt;n quad pts&gt;</span> Number of points for Gaussian quadrature integration. Expect one data set per point.
</div>
<div class="Description">
<span class="Description-entry">xvals &lt;x values&gt;</span> Comma-separated list of X values for integration. Expect one data set per value.
</div>
<div class="Description">
<span class="Description-entry">name &lt;set name&gt;</span> Output data set name.
</div>
<div class="Description">
<span class="Description-entry">out &lt;file&gt;</span> File to write results of integration to.
</div>
<div class="Description">
<span class="Description-entry">curveout &lt;ti curve file&gt;</span> File to write TI curves to.
</div>
<div class="Description">
<span class="Description-entry">nskip &lt;#s to skip&gt;</span> Comma separated list of number of points to skip. For each number given, the TI integration will be repeated.
</div>
<div class="Description">
<span class="Description-entry">avgincrement &lt;#&gt; [avgmax &lt;#&gt;] [avgskip &lt;#&gt;]</span> Starting from point ’avgskip’ (default 0), repeat the TI integration calculation in increments of &lt;#&gt; up to ’avgmax’ (default all points), so ’avgincrement 10’ will do points 0-10, 0-20, etc.
</div>
<div class="Description">
<span class="Description-entry">bs_samples &lt;samples&gt; [bs_points &lt;points&gt;] [bs_seed&lt;#&gt;] [bs_fac &lt;factor&gt;]</span> Estimate error via bootstrap analysis, repeating the TI integration &lt;samples&gt; times using &lt;points&gt; points or &lt;factor&gt; times the total number of points. Randomize with given seed.
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[TIcurve]</span> Raw TI curve. If ’nskip’ index is number of points skipped. If bootstrapping, index is sample index. If ’avgincrement’ the index is the number of points.
</div>
<div class="Description">
<span class="Description-entry">[SD]</span> For bootstrap analysis, standard deviation of average free energy over samples.
</div>

</li>

</ul>
<div class="Unindented">
Calculate free energy using DV/DL energies from thermodynamic integration. The results of integration of the DV/DL curve will be written to <b>&lt;file&gt;</b>, while the curves themselves will be written to <b>&lt;ti curve file&gt;</b>. Use <b>nq</b> to specify number of Gaussian quadrature points; otherwise the lambda values should be specified by <b>xvals</b>, where <b>&lt;x values&gt;</b> is a comma-separated list.
</div>
<div class="Indented">
For example, to perform Gaussian quadrature integration using data sets named ’TIdata’, repeating the calculation for various number of skipped data points:
</div>
<pre class="LyX-Code">
ti TIdata nq 9 name Curve out skip.agr curveout curve.agr nskip 0,5,10,15,20,30,40,50
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.36">12.36</a> timecorr
</h2>
<pre class="LyX-Code">
timecorr vec1 &lt;vecname1&gt; [vec2 &lt;vecname2&gt;] out &lt;filename&gt;
           [order &lt;order&gt;] [tstep &lt;tstep&gt;] [tcorr &lt;tcorr&gt;]
           [dplr] [norm] [drct] [dplrout &lt;dplrfile&gt;] [ptrajformat]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>vec1 &lt;vecname1&gt; [vec2 &lt;vecname2&gt;]</b></span> Vector(s) on which to operate. By default the auto-correlation function will be calculated if one vector is specified, and the cross-correlation function will be calculated if two vectors are specified.
</div>
<div class="Description">
<span class="Description-entry"><b>out &lt;filename&gt;</b></span> Name of file to write output to.
</div>
<div class="Description">
<span class="Description-entry"><b>[order &lt;order&gt;]</b></span> Order of Legendre polynomials to use; default 2.
</div>
<div class="Description">
<span class="Description-entry"><b>[tstep &lt;tstep&gt;]</b></span> Time between snapshots (default 1.0).
</div>
<div class="Description">
<span class="Description-entry"><b>[tcorr &lt;tcorr&gt;]</b></span> Maximum time to calculate correlation functions for (default 10000.0).
</div>
<div class="Description">
<span class="Description-entry"><b>[dplr]</b></span> Output correlation functions <span class="formula"><i>C</i><sub><i>l</i></sub> ≡  &lt; <i>P</i><sub><i>l</i></sub> ⁄ (<i>r</i>(0)<sup>3</sup><i>r</i>(<i>τ</i>)<sup>3</sup>) &gt; </span> and <span class="formula"> &lt; 1 ⁄ (<i>r</i>(0)<sup>3</sup><i>r</i>(<i>τ</i>)<sup>3</sup>) &gt; </span> in addition to the <span class="formula"><i>P</i><sub><i>l</i></sub></span> correlation function.
</div>
<div class="Description">
<span class="Description-entry"><b>[norm]</b></span> Normalize all correlation functions, i.e., <span class="formula"><i>C</i><sub><i>l</i></sub>(<i>t</i> = 0) = <i>P</i><sub><i>l</i></sub>(<i>t</i> = 0) = 1.0</span>.
</div>
<div class="Description">
<span class="Description-entry"><b>[drct]</b></span> Use the direct method to calculate correlations instead of FFT; this will be much slower.
</div>
<div class="Description">
<span class="Description-entry">[dplrout]</span> (<b>dplr</b> only) Write extra information for each vector related to <b>dplr</b> option to &lt;dplrfile&gt;.
</div>
<div class="Description">
<span class="Description-entry">[ptrajformat]</span> Write output in ptraj style (prevents use of data formatting options).
</div>
<div class="Unindented">
DataSet Aspects:
</div>
<div class="Description">
<span class="Description-entry">[P]</span> P&lt;order&gt; correlation function.
</div>
<div class="Description">
<span class="Description-entry">[C]</span> C&lt;order&gt; correlation function (<b>dplr</b> only).
</div>
<div class="Description">
<span class="Description-entry">[R3R3]</span> &lt;1/(r(0)<sup>3</sup>r(t)<sup>3</sup>&gt; correlation function (<b>dplr</b> only).
</div>
<div class="Description">
<span class="Description-entry">[R]</span> (_TC_DIPOLAR_) Average magnitude (&lt;R&gt;).
</div>
<div class="Description">
<span class="Description-entry">[RRIG]</span> (_TC_DIPOLAR_) Sqrt( &lt;R^2&gt; ).
</div>
<div class="Description">
<span class="Description-entry">[R3]</span> (_TC_DIPOLAR_) &lt;1/R^3&gt;.
</div>
<div class="Description">
<span class="Description-entry">[R6]</span> (_TC_DIPOLAR_) &lt;1/R^6&gt;.
</div>
<div class="Description">
<span class="Description-entry">[Name]</span> (_TC_DIPOLAR_) Vector name.
</div>

</li>

</ul>
<div class="Unindented">
Calculate time auto/cross-correlation functions for vectors using spherical harmonics theory. NOTE: To calculate direct correlation functions for vectors just use the <b><i>corr</i></b> analysis command. The <b>norm</b> keyword will normalize the resulting correlation functions. Note that if <b>dplr</b> is specified, a new global data set named _TC_DIPOLAR_ will be created, containing extra data for each vector analyzed with a ’<b><i>timecorr</i></b> <b>dplr</b>’ command.
</div>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--35"></a>Examples
</h3>
<div class="Unindented">
Vectors between atoms 5 and 6 as well as 7 and 8 are calculated below, for which auto and cross time correlation functions are obtained.
</div>
<pre class="LyX-Code">
vector v0 @5 @6
vector v1 @7 @8
timecorr vec1 v0 tstep 1.0 tcorr 100.0 out v0.out order 2
timecorr vec1 v1 tstep 1.0 tcorr 100.0 out v1.out order 2
timecorr vec1 v0 vec2 v1 tstep 1.0 tcorr 100.0 out v0_v1.out order 2
</pre>
<div class="Unindented">
Similarly, a vector perpendicular to the plane through atoms 18, 19, and 20 is obtained and further analyzed.
</div>
<pre class="LyX-Code">
vector v2 @18,@19,@20 corrplane
timecorr vec1 v3 tstep 1.0 tcorr 100.0 out v2.out order 2
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.37">12.37</a> vectormath
</h2>
<pre class="LyX-Code">
vectormath vec1 &lt;vecname1&gt; vec2 &lt;vecname2&gt; [out &lt;filename&gt;] [norm] [name &lt;setname&gt;]
           [ dotproduct | dotangle | crossproduct ]\begin_inset Separator latexpar\end_inset
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry"><b>vec1 &lt;vecname1&gt; vec2 &lt;vecname2&gt;</b></span> Vector(s) on which to operate.
</div>
<div class="Description">
<span class="Description-entry"><b>[out &lt;filename&gt;]</b></span> Name of file to write output to.
</div>
<div class="Description">
<span class="Description-entry">[dotproduct]</span> (Default) Calculate the dot-product of the two vectors.
</div>
<div class="Description">
<span class="Description-entry">[dotangle]</span> Calculate angle from dot-product between the two vectors; vectors will be normalized.
</div>
<div class="Description">
<span class="Description-entry">[crossproduct]</span> Calculate cross-product of the two vectors.
</div>
<div class="Description">
<span class="Description-entry">[norm]</span> Normalize the vectors; this will affect any subsequent calculations with the vectors. This is turned on automatically if <b>dotangle</b> specified.
</div>

</li>

</ul>
<div class="Unindented">
Calculate dot product, angle from dot product (degrees), or cross product for specified vectors. Note that <b>norm</b> normalizes the vectors themselves; the vectors will remain normalized for subsequent calculations or output. Either <b>vec1</b> or <b>vec2</b> can be of size 1; in that case each vector in the set with N frames operates on the single vector. For example, if <b>vec1</b> is size N and <b>vec2</b> is size 1, then each frame of vec1 is operated on the single vector from vec2.
</div>
<div class="Indented">
For example, to get the angles between two previously calculated vectors v1 and v2:
</div>
<pre class="LyX-Code">
vectormath vec1 v1 vec2 v2 dotangle out dotproduct.dat name acos(|V1|*|V2|)
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-12.38">12.38</a> wavelet
</h2>
<pre class="LyX-Code">
wavelet [crdset &lt;set name&gt;] nb &lt;n scaling vals&gt; [s0 &lt;s0&gt;] [ds &lt;ds&gt;]
        [correction &lt;correction&gt;] [chival &lt;chival&gt;] [type &lt;wavelet&gt;]
        [out &lt;filename&gt;] [name &lt;setname&gt;]\begin_inset Separator latexpar\end_inset
        [cluster [minpoints &lt;#&gt;] [epsilon &lt;value&gt;] [clusterout &lt;file&gt;]
           [clustermapout &lt;file&gt;] [cmapdetail] [kdist] [cprefix &lt;PDB prefix&gt;]
           [overlay &lt;trajfile&gt;] [overlayparm &lt;parmfile&gt;]]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[crdset &lt;set name&gt;]</span> COORDS data set to use 
</div>
<div class="Description">
<span class="Description-entry">nb &lt;n scaling vals&gt;</span> Number of scales. The smaller the number the better resolution, but slower to plot. 
</div>
<div class="Description">
<span class="Description-entry">[s0 &lt;s0&gt;]</span> The smallest scale of the wavelet function (default 2dt where dt is time between snapshots in ps ) 
</div>
<div class="Description">
<span class="Description-entry">[ds &lt;ds&gt;]</span> Spacing between discrete scales. (Default is 0.25. Smaller value of ds gives finer resolution. The largest values that give adequate sampling in scale for Morlet and Paul are 0.5 and 1.5, respectively)
</div>
<div class="Description">
<span class="Description-entry">[correction &lt;correction&gt;]</span> The scale-to-wavelength parameter (1.01 for Morlet, 1.389 for Paul). Automatically set based on wavelet if not otherwise specified. 
</div>
<div class="Description">
<span class="Description-entry">[chival &lt;chival&gt;]</span> The value of <span class="formula">χ<span class="scripts"><sup class="script">2</sup><sub class="script">2</sub></span></span>at a particular confidence level 
</div>
<div class="Description">
<span class="Description-entry">[type &lt;wavelet&gt;]</span> Type of wavelet function to use &lt;morlet&gt; or &lt;paul&gt; 
</div>
<div class="Description">
<span class="Description-entry">[out &lt;filename&gt;]</span> Write results to file named &lt;filename&gt; 
</div>
<div class="Description">
<span class="Description-entry">[name &lt;setname&gt;]</span> Store results in data set with name &lt;setname&gt;
</div>
<div class="Description">
<span class="Description-entry">[cluster]</span> Perform wavelet clustering i.e. wavelet feature extraction analysis.
</div>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">[minpoints &lt;#&gt;]</span> Minimum number of points necessary to form a region of interest.
</div>
<div class="Description">
<span class="Description-entry">[epsilon &lt;value&gt;]</span> Minimum region of interest size.
</div>
<div class="Description">
<span class="Description-entry">[clusterout &lt;file&gt;]</span> Output for clustering (see below).
</div>
<div class="Description">
<span class="Description-entry">[clustermapout &lt;file&gt;]</span> Output cluster map (recommended gnuplot format, see below).
</div>
<div class="Description">
<span class="Description-entry">[cmapdetail]</span> Instead of the map being smoothed to cluster regions, show full detail.
</div>
<div class="Description">
<span class="Description-entry">[kdist]</span> Can be used to determine minpoints and epsilon - see below.
</div>
<div class="Description">
<span class="Description-entry">[cprefix &lt;PDB prefix&gt;]</span> Output cluster region PDBs (only containing from minimum to maximum atom and minimum to maximum frame) with given prefix.
</div>
<div class="Description">
<span class="Description-entry">[overlay &lt;trajfile&gt;]</span> Create a trajectory that can be overlaid with the original trajectory to highlight atoms of interest. Atoms in cluster regions will get their normal coordinates - all others are set to the common center of mass. 
</div>
<div class="Description">
<span class="Description-entry">[overlayparm &lt;parmfile&gt;]</span> Topology that can be used with the overlay trajectory.
</div>

</li>

</ul>
<div class="Unindented">
&lt;wavelet&gt;: morlet, paul
</div>

</li>

</ul>
<div class="Unindented">
Perform the wavelet analysis using fast Fourier transform (FFT) algorithm on specified trajectory and write out to a gnuplot-formatted file named &lt;name.gnu&gt;. The created Wavelet map provides a clear picture of the significant motions which are characterized both in time and space. Note that typically the trajectory in question should have rotational and translational movement removed (via e.g. the <b><i>rms</i></b> command); otherwise these will be reflected in the wavelet analysis results.
</div>
<div class="Indented">
Wavelet analysis contains two main steps which performs continues wavelet transform (CWT) and statistical significance testing as proposed by Torrence and Compo<span class="bibcites">[<a class="bibliocite" name="cite-6" href="#biblio-6"><span class="bib-index">6</span></a>]</span>. Analysis is executed on one dimensional (1-D) coordinate which is defined as the displacement from the starting position. For each atom, CWT is calculated over a specified range of scales from <span class="formula"><i>S</i><sub>0</sub></span>up to <span class="formula"><i>S</i><sub>0</sub>2<sup>(<i>nb</i> − 1)<i>ds</i></sup></span>. To obtain the CWT of the trajectory the Fourier transform of atom’s displacement and wavelets which scaled by S ( S is calculated from: <span class="formula"><i>S</i> = <i>S</i><sub>0</sub>2<sup><i>jds</i></sup>; <i>j</i> = 0, 1, 2, …, <i>nb</i> − 1</span>) is computed and then the inverse Fourier transform of the product of Fourier transforms will be calculated as the CWT. After calculating the wavelet coordinates for all atoms, a significance testing is performed to determine the significance of each wavelet coordinate. For doing this test we need to have an appropriate background spectrum to consider as a mean or expected spectrum and compare our wavelet coordinates against this background. In order to calculate the background spectrum since wavelet spectrum (according to the convolution theorem) follows the Fourier spectrum, the Fourier coefficients over every atom’s displacement is calculated using the following formula and a model (<span class="formula"><i>μ</i><sub><i>k</i></sub></span>) is constructed on average which Fourier coefficients fit (<span class="formula"><i>X</i><sub><i>n</i></sub></span>) is the time series which is the atom’s displacement and k is the frequency index<span class="bibcites">[<a class="bibliocite" name="cite-24" href="#biblio-24"><span class="bib-index">24</span></a>]</span>.
</div>
<div class="Indented">
<div class="formula">
<i>f</i><sub><i>k</i> = <span class="fraction"><span class="ignored">(</span><span class="numerator">1</span><span class="ignored">)/(</span><span class="denominator"><i>N</i></span><span class="ignored">)</span></span></sub><span class="limits"><sup class="limit"><i>N</i> − 1</sup><span class="limit">⎲</span><span class="limit">⎳</span><sub class="limit"><i>n</i> = 0</sub></span>exp<span class="array"><span class="arrayrow"><span class="bracket align-left">⎛</span></span><span class="arrayrow"><span class="bracket align-left">⎝</span></span></span><span class="fraction"><span class="ignored">(</span><span class="numerator"> − 2<i>π</i><i>ikn</i></span><span class="ignored">)/(</span><span class="denominator"><i>N</i></span><span class="ignored">)</span></span><span class="array"><span class="arrayrow"><span class="bracket align-right">⎞</span></span><span class="arrayrow"><span class="bracket align-right">⎠</span></span></span><i>X</i><sub><i>n</i></sub>
</div>

</div>
<div class="Indented">
This test is implemented based on the null hypothesis that the assumption is that Fourier coordinates normally distributed around the expected value, then the wavelet coordinates should also be normally distributed. Assuming the expected background spectrum and since the square of a normally distributed variable is chi-square distributed, then the distribution for the square of the absolute values of wavelet coordinates (<span class="formula">|<i>W</i><sub><i>i</i>, <i>k</i></sub>|<sup>2</sup></span> is as follows (<span class="formula"><i>σ</i><sup>2</sup></span>is the variance of the atom’s displacement).
</div>
<div class="Indented">
<div class="formula">
<i>σ</i><sup>2</sup><i>μ</i><sub><i>k</i></sub>χ<span class="scripts"><sup class="script">2</sup><sub class="script">2</sub></span> ⁄ 2
</div>

</div>
<div class="Indented">
Then choosing a confidence level we can determine the minimum acceptable value for <span class="formula">|<i>W</i><sub><i>i</i>, <i>k</i></sub>|<sup>2</sup></span>to be considered as a significant coordinates at that certain confidence level. In the final map the scales of only those wavelet coordinates which are significantly above the expected distribution are stored.
</div>
<div class="Indented">
For example, to perform wavelet analysis on residues 1 to 17 with 40 scaling values starting from scaling of 0.2 with a spacing of 0.25 using the Morlet wavelet:
</div>
<pre class="LyX-Code">
parm nowat.withions.parm7
trajin nowat.image.nc
rms :1-17@C*,N*,O*,P* first mass
wavelet nb 40 s0 0.2 ds 0.25 correction 1.01 chival 1.6094 type morlet \
        :1-17 out wavelet.gnu usemap
</pre>
<h3 class="Subsubsection-">
<a class="toc" name="toc-Subsubsection--36"></a>Wavelet Analysis Feature Extraction
</h3>
<div class="Unindented">
Wavelet analysis feature extraction (WAFEX)<span class="bibcites">[<a class="bibliocite" name="cite-17" href="#biblio-17"><span class="bib-index">17</span></a>]</span> uses a density-based clustering algorithm (a modified version of the DBSCAN algorithm) to highlight physical and temporal regions that have significant motions from wavelet mapsand can extract the specific atoms and frames involved in these motions for further analysis. Cluster regions shown in the map will be smoother by default for easier visualization (unless <b>cmapdetail</b> is specified). Details of the clustering are provided via the <b>clusterout</b> keyword with format:
</div>
<pre class="LyX-Code">
#Cluster [points] [minatm] [maxatm] [minfrm] [maxfrm] [avgval]
</pre>
<ul>
<li class="nested">
<div class="Description">
<span class="Description-entry">#Cluster</span> Cluster region number.
</div>
<div class="Description">
<span class="Description-entry">points</span> Number of points in the cluster.
</div>
<div class="Description">
<span class="Description-entry">minatm</span> Starting atom of the region.
</div>
<div class="Description">
<span class="Description-entry">maxatm</span> End atom of the region.
</div>
<div class="Description">
<span class="Description-entry">minfrm</span> Starting frame of the region.
</div>
<div class="Description">
<span class="Description-entry">maxfrm</span> End frame of the region.
</div>
<div class="Description">
<span class="Description-entry">avgval</span> Average value of points in the region.
</div>

</li>

</ul>
<div class="Unindented">
For example, to create a 2D gnuplot map highlight regions of interest called ’cluster.gnu’ one could use the following input.
</div>
<pre class="LyX-Code">
parm ../DPDP.parm7
trajin ../DPDP.nc
rms @C,CA,N first
wavelet nb 10 s0 2 ds 0.25 type morlet correction 1.01 chival 0.25 \
        :1-22 name DPDP \
        cluster clustermapout cluster.gnu clusterout cluster.dat \
          minpoints 66 epsilon 10.0
datafile cluster.gnu usemap palette kbvyw
</pre>
<div class="Unindented">
Some experimentation with <b>kdist</b> may be required to obtain reasonable values for <b>minpoints</b> and <b>epsilon</b>. See <a class="Reference" href="#subsec:cpptraj_dbscan_Hints">28 on page 1↑</a> as well as the Heidari et al paper for further discussion.
</div>
<h1 class="Section">
<a class="toc" name="toc-Section-13">13</a> Analysis Examples<a class="Label" name="sec:cpptraj-Matrix-Vector-Analysis-Examples"> </a>
</h1>
<div class="Unindented">
Please note that typically for principal component analysis (PCA) the trajectory needs to be aligned against a reference structure to remove overall global and translation motion. Use the <span class="sans"><b><i>rms</i></b></span> command for this. 
</div>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-13.1">13.1</a> <b>Cartesian covariance matrix calculation and projection (PCA)</b>
</h2>
<div class="Unindented">
After calculating modes, snapshots can be projected onto these in an additional pass through the trajectory. It is very important that the snapshots used when projecting are exactly the same as those used to generate the original covariance matrix. This example takes advantage of the COORDS data set functionality in cpptraj to save snapshots for the purposes of projection.
</div>
<pre class="LyX-Code">
<b># Step one. Generate average structure.</b>
<b># RMS-Fit to first frame to remove global translation/rotation.</b>
parm myparm.parm7
trajin mytraj.nc
rms first !@H=
average crdset AVG
run
<b># Step two. RMS-Fit to average structure. Calculate covariance matrix.</b>
<b># Save the fit coordinates.</b>
rms ref AVG !@H=
matrix covar name MyMatrix !@H=
createcrd CRD1
run
<b># Step three. Diagonalize matrix.</b>
runanalysis diagmatrix MyMatrix vecs 2 name MyEvecs
<b># Step four. Project saved fit coordinates along eigenvectors 1 and 2</b>
crdaction CRD1 projection evecs MyEvecs !@H= out project.dat beg 1 end 2
</pre>
<h2 class="Subsection">
<a class="toc" name="toc-Subsection-13.2">13.2</a> Dihedral covariance matrix calculation and projection for backbone phi/psi (PCA)
</h2>
<pre class="LyX-Code">
parm ../1rrb_vac.prmtop 
trajin ../1rrb_vac.mdcrd
<b># Generation of phi/psi dihedral data</b>
multidihedral BB phi psi resrange 2
run
<b># Calculate dihedral covariance matrix and obtain eigenvectors</b>
matrix dihcovar dihedrals BB[*] out dihcovar.dat name DIH
diagmatrix DIH vecs 4 out modes.dihcovar.dat name DIHMODES
run
<b># Project along eigenvectors</b>
projection evecs DIHMODES out dih.project.dat beg 1 end 4 dihedrals BB[*] 
run
</pre>
<div class="Indented">
<h1 class="biblio">
References
</h1>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-1"><span class="bib-index">1</span></a>] </span> <span class="bib-authors">Alex Rodriguez, Alessandro Laio</span>: “<span class="bib-title">Clustering by fast search and find of density peaks</span>”, <i><span class="bib-journal">Science</span></i>, pp. <span class="bib-pages">1492-1496</span>, <span class="bib-year">2014</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-2"><span class="bib-index">2</span></a>] </span> <span class="bib-authors">Daniel R. Roe, Christina Bergonzo, Thomas E. Cheatham III</span>: “<span class="bib-title"> Evaluation of Enhanced Sampling Provided by Accelerated Molecular Dynamics with Hamiltonian Replica Exchange Methods </span>”, <i><span class="bib-journal"> J. Phys. Chem. B </span></i>, pp. <span class="bib-pages">3543-3552</span>, <span class="bib-year">2014</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-3"><span class="bib-index">3</span></a>] </span> <span class="bib-authors">C Altona, M Sundaralingam</span>: “<span class="bib-title">Conformational Analysis of the Sugar Ring in Nucleosides and Nucleotides. A New Description Using the Concept of Pseudorotation</span>”, <i><span class="bib-journal">J Am Chem Soc</span></i>, pp. <span class="bib-pages">8205-8212</span>, <span class="bib-year">1972</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-4"><span class="bib-index">4</span></a>] </span> <span class="bib-authors">M.S. Babcock, E.P.D. Pednault, W.K. Olson</span>: “<span class="bib-title">Nucleic Acid Structure Analysis</span>”, <i><span class="bib-journal">J. Mol. Biol.</span></i>, pp. <span class="bib-pages">125-156</span>, <span class="bib-year">1994</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-5"><span class="bib-index">5</span></a>] </span> <span class="bib-authors">A. Bakan, L.M. Meireles, I. Bahar</span>: “<span class="bib-title">ProDy: Protein Dynamics Inferred from Theory and Experiments</span>”, <i><span class="bib-journal">Bioinformatics</span></i>, pp. <span class="bib-pages">1575-1577</span>, <span class="bib-year">2011</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-6"><span class="bib-index">6</span></a>] </span> <span class="bib-authors">C. Torrence, G. P. Compo</span>: “<span class="bib-title">A practical guide to wavelet analysis</span>”, <i><span class="bib-journal">Bull. Am. Meteorol. Soc.</span></i>, pp. <span class="bib-pages">61-78</span>, <span class="bib-year">1998</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-7"><span class="bib-index">7</span></a>] </span> <span class="bib-authors">Chia-En Chang, Wei Chen, Michael K. Gilson</span>: “<span class="bib-title">Evaluating the Accuracy of the Quasiharmonic Approximation</span>”, <i><span class="bib-journal">Journal of Chemical Theory and Computation</span></i>, pp. <span class="bib-pages">1017-1028</span>, <span class="bib-year">2005</span>. URL <a href=" https://doi.org/10.1021/ct0500904 "><span class="bib-url"> https://doi.org/10.1021/ct0500904 </span></a>. <span class="bib-note">PMID: 26641917</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-8"><span class="bib-index">8</span></a>] </span> <span class="bib-authors">S. Chatterjee, P. G. Debenedetti, F. H. Stillinger, R. M. Lynden-Bell</span>: “<span class="bib-title">A Computational Investigation of Thermodynamics, Structure, Dynamics and Solvation Behavior in Modified Water Models.</span>”, <i><span class="bib-journal">J. Chem. Phys.</span></i>, pp. <span class="bib-pages">124511</span>, <span class="bib-year">2008</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-9"><span class="bib-index">9</span></a>] </span> <span class="bib-authors">J.J. Chou, D.A. Case, A. Bax</span>: “<span class="bib-title">Insights into the mobility of methyl-bearing side chains in proteins</span>”, <i><span class="bib-journal">J. Am. Chem. Soc.</span></i>, pp. <span class="bib-pages">8959-8966</span>, <span class="bib-year">2003</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-10"><span class="bib-index">10</span></a>] </span> <span class="bib-authors">M.L. Connolly</span>: “<span class="bib-title">Analytical molecular surface calculation</span>”, <i><span class="bib-journal">J. Appl. Cryst.</span></i>, pp. <span class="bib-pages">548-558</span>, <span class="bib-year">1983</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-11"><span class="bib-index">11</span></a>] </span> <span class="bib-authors">D Cremer, JA Pople</span>: “<span class="bib-title">A General Definition of Ring Puckering Coordinates</span>”, <i><span class="bib-journal">J Am Chem Soc</span></i>, pp. <span class="bib-pages">1354-1358</span>, <span class="bib-year">1975</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-12"><span class="bib-index">12</span></a>] </span> <span class="bib-authors">G. Cui, J. M. Swails, E. S. Manas</span>: “<span class="bib-title">SPAM: A Simple Approach for Profiling Bound Water Molecules</span>”, <i><span class="bib-journal">J. Chem. Theory Comput.</span></i>, pp. <span class="bib-pages">5539-5549</span>, <span class="bib-year">2013</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-13"><span class="bib-index">13</span></a>] </span> <span class="bib-authors">Donald A. McQuarrie</span>: <i><span class="bib-title">Statistical Thermodynamics</span></i>. <span class="bib-publisher">Harper and Row, New York</span>, <span class="bib-year">1973</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-14"><span class="bib-index">14</span></a>] </span> <span class="bib-authors">M. Ester, H. Kriegel, J. Sander, X. Xu</span>: “<span class="bib-title">A Density-Based Algorithm for Discovering Clusters in Large Spatial Databases with Noise</span>”, <i><span class="bib-journal">Proc. Second Int. Conf. Knowledge Disc. Data Mining (KDD-96)</span></i>, pp. <span class="bib-pages">226-231</span>, <span class="bib-year">1996</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-15"><span class="bib-index">15</span></a>] </span> <span class="bib-authors">Rodrigo Galindo-Murillo, Daniel R. Roe, III Cheatham</span>: “<span class="bib-title">Convergence and reproducibility in molecular dynamics simulations of the DNA duplex d(GCACGAACGAACGAACGC)</span>”, <i><span class="bib-journal">Biochim. Biophys. Acta</span></i>, pp. <span class="bib-pages">1041-1058</span>, <span class="bib-year">2015</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-16"><span class="bib-index">16</span></a>] </span> <span class="bib-authors">SC Harvey, M Prabhakaran</span>: “<span class="bib-title">RIBOSE PUCKERING - STRUCTURE, DYNAMICS, ENERGETICS, AND THE PSEUDOROTATION CYCLE</span>”, <i><span class="bib-journal">J Am Chem Soc</span></i>, pp. <span class="bib-pages">6128-6136</span>, <span class="bib-year">1986</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-17"><span class="bib-index">17</span></a>] </span> <span class="bib-authors">Z. Heidari, D. R. Roe, R. Galindo-Murillo, J. B. Ghasemi, III Cheatham</span>: “<span class="bib-title">Using Wavelet Analysis To Assist in Identification of Significant Events in Molecular Dynamics Simulations</span>”, <i><span class="bib-journal">J. Chem. Inf. Model.</span></i>, pp. <span class="bib-pages">1282-1291</span>, <span class="bib-year">2016</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-18"><span class="bib-index">18</span></a>] </span> <span class="bib-authors">W. Humphrey, A. Dalke, K. Schulten</span>: “<span class="bib-title">VMD Visual Molecular Dynamics.</span>”, <i><span class="bib-journal">J. Molec. Graph.</span></i>, pp. <span class="bib-pages">33-38</span>, <span class="bib-year">1996</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-19"><span class="bib-index">19</span></a>] </span> <span class="bib-authors">P.H. Hunenberger, A.E. Mark, W.F. van Gunsteren</span>: “<span class="bib-title">Fluctuation and Cross-correlation Analysis of Protein Motions Observed in Nanosecond Molecular Dynamics Simulations</span>”, <i><span class="bib-journal">J. Mol. Biol.</span></i>, pp. <span class="bib-pages">492-503</span>, <span class="bib-year">1995</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-20"><span class="bib-index">20</span></a>] </span> <span class="bib-authors">W. Kabsch, C. Sander</span>: “<span class="bib-title">Dictionary of protein secondary structure: pattern recognition of hydrogen-bonded and geometrical features</span>”, <i><span class="bib-journal">Biopolymers</span></i>, pp. <span class="bib-pages">2577-2637</span>, <span class="bib-year">1983</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-21"><span class="bib-index">21</span></a>] </span> <span class="bib-authors">T. Lazaridis</span>: “<span class="bib-title">Inhomogeneous Fluid Approach to Solvation Thermodynamics. 1 Theory</span>”, <i><span class="bib-journal">J. Phys. Chem. B </span></i>, pp. <span class="bib-pages">3531-3541</span>, <span class="bib-year">1998</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-22"><span class="bib-index">22</span></a>] </span> <span class="bib-authors">XJ Lu, WK Olson</span>: “<span class="bib-title">3DNA: a software package for the analysis, rebuilding and visualization of three-dimensional nucleic acid structures</span>”, <i><span class="bib-journal">NUCLEIC ACIDS RESEARCH</span></i>, pp. <span class="bib-pages">5108-5121</span>, <span class="bib-year"> 2003 </span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-23"><span class="bib-index">23</span></a>] </span> <span class="bib-authors">M. A. El Hassan, C. R. Calladine</span>: “<span class="bib-title">Two Distinct Modes of Protein-induced Bending in DNA</span>”, <i><span class="bib-journal">J. Mol. Biol.</span></i>, pp. <span class="bib-pages">331-343</span>, <span class="bib-year">1998</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-24"><span class="bib-index">24</span></a>] </span> <span class="bib-authors">N. C. Benson, V. Dagget</span>: “<span class="bib-title">Wavelet Analysis of Protein Motion</span>”, <i><span class="bib-journal">Int. J. Wavelets Multi.</span></i>, <span class="bib-year">2012</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-25"><span class="bib-index">25</span></a>] </span> <span class="bib-authors">C. N. Nguyen, T. Kurtzman Young, M. K. Gilson</span>: “<span class="bib-title">Grid Inhomogeneous Solvation Theory: Hydration Structure and Thermodynamics of the Miniature Receptor cucurbit[7]uril.</span>”, <i><span class="bib-journal">J. Chem. Phys.</span></i>, pp. <span class="bib-pages">044101</span>, <span class="bib-year">2012</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-26"><span class="bib-index">26</span></a>] </span> <span class="bib-authors">Wilma K. Olson, Manju Bansal, Stephen K. Burley, Richard E. Dickerson, Mark Gerstein, Stephen C. Harvey, Udo Heinemann, Xiang-Jun Lu, Stephen Neidle, Zippora Shakked, Heinz Sklenar, Masashi Suzuki, Chang-Shung Tung, Eric Westhof, Cynthia Wolberger, Helen M. Berman</span>: “<span class="bib-title">A Standard Reference Frame for the Description of Nucleic Acid Base-pair Geometry</span>”, <i><span class="bib-journal">J. Mol. Biol.</span></i>, pp. <span class="bib-pages">229-237</span>, <span class="bib-year">2001</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-27"><span class="bib-index">27</span></a>] </span> <span class="bib-authors">C. Perez, F. Lohr, H. Ruterjans, J.M. Schmidt</span>: “<span class="bib-title">Self-Consistent Karplus Parameterization of (3)J couplings depending on the polypeptide side-chain torsion chi(1)</span>”, <i><span class="bib-journal">J. Am. Chem. Soc.</span></i>, pp. <span class="bib-pages">7081-7093</span>, <span class="bib-year">2001</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-28"><span class="bib-index">28</span></a>] </span> <span class="bib-authors">J.J. Prompers, R. Brüschweiler</span>: “<span class="bib-title">Dynamic and structural analysis of isotropically distributed molecular ensembles</span>”, <i><span class="bib-journal">Proteins</span></i>, pp. <span class="bib-pages">177-189</span>, <span class="bib-year">2002</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-29"><span class="bib-index">29</span></a>] </span> <span class="bib-authors">J.J. Prompers, R. Brüschweiler</span>: “<span class="bib-title">General framework for studying the dynamics of folded and nonfolded proteins by NMR relaxation spectroscopy and MD simulation</span>”, <i><span class="bib-journal">J. Am. Chem. Soc.</span></i>, pp. <span class="bib-pages">4522-4534</span>, <span class="bib-year">2002</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-30"><span class="bib-index">30</span></a>] </span> <span class="bib-authors">Daniel R. Roe, Thomas E. Cheatham III</span>: “<span class="bib-title">Parallelization of CPPTRAJ enables large scale analysis of molecular dynamics trajectory data</span>”, <i><span class="bib-journal">Journal of Computational Chemistry</span></i>, pp. <span class="bib-pages">2110-2117</span>, <span class="bib-year">2018</span>. URL <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/jcc.25382"><span class="bib-url">https://onlinelibrary.wiley.com/doi/abs/10.1002/jcc.25382</span></a>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-31"><span class="bib-index">31</span></a>] </span> <span class="bib-authors">Daniel R. Roe, III Cheatham</span>: “<span class="bib-title">PTRAJ and CPPTRAJ: Software for Processing and Analysis of Molecular Dynamics Trajectory Data</span>”, <i><span class="bib-journal">J. Chem. Theory Comput.</span></i>, pp. <span class="bib-pages">3084-3095</span>, <span class="bib-year">2013</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-32"><span class="bib-index">32</span></a>] </span> <span class="bib-authors">B. Schneider, Z. Moravek, H. M. Berman</span>: “<span class="bib-title">RNA conformational classes</span>”, <i><span class="bib-journal">Nucleic Acids Res.</span></i>, pp. <span class="bib-pages">1666-1677</span>, <span class="bib-year">2004</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-33"><span class="bib-index">33</span></a>] </span> <span class="bib-authors">B. Schneider, S. Neidle, H. M. Berman</span>: “<span class="bib-title">Conformations of the Sugar-Phosphate Backbone in Helical DNA Crystal Structures</span>”, <i><span class="bib-journal">Biopolymers</span></i>, pp. <span class="bib-pages">113-124</span>, <span class="bib-year">1997</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-34"><span class="bib-index">34</span></a>] </span> <span class="bib-authors">D. J. Sindhikara, N. Yoshida, F. Hirata</span>: “<span class="bib-title">Placevent: An Algorithm for Prediction of Explicit Solvent Atom Distribution-Application to HIV-1 Protease and F-ATP Synthase.</span>”, <i><span class="bib-journal">J. Comput. Chem.</span></i>, pp. <span class="bib-pages">1536-1543</span>, <span class="bib-year">2012</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-35"><span class="bib-index">35</span></a>] </span> <span class="bib-authors">J. Weiser, P.S. Shenkin, W.C. Still</span>: “<span class="bib-title">Approximate Atomic Surfaces from Linear Combinations of Pairwise Overlaps (LCPO)</span>”, <i><span class="bib-journal">J. Comput. Chem.</span></i>, pp. <span class="bib-pages">217-230</span>, <span class="bib-year">1999</span>.
</p>
<p class="biblio">
<span class="entry">[<a class="biblioentry" name="biblio-36"><span class="bib-index">36</span></a>] </span> <span class="bib-authors">V. Wong, D.A. Case</span>: “<span class="bib-title">Evaluating rotational diffusion from protein MD simulations</span>”, <i><span class="bib-journal">J. Phys. Chem. B</span></i>, pp. <span class="bib-pages">6013-6024</span>, <span class="bib-year">2008</span>.
</p>

</div>

</div>
</body>
</html>
